{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.Download the datasets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.1 Download the multi-omics data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* parse the multiomics data from the synapse website of the dataset ROSMAP https://www.synapse.org/#!Synapse:syn23446022\n",
    "* Genome Variants https://www.synapse.org/#!Synapse:syn26263118\n",
    "* Methylation https://www.synapse.org/#!Synapse:syn3168763\n",
    "* RNA sequence https://www.synapse.org/#!Synapse:syn3505720\n",
    "* Proteomics https://www.synapse.org/#!Synapse:syn21266454    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.2 Download the clinical data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* parse the clinical data https://www.synapse.org/#!Synapse:syn3191087"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.Read the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from pyensembl import EnsemblRelease"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.1 Read DNA methylation data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylation_value = pd.read_csv(\"./ROSMAP-raw/Methylation/DNA-methylation-array/AMP-AD_ROSMAP_Rush-Broad_IlluminaHumanMethylation450_740_imputed.tsv\",sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TargetID</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>PT-318X</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cg00000165</td>\n",
       "      <td>0.231359</td>\n",
       "      <td>0.157857</td>\n",
       "      <td>0.127105</td>\n",
       "      <td>0.149988</td>\n",
       "      <td>0.130532</td>\n",
       "      <td>0.174451</td>\n",
       "      <td>0.170026</td>\n",
       "      <td>0.165900</td>\n",
       "      <td>0.157745</td>\n",
       "      <td>...</td>\n",
       "      <td>0.124635</td>\n",
       "      <td>0.168002</td>\n",
       "      <td>0.152200</td>\n",
       "      <td>0.230482</td>\n",
       "      <td>0.177287</td>\n",
       "      <td>0.195611</td>\n",
       "      <td>0.151338</td>\n",
       "      <td>0.151508</td>\n",
       "      <td>0.167960</td>\n",
       "      <td>0.136220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cg00000363</td>\n",
       "      <td>0.140664</td>\n",
       "      <td>0.114399</td>\n",
       "      <td>0.140580</td>\n",
       "      <td>0.145805</td>\n",
       "      <td>0.122833</td>\n",
       "      <td>0.117144</td>\n",
       "      <td>0.155559</td>\n",
       "      <td>0.131309</td>\n",
       "      <td>0.157749</td>\n",
       "      <td>...</td>\n",
       "      <td>0.129301</td>\n",
       "      <td>0.126910</td>\n",
       "      <td>0.135973</td>\n",
       "      <td>0.162883</td>\n",
       "      <td>0.122399</td>\n",
       "      <td>0.112684</td>\n",
       "      <td>0.125054</td>\n",
       "      <td>0.118550</td>\n",
       "      <td>0.114618</td>\n",
       "      <td>0.133814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cg00000957</td>\n",
       "      <td>0.776220</td>\n",
       "      <td>0.818279</td>\n",
       "      <td>0.630316</td>\n",
       "      <td>0.849261</td>\n",
       "      <td>0.861136</td>\n",
       "      <td>0.751543</td>\n",
       "      <td>0.778917</td>\n",
       "      <td>0.859892</td>\n",
       "      <td>0.787279</td>\n",
       "      <td>...</td>\n",
       "      <td>0.735520</td>\n",
       "      <td>0.740686</td>\n",
       "      <td>0.748494</td>\n",
       "      <td>0.797072</td>\n",
       "      <td>0.793081</td>\n",
       "      <td>0.692638</td>\n",
       "      <td>0.785597</td>\n",
       "      <td>0.712386</td>\n",
       "      <td>0.839168</td>\n",
       "      <td>0.888304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cg00001349</td>\n",
       "      <td>0.865488</td>\n",
       "      <td>0.919570</td>\n",
       "      <td>0.882256</td>\n",
       "      <td>0.902912</td>\n",
       "      <td>0.907610</td>\n",
       "      <td>0.897496</td>\n",
       "      <td>0.939212</td>\n",
       "      <td>0.919514</td>\n",
       "      <td>0.805626</td>\n",
       "      <td>...</td>\n",
       "      <td>0.885292</td>\n",
       "      <td>0.900684</td>\n",
       "      <td>0.906257</td>\n",
       "      <td>0.756173</td>\n",
       "      <td>0.948341</td>\n",
       "      <td>0.904996</td>\n",
       "      <td>0.927201</td>\n",
       "      <td>0.856637</td>\n",
       "      <td>0.917827</td>\n",
       "      <td>0.943320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cg00001364</td>\n",
       "      <td>0.772899</td>\n",
       "      <td>0.711099</td>\n",
       "      <td>0.694639</td>\n",
       "      <td>0.651986</td>\n",
       "      <td>0.748538</td>\n",
       "      <td>0.706625</td>\n",
       "      <td>0.743491</td>\n",
       "      <td>0.712513</td>\n",
       "      <td>0.722169</td>\n",
       "      <td>...</td>\n",
       "      <td>0.720420</td>\n",
       "      <td>0.724226</td>\n",
       "      <td>0.694462</td>\n",
       "      <td>0.776165</td>\n",
       "      <td>0.758569</td>\n",
       "      <td>0.733207</td>\n",
       "      <td>0.750224</td>\n",
       "      <td>0.737402</td>\n",
       "      <td>0.685959</td>\n",
       "      <td>0.770543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420127</th>\n",
       "      <td>ch.22.772318F</td>\n",
       "      <td>0.233366</td>\n",
       "      <td>0.198370</td>\n",
       "      <td>0.176775</td>\n",
       "      <td>0.141996</td>\n",
       "      <td>0.167610</td>\n",
       "      <td>0.252353</td>\n",
       "      <td>0.153457</td>\n",
       "      <td>0.124837</td>\n",
       "      <td>0.224565</td>\n",
       "      <td>...</td>\n",
       "      <td>0.253596</td>\n",
       "      <td>0.184639</td>\n",
       "      <td>0.175109</td>\n",
       "      <td>0.183692</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.193311</td>\n",
       "      <td>0.149942</td>\n",
       "      <td>0.135414</td>\n",
       "      <td>0.123727</td>\n",
       "      <td>0.121499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420128</th>\n",
       "      <td>ch.22.43177094F</td>\n",
       "      <td>0.173744</td>\n",
       "      <td>0.199095</td>\n",
       "      <td>0.189069</td>\n",
       "      <td>0.226845</td>\n",
       "      <td>0.193875</td>\n",
       "      <td>0.194734</td>\n",
       "      <td>0.194701</td>\n",
       "      <td>0.207784</td>\n",
       "      <td>0.183201</td>\n",
       "      <td>...</td>\n",
       "      <td>0.225843</td>\n",
       "      <td>0.172531</td>\n",
       "      <td>0.234507</td>\n",
       "      <td>0.169050</td>\n",
       "      <td>0.177599</td>\n",
       "      <td>0.205192</td>\n",
       "      <td>0.176225</td>\n",
       "      <td>0.191057</td>\n",
       "      <td>0.204723</td>\n",
       "      <td>0.179136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420129</th>\n",
       "      <td>ch.22.909671F</td>\n",
       "      <td>0.409177</td>\n",
       "      <td>0.463476</td>\n",
       "      <td>0.307656</td>\n",
       "      <td>0.302488</td>\n",
       "      <td>0.426867</td>\n",
       "      <td>0.425683</td>\n",
       "      <td>0.335775</td>\n",
       "      <td>0.361516</td>\n",
       "      <td>0.385871</td>\n",
       "      <td>...</td>\n",
       "      <td>0.431509</td>\n",
       "      <td>0.390838</td>\n",
       "      <td>0.402118</td>\n",
       "      <td>0.382906</td>\n",
       "      <td>0.410532</td>\n",
       "      <td>0.472419</td>\n",
       "      <td>0.340013</td>\n",
       "      <td>0.436682</td>\n",
       "      <td>0.413712</td>\n",
       "      <td>0.447053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420130</th>\n",
       "      <td>ch.22.46830341F</td>\n",
       "      <td>0.225070</td>\n",
       "      <td>0.279521</td>\n",
       "      <td>0.224084</td>\n",
       "      <td>0.286071</td>\n",
       "      <td>0.272743</td>\n",
       "      <td>0.322463</td>\n",
       "      <td>0.271097</td>\n",
       "      <td>0.268207</td>\n",
       "      <td>0.250027</td>\n",
       "      <td>...</td>\n",
       "      <td>0.267995</td>\n",
       "      <td>0.279551</td>\n",
       "      <td>0.333601</td>\n",
       "      <td>0.305278</td>\n",
       "      <td>0.280626</td>\n",
       "      <td>0.344297</td>\n",
       "      <td>0.251353</td>\n",
       "      <td>0.233328</td>\n",
       "      <td>0.280114</td>\n",
       "      <td>0.306628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420131</th>\n",
       "      <td>ch.22.1008279F</td>\n",
       "      <td>0.067878</td>\n",
       "      <td>0.072976</td>\n",
       "      <td>0.090009</td>\n",
       "      <td>0.051107</td>\n",
       "      <td>0.072089</td>\n",
       "      <td>0.110484</td>\n",
       "      <td>0.059588</td>\n",
       "      <td>0.036466</td>\n",
       "      <td>0.044706</td>\n",
       "      <td>...</td>\n",
       "      <td>0.053982</td>\n",
       "      <td>0.068004</td>\n",
       "      <td>0.079603</td>\n",
       "      <td>0.048614</td>\n",
       "      <td>0.080189</td>\n",
       "      <td>0.095757</td>\n",
       "      <td>0.051977</td>\n",
       "      <td>0.057825</td>\n",
       "      <td>0.071150</td>\n",
       "      <td>0.055774</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>420132 rows × 741 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               TargetID  TBI-AUTO73325-PT-3149   PT-BZHL   PT-BZCH   PT-BY9H  \\\n",
       "0            cg00000165               0.231359  0.157857  0.127105  0.149988   \n",
       "1            cg00000363               0.140664  0.114399  0.140580  0.145805   \n",
       "2            cg00000957               0.776220  0.818279  0.630316  0.849261   \n",
       "3            cg00001349               0.865488  0.919570  0.882256  0.902912   \n",
       "4            cg00001364               0.772899  0.711099  0.694639  0.651986   \n",
       "...                 ...                    ...       ...       ...       ...   \n",
       "420127    ch.22.772318F               0.233366  0.198370  0.176775  0.141996   \n",
       "420128  ch.22.43177094F               0.173744  0.199095  0.189069  0.226845   \n",
       "420129    ch.22.909671F               0.409177  0.463476  0.307656  0.302488   \n",
       "420130  ch.22.46830341F               0.225070  0.279521  0.224084  0.286071   \n",
       "420131   ch.22.1008279F               0.067878  0.072976  0.090009  0.051107   \n",
       "\n",
       "        TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD   PT-BZI5   PT-BZ1A  \\\n",
       "0                    0.130532               0.174451  0.170026  0.165900   \n",
       "1                    0.122833               0.117144  0.155559  0.131309   \n",
       "2                    0.861136               0.751543  0.778917  0.859892   \n",
       "3                    0.907610               0.897496  0.939212  0.919514   \n",
       "4                    0.748538               0.706625  0.743491  0.712513   \n",
       "...                       ...                    ...       ...       ...   \n",
       "420127               0.167610               0.252353  0.153457  0.124837   \n",
       "420128               0.193875               0.194734  0.194701  0.207784   \n",
       "420129               0.426867               0.425683  0.335775  0.361516   \n",
       "420130               0.272743               0.322463  0.271097  0.268207   \n",
       "420131               0.072089               0.110484  0.059588  0.036466   \n",
       "\n",
       "         PT-318X  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD   PT-BZD7  \\\n",
       "0       0.157745  ...               0.124635               0.168002  0.152200   \n",
       "1       0.157749  ...               0.129301               0.126910  0.135973   \n",
       "2       0.787279  ...               0.735520               0.740686  0.748494   \n",
       "3       0.805626  ...               0.885292               0.900684  0.906257   \n",
       "4       0.722169  ...               0.720420               0.724226  0.694462   \n",
       "...          ...  ...                    ...                    ...       ...   \n",
       "420127  0.224565  ...               0.253596               0.184639  0.175109   \n",
       "420128  0.183201  ...               0.225843               0.172531  0.234507   \n",
       "420129  0.385871  ...               0.431509               0.390838  0.402118   \n",
       "420130  0.250027  ...               0.267995               0.279551  0.333601   \n",
       "420131  0.044706  ...               0.053982               0.068004  0.079603   \n",
       "\n",
       "         PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0       0.230482  0.177287  0.195611  0.151338  0.151508   \n",
       "1       0.162883  0.122399  0.112684  0.125054  0.118550   \n",
       "2       0.797072  0.793081  0.692638  0.785597  0.712386   \n",
       "3       0.756173  0.948341  0.904996  0.927201  0.856637   \n",
       "4       0.776165  0.758569  0.733207  0.750224  0.737402   \n",
       "...          ...       ...       ...       ...       ...   \n",
       "420127  0.183692  0.000000  0.193311  0.149942  0.135414   \n",
       "420128  0.169050  0.177599  0.205192  0.176225  0.191057   \n",
       "420129  0.382906  0.410532  0.472419  0.340013  0.436682   \n",
       "420130  0.305278  0.280626  0.344297  0.251353  0.233328   \n",
       "420131  0.048614  0.080189  0.095757  0.051977  0.057825   \n",
       "\n",
       "        TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                    0.167960  0.136220  \n",
       "1                    0.114618  0.133814  \n",
       "2                    0.839168  0.888304  \n",
       "3                    0.917827  0.943320  \n",
       "4                    0.685959  0.770543  \n",
       "...                       ...       ...  \n",
       "420127               0.123727  0.121499  \n",
       "420128               0.204723  0.179136  \n",
       "420129               0.413712  0.447053  \n",
       "420130               0.280114  0.306628  \n",
       "420131               0.071150  0.055774  \n",
       "\n",
       "[420132 rows x 741 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylation_value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.2 Read Platform annotations for 450k methylation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\1730697995.py:3: DtypeWarning: Columns (16,17) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  annotation = pd.read_table('./ROSMAP-raw/Methylation/DNA-methylation-array/GPL16304-47833.txt', delimiter='\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>MAPINFO-1</th>\n",
       "      <th>MAPINFO+1</th>\n",
       "      <th>Probe_start</th>\n",
       "      <th>Probe_end</th>\n",
       "      <th>Target CpG SNP</th>\n",
       "      <th>n_target CpG SNP</th>\n",
       "      <th>SNPprobe</th>\n",
       "      <th>n_SNPprobe</th>\n",
       "      <th>HIL_CpG_class</th>\n",
       "      <th>...</th>\n",
       "      <th>AlleleA_Hits</th>\n",
       "      <th>AlleleB_Hits</th>\n",
       "      <th>XY_Hits</th>\n",
       "      <th>Autosomal_Hits</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>Closest_TSS_1</th>\n",
       "      <th>Distance_closest_TSS</th>\n",
       "      <th>Closest_TSS_gene_name</th>\n",
       "      <th>Closest_TSS_Transcript</th>\n",
       "      <th>SPOT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cg00000029</td>\n",
       "      <td>53468111</td>\n",
       "      <td>53468113</td>\n",
       "      <td>53468112</td>\n",
       "      <td>53468162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>53468350</td>\n",
       "      <td>53468351</td>\n",
       "      <td>-238</td>\n",
       "      <td>RBL2</td>\n",
       "      <td>NM_005611</td>\n",
       "      <td>cg00000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cg00000108</td>\n",
       "      <td>37459205</td>\n",
       "      <td>37459207</td>\n",
       "      <td>37459206</td>\n",
       "      <td>37459256</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs9857774</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>37458757</td>\n",
       "      <td>37458758</td>\n",
       "      <td>449</td>\n",
       "      <td>C3orf35</td>\n",
       "      <td>CCDS46792</td>\n",
       "      <td>cg00000108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cg00000109</td>\n",
       "      <td>171916036</td>\n",
       "      <td>171916038</td>\n",
       "      <td>171916037</td>\n",
       "      <td>171916087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs9864492</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>171851260</td>\n",
       "      <td>171851261</td>\n",
       "      <td>64777</td>\n",
       "      <td>FNDC3B</td>\n",
       "      <td>AY358367</td>\n",
       "      <td>cg00000109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cg00000165</td>\n",
       "      <td>91194673</td>\n",
       "      <td>91194675</td>\n",
       "      <td>91194624</td>\n",
       "      <td>91194674</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs76771611</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ICshore</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>91182793</td>\n",
       "      <td>91182794</td>\n",
       "      <td>-11880</td>\n",
       "      <td>BARHL2</td>\n",
       "      <td>NM_020063</td>\n",
       "      <td>cg00000165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cg00000236</td>\n",
       "      <td>42263293</td>\n",
       "      <td>42263295</td>\n",
       "      <td>42263244</td>\n",
       "      <td>42263294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>42251727</td>\n",
       "      <td>42251728</td>\n",
       "      <td>11567</td>\n",
       "      <td>VDAC3</td>\n",
       "      <td>CCDS47850</td>\n",
       "      <td>cg00000236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485507</th>\n",
       "      <td>ch.X.97129969R</td>\n",
       "      <td>97243312</td>\n",
       "      <td>97243314</td>\n",
       "      <td>97243313</td>\n",
       "      <td>97243363</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>97607750</td>\n",
       "      <td>97607751</td>\n",
       "      <td>-364437</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>.</td>\n",
       "      <td>ch.X.97129969R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485508</th>\n",
       "      <td>ch.X.97133160R</td>\n",
       "      <td>97246503</td>\n",
       "      <td>97246505</td>\n",
       "      <td>97246504</td>\n",
       "      <td>97246554</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_YES</td>\n",
       "      <td>97607750</td>\n",
       "      <td>97607751</td>\n",
       "      <td>-361246</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>.</td>\n",
       "      <td>ch.X.97133160R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485509</th>\n",
       "      <td>ch.X.97651759F</td>\n",
       "      <td>97765102</td>\n",
       "      <td>97765104</td>\n",
       "      <td>97765103</td>\n",
       "      <td>97765153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>97607750</td>\n",
       "      <td>97607751</td>\n",
       "      <td>157353</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>.</td>\n",
       "      <td>ch.X.97651759F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485510</th>\n",
       "      <td>ch.X.97737721F</td>\n",
       "      <td>97851064</td>\n",
       "      <td>97851066</td>\n",
       "      <td>97851065</td>\n",
       "      <td>97851115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_YES</td>\n",
       "      <td>97607750</td>\n",
       "      <td>97607751</td>\n",
       "      <td>243315</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>.</td>\n",
       "      <td>ch.X.97737721F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485511</th>\n",
       "      <td>ch.X.98007042R</td>\n",
       "      <td>98120385</td>\n",
       "      <td>98120387</td>\n",
       "      <td>98120386</td>\n",
       "      <td>98120436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LC</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>XY_NO</td>\n",
       "      <td>A_NO</td>\n",
       "      <td>97607750</td>\n",
       "      <td>97607751</td>\n",
       "      <td>512636</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>.</td>\n",
       "      <td>ch.X.98007042R</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>485493 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID  MAPINFO-1  MAPINFO+1  Probe_start  Probe_end  \\\n",
       "0           cg00000029   53468111   53468113     53468112   53468162   \n",
       "1           cg00000108   37459205   37459207     37459206   37459256   \n",
       "2           cg00000109  171916036  171916038    171916037  171916087   \n",
       "3           cg00000165   91194673   91194675     91194624   91194674   \n",
       "4           cg00000236   42263293   42263295     42263244   42263294   \n",
       "...                ...        ...        ...          ...        ...   \n",
       "485507  ch.X.97129969R   97243312   97243314     97243313   97243363   \n",
       "485508  ch.X.97133160R   97246503   97246505     97246504   97246554   \n",
       "485509  ch.X.97651759F   97765102   97765104     97765103   97765153   \n",
       "485510  ch.X.97737721F   97851064   97851066     97851065   97851115   \n",
       "485511  ch.X.98007042R   98120385   98120387     98120386   98120436   \n",
       "\n",
       "       Target CpG SNP  n_target CpG SNP    SNPprobe  n_SNPprobe HIL_CpG_class  \\\n",
       "0                 NaN               NaN         NaN         NaN            HC   \n",
       "1                 NaN               NaN   rs9857774         1.0            LC   \n",
       "2                 NaN               NaN   rs9864492         1.0            LC   \n",
       "3                 NaN               NaN  rs76771611         1.0       ICshore   \n",
       "4                 NaN               NaN         NaN         NaN            IC   \n",
       "...               ...               ...         ...         ...           ...   \n",
       "485507            NaN               NaN         NaN         NaN            LC   \n",
       "485508            NaN               NaN         NaN         NaN            LC   \n",
       "485509            NaN               NaN         NaN         NaN            LC   \n",
       "485510            NaN               NaN         NaN         NaN            LC   \n",
       "485511            NaN               NaN         NaN         NaN            LC   \n",
       "\n",
       "        ... AlleleA_Hits  AlleleB_Hits  XY_Hits  Autosomal_Hits Closest_TSS  \\\n",
       "0       ...            1             0    XY_NO            A_NO    53468350   \n",
       "1       ...            1             0    XY_NO            A_NO    37458757   \n",
       "2       ...            1             0    XY_NO            A_NO   171851260   \n",
       "3       ...            1             0    XY_NO            A_NO    91182793   \n",
       "4       ...            1             0    XY_NO            A_NO    42251727   \n",
       "...     ...          ...           ...      ...             ...         ...   \n",
       "485507  ...            1             0    XY_NO            A_NO    97607750   \n",
       "485508  ...            2             0    XY_NO           A_YES    97607750   \n",
       "485509  ...            1             0    XY_NO            A_NO    97607750   \n",
       "485510  ...           21             0    XY_NO           A_YES    97607750   \n",
       "485511  ...            1             0    XY_NO            A_NO    97607750   \n",
       "\n",
       "       Closest_TSS_1 Distance_closest_TSS Closest_TSS_gene_name  \\\n",
       "0           53468351                 -238                  RBL2   \n",
       "1           37458758                  449               C3orf35   \n",
       "2          171851261                64777                FNDC3B   \n",
       "3           91182794               -11880                BARHL2   \n",
       "4           42251728                11567                 VDAC3   \n",
       "...              ...                  ...                   ...   \n",
       "485507      97607751              -364437               Mir_340   \n",
       "485508      97607751              -361246               Mir_340   \n",
       "485509      97607751               157353               Mir_340   \n",
       "485510      97607751               243315               Mir_340   \n",
       "485511      97607751               512636               Mir_340   \n",
       "\n",
       "        Closest_TSS_Transcript         SPOT_ID  \n",
       "0                    NM_005611      cg00000029  \n",
       "1                    CCDS46792      cg00000108  \n",
       "2                     AY358367      cg00000109  \n",
       "3                    NM_020063      cg00000165  \n",
       "4                    CCDS47850      cg00000236  \n",
       "...                        ...             ...  \n",
       "485507                       .  ch.X.97129969R  \n",
       "485508                       .  ch.X.97133160R  \n",
       "485509                       .  ch.X.97651759F  \n",
       "485510                       .  ch.X.97737721F  \n",
       "485511                       .  ch.X.98007042R  \n",
       "\n",
       "[485493 rows x 22 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reading and processing basic data\n",
    "try:\n",
    "    annotation = pd.read_table('./ROSMAP-raw/Methylation/DNA-methylation-array/GPL16304-47833.txt', delimiter='\\t')\n",
    "    annotation['Distance_closest_TSS'] = annotation['Distance_closest_TSS'].astype(int)\n",
    "    annotation = annotation[~annotation['Closest_TSS'].apply(lambda x: len(str(x).split(';')) > 1)]\n",
    "except ValueError as e:\n",
    "    print(f\"Unable to convert 'Closest_TSS' column to integer: {e}\")\n",
    "    problematic_rows = annotation['Distance_closest_TSS'].apply(lambda x: not str(x).isnumeric())\n",
    "    print(\"Problematic rows:\")\n",
    "    print(annotation.loc[problematic_rows])\n",
    "\n",
    "annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>Closest_TSS_gene_name</th>\n",
       "      <th>Distance_closest_TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cg00000029</td>\n",
       "      <td>53468350</td>\n",
       "      <td>RBL2</td>\n",
       "      <td>-238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cg00000108</td>\n",
       "      <td>37458757</td>\n",
       "      <td>C3orf35</td>\n",
       "      <td>449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cg00000109</td>\n",
       "      <td>171851260</td>\n",
       "      <td>FNDC3B</td>\n",
       "      <td>64777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cg00000165</td>\n",
       "      <td>91182793</td>\n",
       "      <td>BARHL2</td>\n",
       "      <td>-11880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cg00000236</td>\n",
       "      <td>42251727</td>\n",
       "      <td>VDAC3</td>\n",
       "      <td>11567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485507</th>\n",
       "      <td>ch.X.97129969R</td>\n",
       "      <td>97607750</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>-364437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485508</th>\n",
       "      <td>ch.X.97133160R</td>\n",
       "      <td>97607750</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>-361246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485509</th>\n",
       "      <td>ch.X.97651759F</td>\n",
       "      <td>97607750</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>157353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485510</th>\n",
       "      <td>ch.X.97737721F</td>\n",
       "      <td>97607750</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>243315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485511</th>\n",
       "      <td>ch.X.98007042R</td>\n",
       "      <td>97607750</td>\n",
       "      <td>Mir_340</td>\n",
       "      <td>512636</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>485493 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID Closest_TSS Closest_TSS_gene_name  Distance_closest_TSS\n",
       "0           cg00000029    53468350                  RBL2                  -238\n",
       "1           cg00000108    37458757               C3orf35                   449\n",
       "2           cg00000109   171851260                FNDC3B                 64777\n",
       "3           cg00000165    91182793                BARHL2                -11880\n",
       "4           cg00000236    42251727                 VDAC3                 11567\n",
       "...                ...         ...                   ...                   ...\n",
       "485507  ch.X.97129969R    97607750               Mir_340               -364437\n",
       "485508  ch.X.97133160R    97607750               Mir_340               -361246\n",
       "485509  ch.X.97651759F    97607750               Mir_340                157353\n",
       "485510  ch.X.97737721F    97607750               Mir_340                243315\n",
       "485511  ch.X.98007042R    97607750               Mir_340                512636\n",
       "\n",
       "[485493 rows x 4 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "map_annotation= annotation[['ID', 'Closest_TSS','Closest_TSS_gene_name', 'Distance_closest_TSS']]\n",
    "map_annotation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.3 Read gene expression data, mapping information and substitue the the gene name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tracking_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>...</th>\n",
       "      <th>831_130725_8</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000167578.11</td>\n",
       "      <td>ENSG00000167578.11</td>\n",
       "      <td>60.84</td>\n",
       "      <td>65.45</td>\n",
       "      <td>69.18</td>\n",
       "      <td>51.60</td>\n",
       "      <td>48.61</td>\n",
       "      <td>55.65</td>\n",
       "      <td>51.18</td>\n",
       "      <td>60.63</td>\n",
       "      <td>...</td>\n",
       "      <td>44.85</td>\n",
       "      <td>45.02</td>\n",
       "      <td>33.71</td>\n",
       "      <td>38.30</td>\n",
       "      <td>28.27</td>\n",
       "      <td>51.02</td>\n",
       "      <td>31.87</td>\n",
       "      <td>37.14</td>\n",
       "      <td>41.77</td>\n",
       "      <td>44.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000242268.1</td>\n",
       "      <td>ENSG00000242268.1</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.11</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000078237.4</td>\n",
       "      <td>ENSG00000078237.4</td>\n",
       "      <td>4.39</td>\n",
       "      <td>4.49</td>\n",
       "      <td>2.51</td>\n",
       "      <td>2.90</td>\n",
       "      <td>2.67</td>\n",
       "      <td>5.50</td>\n",
       "      <td>3.36</td>\n",
       "      <td>3.43</td>\n",
       "      <td>...</td>\n",
       "      <td>8.15</td>\n",
       "      <td>4.05</td>\n",
       "      <td>6.06</td>\n",
       "      <td>6.03</td>\n",
       "      <td>4.34</td>\n",
       "      <td>5.16</td>\n",
       "      <td>2.70</td>\n",
       "      <td>5.00</td>\n",
       "      <td>3.82</td>\n",
       "      <td>3.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000263642.1</td>\n",
       "      <td>ENSG00000263642.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000225275.4</td>\n",
       "      <td>ENSG00000225275.4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55884</th>\n",
       "      <td>ENSG00000265520.1</td>\n",
       "      <td>ENSG00000265520.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55885</th>\n",
       "      <td>ENSG00000231119.2</td>\n",
       "      <td>ENSG00000231119.2</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.26</td>\n",
       "      <td>...</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55886</th>\n",
       "      <td>ENSG00000105063.14</td>\n",
       "      <td>ENSG00000105063.14</td>\n",
       "      <td>40.59</td>\n",
       "      <td>39.26</td>\n",
       "      <td>34.70</td>\n",
       "      <td>36.74</td>\n",
       "      <td>39.85</td>\n",
       "      <td>35.28</td>\n",
       "      <td>24.78</td>\n",
       "      <td>31.14</td>\n",
       "      <td>...</td>\n",
       "      <td>28.82</td>\n",
       "      <td>24.08</td>\n",
       "      <td>24.10</td>\n",
       "      <td>24.81</td>\n",
       "      <td>27.30</td>\n",
       "      <td>22.24</td>\n",
       "      <td>20.08</td>\n",
       "      <td>23.68</td>\n",
       "      <td>17.91</td>\n",
       "      <td>26.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55887</th>\n",
       "      <td>ENSG00000123685.4</td>\n",
       "      <td>ENSG00000123685.4</td>\n",
       "      <td>4.46</td>\n",
       "      <td>4.51</td>\n",
       "      <td>5.27</td>\n",
       "      <td>5.11</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.31</td>\n",
       "      <td>4.63</td>\n",
       "      <td>4.40</td>\n",
       "      <td>...</td>\n",
       "      <td>3.03</td>\n",
       "      <td>5.06</td>\n",
       "      <td>2.05</td>\n",
       "      <td>4.01</td>\n",
       "      <td>2.89</td>\n",
       "      <td>2.69</td>\n",
       "      <td>3.42</td>\n",
       "      <td>4.25</td>\n",
       "      <td>5.46</td>\n",
       "      <td>3.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55888</th>\n",
       "      <td>ENSG00000181518.2</td>\n",
       "      <td>ENSG00000181518.2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55889 rows × 642 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              tracking_id             gene_id  525_120515_0  383_120503_0  \\\n",
       "0      ENSG00000167578.11  ENSG00000167578.11         60.84         65.45   \n",
       "1       ENSG00000242268.1   ENSG00000242268.1          0.08          0.05   \n",
       "2       ENSG00000078237.4   ENSG00000078237.4          4.39          4.49   \n",
       "3       ENSG00000263642.1   ENSG00000263642.1          0.00          0.00   \n",
       "4       ENSG00000225275.4   ENSG00000225275.4          0.00          0.00   \n",
       "...                   ...                 ...           ...           ...   \n",
       "55884   ENSG00000265520.1   ENSG00000265520.1          0.00          0.00   \n",
       "55885   ENSG00000231119.2   ENSG00000231119.2          0.20          0.20   \n",
       "55886  ENSG00000105063.14  ENSG00000105063.14         40.59         39.26   \n",
       "55887   ENSG00000123685.4   ENSG00000123685.4          4.46          4.51   \n",
       "55888   ENSG00000181518.2   ENSG00000181518.2          0.00          0.00   \n",
       "\n",
       "       93_120417_0  610_120523_0  560_120517_0  492_120515_0  576_120521_0  \\\n",
       "0            69.18         51.60         48.61         55.65         51.18   \n",
       "1             0.08          0.08          0.10          0.08          0.06   \n",
       "2             2.51          2.90          2.67          5.50          3.36   \n",
       "3             0.00          0.00          0.00          0.00          0.00   \n",
       "4             0.00          0.00          0.00          0.00          0.00   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "55884         0.00          0.00          0.00          0.00          0.00   \n",
       "55885         0.24          0.21          0.09          0.14          0.42   \n",
       "55886        34.70         36.74         39.85         35.28         24.78   \n",
       "55887         5.27          5.11          3.77          4.31          4.63   \n",
       "55888         0.05          0.05          0.00          0.00          0.02   \n",
       "\n",
       "       150_120419_0  ...  831_130725_8  901_131010_8  894_130923_8  \\\n",
       "0             60.63  ...         44.85         45.02         33.71   \n",
       "1              0.11  ...          0.00          0.10          0.20   \n",
       "2              3.43  ...          8.15          4.05          6.06   \n",
       "3              0.00  ...          0.00          0.00          0.00   \n",
       "4              0.00  ...          0.00          0.00          0.00   \n",
       "...             ...  ...           ...           ...           ...   \n",
       "55884          0.00  ...          0.00          0.00          0.00   \n",
       "55885          0.26  ...          0.22          0.11          0.18   \n",
       "55886         31.14  ...         28.82         24.08         24.10   \n",
       "55887          4.40  ...          3.03          5.06          2.05   \n",
       "55888          0.00  ...          0.20          0.00          0.00   \n",
       "\n",
       "       938_131101_8  942_131101_8  939_131101_8  895_130923_8  829_130725_8  \\\n",
       "0             38.30         28.27         51.02         31.87         37.14   \n",
       "1              0.00          0.00          0.00          0.00          0.00   \n",
       "2              6.03          4.34          5.16          2.70          5.00   \n",
       "3              0.00          0.00          0.00          0.00          0.00   \n",
       "4              0.00          0.00          0.00          0.00          0.00   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "55884          0.00          0.00          0.00          0.00          0.00   \n",
       "55885          0.24          0.14          0.28          0.24          0.15   \n",
       "55886         24.81         27.30         22.24         20.08         23.68   \n",
       "55887          4.01          2.89          2.69          3.42          4.25   \n",
       "55888          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       944_131107_8  775_130528_8  \n",
       "0             41.77         44.01  \n",
       "1              0.09          0.10  \n",
       "2              3.82          3.69  \n",
       "3              0.00          0.00  \n",
       "4              0.00          0.00  \n",
       "...             ...           ...  \n",
       "55884          0.00          0.00  \n",
       "55885          0.23          0.15  \n",
       "55886         17.91         26.37  \n",
       "55887          5.46          3.65  \n",
       "55888          0.00          0.00  \n",
       "\n",
       "[55889 rows x 642 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_expression = pd.read_csv('./ROSMAP-raw/RNASeq/ROSMAP_RNAseq_FPKM_gene.tsv', sep='\\t')\n",
    "gene_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tracking_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>...</th>\n",
       "      <th>831_130725_8</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000167578.11</td>\n",
       "      <td>ENSG00000167578</td>\n",
       "      <td>60.84</td>\n",
       "      <td>65.45</td>\n",
       "      <td>69.18</td>\n",
       "      <td>51.60</td>\n",
       "      <td>48.61</td>\n",
       "      <td>55.65</td>\n",
       "      <td>51.18</td>\n",
       "      <td>60.63</td>\n",
       "      <td>...</td>\n",
       "      <td>44.85</td>\n",
       "      <td>45.02</td>\n",
       "      <td>33.71</td>\n",
       "      <td>38.30</td>\n",
       "      <td>28.27</td>\n",
       "      <td>51.02</td>\n",
       "      <td>31.87</td>\n",
       "      <td>37.14</td>\n",
       "      <td>41.77</td>\n",
       "      <td>44.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000242268.1</td>\n",
       "      <td>ENSG00000242268</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.11</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000078237.4</td>\n",
       "      <td>ENSG00000078237</td>\n",
       "      <td>4.39</td>\n",
       "      <td>4.49</td>\n",
       "      <td>2.51</td>\n",
       "      <td>2.90</td>\n",
       "      <td>2.67</td>\n",
       "      <td>5.50</td>\n",
       "      <td>3.36</td>\n",
       "      <td>3.43</td>\n",
       "      <td>...</td>\n",
       "      <td>8.15</td>\n",
       "      <td>4.05</td>\n",
       "      <td>6.06</td>\n",
       "      <td>6.03</td>\n",
       "      <td>4.34</td>\n",
       "      <td>5.16</td>\n",
       "      <td>2.70</td>\n",
       "      <td>5.00</td>\n",
       "      <td>3.82</td>\n",
       "      <td>3.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000263642.1</td>\n",
       "      <td>ENSG00000263642</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000225275.4</td>\n",
       "      <td>ENSG00000225275</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55884</th>\n",
       "      <td>ENSG00000265520.1</td>\n",
       "      <td>ENSG00000265520</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55885</th>\n",
       "      <td>ENSG00000231119.2</td>\n",
       "      <td>ENSG00000231119</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.26</td>\n",
       "      <td>...</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55886</th>\n",
       "      <td>ENSG00000105063.14</td>\n",
       "      <td>ENSG00000105063</td>\n",
       "      <td>40.59</td>\n",
       "      <td>39.26</td>\n",
       "      <td>34.70</td>\n",
       "      <td>36.74</td>\n",
       "      <td>39.85</td>\n",
       "      <td>35.28</td>\n",
       "      <td>24.78</td>\n",
       "      <td>31.14</td>\n",
       "      <td>...</td>\n",
       "      <td>28.82</td>\n",
       "      <td>24.08</td>\n",
       "      <td>24.10</td>\n",
       "      <td>24.81</td>\n",
       "      <td>27.30</td>\n",
       "      <td>22.24</td>\n",
       "      <td>20.08</td>\n",
       "      <td>23.68</td>\n",
       "      <td>17.91</td>\n",
       "      <td>26.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55887</th>\n",
       "      <td>ENSG00000123685.4</td>\n",
       "      <td>ENSG00000123685</td>\n",
       "      <td>4.46</td>\n",
       "      <td>4.51</td>\n",
       "      <td>5.27</td>\n",
       "      <td>5.11</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.31</td>\n",
       "      <td>4.63</td>\n",
       "      <td>4.40</td>\n",
       "      <td>...</td>\n",
       "      <td>3.03</td>\n",
       "      <td>5.06</td>\n",
       "      <td>2.05</td>\n",
       "      <td>4.01</td>\n",
       "      <td>2.89</td>\n",
       "      <td>2.69</td>\n",
       "      <td>3.42</td>\n",
       "      <td>4.25</td>\n",
       "      <td>5.46</td>\n",
       "      <td>3.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55888</th>\n",
       "      <td>ENSG00000181518.2</td>\n",
       "      <td>ENSG00000181518</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55889 rows × 642 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              tracking_id          gene_id  525_120515_0  383_120503_0  \\\n",
       "0      ENSG00000167578.11  ENSG00000167578         60.84         65.45   \n",
       "1       ENSG00000242268.1  ENSG00000242268          0.08          0.05   \n",
       "2       ENSG00000078237.4  ENSG00000078237          4.39          4.49   \n",
       "3       ENSG00000263642.1  ENSG00000263642          0.00          0.00   \n",
       "4       ENSG00000225275.4  ENSG00000225275          0.00          0.00   \n",
       "...                   ...              ...           ...           ...   \n",
       "55884   ENSG00000265520.1  ENSG00000265520          0.00          0.00   \n",
       "55885   ENSG00000231119.2  ENSG00000231119          0.20          0.20   \n",
       "55886  ENSG00000105063.14  ENSG00000105063         40.59         39.26   \n",
       "55887   ENSG00000123685.4  ENSG00000123685          4.46          4.51   \n",
       "55888   ENSG00000181518.2  ENSG00000181518          0.00          0.00   \n",
       "\n",
       "       93_120417_0  610_120523_0  560_120517_0  492_120515_0  576_120521_0  \\\n",
       "0            69.18         51.60         48.61         55.65         51.18   \n",
       "1             0.08          0.08          0.10          0.08          0.06   \n",
       "2             2.51          2.90          2.67          5.50          3.36   \n",
       "3             0.00          0.00          0.00          0.00          0.00   \n",
       "4             0.00          0.00          0.00          0.00          0.00   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "55884         0.00          0.00          0.00          0.00          0.00   \n",
       "55885         0.24          0.21          0.09          0.14          0.42   \n",
       "55886        34.70         36.74         39.85         35.28         24.78   \n",
       "55887         5.27          5.11          3.77          4.31          4.63   \n",
       "55888         0.05          0.05          0.00          0.00          0.02   \n",
       "\n",
       "       150_120419_0  ...  831_130725_8  901_131010_8  894_130923_8  \\\n",
       "0             60.63  ...         44.85         45.02         33.71   \n",
       "1              0.11  ...          0.00          0.10          0.20   \n",
       "2              3.43  ...          8.15          4.05          6.06   \n",
       "3              0.00  ...          0.00          0.00          0.00   \n",
       "4              0.00  ...          0.00          0.00          0.00   \n",
       "...             ...  ...           ...           ...           ...   \n",
       "55884          0.00  ...          0.00          0.00          0.00   \n",
       "55885          0.26  ...          0.22          0.11          0.18   \n",
       "55886         31.14  ...         28.82         24.08         24.10   \n",
       "55887          4.40  ...          3.03          5.06          2.05   \n",
       "55888          0.00  ...          0.20          0.00          0.00   \n",
       "\n",
       "       938_131101_8  942_131101_8  939_131101_8  895_130923_8  829_130725_8  \\\n",
       "0             38.30         28.27         51.02         31.87         37.14   \n",
       "1              0.00          0.00          0.00          0.00          0.00   \n",
       "2              6.03          4.34          5.16          2.70          5.00   \n",
       "3              0.00          0.00          0.00          0.00          0.00   \n",
       "4              0.00          0.00          0.00          0.00          0.00   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "55884          0.00          0.00          0.00          0.00          0.00   \n",
       "55885          0.24          0.14          0.28          0.24          0.15   \n",
       "55886         24.81         27.30         22.24         20.08         23.68   \n",
       "55887          4.01          2.89          2.69          3.42          4.25   \n",
       "55888          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       944_131107_8  775_130528_8  \n",
       "0             41.77         44.01  \n",
       "1              0.09          0.10  \n",
       "2              3.82          3.69  \n",
       "3              0.00          0.00  \n",
       "4              0.00          0.00  \n",
       "...             ...           ...  \n",
       "55884          0.00          0.00  \n",
       "55885          0.23          0.15  \n",
       "55886         17.91         26.37  \n",
       "55887          5.46          3.65  \n",
       "55888          0.00          0.00  \n",
       "\n",
       "[55889 rows x 642 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl_gene_ids = gene_expression['gene_id'].apply(lambda x: x.split('.')[0]).tolist()\n",
    "gene_expression['gene_id'] = ensembl_gene_ids\n",
    "gene_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48105</th>\n",
       "      <td>ZNF593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48106</th>\n",
       "      <td>CNKSR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48107</th>\n",
       "      <td>ZPLD2P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48108</th>\n",
       "      <td>DDX11L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48109</th>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48110 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene name\n",
       "0         MT-TF\n",
       "1       MT-RNR1\n",
       "2         MT-TV\n",
       "3       MT-RNR2\n",
       "4        MT-TL1\n",
       "...         ...\n",
       "48105    ZNF593\n",
       "48106    CNKSR1\n",
       "48107    ZPLD2P\n",
       "48108   DDX11L2\n",
       "48109    WASH7P\n",
       "\n",
       "[48110 rows x 1 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl_data_unique_gene = pd.read_csv(\"./ROSMAP-raw/Meta-Data/mart_export_unique_gene.txt\")\n",
    "ensembl_data_unique_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70706</th>\n",
       "      <td>ENSG00000288629</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70707</th>\n",
       "      <td>ENSG00000288678</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70708</th>\n",
       "      <td>ENSG00000290825</td>\n",
       "      <td>DDX11L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70709</th>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70710</th>\n",
       "      <td>ENSG00000290826</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70711 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id Gene name\n",
       "0      ENSG00000210049     MT-TF\n",
       "1      ENSG00000211459   MT-RNR1\n",
       "2      ENSG00000210077     MT-TV\n",
       "3      ENSG00000210082   MT-RNR2\n",
       "4      ENSG00000209082    MT-TL1\n",
       "...                ...       ...\n",
       "70706  ENSG00000288629       NaN\n",
       "70707  ENSG00000288678       NaN\n",
       "70708  ENSG00000290825   DDX11L2\n",
       "70709  ENSG00000227232    WASH7P\n",
       "70710  ENSG00000290826       NaN\n",
       "\n",
       "[70711 rows x 2 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl_data = pd.read_csv(\"./ROSMAP-raw/Meta-Data/mart_export_genename.txt\")\n",
    "ensembl_data= ensembl_data.rename(columns={'Gene stable ID': 'gene_id'})\n",
    "ensembl_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70701</th>\n",
       "      <td>ENSG00000142684</td>\n",
       "      <td>ZNF593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70702</th>\n",
       "      <td>ENSG00000142675</td>\n",
       "      <td>CNKSR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70703</th>\n",
       "      <td>ENSG00000293494</td>\n",
       "      <td>ZPLD2P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70708</th>\n",
       "      <td>ENSG00000290825</td>\n",
       "      <td>DDX11L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70709</th>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48110 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id Gene name\n",
       "0      ENSG00000210049     MT-TF\n",
       "1      ENSG00000211459   MT-RNR1\n",
       "2      ENSG00000210077     MT-TV\n",
       "3      ENSG00000210082   MT-RNR2\n",
       "4      ENSG00000209082    MT-TL1\n",
       "...                ...       ...\n",
       "70701  ENSG00000142684    ZNF593\n",
       "70702  ENSG00000142675    CNKSR1\n",
       "70703  ENSG00000293494    ZPLD2P\n",
       "70708  ENSG00000290825   DDX11L2\n",
       "70709  ENSG00000227232    WASH7P\n",
       "\n",
       "[48110 rows x 2 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl_data = ensembl_data.dropna()\n",
    "ensembl_data = ensembl_data.drop_duplicates()\n",
    "ensembl_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tracking_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>...</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000167578.11</td>\n",
       "      <td>ENSG00000167578</td>\n",
       "      <td>60.84</td>\n",
       "      <td>65.45</td>\n",
       "      <td>69.18</td>\n",
       "      <td>51.60</td>\n",
       "      <td>48.61</td>\n",
       "      <td>55.65</td>\n",
       "      <td>51.18</td>\n",
       "      <td>60.63</td>\n",
       "      <td>...</td>\n",
       "      <td>45.02</td>\n",
       "      <td>33.71</td>\n",
       "      <td>38.30</td>\n",
       "      <td>28.27</td>\n",
       "      <td>51.02</td>\n",
       "      <td>31.87</td>\n",
       "      <td>37.14</td>\n",
       "      <td>41.77</td>\n",
       "      <td>44.01</td>\n",
       "      <td>RAB4B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000242268.1</td>\n",
       "      <td>ENSG00000242268</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.11</td>\n",
       "      <td>...</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.10</td>\n",
       "      <td>LINC02082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000078237.4</td>\n",
       "      <td>ENSG00000078237</td>\n",
       "      <td>4.39</td>\n",
       "      <td>4.49</td>\n",
       "      <td>2.51</td>\n",
       "      <td>2.90</td>\n",
       "      <td>2.67</td>\n",
       "      <td>5.50</td>\n",
       "      <td>3.36</td>\n",
       "      <td>3.43</td>\n",
       "      <td>...</td>\n",
       "      <td>4.05</td>\n",
       "      <td>6.06</td>\n",
       "      <td>6.03</td>\n",
       "      <td>4.34</td>\n",
       "      <td>5.16</td>\n",
       "      <td>2.70</td>\n",
       "      <td>5.00</td>\n",
       "      <td>3.82</td>\n",
       "      <td>3.69</td>\n",
       "      <td>TIGAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000263642.1</td>\n",
       "      <td>ENSG00000263642</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>MIR4802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000225275.4</td>\n",
       "      <td>ENSG00000225275</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NUP210P2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38777</th>\n",
       "      <td>ENSG00000148943.7</td>\n",
       "      <td>ENSG00000148943</td>\n",
       "      <td>10.02</td>\n",
       "      <td>8.76</td>\n",
       "      <td>6.28</td>\n",
       "      <td>8.32</td>\n",
       "      <td>8.68</td>\n",
       "      <td>8.93</td>\n",
       "      <td>9.85</td>\n",
       "      <td>10.67</td>\n",
       "      <td>...</td>\n",
       "      <td>11.47</td>\n",
       "      <td>14.18</td>\n",
       "      <td>13.93</td>\n",
       "      <td>15.96</td>\n",
       "      <td>14.90</td>\n",
       "      <td>11.57</td>\n",
       "      <td>11.81</td>\n",
       "      <td>15.92</td>\n",
       "      <td>12.16</td>\n",
       "      <td>LIN7C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38778</th>\n",
       "      <td>ENSG00000265520.1</td>\n",
       "      <td>ENSG00000265520</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>MIR548V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38779</th>\n",
       "      <td>ENSG00000105063.14</td>\n",
       "      <td>ENSG00000105063</td>\n",
       "      <td>40.59</td>\n",
       "      <td>39.26</td>\n",
       "      <td>34.70</td>\n",
       "      <td>36.74</td>\n",
       "      <td>39.85</td>\n",
       "      <td>35.28</td>\n",
       "      <td>24.78</td>\n",
       "      <td>31.14</td>\n",
       "      <td>...</td>\n",
       "      <td>24.08</td>\n",
       "      <td>24.10</td>\n",
       "      <td>24.81</td>\n",
       "      <td>27.30</td>\n",
       "      <td>22.24</td>\n",
       "      <td>20.08</td>\n",
       "      <td>23.68</td>\n",
       "      <td>17.91</td>\n",
       "      <td>26.37</td>\n",
       "      <td>PPP6R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38780</th>\n",
       "      <td>ENSG00000123685.4</td>\n",
       "      <td>ENSG00000123685</td>\n",
       "      <td>4.46</td>\n",
       "      <td>4.51</td>\n",
       "      <td>5.27</td>\n",
       "      <td>5.11</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.31</td>\n",
       "      <td>4.63</td>\n",
       "      <td>4.40</td>\n",
       "      <td>...</td>\n",
       "      <td>5.06</td>\n",
       "      <td>2.05</td>\n",
       "      <td>4.01</td>\n",
       "      <td>2.89</td>\n",
       "      <td>2.69</td>\n",
       "      <td>3.42</td>\n",
       "      <td>4.25</td>\n",
       "      <td>5.46</td>\n",
       "      <td>3.65</td>\n",
       "      <td>BATF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38781</th>\n",
       "      <td>ENSG00000181518.2</td>\n",
       "      <td>ENSG00000181518</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>OR8D4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38782 rows × 643 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              tracking_id          gene_id  525_120515_0  383_120503_0  \\\n",
       "0      ENSG00000167578.11  ENSG00000167578         60.84         65.45   \n",
       "1       ENSG00000242268.1  ENSG00000242268          0.08          0.05   \n",
       "2       ENSG00000078237.4  ENSG00000078237          4.39          4.49   \n",
       "3       ENSG00000263642.1  ENSG00000263642          0.00          0.00   \n",
       "4       ENSG00000225275.4  ENSG00000225275          0.00          0.00   \n",
       "...                   ...              ...           ...           ...   \n",
       "38777   ENSG00000148943.7  ENSG00000148943         10.02          8.76   \n",
       "38778   ENSG00000265520.1  ENSG00000265520          0.00          0.00   \n",
       "38779  ENSG00000105063.14  ENSG00000105063         40.59         39.26   \n",
       "38780   ENSG00000123685.4  ENSG00000123685          4.46          4.51   \n",
       "38781   ENSG00000181518.2  ENSG00000181518          0.00          0.00   \n",
       "\n",
       "       93_120417_0  610_120523_0  560_120517_0  492_120515_0  576_120521_0  \\\n",
       "0            69.18         51.60         48.61         55.65         51.18   \n",
       "1             0.08          0.08          0.10          0.08          0.06   \n",
       "2             2.51          2.90          2.67          5.50          3.36   \n",
       "3             0.00          0.00          0.00          0.00          0.00   \n",
       "4             0.00          0.00          0.00          0.00          0.00   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "38777         6.28          8.32          8.68          8.93          9.85   \n",
       "38778         0.00          0.00          0.00          0.00          0.00   \n",
       "38779        34.70         36.74         39.85         35.28         24.78   \n",
       "38780         5.27          5.11          3.77          4.31          4.63   \n",
       "38781         0.05          0.05          0.00          0.00          0.02   \n",
       "\n",
       "       150_120419_0  ...  901_131010_8  894_130923_8  938_131101_8  \\\n",
       "0             60.63  ...         45.02         33.71         38.30   \n",
       "1              0.11  ...          0.10          0.20          0.00   \n",
       "2              3.43  ...          4.05          6.06          6.03   \n",
       "3              0.00  ...          0.00          0.00          0.00   \n",
       "4              0.00  ...          0.00          0.00          0.00   \n",
       "...             ...  ...           ...           ...           ...   \n",
       "38777         10.67  ...         11.47         14.18         13.93   \n",
       "38778          0.00  ...          0.00          0.00          0.00   \n",
       "38779         31.14  ...         24.08         24.10         24.81   \n",
       "38780          4.40  ...          5.06          2.05          4.01   \n",
       "38781          0.00  ...          0.00          0.00          0.00   \n",
       "\n",
       "       942_131101_8  939_131101_8  895_130923_8  829_130725_8  944_131107_8  \\\n",
       "0             28.27         51.02         31.87         37.14         41.77   \n",
       "1              0.00          0.00          0.00          0.00          0.09   \n",
       "2              4.34          5.16          2.70          5.00          3.82   \n",
       "3              0.00          0.00          0.00          0.00          0.00   \n",
       "4              0.00          0.00          0.00          0.00          0.00   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "38777         15.96         14.90         11.57         11.81         15.92   \n",
       "38778          0.00          0.00          0.00          0.00          0.00   \n",
       "38779         27.30         22.24         20.08         23.68         17.91   \n",
       "38780          2.89          2.69          3.42          4.25          5.46   \n",
       "38781          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       775_130528_8  Gene name  \n",
       "0             44.01      RAB4B  \n",
       "1              0.10  LINC02082  \n",
       "2              3.69      TIGAR  \n",
       "3              0.00    MIR4802  \n",
       "4              0.00   NUP210P2  \n",
       "...             ...        ...  \n",
       "38777         12.16      LIN7C  \n",
       "38778          0.00    MIR548V  \n",
       "38779         26.37     PPP6R1  \n",
       "38780          3.65      BATF3  \n",
       "38781          0.00      OR8D4  \n",
       "\n",
       "[38782 rows x 643 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now, merge the two dataframes on the 'gene_name' column\n",
    "merged_data = pd.merge(gene_expression, ensembl_data, on='gene_id', how='inner')\n",
    "merged_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tracking_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>...</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000167578.11</td>\n",
       "      <td>ENSG00000167578</td>\n",
       "      <td>60.84</td>\n",
       "      <td>65.45</td>\n",
       "      <td>69.18</td>\n",
       "      <td>51.60</td>\n",
       "      <td>48.61</td>\n",
       "      <td>55.65</td>\n",
       "      <td>51.18</td>\n",
       "      <td>60.63</td>\n",
       "      <td>...</td>\n",
       "      <td>45.02</td>\n",
       "      <td>33.71</td>\n",
       "      <td>38.30</td>\n",
       "      <td>28.27</td>\n",
       "      <td>51.02</td>\n",
       "      <td>31.87</td>\n",
       "      <td>37.14</td>\n",
       "      <td>41.77</td>\n",
       "      <td>44.01</td>\n",
       "      <td>RAB4B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000242268.1</td>\n",
       "      <td>ENSG00000242268</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.11</td>\n",
       "      <td>...</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.10</td>\n",
       "      <td>LINC02082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000078237.4</td>\n",
       "      <td>ENSG00000078237</td>\n",
       "      <td>4.39</td>\n",
       "      <td>4.49</td>\n",
       "      <td>2.51</td>\n",
       "      <td>2.90</td>\n",
       "      <td>2.67</td>\n",
       "      <td>5.50</td>\n",
       "      <td>3.36</td>\n",
       "      <td>3.43</td>\n",
       "      <td>...</td>\n",
       "      <td>4.05</td>\n",
       "      <td>6.06</td>\n",
       "      <td>6.03</td>\n",
       "      <td>4.34</td>\n",
       "      <td>5.16</td>\n",
       "      <td>2.70</td>\n",
       "      <td>5.00</td>\n",
       "      <td>3.82</td>\n",
       "      <td>3.69</td>\n",
       "      <td>TIGAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000263642.1</td>\n",
       "      <td>ENSG00000263642</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>MIR4802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000225275.4</td>\n",
       "      <td>ENSG00000225275</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NUP210P2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38777</th>\n",
       "      <td>ENSG00000148943.7</td>\n",
       "      <td>ENSG00000148943</td>\n",
       "      <td>10.02</td>\n",
       "      <td>8.76</td>\n",
       "      <td>6.28</td>\n",
       "      <td>8.32</td>\n",
       "      <td>8.68</td>\n",
       "      <td>8.93</td>\n",
       "      <td>9.85</td>\n",
       "      <td>10.67</td>\n",
       "      <td>...</td>\n",
       "      <td>11.47</td>\n",
       "      <td>14.18</td>\n",
       "      <td>13.93</td>\n",
       "      <td>15.96</td>\n",
       "      <td>14.90</td>\n",
       "      <td>11.57</td>\n",
       "      <td>11.81</td>\n",
       "      <td>15.92</td>\n",
       "      <td>12.16</td>\n",
       "      <td>LIN7C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38778</th>\n",
       "      <td>ENSG00000265520.1</td>\n",
       "      <td>ENSG00000265520</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>MIR548V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38779</th>\n",
       "      <td>ENSG00000105063.14</td>\n",
       "      <td>ENSG00000105063</td>\n",
       "      <td>40.59</td>\n",
       "      <td>39.26</td>\n",
       "      <td>34.70</td>\n",
       "      <td>36.74</td>\n",
       "      <td>39.85</td>\n",
       "      <td>35.28</td>\n",
       "      <td>24.78</td>\n",
       "      <td>31.14</td>\n",
       "      <td>...</td>\n",
       "      <td>24.08</td>\n",
       "      <td>24.10</td>\n",
       "      <td>24.81</td>\n",
       "      <td>27.30</td>\n",
       "      <td>22.24</td>\n",
       "      <td>20.08</td>\n",
       "      <td>23.68</td>\n",
       "      <td>17.91</td>\n",
       "      <td>26.37</td>\n",
       "      <td>PPP6R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38780</th>\n",
       "      <td>ENSG00000123685.4</td>\n",
       "      <td>ENSG00000123685</td>\n",
       "      <td>4.46</td>\n",
       "      <td>4.51</td>\n",
       "      <td>5.27</td>\n",
       "      <td>5.11</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.31</td>\n",
       "      <td>4.63</td>\n",
       "      <td>4.40</td>\n",
       "      <td>...</td>\n",
       "      <td>5.06</td>\n",
       "      <td>2.05</td>\n",
       "      <td>4.01</td>\n",
       "      <td>2.89</td>\n",
       "      <td>2.69</td>\n",
       "      <td>3.42</td>\n",
       "      <td>4.25</td>\n",
       "      <td>5.46</td>\n",
       "      <td>3.65</td>\n",
       "      <td>BATF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38781</th>\n",
       "      <td>ENSG00000181518.2</td>\n",
       "      <td>ENSG00000181518</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>OR8D4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38782 rows × 643 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              tracking_id          gene_id  525_120515_0  383_120503_0  \\\n",
       "0      ENSG00000167578.11  ENSG00000167578         60.84         65.45   \n",
       "1       ENSG00000242268.1  ENSG00000242268          0.08          0.05   \n",
       "2       ENSG00000078237.4  ENSG00000078237          4.39          4.49   \n",
       "3       ENSG00000263642.1  ENSG00000263642          0.00          0.00   \n",
       "4       ENSG00000225275.4  ENSG00000225275          0.00          0.00   \n",
       "...                   ...              ...           ...           ...   \n",
       "38777   ENSG00000148943.7  ENSG00000148943         10.02          8.76   \n",
       "38778   ENSG00000265520.1  ENSG00000265520          0.00          0.00   \n",
       "38779  ENSG00000105063.14  ENSG00000105063         40.59         39.26   \n",
       "38780   ENSG00000123685.4  ENSG00000123685          4.46          4.51   \n",
       "38781   ENSG00000181518.2  ENSG00000181518          0.00          0.00   \n",
       "\n",
       "       93_120417_0  610_120523_0  560_120517_0  492_120515_0  576_120521_0  \\\n",
       "0            69.18         51.60         48.61         55.65         51.18   \n",
       "1             0.08          0.08          0.10          0.08          0.06   \n",
       "2             2.51          2.90          2.67          5.50          3.36   \n",
       "3             0.00          0.00          0.00          0.00          0.00   \n",
       "4             0.00          0.00          0.00          0.00          0.00   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "38777         6.28          8.32          8.68          8.93          9.85   \n",
       "38778         0.00          0.00          0.00          0.00          0.00   \n",
       "38779        34.70         36.74         39.85         35.28         24.78   \n",
       "38780         5.27          5.11          3.77          4.31          4.63   \n",
       "38781         0.05          0.05          0.00          0.00          0.02   \n",
       "\n",
       "       150_120419_0  ...  901_131010_8  894_130923_8  938_131101_8  \\\n",
       "0             60.63  ...         45.02         33.71         38.30   \n",
       "1              0.11  ...          0.10          0.20          0.00   \n",
       "2              3.43  ...          4.05          6.06          6.03   \n",
       "3              0.00  ...          0.00          0.00          0.00   \n",
       "4              0.00  ...          0.00          0.00          0.00   \n",
       "...             ...  ...           ...           ...           ...   \n",
       "38777         10.67  ...         11.47         14.18         13.93   \n",
       "38778          0.00  ...          0.00          0.00          0.00   \n",
       "38779         31.14  ...         24.08         24.10         24.81   \n",
       "38780          4.40  ...          5.06          2.05          4.01   \n",
       "38781          0.00  ...          0.00          0.00          0.00   \n",
       "\n",
       "       942_131101_8  939_131101_8  895_130923_8  829_130725_8  944_131107_8  \\\n",
       "0             28.27         51.02         31.87         37.14         41.77   \n",
       "1              0.00          0.00          0.00          0.00          0.09   \n",
       "2              4.34          5.16          2.70          5.00          3.82   \n",
       "3              0.00          0.00          0.00          0.00          0.00   \n",
       "4              0.00          0.00          0.00          0.00          0.00   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "38777         15.96         14.90         11.57         11.81         15.92   \n",
       "38778          0.00          0.00          0.00          0.00          0.00   \n",
       "38779         27.30         22.24         20.08         23.68         17.91   \n",
       "38780          2.89          2.69          3.42          4.25          5.46   \n",
       "38781          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       775_130528_8  gene_name  \n",
       "0             44.01      RAB4B  \n",
       "1              0.10  LINC02082  \n",
       "2              3.69      TIGAR  \n",
       "3              0.00    MIR4802  \n",
       "4              0.00   NUP210P2  \n",
       "...             ...        ...  \n",
       "38777         12.16      LIN7C  \n",
       "38778          0.00    MIR548V  \n",
       "38779         26.37     PPP6R1  \n",
       "38780          3.65      BATF3  \n",
       "38781          0.00      OR8D4  \n",
       "\n",
       "[38782 rows x 643 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_data = merged_data.rename(columns={'Gene name': 'gene_name'})\n",
    "merged_data.replace('Nan', pd.NA, inplace=True)  # Replace 'Nan' string with actual NaN\n",
    "merged_data.dropna(subset=['gene_name'], inplace=True)  # Drop rows with NaN values\n",
    "merged_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>416_120503_0</th>\n",
       "      <th>...</th>\n",
       "      <th>831_130725_8</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7SK</td>\n",
       "      <td>233.11</td>\n",
       "      <td>148.79</td>\n",
       "      <td>177.12</td>\n",
       "      <td>197.38</td>\n",
       "      <td>238.15</td>\n",
       "      <td>156.24</td>\n",
       "      <td>340.67</td>\n",
       "      <td>111.16</td>\n",
       "      <td>212.84</td>\n",
       "      <td>...</td>\n",
       "      <td>63.46</td>\n",
       "      <td>70.21</td>\n",
       "      <td>89.59</td>\n",
       "      <td>85.90</td>\n",
       "      <td>140.91</td>\n",
       "      <td>132.21</td>\n",
       "      <td>125.26</td>\n",
       "      <td>67.56</td>\n",
       "      <td>111.18</td>\n",
       "      <td>89.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>2.93</td>\n",
       "      <td>2.44</td>\n",
       "      <td>3.78</td>\n",
       "      <td>2.82</td>\n",
       "      <td>3.26</td>\n",
       "      <td>3.19</td>\n",
       "      <td>3.40</td>\n",
       "      <td>3.62</td>\n",
       "      <td>2.82</td>\n",
       "      <td>...</td>\n",
       "      <td>1.23</td>\n",
       "      <td>1.86</td>\n",
       "      <td>1.66</td>\n",
       "      <td>1.59</td>\n",
       "      <td>1.32</td>\n",
       "      <td>1.59</td>\n",
       "      <td>1.03</td>\n",
       "      <td>1.95</td>\n",
       "      <td>3.23</td>\n",
       "      <td>1.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>15.37</td>\n",
       "      <td>16.15</td>\n",
       "      <td>31.83</td>\n",
       "      <td>32.41</td>\n",
       "      <td>29.42</td>\n",
       "      <td>33.33</td>\n",
       "      <td>46.22</td>\n",
       "      <td>30.92</td>\n",
       "      <td>42.70</td>\n",
       "      <td>...</td>\n",
       "      <td>26.35</td>\n",
       "      <td>33.72</td>\n",
       "      <td>46.68</td>\n",
       "      <td>34.92</td>\n",
       "      <td>32.48</td>\n",
       "      <td>26.25</td>\n",
       "      <td>62.94</td>\n",
       "      <td>25.71</td>\n",
       "      <td>21.72</td>\n",
       "      <td>34.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.45</td>\n",
       "      <td>...</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.76</td>\n",
       "      <td>0.45</td>\n",
       "      <td>1.40</td>\n",
       "      <td>0.75</td>\n",
       "      <td>1.09</td>\n",
       "      <td>1.52</td>\n",
       "      <td>0.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37893</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>10.24</td>\n",
       "      <td>7.94</td>\n",
       "      <td>6.20</td>\n",
       "      <td>6.08</td>\n",
       "      <td>5.98</td>\n",
       "      <td>17.90</td>\n",
       "      <td>5.65</td>\n",
       "      <td>10.92</td>\n",
       "      <td>7.60</td>\n",
       "      <td>...</td>\n",
       "      <td>17.64</td>\n",
       "      <td>16.46</td>\n",
       "      <td>14.33</td>\n",
       "      <td>13.96</td>\n",
       "      <td>11.23</td>\n",
       "      <td>14.36</td>\n",
       "      <td>8.10</td>\n",
       "      <td>12.67</td>\n",
       "      <td>14.02</td>\n",
       "      <td>11.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37894</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>55.86</td>\n",
       "      <td>62.33</td>\n",
       "      <td>77.69</td>\n",
       "      <td>54.95</td>\n",
       "      <td>55.05</td>\n",
       "      <td>49.34</td>\n",
       "      <td>62.83</td>\n",
       "      <td>86.67</td>\n",
       "      <td>69.39</td>\n",
       "      <td>...</td>\n",
       "      <td>42.19</td>\n",
       "      <td>50.30</td>\n",
       "      <td>47.13</td>\n",
       "      <td>36.20</td>\n",
       "      <td>32.47</td>\n",
       "      <td>41.14</td>\n",
       "      <td>33.11</td>\n",
       "      <td>55.77</td>\n",
       "      <td>35.80</td>\n",
       "      <td>44.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37895</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>5.57</td>\n",
       "      <td>6.09</td>\n",
       "      <td>5.19</td>\n",
       "      <td>5.36</td>\n",
       "      <td>5.53</td>\n",
       "      <td>8.74</td>\n",
       "      <td>9.64</td>\n",
       "      <td>9.37</td>\n",
       "      <td>6.14</td>\n",
       "      <td>...</td>\n",
       "      <td>5.91</td>\n",
       "      <td>8.76</td>\n",
       "      <td>6.18</td>\n",
       "      <td>5.91</td>\n",
       "      <td>6.18</td>\n",
       "      <td>5.03</td>\n",
       "      <td>4.97</td>\n",
       "      <td>5.45</td>\n",
       "      <td>6.13</td>\n",
       "      <td>6.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37896</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>4.73</td>\n",
       "      <td>5.13</td>\n",
       "      <td>3.41</td>\n",
       "      <td>4.87</td>\n",
       "      <td>5.21</td>\n",
       "      <td>6.66</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.12</td>\n",
       "      <td>4.76</td>\n",
       "      <td>...</td>\n",
       "      <td>8.89</td>\n",
       "      <td>6.25</td>\n",
       "      <td>8.17</td>\n",
       "      <td>8.35</td>\n",
       "      <td>8.74</td>\n",
       "      <td>7.51</td>\n",
       "      <td>6.41</td>\n",
       "      <td>6.73</td>\n",
       "      <td>9.76</td>\n",
       "      <td>8.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37897</th>\n",
       "      <td>hsa-mir-423</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37898 rows × 641 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene_name  525_120515_0  383_120503_0  93_120417_0  610_120523_0  \\\n",
       "0              7SK        233.11        148.79       177.12        197.38   \n",
       "1             A1BG          2.93          2.44         3.78          2.82   \n",
       "2             A1CF          0.00          0.00         0.00          0.00   \n",
       "3              A2M         15.37         16.15        31.83         32.41   \n",
       "4          A2M-AS1          0.59          0.79         0.71          0.63   \n",
       "...            ...           ...           ...          ...           ...   \n",
       "37893       ZYG11B         10.24          7.94         6.20          6.08   \n",
       "37894          ZYX         55.86         62.33        77.69         54.95   \n",
       "37895        ZZEF1          5.57          6.09         5.19          5.36   \n",
       "37896         ZZZ3          4.73          5.13         3.41          4.87   \n",
       "37897  hsa-mir-423          0.00          0.00         0.00          0.00   \n",
       "\n",
       "       560_120517_0  492_120515_0  576_120521_0  150_120419_0  416_120503_0  \\\n",
       "0            238.15        156.24        340.67        111.16        212.84   \n",
       "1              3.26          3.19          3.40          3.62          2.82   \n",
       "2              0.00          0.00          0.00          0.00          0.00   \n",
       "3             29.42         33.33         46.22         30.92         42.70   \n",
       "4              0.39          0.77          0.34          0.57          0.45   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "37893          5.98         17.90          5.65         10.92          7.60   \n",
       "37894         55.05         49.34         62.83         86.67         69.39   \n",
       "37895          5.53          8.74          9.64          9.37          6.14   \n",
       "37896          5.21          6.66          5.09          5.12          4.76   \n",
       "37897          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       ...  831_130725_8  901_131010_8  894_130923_8  938_131101_8  \\\n",
       "0      ...         63.46         70.21         89.59         85.90   \n",
       "1      ...          1.23          1.86          1.66          1.59   \n",
       "2      ...          0.00          0.00          0.00          0.00   \n",
       "3      ...         26.35         33.72         46.68         34.92   \n",
       "4      ...          1.18          0.95          0.57          0.76   \n",
       "...    ...           ...           ...           ...           ...   \n",
       "37893  ...         17.64         16.46         14.33         13.96   \n",
       "37894  ...         42.19         50.30         47.13         36.20   \n",
       "37895  ...          5.91          8.76          6.18          5.91   \n",
       "37896  ...          8.89          6.25          8.17          8.35   \n",
       "37897  ...          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       942_131101_8  939_131101_8  895_130923_8  829_130725_8  944_131107_8  \\\n",
       "0            140.91        132.21        125.26         67.56        111.18   \n",
       "1              1.32          1.59          1.03          1.95          3.23   \n",
       "2              0.00          0.00          0.00          0.00          0.00   \n",
       "3             32.48         26.25         62.94         25.71         21.72   \n",
       "4              0.45          1.40          0.75          1.09          1.52   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "37893         11.23         14.36          8.10         12.67         14.02   \n",
       "37894         32.47         41.14         33.11         55.77         35.80   \n",
       "37895          6.18          5.03          4.97          5.45          6.13   \n",
       "37896          8.74          7.51          6.41          6.73          9.76   \n",
       "37897          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       775_130528_8  \n",
       "0             89.51  \n",
       "1              1.98  \n",
       "2              0.00  \n",
       "3             34.44  \n",
       "4              0.92  \n",
       "...             ...  \n",
       "37893         11.08  \n",
       "37894         44.74  \n",
       "37895          6.53  \n",
       "37896          8.06  \n",
       "37897          0.00  \n",
       "\n",
       "[37898 rows x 641 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "cols = ['gene_name'] + [col for col in merged_data if col != 'gene_name']\n",
    "gene_expression_new = merged_data[cols]\n",
    "numeric_cols = gene_expression_new.select_dtypes(include=[np.number]).columns.tolist()\n",
    "\n",
    "# Now, we group by 'gene_name' and calculate the mean of all numeric columns\n",
    "gene_expression_new = gene_expression_new.groupby('gene_name')[numeric_cols].mean()\n",
    "\n",
    "# Reset the index to turn the grouped keys (gene_name) back into a column\n",
    "gene_expression_new.reset_index(inplace=True)\n",
    "gene_expression_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import pandas as pd\n",
    "# from pyensembl import EnsemblRelease\n",
    "\n",
    "# # Function to fetch the requested information for a gene ID\n",
    "# def fetch_gene_info(gene_id_with_version, ensembl_data, attributes):\n",
    "#     # Split the gene_id to exclude the version number\n",
    "#     gene_id = gene_id_with_version.split('.')[0]\n",
    "#     info = {}\n",
    "#     try:\n",
    "#         gene = ensembl_data.gene_by_id(gene_id)\n",
    "#         for attribute in attributes:\n",
    "#             value = None\n",
    "#             if hasattr(gene, attribute):\n",
    "#                 value = getattr(gene, attribute)\n",
    "#             elif attribute == 'transcript_names':\n",
    "#                 value = ','.join([t.name for t in gene.transcripts])\n",
    "#             elif attribute == 'transcript_ids':\n",
    "#                 value = ','.join([t.transcript_id for t in gene.transcripts])\n",
    "#             elif attribute == 'exon_ids':\n",
    "#                 exons = [exon.exon_id for transcript in gene.transcripts for exon in transcript.exons]\n",
    "#                 value = ','.join(exons)\n",
    "#             info[attribute] = value\n",
    "#     except ValueError:\n",
    "#         # If gene ID is not found, fill all attributes with \"ValueError\"\n",
    "#         for attribute in attributes:\n",
    "#             info[attribute] = \"ValueError\"\n",
    "#     return info\n",
    "\n",
    "# # Main function to update the CSV file with the requested gene information\n",
    "# def update_csv_with_gene_attributes(csv_path, attributes):\n",
    "#     # Load the CSV file\n",
    "#     data = pd.read_csv(csv_path, sep='\\t')\n",
    "\n",
    "#     # Initialize the Ensembl data\n",
    "#     ensembl_data = EnsemblRelease()\n",
    "#     ensembl_data.download()\n",
    "#     ensembl_data.index()\n",
    "\n",
    "#     # Fetch and add the requested gene-related information\n",
    "#     for attribute in attributes:\n",
    "#         # Apply the fetch_gene_info function for each gene_id and extract the attribute\n",
    "#         data[attribute] = data['gene_id'].apply(lambda gene_id: fetch_gene_info(gene_id, ensembl_data, attributes)[attribute])\n",
    "\n",
    "#     # Save the updated CSV file\n",
    "#     updated_csv_path = csv_path.replace('.tsv', '_with_attributes.tsv')\n",
    "#     data.to_csv(updated_csv_path, sep='\\t', index=False)\n",
    "#     return updated_csv_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # input the file path and additional attributes you want:\n",
    "# csv_file_path = './ROSMAP-raw/RNASeq/ROSMAP_RNAseq_FPKM_gene.tsv' \n",
    "# attributes_to_add = ['gene_name'] \n",
    "\n",
    "# # Call the function with the CSV path and the list of attributes\n",
    "# updated_file_path = update_csv_with_gene_attributes(csv_file_path, attributes_to_add)\n",
    "# print(f\"Updated CSV file is saved at {updated_file_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Read the new gene_expression with gene_name\n",
    "# gene_expression = pd.read_csv('./ROSMAP-raw/RNASeq/ROSMAP_RNAseq_FPKM_gene_with_attributes.tsv', sep='\\t')\n",
    "# gene_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Check if any gene names have not been found\n",
    "# count = gene_expression[gene_expression['gene_name'] == 'Gene name not found'].shape[0]\n",
    "# count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # set gene to the first column\n",
    "# cols = ['gene_name'] + [col for col in gene_expression if col != 'gene_name']\n",
    "# gene_expression_new = gene_expression[cols]\n",
    "# gene_expression_new.drop(columns=['tracking_id', 'gene_id'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gene_expression_new = gene_expression_new.dropna(subset=['gene_name'])\n",
    "# gene_expression_new = gene_expression_new.groupby('gene_name', as_index=False).mean()\n",
    "# gene_expression_new = gene_expression_new.reset_index(drop=True)\n",
    "# gene_expression_new"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.4 Read clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_first_ad_dx</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_first_ad_dx</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "      <th>individualID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10101589</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R6939144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>86767530</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R3893503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9650662</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.850000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R8937093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>50402855</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R7139444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20544321</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R4971237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3579</th>\n",
       "      <td>22207815</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>57.653661875427787</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R5306025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3580</th>\n",
       "      <td>22207941</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>56.651608487337441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R6142763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3581</th>\n",
       "      <td>49333806</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.599589322381931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R4468842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3582</th>\n",
       "      <td>59720188</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.622861054072551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R9446033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3583</th>\n",
       "      <td>90214403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R5615105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3584 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0     10101589   ROS   1.0  20.0   1.0      2.0           34.0   \n",
       "1     86767530   MAP   0.0  10.0   1.0      2.0           33.0   \n",
       "2      9650662   MAP   0.0  15.0   1.0      2.0           23.0   \n",
       "3     50402855   MAP   0.0  21.0   1.0      2.0           33.0   \n",
       "4     20544321   ROS   0.0  16.0   1.0      2.0           23.0   \n",
       "...        ...   ...   ...   ...   ...      ...            ...   \n",
       "3579  22207815   ROS   0.0  18.0   2.0      2.0           23.0   \n",
       "3580  22207941   ROS   0.0  16.0   2.0      2.0           34.0   \n",
       "3581  49333806   MAP   0.0  12.0   2.0      2.0            NaN   \n",
       "3582  59720188   MAP   0.0  13.0   1.0      1.0            NaN   \n",
       "3583  90214403   NaN   NaN   NaN   NaN      NaN            NaN   \n",
       "\n",
       "        age_at_visit_max age_first_ad_dx age_death  cts_mmse30_first_ad_dx  \\\n",
       "0                    90+             90+       90+                    18.0   \n",
       "1                    90+             90+       90+                    18.0   \n",
       "2                    90+             90+       90+                     0.0   \n",
       "3                    90+             NaN       NaN                     NaN   \n",
       "4                    90+             90+       NaN                    13.0   \n",
       "...                  ...             ...       ...                     ...   \n",
       "3579  57.653661875427787             NaN       NaN                     NaN   \n",
       "3580  56.651608487337441             NaN       NaN                     NaN   \n",
       "3581  56.599589322381931             NaN       NaN                     NaN   \n",
       "3582  54.622861054072551             NaN       NaN                     NaN   \n",
       "3583                 NaN             NaN       NaN                     NaN   \n",
       "\n",
       "      cts_mmse30_lv       pmi  braaksc  ceradsc  cogdx  dcfdx_lv individualID  \n",
       "0               5.0  9.916667      4.0      2.0    4.0       4.0     R6939144  \n",
       "1              10.0  6.500000      4.0      2.0    4.0       4.0     R3893503  \n",
       "2               0.0  3.850000      3.0      2.0    4.0       4.0     R8937093  \n",
       "3              27.0       NaN      NaN      NaN    NaN       1.0     R7139444  \n",
       "4              14.0       NaN      NaN      NaN    NaN       4.0     R4971237  \n",
       "...             ...       ...      ...      ...    ...       ...          ...  \n",
       "3579           29.0       NaN      NaN      NaN    NaN       1.0     R5306025  \n",
       "3580           27.0       NaN      NaN      NaN    NaN       1.0     R6142763  \n",
       "3581           30.0       NaN      NaN      NaN    NaN       1.0     R4468842  \n",
       "3582           29.0       NaN      NaN      NaN    NaN       1.0     R9446033  \n",
       "3583            NaN       NaN      NaN      NaN    NaN       NaN     R5615105  \n",
       "\n",
       "[3584 rows x 18 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "survival = pd.read_csv('./ROSMAP-raw/ROSMAP_clinical/ROSMAP_clinical.csv', sep=',')\n",
    "survival"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.5 Read proteomics data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>b01.127C</th>\n",
       "      <th>b01.127N</th>\n",
       "      <th>b01.128C</th>\n",
       "      <th>b01.128N</th>\n",
       "      <th>b01.129C</th>\n",
       "      <th>b01.129N</th>\n",
       "      <th>b01.130C</th>\n",
       "      <th>b01.130N</th>\n",
       "      <th>b02.127C</th>\n",
       "      <th>...</th>\n",
       "      <th>b49.130C</th>\n",
       "      <th>b49.130N</th>\n",
       "      <th>b50.127C</th>\n",
       "      <th>b50.127N</th>\n",
       "      <th>b50.128C</th>\n",
       "      <th>b50.128N</th>\n",
       "      <th>b50.129C</th>\n",
       "      <th>b50.129N</th>\n",
       "      <th>b50.130C</th>\n",
       "      <th>b50.130N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VAMP1|P23763</td>\n",
       "      <td>-0.278255</td>\n",
       "      <td>0.410722</td>\n",
       "      <td>-0.112329</td>\n",
       "      <td>-0.162641</td>\n",
       "      <td>-0.071647</td>\n",
       "      <td>0.117440</td>\n",
       "      <td>-0.042157</td>\n",
       "      <td>0.187085</td>\n",
       "      <td>-0.835022</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.073641</td>\n",
       "      <td>0.081353</td>\n",
       "      <td>0.084443</td>\n",
       "      <td>-0.220764</td>\n",
       "      <td>0.231834</td>\n",
       "      <td>-0.076819</td>\n",
       "      <td>-0.130791</td>\n",
       "      <td>-0.173961</td>\n",
       "      <td>-0.043709</td>\n",
       "      <td>-0.054343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KCTD13|Q8WZ19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.225794</td>\n",
       "      <td>0.019362</td>\n",
       "      <td>0.393129</td>\n",
       "      <td>0.238450</td>\n",
       "      <td>-0.039415</td>\n",
       "      <td>0.313539</td>\n",
       "      <td>0.335718</td>\n",
       "      <td>0.197204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TXNDC12|O95881</td>\n",
       "      <td>-0.366508</td>\n",
       "      <td>-0.498610</td>\n",
       "      <td>-0.276421</td>\n",
       "      <td>-0.354865</td>\n",
       "      <td>-0.238212</td>\n",
       "      <td>-0.321546</td>\n",
       "      <td>-0.213956</td>\n",
       "      <td>-0.532379</td>\n",
       "      <td>-0.742539</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.521676</td>\n",
       "      <td>-0.289618</td>\n",
       "      <td>-0.295169</td>\n",
       "      <td>-0.506661</td>\n",
       "      <td>-0.268946</td>\n",
       "      <td>-0.403568</td>\n",
       "      <td>-0.320117</td>\n",
       "      <td>-0.466931</td>\n",
       "      <td>-0.289946</td>\n",
       "      <td>-0.373653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDHX|O00330</td>\n",
       "      <td>-0.144993</td>\n",
       "      <td>0.155336</td>\n",
       "      <td>0.050900</td>\n",
       "      <td>0.079455</td>\n",
       "      <td>-0.021996</td>\n",
       "      <td>-0.071422</td>\n",
       "      <td>0.081380</td>\n",
       "      <td>0.089962</td>\n",
       "      <td>0.058278</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.132236</td>\n",
       "      <td>-0.095648</td>\n",
       "      <td>-0.145319</td>\n",
       "      <td>0.016944</td>\n",
       "      <td>0.065491</td>\n",
       "      <td>0.135842</td>\n",
       "      <td>0.147856</td>\n",
       "      <td>-0.020226</td>\n",
       "      <td>0.083475</td>\n",
       "      <td>0.047506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>APIP|Q96GX9</td>\n",
       "      <td>0.145339</td>\n",
       "      <td>-0.417854</td>\n",
       "      <td>-0.843172</td>\n",
       "      <td>0.577196</td>\n",
       "      <td>-1.468350</td>\n",
       "      <td>0.221551</td>\n",
       "      <td>-0.136442</td>\n",
       "      <td>-0.323159</td>\n",
       "      <td>0.976332</td>\n",
       "      <td>...</td>\n",
       "      <td>0.713746</td>\n",
       "      <td>-0.831854</td>\n",
       "      <td>0.236361</td>\n",
       "      <td>-0.814508</td>\n",
       "      <td>-0.009136</td>\n",
       "      <td>-1.053596</td>\n",
       "      <td>-0.009197</td>\n",
       "      <td>-1.010743</td>\n",
       "      <td>-0.467762</td>\n",
       "      <td>0.403568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8812</th>\n",
       "      <td>S1PR5|Q9H228</td>\n",
       "      <td>0.505338</td>\n",
       "      <td>0.391576</td>\n",
       "      <td>-0.219065</td>\n",
       "      <td>-0.239153</td>\n",
       "      <td>0.426485</td>\n",
       "      <td>-0.055620</td>\n",
       "      <td>-0.359028</td>\n",
       "      <td>0.013233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.560319</td>\n",
       "      <td>0.031164</td>\n",
       "      <td>-0.026439</td>\n",
       "      <td>0.217259</td>\n",
       "      <td>0.358087</td>\n",
       "      <td>-0.165559</td>\n",
       "      <td>-0.033307</td>\n",
       "      <td>-0.419110</td>\n",
       "      <td>-0.010325</td>\n",
       "      <td>-0.054859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8813</th>\n",
       "      <td>GIGYF2|Q6Y7W6</td>\n",
       "      <td>-0.449736</td>\n",
       "      <td>-0.411664</td>\n",
       "      <td>-0.432484</td>\n",
       "      <td>-0.488819</td>\n",
       "      <td>-0.458550</td>\n",
       "      <td>-0.421006</td>\n",
       "      <td>-0.526388</td>\n",
       "      <td>-0.527799</td>\n",
       "      <td>-0.554216</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.482969</td>\n",
       "      <td>-0.416386</td>\n",
       "      <td>-0.507107</td>\n",
       "      <td>-0.374713</td>\n",
       "      <td>-0.455986</td>\n",
       "      <td>-0.407432</td>\n",
       "      <td>-0.529108</td>\n",
       "      <td>-0.483092</td>\n",
       "      <td>-0.439813</td>\n",
       "      <td>-0.469659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8814</th>\n",
       "      <td>YME1L1|Q96TA2</td>\n",
       "      <td>0.462210</td>\n",
       "      <td>0.035486</td>\n",
       "      <td>0.035512</td>\n",
       "      <td>-0.117947</td>\n",
       "      <td>0.494874</td>\n",
       "      <td>-0.018295</td>\n",
       "      <td>-0.081191</td>\n",
       "      <td>-0.002421</td>\n",
       "      <td>-0.047577</td>\n",
       "      <td>...</td>\n",
       "      <td>0.277916</td>\n",
       "      <td>-0.021786</td>\n",
       "      <td>0.104153</td>\n",
       "      <td>0.092986</td>\n",
       "      <td>0.206298</td>\n",
       "      <td>-0.137310</td>\n",
       "      <td>-0.037395</td>\n",
       "      <td>-0.114052</td>\n",
       "      <td>0.053101</td>\n",
       "      <td>-0.077162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8815</th>\n",
       "      <td>PPP1R3F|Q6ZSY5</td>\n",
       "      <td>-0.177028</td>\n",
       "      <td>-0.080174</td>\n",
       "      <td>-0.145297</td>\n",
       "      <td>0.049689</td>\n",
       "      <td>-0.064137</td>\n",
       "      <td>-2.274703</td>\n",
       "      <td>0.017258</td>\n",
       "      <td>-0.120344</td>\n",
       "      <td>0.063776</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.264418</td>\n",
       "      <td>-0.035055</td>\n",
       "      <td>-0.124822</td>\n",
       "      <td>-0.035162</td>\n",
       "      <td>-0.159635</td>\n",
       "      <td>-0.145910</td>\n",
       "      <td>-0.093056</td>\n",
       "      <td>-0.001350</td>\n",
       "      <td>-0.129813</td>\n",
       "      <td>-0.086022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8816</th>\n",
       "      <td>CACNG4|Q9UBN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.075949</td>\n",
       "      <td>0.111414</td>\n",
       "      <td>0.132279</td>\n",
       "      <td>-0.117680</td>\n",
       "      <td>0.437808</td>\n",
       "      <td>0.240843</td>\n",
       "      <td>0.034081</td>\n",
       "      <td>0.051218</td>\n",
       "      <td>0.090312</td>\n",
       "      <td>-0.070310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8817 rows × 401 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Unnamed: 0  b01.127C  b01.127N  b01.128C  b01.128N  b01.129C  \\\n",
       "0       VAMP1|P23763 -0.278255  0.410722 -0.112329 -0.162641 -0.071647   \n",
       "1      KCTD13|Q8WZ19       NaN       NaN       NaN       NaN       NaN   \n",
       "2     TXNDC12|O95881 -0.366508 -0.498610 -0.276421 -0.354865 -0.238212   \n",
       "3        PDHX|O00330 -0.144993  0.155336  0.050900  0.079455 -0.021996   \n",
       "4        APIP|Q96GX9  0.145339 -0.417854 -0.843172  0.577196 -1.468350   \n",
       "...              ...       ...       ...       ...       ...       ...   \n",
       "8812    S1PR5|Q9H228  0.505338  0.391576 -0.219065 -0.239153  0.426485   \n",
       "8813   GIGYF2|Q6Y7W6 -0.449736 -0.411664 -0.432484 -0.488819 -0.458550   \n",
       "8814   YME1L1|Q96TA2  0.462210  0.035486  0.035512 -0.117947  0.494874   \n",
       "8815  PPP1R3F|Q6ZSY5 -0.177028 -0.080174 -0.145297  0.049689 -0.064137   \n",
       "8816   CACNG4|Q9UBN1       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "      b01.129N  b01.130C  b01.130N  b02.127C  ...  b49.130C  b49.130N  \\\n",
       "0     0.117440 -0.042157  0.187085 -0.835022  ... -0.073641  0.081353   \n",
       "1          NaN       NaN       NaN       NaN  ...       NaN       NaN   \n",
       "2    -0.321546 -0.213956 -0.532379 -0.742539  ... -0.521676 -0.289618   \n",
       "3    -0.071422  0.081380  0.089962  0.058278  ... -0.132236 -0.095648   \n",
       "4     0.221551 -0.136442 -0.323159  0.976332  ...  0.713746 -0.831854   \n",
       "...        ...       ...       ...       ...  ...       ...       ...   \n",
       "8812 -0.055620 -0.359028  0.013233       NaN  ...  0.560319  0.031164   \n",
       "8813 -0.421006 -0.526388 -0.527799 -0.554216  ... -0.482969 -0.416386   \n",
       "8814 -0.018295 -0.081191 -0.002421 -0.047577  ...  0.277916 -0.021786   \n",
       "8815 -2.274703  0.017258 -0.120344  0.063776  ... -0.264418 -0.035055   \n",
       "8816       NaN       NaN       NaN       NaN  ... -0.075949  0.111414   \n",
       "\n",
       "      b50.127C  b50.127N  b50.128C  b50.128N  b50.129C  b50.129N  b50.130C  \\\n",
       "0     0.084443 -0.220764  0.231834 -0.076819 -0.130791 -0.173961 -0.043709   \n",
       "1     0.225794  0.019362  0.393129  0.238450 -0.039415  0.313539  0.335718   \n",
       "2    -0.295169 -0.506661 -0.268946 -0.403568 -0.320117 -0.466931 -0.289946   \n",
       "3    -0.145319  0.016944  0.065491  0.135842  0.147856 -0.020226  0.083475   \n",
       "4     0.236361 -0.814508 -0.009136 -1.053596 -0.009197 -1.010743 -0.467762   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "8812 -0.026439  0.217259  0.358087 -0.165559 -0.033307 -0.419110 -0.010325   \n",
       "8813 -0.507107 -0.374713 -0.455986 -0.407432 -0.529108 -0.483092 -0.439813   \n",
       "8814  0.104153  0.092986  0.206298 -0.137310 -0.037395 -0.114052  0.053101   \n",
       "8815 -0.124822 -0.035162 -0.159635 -0.145910 -0.093056 -0.001350 -0.129813   \n",
       "8816  0.132279 -0.117680  0.437808  0.240843  0.034081  0.051218  0.090312   \n",
       "\n",
       "      b50.130N  \n",
       "0    -0.054343  \n",
       "1     0.197204  \n",
       "2    -0.373653  \n",
       "3     0.047506  \n",
       "4     0.403568  \n",
       "...        ...  \n",
       "8812 -0.054859  \n",
       "8813 -0.469659  \n",
       "8814 -0.077162  \n",
       "8815 -0.086022  \n",
       "8816 -0.070310  \n",
       "\n",
       "[8817 rows x 401 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein = pd.read_csv(\"./ROSMAP-raw/Proteomics/C2.median_polish_corrected_log2(abundanceRatioCenteredOnMedianOfBatchMediansPerProtein)-8817x400.csv\",sep=',')\n",
    "protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>b01.127C</th>\n",
       "      <th>b01.127N</th>\n",
       "      <th>b01.128C</th>\n",
       "      <th>b01.128N</th>\n",
       "      <th>b01.129C</th>\n",
       "      <th>b01.129N</th>\n",
       "      <th>b01.130C</th>\n",
       "      <th>b01.130N</th>\n",
       "      <th>b02.127C</th>\n",
       "      <th>...</th>\n",
       "      <th>b49.130C</th>\n",
       "      <th>b49.130N</th>\n",
       "      <th>b50.127C</th>\n",
       "      <th>b50.127N</th>\n",
       "      <th>b50.128C</th>\n",
       "      <th>b50.128N</th>\n",
       "      <th>b50.129C</th>\n",
       "      <th>b50.129N</th>\n",
       "      <th>b50.130C</th>\n",
       "      <th>b50.130N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VAMP1</td>\n",
       "      <td>-0.278255</td>\n",
       "      <td>0.410722</td>\n",
       "      <td>-0.112329</td>\n",
       "      <td>-0.162641</td>\n",
       "      <td>-0.071647</td>\n",
       "      <td>0.117440</td>\n",
       "      <td>-0.042157</td>\n",
       "      <td>0.187085</td>\n",
       "      <td>-0.835022</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.073641</td>\n",
       "      <td>0.081353</td>\n",
       "      <td>0.084443</td>\n",
       "      <td>-0.220764</td>\n",
       "      <td>0.231834</td>\n",
       "      <td>-0.076819</td>\n",
       "      <td>-0.130791</td>\n",
       "      <td>-0.173961</td>\n",
       "      <td>-0.043709</td>\n",
       "      <td>-0.054343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KCTD13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.225794</td>\n",
       "      <td>0.019362</td>\n",
       "      <td>0.393129</td>\n",
       "      <td>0.238450</td>\n",
       "      <td>-0.039415</td>\n",
       "      <td>0.313539</td>\n",
       "      <td>0.335718</td>\n",
       "      <td>0.197204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TXNDC12</td>\n",
       "      <td>-0.366508</td>\n",
       "      <td>-0.498610</td>\n",
       "      <td>-0.276421</td>\n",
       "      <td>-0.354865</td>\n",
       "      <td>-0.238212</td>\n",
       "      <td>-0.321546</td>\n",
       "      <td>-0.213956</td>\n",
       "      <td>-0.532379</td>\n",
       "      <td>-0.742539</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.521676</td>\n",
       "      <td>-0.289618</td>\n",
       "      <td>-0.295169</td>\n",
       "      <td>-0.506661</td>\n",
       "      <td>-0.268946</td>\n",
       "      <td>-0.403568</td>\n",
       "      <td>-0.320117</td>\n",
       "      <td>-0.466931</td>\n",
       "      <td>-0.289946</td>\n",
       "      <td>-0.373653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDHX</td>\n",
       "      <td>-0.144993</td>\n",
       "      <td>0.155336</td>\n",
       "      <td>0.050900</td>\n",
       "      <td>0.079455</td>\n",
       "      <td>-0.021996</td>\n",
       "      <td>-0.071422</td>\n",
       "      <td>0.081380</td>\n",
       "      <td>0.089962</td>\n",
       "      <td>0.058278</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.132236</td>\n",
       "      <td>-0.095648</td>\n",
       "      <td>-0.145319</td>\n",
       "      <td>0.016944</td>\n",
       "      <td>0.065491</td>\n",
       "      <td>0.135842</td>\n",
       "      <td>0.147856</td>\n",
       "      <td>-0.020226</td>\n",
       "      <td>0.083475</td>\n",
       "      <td>0.047506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>APIP</td>\n",
       "      <td>0.145339</td>\n",
       "      <td>-0.417854</td>\n",
       "      <td>-0.843172</td>\n",
       "      <td>0.577196</td>\n",
       "      <td>-1.468350</td>\n",
       "      <td>0.221551</td>\n",
       "      <td>-0.136442</td>\n",
       "      <td>-0.323159</td>\n",
       "      <td>0.976332</td>\n",
       "      <td>...</td>\n",
       "      <td>0.713746</td>\n",
       "      <td>-0.831854</td>\n",
       "      <td>0.236361</td>\n",
       "      <td>-0.814508</td>\n",
       "      <td>-0.009136</td>\n",
       "      <td>-1.053596</td>\n",
       "      <td>-0.009197</td>\n",
       "      <td>-1.010743</td>\n",
       "      <td>-0.467762</td>\n",
       "      <td>0.403568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8812</th>\n",
       "      <td>S1PR5</td>\n",
       "      <td>0.505338</td>\n",
       "      <td>0.391576</td>\n",
       "      <td>-0.219065</td>\n",
       "      <td>-0.239153</td>\n",
       "      <td>0.426485</td>\n",
       "      <td>-0.055620</td>\n",
       "      <td>-0.359028</td>\n",
       "      <td>0.013233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.560319</td>\n",
       "      <td>0.031164</td>\n",
       "      <td>-0.026439</td>\n",
       "      <td>0.217259</td>\n",
       "      <td>0.358087</td>\n",
       "      <td>-0.165559</td>\n",
       "      <td>-0.033307</td>\n",
       "      <td>-0.419110</td>\n",
       "      <td>-0.010325</td>\n",
       "      <td>-0.054859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8813</th>\n",
       "      <td>GIGYF2</td>\n",
       "      <td>-0.449736</td>\n",
       "      <td>-0.411664</td>\n",
       "      <td>-0.432484</td>\n",
       "      <td>-0.488819</td>\n",
       "      <td>-0.458550</td>\n",
       "      <td>-0.421006</td>\n",
       "      <td>-0.526388</td>\n",
       "      <td>-0.527799</td>\n",
       "      <td>-0.554216</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.482969</td>\n",
       "      <td>-0.416386</td>\n",
       "      <td>-0.507107</td>\n",
       "      <td>-0.374713</td>\n",
       "      <td>-0.455986</td>\n",
       "      <td>-0.407432</td>\n",
       "      <td>-0.529108</td>\n",
       "      <td>-0.483092</td>\n",
       "      <td>-0.439813</td>\n",
       "      <td>-0.469659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8814</th>\n",
       "      <td>YME1L1</td>\n",
       "      <td>0.462210</td>\n",
       "      <td>0.035486</td>\n",
       "      <td>0.035512</td>\n",
       "      <td>-0.117947</td>\n",
       "      <td>0.494874</td>\n",
       "      <td>-0.018295</td>\n",
       "      <td>-0.081191</td>\n",
       "      <td>-0.002421</td>\n",
       "      <td>-0.047577</td>\n",
       "      <td>...</td>\n",
       "      <td>0.277916</td>\n",
       "      <td>-0.021786</td>\n",
       "      <td>0.104153</td>\n",
       "      <td>0.092986</td>\n",
       "      <td>0.206298</td>\n",
       "      <td>-0.137310</td>\n",
       "      <td>-0.037395</td>\n",
       "      <td>-0.114052</td>\n",
       "      <td>0.053101</td>\n",
       "      <td>-0.077162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8815</th>\n",
       "      <td>PPP1R3F</td>\n",
       "      <td>-0.177028</td>\n",
       "      <td>-0.080174</td>\n",
       "      <td>-0.145297</td>\n",
       "      <td>0.049689</td>\n",
       "      <td>-0.064137</td>\n",
       "      <td>-2.274703</td>\n",
       "      <td>0.017258</td>\n",
       "      <td>-0.120344</td>\n",
       "      <td>0.063776</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.264418</td>\n",
       "      <td>-0.035055</td>\n",
       "      <td>-0.124822</td>\n",
       "      <td>-0.035162</td>\n",
       "      <td>-0.159635</td>\n",
       "      <td>-0.145910</td>\n",
       "      <td>-0.093056</td>\n",
       "      <td>-0.001350</td>\n",
       "      <td>-0.129813</td>\n",
       "      <td>-0.086022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8816</th>\n",
       "      <td>CACNG4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.075949</td>\n",
       "      <td>0.111414</td>\n",
       "      <td>0.132279</td>\n",
       "      <td>-0.117680</td>\n",
       "      <td>0.437808</td>\n",
       "      <td>0.240843</td>\n",
       "      <td>0.034081</td>\n",
       "      <td>0.051218</td>\n",
       "      <td>0.090312</td>\n",
       "      <td>-0.070310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8817 rows × 401 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  b01.127C  b01.127N  b01.128C  b01.128N  b01.129C  b01.129N  \\\n",
       "0        VAMP1 -0.278255  0.410722 -0.112329 -0.162641 -0.071647  0.117440   \n",
       "1       KCTD13       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "2      TXNDC12 -0.366508 -0.498610 -0.276421 -0.354865 -0.238212 -0.321546   \n",
       "3         PDHX -0.144993  0.155336  0.050900  0.079455 -0.021996 -0.071422   \n",
       "4         APIP  0.145339 -0.417854 -0.843172  0.577196 -1.468350  0.221551   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "8812     S1PR5  0.505338  0.391576 -0.219065 -0.239153  0.426485 -0.055620   \n",
       "8813    GIGYF2 -0.449736 -0.411664 -0.432484 -0.488819 -0.458550 -0.421006   \n",
       "8814    YME1L1  0.462210  0.035486  0.035512 -0.117947  0.494874 -0.018295   \n",
       "8815   PPP1R3F -0.177028 -0.080174 -0.145297  0.049689 -0.064137 -2.274703   \n",
       "8816    CACNG4       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "      b01.130C  b01.130N  b02.127C  ...  b49.130C  b49.130N  b50.127C  \\\n",
       "0    -0.042157  0.187085 -0.835022  ... -0.073641  0.081353  0.084443   \n",
       "1          NaN       NaN       NaN  ...       NaN       NaN  0.225794   \n",
       "2    -0.213956 -0.532379 -0.742539  ... -0.521676 -0.289618 -0.295169   \n",
       "3     0.081380  0.089962  0.058278  ... -0.132236 -0.095648 -0.145319   \n",
       "4    -0.136442 -0.323159  0.976332  ...  0.713746 -0.831854  0.236361   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "8812 -0.359028  0.013233       NaN  ...  0.560319  0.031164 -0.026439   \n",
       "8813 -0.526388 -0.527799 -0.554216  ... -0.482969 -0.416386 -0.507107   \n",
       "8814 -0.081191 -0.002421 -0.047577  ...  0.277916 -0.021786  0.104153   \n",
       "8815  0.017258 -0.120344  0.063776  ... -0.264418 -0.035055 -0.124822   \n",
       "8816       NaN       NaN       NaN  ... -0.075949  0.111414  0.132279   \n",
       "\n",
       "      b50.127N  b50.128C  b50.128N  b50.129C  b50.129N  b50.130C  b50.130N  \n",
       "0    -0.220764  0.231834 -0.076819 -0.130791 -0.173961 -0.043709 -0.054343  \n",
       "1     0.019362  0.393129  0.238450 -0.039415  0.313539  0.335718  0.197204  \n",
       "2    -0.506661 -0.268946 -0.403568 -0.320117 -0.466931 -0.289946 -0.373653  \n",
       "3     0.016944  0.065491  0.135842  0.147856 -0.020226  0.083475  0.047506  \n",
       "4    -0.814508 -0.009136 -1.053596 -0.009197 -1.010743 -0.467762  0.403568  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "8812  0.217259  0.358087 -0.165559 -0.033307 -0.419110 -0.010325 -0.054859  \n",
       "8813 -0.374713 -0.455986 -0.407432 -0.529108 -0.483092 -0.439813 -0.469659  \n",
       "8814  0.092986  0.206298 -0.137310 -0.037395 -0.114052  0.053101 -0.077162  \n",
       "8815 -0.035162 -0.159635 -0.145910 -0.093056 -0.001350 -0.129813 -0.086022  \n",
       "8816 -0.117680  0.437808  0.240843  0.034081  0.051218  0.090312 -0.070310  \n",
       "\n",
       "[8817 rows x 401 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein.iloc[:, 0] = protein.iloc[:, 0].str.split('|').str[0]\n",
    "protein.columns = ['gene_name'] + protein.columns[1:].tolist()\n",
    "protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0       0.000000\n",
      "1       0.458853\n",
      "2       0.000000\n",
      "3       0.000000\n",
      "4       0.019950\n",
      "          ...   \n",
      "8812    0.159601\n",
      "8813    0.000000\n",
      "8814    0.000000\n",
      "8815    0.000000\n",
      "8816    0.438903\n",
      "Length: 8817, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "#calculate the NaN proportion of each row\n",
    "nan_proportions = protein.isna().mean(axis=1)\n",
    "# Display the results\n",
    "print(nan_proportions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>b01.127C</th>\n",
       "      <th>b01.127N</th>\n",
       "      <th>b01.128C</th>\n",
       "      <th>b01.128N</th>\n",
       "      <th>b01.129C</th>\n",
       "      <th>b01.129N</th>\n",
       "      <th>b01.130C</th>\n",
       "      <th>b01.130N</th>\n",
       "      <th>b02.127C</th>\n",
       "      <th>...</th>\n",
       "      <th>b49.130C</th>\n",
       "      <th>b49.130N</th>\n",
       "      <th>b50.127C</th>\n",
       "      <th>b50.127N</th>\n",
       "      <th>b50.128C</th>\n",
       "      <th>b50.128N</th>\n",
       "      <th>b50.129C</th>\n",
       "      <th>b50.129N</th>\n",
       "      <th>b50.130C</th>\n",
       "      <th>b50.130N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0.056825</td>\n",
       "      <td>-0.191644</td>\n",
       "      <td>0.329755</td>\n",
       "      <td>-0.105563</td>\n",
       "      <td>0.065585</td>\n",
       "      <td>-0.003961</td>\n",
       "      <td>0.068675</td>\n",
       "      <td>0.624928</td>\n",
       "      <td>0.488592</td>\n",
       "      <td>...</td>\n",
       "      <td>0.208725</td>\n",
       "      <td>-0.058407</td>\n",
       "      <td>-0.238951</td>\n",
       "      <td>-0.082743</td>\n",
       "      <td>0.131467</td>\n",
       "      <td>-0.177005</td>\n",
       "      <td>0.027383</td>\n",
       "      <td>-0.132419</td>\n",
       "      <td>-0.119505</td>\n",
       "      <td>0.101175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.078637</td>\n",
       "      <td>-0.300053</td>\n",
       "      <td>0.241096</td>\n",
       "      <td>-0.696597</td>\n",
       "      <td>0.089856</td>\n",
       "      <td>-0.891969</td>\n",
       "      <td>-0.121492</td>\n",
       "      <td>-0.674060</td>\n",
       "      <td>-0.360159</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018837</td>\n",
       "      <td>-0.157993</td>\n",
       "      <td>-0.359336</td>\n",
       "      <td>-0.013279</td>\n",
       "      <td>-0.265634</td>\n",
       "      <td>-0.247393</td>\n",
       "      <td>-0.237626</td>\n",
       "      <td>-0.201277</td>\n",
       "      <td>0.146902</td>\n",
       "      <td>-0.027162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.024694</td>\n",
       "      <td>-0.636475</td>\n",
       "      <td>-0.028690</td>\n",
       "      <td>-0.582275</td>\n",
       "      <td>-0.084713</td>\n",
       "      <td>-0.661039</td>\n",
       "      <td>-0.127533</td>\n",
       "      <td>-0.136789</td>\n",
       "      <td>0.071652</td>\n",
       "      <td>...</td>\n",
       "      <td>0.199730</td>\n",
       "      <td>-0.142154</td>\n",
       "      <td>-0.202204</td>\n",
       "      <td>0.249200</td>\n",
       "      <td>-0.125857</td>\n",
       "      <td>-0.223538</td>\n",
       "      <td>-0.155823</td>\n",
       "      <td>-0.227838</td>\n",
       "      <td>0.267179</td>\n",
       "      <td>0.201879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>-0.233335</td>\n",
       "      <td>-0.130781</td>\n",
       "      <td>-0.255823</td>\n",
       "      <td>-0.217027</td>\n",
       "      <td>-0.153392</td>\n",
       "      <td>-0.093743</td>\n",
       "      <td>-0.059018</td>\n",
       "      <td>-0.028069</td>\n",
       "      <td>-0.424658</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.131372</td>\n",
       "      <td>-0.143658</td>\n",
       "      <td>0.001387</td>\n",
       "      <td>-0.216282</td>\n",
       "      <td>-0.055581</td>\n",
       "      <td>-0.137365</td>\n",
       "      <td>-0.114133</td>\n",
       "      <td>-0.164167</td>\n",
       "      <td>-0.251599</td>\n",
       "      <td>-0.204984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS</td>\n",
       "      <td>-0.628114</td>\n",
       "      <td>0.565258</td>\n",
       "      <td>-0.414934</td>\n",
       "      <td>0.312393</td>\n",
       "      <td>-0.035785</td>\n",
       "      <td>0.073530</td>\n",
       "      <td>0.376492</td>\n",
       "      <td>0.143698</td>\n",
       "      <td>-0.303845</td>\n",
       "      <td>...</td>\n",
       "      <td>0.111698</td>\n",
       "      <td>0.113708</td>\n",
       "      <td>0.110016</td>\n",
       "      <td>0.089854</td>\n",
       "      <td>0.144114</td>\n",
       "      <td>0.233087</td>\n",
       "      <td>0.140215</td>\n",
       "      <td>0.071328</td>\n",
       "      <td>-0.076422</td>\n",
       "      <td>-0.218106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8247</th>\n",
       "      <td>ZW10</td>\n",
       "      <td>-0.197293</td>\n",
       "      <td>0.005219</td>\n",
       "      <td>-0.214280</td>\n",
       "      <td>-0.044056</td>\n",
       "      <td>-0.040142</td>\n",
       "      <td>-0.132064</td>\n",
       "      <td>-0.103338</td>\n",
       "      <td>-0.061988</td>\n",
       "      <td>0.342361</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.097979</td>\n",
       "      <td>-0.138800</td>\n",
       "      <td>-0.128826</td>\n",
       "      <td>-0.173144</td>\n",
       "      <td>-0.041287</td>\n",
       "      <td>-0.075898</td>\n",
       "      <td>-0.086471</td>\n",
       "      <td>-0.158522</td>\n",
       "      <td>-0.096213</td>\n",
       "      <td>-0.051069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8248</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>-0.252162</td>\n",
       "      <td>0.008941</td>\n",
       "      <td>-0.085900</td>\n",
       "      <td>-0.106210</td>\n",
       "      <td>-0.172175</td>\n",
       "      <td>-0.027401</td>\n",
       "      <td>-0.167308</td>\n",
       "      <td>-0.051284</td>\n",
       "      <td>-0.180830</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.154005</td>\n",
       "      <td>-0.115725</td>\n",
       "      <td>-0.101206</td>\n",
       "      <td>-0.147403</td>\n",
       "      <td>-0.108262</td>\n",
       "      <td>-0.172407</td>\n",
       "      <td>-0.214812</td>\n",
       "      <td>-0.051132</td>\n",
       "      <td>-0.082856</td>\n",
       "      <td>-0.043790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8249</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.410138</td>\n",
       "      <td>0.519709</td>\n",
       "      <td>0.439445</td>\n",
       "      <td>0.243174</td>\n",
       "      <td>0.210981</td>\n",
       "      <td>0.235149</td>\n",
       "      <td>0.297345</td>\n",
       "      <td>0.214159</td>\n",
       "      <td>0.008757</td>\n",
       "      <td>...</td>\n",
       "      <td>0.283333</td>\n",
       "      <td>0.228445</td>\n",
       "      <td>0.224010</td>\n",
       "      <td>0.288499</td>\n",
       "      <td>0.331679</td>\n",
       "      <td>0.267056</td>\n",
       "      <td>0.402636</td>\n",
       "      <td>0.243234</td>\n",
       "      <td>0.256104</td>\n",
       "      <td>0.083366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8250</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.048636</td>\n",
       "      <td>0.137780</td>\n",
       "      <td>-0.018630</td>\n",
       "      <td>0.079965</td>\n",
       "      <td>0.200536</td>\n",
       "      <td>0.018601</td>\n",
       "      <td>0.052656</td>\n",
       "      <td>0.092258</td>\n",
       "      <td>0.133819</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034985</td>\n",
       "      <td>-0.021704</td>\n",
       "      <td>0.084270</td>\n",
       "      <td>0.017890</td>\n",
       "      <td>0.099436</td>\n",
       "      <td>0.039346</td>\n",
       "      <td>0.021192</td>\n",
       "      <td>0.042612</td>\n",
       "      <td>0.072650</td>\n",
       "      <td>0.096722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8251</th>\n",
       "      <td>uncharacterized</td>\n",
       "      <td>-0.435451</td>\n",
       "      <td>-0.077959</td>\n",
       "      <td>-0.448560</td>\n",
       "      <td>-0.175047</td>\n",
       "      <td>-0.368123</td>\n",
       "      <td>-0.207729</td>\n",
       "      <td>-0.378900</td>\n",
       "      <td>-0.116765</td>\n",
       "      <td>-0.144452</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.395805</td>\n",
       "      <td>-0.269467</td>\n",
       "      <td>-0.347543</td>\n",
       "      <td>-0.228417</td>\n",
       "      <td>-0.142008</td>\n",
       "      <td>-0.428470</td>\n",
       "      <td>-0.324132</td>\n",
       "      <td>-0.302110</td>\n",
       "      <td>-0.291100</td>\n",
       "      <td>-0.169623</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8252 rows × 401 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene_name  b01.127C  b01.127N  b01.128C  b01.128N  b01.129C  \\\n",
       "0                   0  0.056825 -0.191644  0.329755 -0.105563  0.065585   \n",
       "1                A1BG  0.078637 -0.300053  0.241096 -0.696597  0.089856   \n",
       "2                 A2M  0.024694 -0.636475 -0.028690 -0.582275 -0.084713   \n",
       "3                AAAS -0.233335 -0.130781 -0.255823 -0.217027 -0.153392   \n",
       "4                AACS -0.628114  0.565258 -0.414934  0.312393 -0.035785   \n",
       "...               ...       ...       ...       ...       ...       ...   \n",
       "8247             ZW10 -0.197293  0.005219 -0.214280 -0.044056 -0.040142   \n",
       "8248           ZYG11B -0.252162  0.008941 -0.085900 -0.106210 -0.172175   \n",
       "8249              ZYX  0.410138  0.519709  0.439445  0.243174  0.210981   \n",
       "8250            ZZEF1  0.048636  0.137780 -0.018630  0.079965  0.200536   \n",
       "8251  uncharacterized -0.435451 -0.077959 -0.448560 -0.175047 -0.368123   \n",
       "\n",
       "      b01.129N  b01.130C  b01.130N  b02.127C  ...  b49.130C  b49.130N  \\\n",
       "0    -0.003961  0.068675  0.624928  0.488592  ...  0.208725 -0.058407   \n",
       "1    -0.891969 -0.121492 -0.674060 -0.360159  ...  0.018837 -0.157993   \n",
       "2    -0.661039 -0.127533 -0.136789  0.071652  ...  0.199730 -0.142154   \n",
       "3    -0.093743 -0.059018 -0.028069 -0.424658  ... -0.131372 -0.143658   \n",
       "4     0.073530  0.376492  0.143698 -0.303845  ...  0.111698  0.113708   \n",
       "...        ...       ...       ...       ...  ...       ...       ...   \n",
       "8247 -0.132064 -0.103338 -0.061988  0.342361  ... -0.097979 -0.138800   \n",
       "8248 -0.027401 -0.167308 -0.051284 -0.180830  ... -0.154005 -0.115725   \n",
       "8249  0.235149  0.297345  0.214159  0.008757  ...  0.283333  0.228445   \n",
       "8250  0.018601  0.052656  0.092258  0.133819  ...  0.034985 -0.021704   \n",
       "8251 -0.207729 -0.378900 -0.116765 -0.144452  ... -0.395805 -0.269467   \n",
       "\n",
       "      b50.127C  b50.127N  b50.128C  b50.128N  b50.129C  b50.129N  b50.130C  \\\n",
       "0    -0.238951 -0.082743  0.131467 -0.177005  0.027383 -0.132419 -0.119505   \n",
       "1    -0.359336 -0.013279 -0.265634 -0.247393 -0.237626 -0.201277  0.146902   \n",
       "2    -0.202204  0.249200 -0.125857 -0.223538 -0.155823 -0.227838  0.267179   \n",
       "3     0.001387 -0.216282 -0.055581 -0.137365 -0.114133 -0.164167 -0.251599   \n",
       "4     0.110016  0.089854  0.144114  0.233087  0.140215  0.071328 -0.076422   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "8247 -0.128826 -0.173144 -0.041287 -0.075898 -0.086471 -0.158522 -0.096213   \n",
       "8248 -0.101206 -0.147403 -0.108262 -0.172407 -0.214812 -0.051132 -0.082856   \n",
       "8249  0.224010  0.288499  0.331679  0.267056  0.402636  0.243234  0.256104   \n",
       "8250  0.084270  0.017890  0.099436  0.039346  0.021192  0.042612  0.072650   \n",
       "8251 -0.347543 -0.228417 -0.142008 -0.428470 -0.324132 -0.302110 -0.291100   \n",
       "\n",
       "      b50.130N  \n",
       "0     0.101175  \n",
       "1    -0.027162  \n",
       "2     0.201879  \n",
       "3    -0.204984  \n",
       "4    -0.218106  \n",
       "...        ...  \n",
       "8247 -0.051069  \n",
       "8248 -0.043790  \n",
       "8249  0.083366  \n",
       "8250  0.096722  \n",
       "8251 -0.169623  \n",
       "\n",
       "[8252 rows x 401 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Fill NaN values with 0 in the remaining rows\n",
    "# protein = protein[nan_proportions <= 1/3]\n",
    "protein = protein.groupby('gene_name', as_index=False).mean()\n",
    "protein = protein.fillna(0)\n",
    "protein"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.6 Read Genome Variants data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>NAME</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>Quality</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTDVK</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>766593</td>\n",
       "      <td>769113</td>\n",
       "      <td>mCNV39</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>776769</td>\n",
       "      <td>791879</td>\n",
       "      <td>mCNV40</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>830676</td>\n",
       "      <td>834492</td>\n",
       "      <td>mCNV42</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1139154</td>\n",
       "      <td>1140550</td>\n",
       "      <td>mCNV53</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1238076</td>\n",
       "      <td>1239491</td>\n",
       "      <td>mCNV57</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>22</td>\n",
       "      <td>49765259</td>\n",
       "      <td>49767008</td>\n",
       "      <td>mCNV48272</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9898</th>\n",
       "      <td>22</td>\n",
       "      <td>49780301</td>\n",
       "      <td>49781900</td>\n",
       "      <td>DUP48274</td>\n",
       "      <td>DUP</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9899</th>\n",
       "      <td>22</td>\n",
       "      <td>49840021</td>\n",
       "      <td>49908322</td>\n",
       "      <td>mCNV48276</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9900</th>\n",
       "      <td>22</td>\n",
       "      <td>51082701</td>\n",
       "      <td>51121100</td>\n",
       "      <td>mCNV48291</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9901</th>\n",
       "      <td>22</td>\n",
       "      <td>51119846</td>\n",
       "      <td>51121237</td>\n",
       "      <td>mCNV48292</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9902 rows × 1133 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM     START       END       NAME SVTYPE        Quality  SM-CTDSC  \\\n",
       "0         1    766593    769113     mCNV39   mCNV  Consensus.III         2   \n",
       "1         1    776769    791879     mCNV40   mCNV  Consensus.III         2   \n",
       "2         1    830676    834492     mCNV42   mCNV  Consensus.III         2   \n",
       "3         1   1139154   1140550     mCNV53   mCNV  Consensus.III         2   \n",
       "4         1   1238076   1239491     mCNV57   mCNV  Consensus.III         2   \n",
       "...     ...       ...       ...        ...    ...            ...       ...   \n",
       "9897     22  49765259  49767008  mCNV48272   mCNV  Consensus.III         2   \n",
       "9898     22  49780301  49781900   DUP48274    DUP  Consensus.III         3   \n",
       "9899     22  49840021  49908322  mCNV48276   mCNV  Consensus.III         2   \n",
       "9900     22  51082701  51121100  mCNV48291   mCNV  Consensus.III         2   \n",
       "9901     22  51119846  51121237  mCNV48292   mCNV  Consensus.III         1   \n",
       "\n",
       "      SM-CJGMZ  SM-CJFM3  SM-CTEET  ...  SM-CTDVK  SM-CTED3  SM-CJK4V  \\\n",
       "0            2         2         2  ...         2         2         2   \n",
       "1            2         2         2  ...         2         2         2   \n",
       "2            2         2         2  ...         2         2         1   \n",
       "3            2         2         2  ...         2         2         2   \n",
       "4            2         2         2  ...         2         2         2   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "9897         2         2         2  ...         2         2         2   \n",
       "9898         3         4         3  ...         3         4         3   \n",
       "9899         2         2         2  ...         2         2         2   \n",
       "9900         2         2         2  ...         2         2         1   \n",
       "9901         3         2         3  ...         3         3         1   \n",
       "\n",
       "      SM-CJK5F  SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \n",
       "0            2         2         2         2         2         2         2  \n",
       "1            2         2         2         2         2         2         2  \n",
       "2            2         2         2         2         2         2         2  \n",
       "3            2         2         2         2         2         2         2  \n",
       "4            2         2         2         2         2         2         2  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "9897         2         2         2         2         1         2         2  \n",
       "9898         2         4         3         4         3         3         3  \n",
       "9899         2         2         2         2         2         2         2  \n",
       "9900         2         2         2         2         2         2         2  \n",
       "9901         1         2         1         3         1         2         3  \n",
       "\n",
       "[9902 rows x 1133 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_data = pd.read_csv('./ROSMAP-raw/GenomeVariants/ROSMAP.CNV.Matrix.txt',sep='\\t')\n",
    "cnv_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:pyensembl.sequence_data:Loaded sequence dictionary from C:\\Users\\Administrator\\AppData\\Local\\pyensembl\\GRCh38\\ensembl110\\pyensembl\\GRCh38\\ensembl110\\Cache\\Homo_sapiens.GRCh38.cdna.all.fa.gz.pickle\n",
      "INFO:pyensembl.sequence_data:Loaded sequence dictionary from C:\\Users\\Administrator\\AppData\\Local\\pyensembl\\GRCh38\\ensembl110\\pyensembl\\GRCh38\\ensembl110\\Cache\\Homo_sapiens.GRCh38.ncrna.fa.gz.pickle\n",
      "INFO:pyensembl.sequence_data:Loaded sequence dictionary from C:\\Users\\Administrator\\AppData\\Local\\pyensembl\\GRCh38\\ensembl110\\pyensembl\\GRCh38\\ensembl110\\Cache\\Homo_sapiens.GRCh38.pep.all.fa.gz.pickle\n"
     ]
    }
   ],
   "source": [
    "ensembl_data = EnsemblRelease()\n",
    "ensembl_data.download()\n",
    "ensembl_data.index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>NAME</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>Quality</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>766593</td>\n",
       "      <td>769113</td>\n",
       "      <td>mCNV39</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>, ,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>776769</td>\n",
       "      <td>791879</td>\n",
       "      <td>mCNV40</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>, LINC01409, ,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>830676</td>\n",
       "      <td>834492</td>\n",
       "      <td>mCNV42</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1139154</td>\n",
       "      <td>1140550</td>\n",
       "      <td>mCNV53</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1238076</td>\n",
       "      <td>1239491</td>\n",
       "      <td>mCNV57</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>22</td>\n",
       "      <td>49765259</td>\n",
       "      <td>49767008</td>\n",
       "      <td>mCNV48272</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9898</th>\n",
       "      <td>22</td>\n",
       "      <td>49780301</td>\n",
       "      <td>49781900</td>\n",
       "      <td>DUP48274</td>\n",
       "      <td>DUP</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>BRD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9899</th>\n",
       "      <td>22</td>\n",
       "      <td>49840021</td>\n",
       "      <td>49908322</td>\n",
       "      <td>mCNV48276</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ZBED4, ALG12, , , Metazoa_SRP,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9900</th>\n",
       "      <td>22</td>\n",
       "      <td>51082701</td>\n",
       "      <td>51121100</td>\n",
       "      <td>mCNV48291</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9901</th>\n",
       "      <td>22</td>\n",
       "      <td>51119846</td>\n",
       "      <td>51121237</td>\n",
       "      <td>mCNV48292</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9902 rows × 1134 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM     START       END       NAME SVTYPE        Quality  SM-CTDSC  \\\n",
       "0         1    766593    769113     mCNV39   mCNV  Consensus.III         2   \n",
       "1         1    776769    791879     mCNV40   mCNV  Consensus.III         2   \n",
       "2         1    830676    834492     mCNV42   mCNV  Consensus.III         2   \n",
       "3         1   1139154   1140550     mCNV53   mCNV  Consensus.III         2   \n",
       "4         1   1238076   1239491     mCNV57   mCNV  Consensus.III         2   \n",
       "...     ...       ...       ...        ...    ...            ...       ...   \n",
       "9897     22  49765259  49767008  mCNV48272   mCNV  Consensus.III         2   \n",
       "9898     22  49780301  49781900   DUP48274    DUP  Consensus.III         3   \n",
       "9899     22  49840021  49908322  mCNV48276   mCNV  Consensus.III         2   \n",
       "9900     22  51082701  51121100  mCNV48291   mCNV  Consensus.III         2   \n",
       "9901     22  51119846  51121237  mCNV48292   mCNV  Consensus.III         1   \n",
       "\n",
       "      SM-CJGMZ  SM-CJFM3  SM-CTEET  ...  SM-CTED3  SM-CJK4V  SM-CJK5F  \\\n",
       "0            2         2         2  ...         2         2         2   \n",
       "1            2         2         2  ...         2         2         2   \n",
       "2            2         2         2  ...         2         1         2   \n",
       "3            2         2         2  ...         2         2         2   \n",
       "4            2         2         2  ...         2         2         2   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "9897         2         2         2  ...         2         2         2   \n",
       "9898         3         4         3  ...         4         3         2   \n",
       "9899         2         2         2  ...         2         2         2   \n",
       "9900         2         2         2  ...         2         1         2   \n",
       "9901         3         2         3  ...         3         1         1   \n",
       "\n",
       "      SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \\\n",
       "0            2         2         2         2         2         2   \n",
       "1            2         2         2         2         2         2   \n",
       "2            2         2         2         2         2         2   \n",
       "3            2         2         2         2         2         2   \n",
       "4            2         2         2         2         2         2   \n",
       "...        ...       ...       ...       ...       ...       ...   \n",
       "9897         2         2         2         1         2         2   \n",
       "9898         4         3         4         3         3         3   \n",
       "9899         2         2         2         2         2         2   \n",
       "9900         2         2         2         2         2         2   \n",
       "9901         2         1         3         1         2         3   \n",
       "\n",
       "                            gene_name  \n",
       "0                                , ,   \n",
       "1                     , LINC01409, ,   \n",
       "2                           LINC01128  \n",
       "3                           LINC01342  \n",
       "4                                None  \n",
       "...                               ...  \n",
       "9897                             None  \n",
       "9898                             BRD1  \n",
       "9899  ZBED4, ALG12, , , Metazoa_SRP,   \n",
       "9900                             None  \n",
       "9901                             None  \n",
       "\n",
       "[9902 rows x 1134 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use pyensemb to map the gene name\n",
    "def get_gene_names(chromosome, start, end):\n",
    "    genes = ensembl_data.genes_at_locus(contig=chromosome, position=start, end=end)\n",
    "    gene_names = [gene.gene_name for gene in genes]\n",
    "    return ', '.join(gene_names) if gene_names else None\n",
    "\n",
    "# Apply the function to each row in the DataFrame\n",
    "cnv_data['gene_name'] = cnv_data.apply(\n",
    "    lambda row: get_gene_names(str(row['CHROM']), row['START'], row['END']), axis=1\n",
    ")\n",
    "cnv_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>NAME</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>Quality</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>766593</td>\n",
       "      <td>769113</td>\n",
       "      <td>mCNV39</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>776769</td>\n",
       "      <td>791879</td>\n",
       "      <td>mCNV40</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>830676</td>\n",
       "      <td>834492</td>\n",
       "      <td>mCNV42</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1139154</td>\n",
       "      <td>1140550</td>\n",
       "      <td>mCNV53</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1238076</td>\n",
       "      <td>1239491</td>\n",
       "      <td>mCNV57</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>22</td>\n",
       "      <td>49765259</td>\n",
       "      <td>49767008</td>\n",
       "      <td>mCNV48272</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9898</th>\n",
       "      <td>22</td>\n",
       "      <td>49780301</td>\n",
       "      <td>49781900</td>\n",
       "      <td>DUP48274</td>\n",
       "      <td>DUP</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>BRD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9899</th>\n",
       "      <td>22</td>\n",
       "      <td>49840021</td>\n",
       "      <td>49908322</td>\n",
       "      <td>mCNV48276</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ZBED4, ALG12, Metazoa_SRP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9900</th>\n",
       "      <td>22</td>\n",
       "      <td>51082701</td>\n",
       "      <td>51121100</td>\n",
       "      <td>mCNV48291</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9901</th>\n",
       "      <td>22</td>\n",
       "      <td>51119846</td>\n",
       "      <td>51121237</td>\n",
       "      <td>mCNV48292</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9902 rows × 1134 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM     START       END       NAME SVTYPE        Quality  SM-CTDSC  \\\n",
       "0         1    766593    769113     mCNV39   mCNV  Consensus.III         2   \n",
       "1         1    776769    791879     mCNV40   mCNV  Consensus.III         2   \n",
       "2         1    830676    834492     mCNV42   mCNV  Consensus.III         2   \n",
       "3         1   1139154   1140550     mCNV53   mCNV  Consensus.III         2   \n",
       "4         1   1238076   1239491     mCNV57   mCNV  Consensus.III         2   \n",
       "...     ...       ...       ...        ...    ...            ...       ...   \n",
       "9897     22  49765259  49767008  mCNV48272   mCNV  Consensus.III         2   \n",
       "9898     22  49780301  49781900   DUP48274    DUP  Consensus.III         3   \n",
       "9899     22  49840021  49908322  mCNV48276   mCNV  Consensus.III         2   \n",
       "9900     22  51082701  51121100  mCNV48291   mCNV  Consensus.III         2   \n",
       "9901     22  51119846  51121237  mCNV48292   mCNV  Consensus.III         1   \n",
       "\n",
       "      SM-CJGMZ  SM-CJFM3  SM-CTEET  ...  SM-CTED3  SM-CJK4V  SM-CJK5F  \\\n",
       "0            2         2         2  ...         2         2         2   \n",
       "1            2         2         2  ...         2         2         2   \n",
       "2            2         2         2  ...         2         1         2   \n",
       "3            2         2         2  ...         2         2         2   \n",
       "4            2         2         2  ...         2         2         2   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "9897         2         2         2  ...         2         2         2   \n",
       "9898         3         4         3  ...         4         3         2   \n",
       "9899         2         2         2  ...         2         2         2   \n",
       "9900         2         2         2  ...         2         1         2   \n",
       "9901         3         2         3  ...         3         1         1   \n",
       "\n",
       "      SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \\\n",
       "0            2         2         2         2         2         2   \n",
       "1            2         2         2         2         2         2   \n",
       "2            2         2         2         2         2         2   \n",
       "3            2         2         2         2         2         2   \n",
       "4            2         2         2         2         2         2   \n",
       "...        ...       ...       ...       ...       ...       ...   \n",
       "9897         2         2         2         1         2         2   \n",
       "9898         4         3         4         3         3         3   \n",
       "9899         2         2         2         2         2         2   \n",
       "9900         2         2         2         2         2         2   \n",
       "9901         2         1         3         1         2         3   \n",
       "\n",
       "                      gene_name  \n",
       "0                                \n",
       "1                     LINC01409  \n",
       "2                     LINC01128  \n",
       "3                     LINC01342  \n",
       "4                 No gene found  \n",
       "...                         ...  \n",
       "9897              No gene found  \n",
       "9898                       BRD1  \n",
       "9899  ZBED4, ALG12, Metazoa_SRP  \n",
       "9900              No gene found  \n",
       "9901              No gene found  \n",
       "\n",
       "[9902 rows x 1134 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Function to clean gene names\n",
    "def clean_gene_names(gene_names):\n",
    "    if gene_names:\n",
    "        # Split the string by comma, strip whitespace and periods, then filter out empty strings\n",
    "        genes = [gene.strip(' .') for gene in gene_names.split(',') if gene.strip(' .')]\n",
    "        # Join the cleaned gene names with a comma\n",
    "        return ', '.join(genes)\n",
    "    else:\n",
    "        # If gene_names is None or empty, return a placeholder\n",
    "        return 'No gene found'\n",
    "\n",
    "# Apply the function to clean up the gene names\n",
    "cnv_data['gene_name'] = cnv_data['gene_name'].apply(clean_gene_names)\n",
    "cnv_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>NAME</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>Quality</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>766593</td>\n",
       "      <td>769113</td>\n",
       "      <td>mCNV39</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>776769</td>\n",
       "      <td>791879</td>\n",
       "      <td>mCNV40</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>830676</td>\n",
       "      <td>834492</td>\n",
       "      <td>mCNV42</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1139154</td>\n",
       "      <td>1140550</td>\n",
       "      <td>mCNV53</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>LINC01342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1238076</td>\n",
       "      <td>1239491</td>\n",
       "      <td>mCNV57</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9899</th>\n",
       "      <td>22</td>\n",
       "      <td>49840021</td>\n",
       "      <td>49908322</td>\n",
       "      <td>mCNV48276</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ZBED4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9899</th>\n",
       "      <td>22</td>\n",
       "      <td>49840021</td>\n",
       "      <td>49908322</td>\n",
       "      <td>mCNV48276</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ALG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9899</th>\n",
       "      <td>22</td>\n",
       "      <td>49840021</td>\n",
       "      <td>49908322</td>\n",
       "      <td>mCNV48276</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Metazoa_SRP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9900</th>\n",
       "      <td>22</td>\n",
       "      <td>51082701</td>\n",
       "      <td>51121100</td>\n",
       "      <td>mCNV48291</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9901</th>\n",
       "      <td>22</td>\n",
       "      <td>51119846</td>\n",
       "      <td>51121237</td>\n",
       "      <td>mCNV48292</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>Consensus.III</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>No gene found</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13549 rows × 1134 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM     START       END       NAME SVTYPE        Quality  SM-CTDSC  \\\n",
       "0         1    766593    769113     mCNV39   mCNV  Consensus.III         2   \n",
       "1         1    776769    791879     mCNV40   mCNV  Consensus.III         2   \n",
       "2         1    830676    834492     mCNV42   mCNV  Consensus.III         2   \n",
       "3         1   1139154   1140550     mCNV53   mCNV  Consensus.III         2   \n",
       "4         1   1238076   1239491     mCNV57   mCNV  Consensus.III         2   \n",
       "...     ...       ...       ...        ...    ...            ...       ...   \n",
       "9899     22  49840021  49908322  mCNV48276   mCNV  Consensus.III         2   \n",
       "9899     22  49840021  49908322  mCNV48276   mCNV  Consensus.III         2   \n",
       "9899     22  49840021  49908322  mCNV48276   mCNV  Consensus.III         2   \n",
       "9900     22  51082701  51121100  mCNV48291   mCNV  Consensus.III         2   \n",
       "9901     22  51119846  51121237  mCNV48292   mCNV  Consensus.III         1   \n",
       "\n",
       "      SM-CJGMZ  SM-CJFM3  SM-CTEET  ...  SM-CTED3  SM-CJK4V  SM-CJK5F  \\\n",
       "0            2         2         2  ...         2         2         2   \n",
       "1            2         2         2  ...         2         2         2   \n",
       "2            2         2         2  ...         2         1         2   \n",
       "3            2         2         2  ...         2         2         2   \n",
       "4            2         2         2  ...         2         2         2   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "9899         2         2         2  ...         2         2         2   \n",
       "9899         2         2         2  ...         2         2         2   \n",
       "9899         2         2         2  ...         2         2         2   \n",
       "9900         2         2         2  ...         2         1         2   \n",
       "9901         3         2         3  ...         3         1         1   \n",
       "\n",
       "      SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \\\n",
       "0            2         2         2         2         2         2   \n",
       "1            2         2         2         2         2         2   \n",
       "2            2         2         2         2         2         2   \n",
       "3            2         2         2         2         2         2   \n",
       "4            2         2         2         2         2         2   \n",
       "...        ...       ...       ...       ...       ...       ...   \n",
       "9899         2         2         2         2         2         2   \n",
       "9899         2         2         2         2         2         2   \n",
       "9899         2         2         2         2         2         2   \n",
       "9900         2         2         2         2         2         2   \n",
       "9901         2         1         3         1         2         3   \n",
       "\n",
       "          gene_name  \n",
       "0                    \n",
       "1         LINC01409  \n",
       "2         LINC01128  \n",
       "3         LINC01342  \n",
       "4     No gene found  \n",
       "...             ...  \n",
       "9899          ZBED4  \n",
       "9899          ALG12  \n",
       "9899    Metazoa_SRP  \n",
       "9900  No gene found  \n",
       "9901  No gene found  \n",
       "\n",
       "[13549 rows x 1134 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_data['gene_name'] = cnv_data['gene_name'].str.split(', ')\n",
    "cnv_data_exploded = cnv_data.explode('gene_name')\n",
    "cnv_data_exploded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>SM-CJK4L</th>\n",
       "      <th>SM-CJIZ4</th>\n",
       "      <th>SM-CJK3I</th>\n",
       "      <th>SM-CJGN4</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTDVK</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5S_rRNA</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5S_rRNA</td>\n",
       "      <td>DUP</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ML1-AS1</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7300</th>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7301</th>\n",
       "      <td>ZSWIM9</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7302</th>\n",
       "      <td>ZUP1</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7303</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7304</th>\n",
       "      <td>hsa-mir-423</td>\n",
       "      <td>mCNV</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7305 rows × 1129 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_name SVTYPE  SM-CTDSC  SM-CJGMZ  SM-CJFM3  SM-CTEET  SM-CJK4L  \\\n",
       "0         5S_rRNA    DEL         2         2         2         2         2   \n",
       "1         5S_rRNA    DUP         2         2         2         2         2   \n",
       "2       A2ML1-AS1    DEL         2         2         2         2         2   \n",
       "3          A4GALT   mCNV         2         3         2         3         2   \n",
       "4            AAAS   mCNV         2         2         2         2         2   \n",
       "...           ...    ...       ...       ...       ...       ...       ...   \n",
       "7300       ZSWIM6    DEL         2         2         2         2         2   \n",
       "7301       ZSWIM9   mCNV         6         6         6         6         6   \n",
       "7302         ZUP1   mCNV         2         2         2         2         2   \n",
       "7303         ZZZ3    DEL         2         2         2         2         2   \n",
       "7304  hsa-mir-423   mCNV         2         2         2         2         2   \n",
       "\n",
       "      SM-CJIZ4  SM-CJK3I  SM-CJGN4  ...  SM-CTDVK  SM-CTED3  SM-CJK4V  \\\n",
       "0            2         2         2  ...         2         2         2   \n",
       "1            2         2         2  ...         2         2         2   \n",
       "2            2         2         2  ...         2         2         1   \n",
       "3            3         2         3  ...         2         3         2   \n",
       "4            2         2         2  ...         2         2         1   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "7300         2         2         2  ...         2         2         2   \n",
       "7301         6         6         5  ...         6         6         3   \n",
       "7302         2         2         2  ...         2         2         2   \n",
       "7303         2         2         2  ...         2         2         2   \n",
       "7304         2         2         1  ...         2         2         1   \n",
       "\n",
       "      SM-CJK5F  SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \n",
       "0            2         2         2         2         2         2         2  \n",
       "1            2         2         2         2         2         2         2  \n",
       "2            2         2         2         2         2         2         2  \n",
       "3            2         2         2         2         1         3         3  \n",
       "4            2         2         2         1         2         2         2  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "7300         1         2         2         2         2         2         2  \n",
       "7301         3         5         6         3         6         6         6  \n",
       "7302         2         2         2         2         2         2         2  \n",
       "7303         2         2         2         2         2         2         2  \n",
       "7304         1         1         2         1         2         2         2  \n",
       "\n",
       "[7305 rows x 1129 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter out rows with empty 'Gene_Names' or 'No gene found'\n",
    "cnv_data_exploded = cnv_data_exploded[(cnv_data_exploded['gene_name'] != '') & (cnv_data_exploded['gene_name'] != 'No gene found')]\n",
    "\n",
    "sample_columns = cnv_data_exploded.columns[6:-1] \n",
    "cnv_aggregated = cnv_data_exploded.groupby(['gene_name','SVTYPE'])[sample_columns].sum().reset_index()\n",
    "# cnv_aggregated = cnv_data_exploded.groupby('gene_name')[sample_columns].sum().reset_index()\n",
    "cnv_aggregated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of cnv_aggregated_del: (2483, 1129)\n",
      "Shape of cnv_aggregated_dup: (1096, 1129)\n",
      "Shape of cnv_aggregated_mcnv: (3726, 1129)\n"
     ]
    }
   ],
   "source": [
    "cnv_aggregated_del = cnv_aggregated[cnv_aggregated['SVTYPE'] == 'DEL']\n",
    "cnv_aggregated_dup = cnv_aggregated[cnv_aggregated['SVTYPE'] == 'DUP']\n",
    "cnv_aggregated_mcnv = cnv_aggregated[cnv_aggregated['SVTYPE'] == 'mCNV']\n",
    "print(f\"Shape of cnv_aggregated_del: {cnv_aggregated_del.shape}\")\n",
    "print(f\"Shape of cnv_aggregated_dup: {cnv_aggregated_dup.shape}\")\n",
    "print(f\"Shape of cnv_aggregated_mcnv: {cnv_aggregated_mcnv.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of cnv_aggregated_del: (6423, 1128)\n",
      "Shape of cnv_aggregated_dup: (6423, 1128)\n",
      "Shape of cnv_aggregated_mcnv: (6423, 1128)\n"
     ]
    }
   ],
   "source": [
    "# Create a DataFrame from the union set of all gene names\n",
    "cnv_all_genes = pd.DataFrame({'gene_name': list(set(cnv_aggregated_del['gene_name']).union(set(cnv_aggregated_dup['gene_name']), set(cnv_aggregated_mcnv['gene_name'])))})\n",
    "\n",
    "# Merge each aggregated DataFrame with the all genes DataFrame\n",
    "# This will align all genes across all DataFrames, and fill missing entries with 0\n",
    "cnv_aggregated_del = pd.merge(cnv_all_genes, cnv_aggregated_del, on='gene_name', how='outer').fillna(-1)\n",
    "cnv_aggregated_dup = pd.merge(cnv_all_genes, cnv_aggregated_dup, on='gene_name', how='outer').fillna(-1)\n",
    "cnv_aggregated_mcnv = pd.merge(cnv_all_genes, cnv_aggregated_mcnv, on='gene_name', how='outer').fillna(-1)\n",
    "\n",
    "# Drop the 'SVTYPE' column from each merged DataFrame\n",
    "cnv_aggregated_del.drop(columns='SVTYPE', inplace=True)\n",
    "cnv_aggregated_dup.drop(columns='SVTYPE', inplace=True)\n",
    "cnv_aggregated_mcnv.drop(columns='SVTYPE', inplace=True)\n",
    "print(f\"Shape of cnv_aggregated_del: {cnv_aggregated_del.shape}\")\n",
    "print(f\"Shape of cnv_aggregated_dup: {cnv_aggregated_dup.shape}\")\n",
    "print(f\"Shape of cnv_aggregated_mcnv: {cnv_aggregated_mcnv.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>SM-CJK4L</th>\n",
       "      <th>SM-CJIZ4</th>\n",
       "      <th>SM-CJK3I</th>\n",
       "      <th>SM-CJGN4</th>\n",
       "      <th>SM-CTDSM</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTDVK</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XRCC1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LRP10</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALB</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIR4509-3</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LINC02556</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6418</th>\n",
       "      <td>TP53TG5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6419</th>\n",
       "      <td>ERCC1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6420</th>\n",
       "      <td>ETHE1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6421</th>\n",
       "      <td>SSC4D</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6422</th>\n",
       "      <td>VPS13B</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6423 rows × 1128 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_name  SM-CTDSC  SM-CJGMZ  SM-CJFM3  SM-CTEET  SM-CJK4L  SM-CJIZ4  \\\n",
       "0         XRCC1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "1         LRP10       2.0       1.0       2.0       2.0       2.0       2.0   \n",
       "2           ALB      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "3     MIR4509-3      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "4     LINC02556       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "...         ...       ...       ...       ...       ...       ...       ...   \n",
       "6418    TP53TG5       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "6419      ERCC1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "6420      ETHE1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "6421      SSC4D       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "6422     VPS13B       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "\n",
       "      SM-CJK3I  SM-CJGN4  SM-CTDSM  ...  SM-CTDVK  SM-CTED3  SM-CJK4V  \\\n",
       "0         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "1          1.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "2         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "3         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "4          2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "6418       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "6419      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "6420      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "6421       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "6422       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "\n",
       "      SM-CJK5F  SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \n",
       "0         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "1          2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "2         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "3         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "4          2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "6418       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "6419      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "6420      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "6421       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "6422       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "\n",
       "[6423 rows x 1128 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_aggregated_del"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cnv_aggregated.iloc[:, 1:] = cnv_aggregated.iloc[:, 1:].where(cnv_aggregated.iloc[:, 1:] == 0, 1)\n",
    "# cnv_aggregated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# snphead = pd.read_csv('./ROSMAP-raw/GenomeVariants/SNP-Array-raw/AMP-AD_ROSMAP_Rush-Broad_AffymetrixGenechip6_Imputed.fam', sep='\\t',header=None)\n",
    "# snphead"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# survival_gwas_id = pd.DataFrame(survival.iloc[:, 1].astype(str) + survival.iloc[:, 0].astype(str), columns=['gwas_id'])\n",
    "# survival_gwas_id['individualID'] = survival['individualID']\n",
    "\n",
    "# # Displaying the new DataFrame\n",
    "# survival_gwas_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gwas_id_list = survival_gwas_id['gwas_id'].tolist()\n",
    "\n",
    "# snphead['extracted_gwas_id'] = snphead.iloc[:, 0].apply(\n",
    "#     lambda x: next((gwas_id for gwas_id in gwas_id_list if gwas_id in x), np.nan)\n",
    "# )\n",
    "\n",
    "# # Perform a left merge with 'survival_gwas_id' on the extracted 'gwas_id' to keep all rows from 'snphead'\n",
    "# # and to get the corresponding 'individualID' where there is a match\n",
    "# merged_result = pd.merge(\n",
    "#     snphead, \n",
    "#     survival_gwas_id, \n",
    "#     how='left', \n",
    "#     left_on='extracted_gwas_id', \n",
    "#     right_on='gwas_id'\n",
    "# )\n",
    "\n",
    "# # Select only the 'gwas_id' and 'individualID' columns for the final result\n",
    "# # Fill missing values with 'NA' to indicate unmatched rows\n",
    "# snphead_updated = merged_result[['gwas_id', 'individualID']].fillna('NA')\n",
    "\n",
    "# # Display the head of the final DataFrame\n",
    "# snphead_updated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# snphead_updated_withoutNA = snphead_updated[snphead_updated.iloc[:, 0] != \"NA\"]\n",
    "# snphead_updated_withoutNA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3.Methylation data process"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.1 Define methylation region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "#Define vectorized area determination function for methylation data\n",
    "def vectorized_determine_region(distances):\n",
    "    regions = ['Upstream', 'Distal Promoter', 'Proximal Promoter', 'Core Promoter', 'Downstream']\n",
    "    conditions = [\n",
    "        (-6000 <= distances) & (distances < -3000),\n",
    "        (-3000 <= distances) & (distances < -250),\n",
    "        (-250 <= distances) & (distances < -50),\n",
    "        (-50 <= distances) & (distances <= 0),\n",
    "        (0 < distances) & (distances <= 3000)\n",
    "    ]\n",
    "    return np.select(conditions, regions, default=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.2 Merge the annotation files to the methylation data and apply region function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merging basic data and methylation data\n",
    "methylation_merged_df = pd.merge(map_annotation, methylation_value, left_on='ID', right_on='TargetID', how='right')\n",
    "\n",
    "# Determining the region for each row outside the loop\n",
    "methylation_merged_df['Region'] = vectorized_determine_region(methylation_merged_df['Distance_closest_TSS'])\n",
    "\n",
    "methylation_merged_df = methylation_merged_df.dropna(subset=['Region'])  # Remove rows without a region\n",
    "\n",
    "# Initializing a dictionary to store data for each region\n",
    "regions_data = {region: pd.DataFrame() for region in [\"Upstream\", \"Distal Promoter\", \"Proximal Promoter\", \"Core Promoter\", \"Downstream\"]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>Closest_TSS_gene_name</th>\n",
       "      <th>Distance_closest_TSS</th>\n",
       "      <th>TargetID</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "      <th>Region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cg00001349</td>\n",
       "      <td>166958518</td>\n",
       "      <td>MAEL</td>\n",
       "      <td>-79.0</td>\n",
       "      <td>cg00001349</td>\n",
       "      <td>0.865488</td>\n",
       "      <td>0.919570</td>\n",
       "      <td>0.882256</td>\n",
       "      <td>0.902912</td>\n",
       "      <td>0.907610</td>\n",
       "      <td>...</td>\n",
       "      <td>0.900684</td>\n",
       "      <td>0.906257</td>\n",
       "      <td>0.756173</td>\n",
       "      <td>0.948341</td>\n",
       "      <td>0.904996</td>\n",
       "      <td>0.927201</td>\n",
       "      <td>0.856637</td>\n",
       "      <td>0.917827</td>\n",
       "      <td>0.943320</td>\n",
       "      <td>Proximal Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>cg00001446</td>\n",
       "      <td>43833698</td>\n",
       "      <td>ELOVL1</td>\n",
       "      <td>2658.0</td>\n",
       "      <td>cg00001446</td>\n",
       "      <td>0.842025</td>\n",
       "      <td>0.831457</td>\n",
       "      <td>0.835075</td>\n",
       "      <td>0.834157</td>\n",
       "      <td>0.839395</td>\n",
       "      <td>...</td>\n",
       "      <td>0.847344</td>\n",
       "      <td>0.825179</td>\n",
       "      <td>0.887574</td>\n",
       "      <td>0.881636</td>\n",
       "      <td>0.852642</td>\n",
       "      <td>0.833056</td>\n",
       "      <td>0.842108</td>\n",
       "      <td>0.841747</td>\n",
       "      <td>0.833336</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>cg00001583</td>\n",
       "      <td>200011716</td>\n",
       "      <td>NR5A2</td>\n",
       "      <td>70.0</td>\n",
       "      <td>cg00001583</td>\n",
       "      <td>0.099888</td>\n",
       "      <td>0.076240</td>\n",
       "      <td>0.075861</td>\n",
       "      <td>0.101837</td>\n",
       "      <td>0.071947</td>\n",
       "      <td>...</td>\n",
       "      <td>0.084641</td>\n",
       "      <td>0.092234</td>\n",
       "      <td>0.066651</td>\n",
       "      <td>0.040554</td>\n",
       "      <td>0.059607</td>\n",
       "      <td>0.116837</td>\n",
       "      <td>0.069391</td>\n",
       "      <td>0.061016</td>\n",
       "      <td>0.035535</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>cg00001593</td>\n",
       "      <td>170489886</td>\n",
       "      <td>AK096329</td>\n",
       "      <td>-547.0</td>\n",
       "      <td>cg00001593</td>\n",
       "      <td>0.918081</td>\n",
       "      <td>0.955305</td>\n",
       "      <td>0.896318</td>\n",
       "      <td>0.941423</td>\n",
       "      <td>0.940628</td>\n",
       "      <td>...</td>\n",
       "      <td>0.918949</td>\n",
       "      <td>0.929937</td>\n",
       "      <td>0.952355</td>\n",
       "      <td>0.968486</td>\n",
       "      <td>0.972349</td>\n",
       "      <td>0.944855</td>\n",
       "      <td>0.924879</td>\n",
       "      <td>0.885508</td>\n",
       "      <td>0.939249</td>\n",
       "      <td>Distal Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>cg00002028</td>\n",
       "      <td>20959947</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>63.0</td>\n",
       "      <td>cg00002028</td>\n",
       "      <td>0.026821</td>\n",
       "      <td>0.035839</td>\n",
       "      <td>0.020016</td>\n",
       "      <td>0.014101</td>\n",
       "      <td>0.031390</td>\n",
       "      <td>...</td>\n",
       "      <td>0.021462</td>\n",
       "      <td>0.033637</td>\n",
       "      <td>0.024296</td>\n",
       "      <td>0.033715</td>\n",
       "      <td>0.019535</td>\n",
       "      <td>0.011352</td>\n",
       "      <td>0.020888</td>\n",
       "      <td>0.035047</td>\n",
       "      <td>0.026995</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420108</th>\n",
       "      <td>cg27652464</td>\n",
       "      <td>48885287</td>\n",
       "      <td>FAM19A5</td>\n",
       "      <td>1488.0</td>\n",
       "      <td>cg27652464</td>\n",
       "      <td>0.077791</td>\n",
       "      <td>0.081391</td>\n",
       "      <td>0.066050</td>\n",
       "      <td>0.076101</td>\n",
       "      <td>0.081930</td>\n",
       "      <td>...</td>\n",
       "      <td>0.082855</td>\n",
       "      <td>0.078940</td>\n",
       "      <td>0.033035</td>\n",
       "      <td>0.029362</td>\n",
       "      <td>0.041915</td>\n",
       "      <td>0.059740</td>\n",
       "      <td>0.060660</td>\n",
       "      <td>0.040806</td>\n",
       "      <td>0.044582</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420111</th>\n",
       "      <td>cg27657537</td>\n",
       "      <td>20862338</td>\n",
       "      <td>MED15</td>\n",
       "      <td>1424.0</td>\n",
       "      <td>cg27657537</td>\n",
       "      <td>0.246730</td>\n",
       "      <td>0.273854</td>\n",
       "      <td>0.250933</td>\n",
       "      <td>0.297172</td>\n",
       "      <td>0.256656</td>\n",
       "      <td>...</td>\n",
       "      <td>0.296850</td>\n",
       "      <td>0.295459</td>\n",
       "      <td>0.266729</td>\n",
       "      <td>0.301941</td>\n",
       "      <td>0.265106</td>\n",
       "      <td>0.263730</td>\n",
       "      <td>0.268427</td>\n",
       "      <td>0.252280</td>\n",
       "      <td>0.243976</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420112</th>\n",
       "      <td>cg27662611</td>\n",
       "      <td>38599026</td>\n",
       "      <td>MAFF</td>\n",
       "      <td>-45.0</td>\n",
       "      <td>cg27662611</td>\n",
       "      <td>0.091869</td>\n",
       "      <td>0.098040</td>\n",
       "      <td>0.066290</td>\n",
       "      <td>0.094694</td>\n",
       "      <td>0.069012</td>\n",
       "      <td>...</td>\n",
       "      <td>0.080595</td>\n",
       "      <td>0.067712</td>\n",
       "      <td>0.072670</td>\n",
       "      <td>0.069177</td>\n",
       "      <td>0.084736</td>\n",
       "      <td>0.071235</td>\n",
       "      <td>0.072670</td>\n",
       "      <td>0.083982</td>\n",
       "      <td>0.062013</td>\n",
       "      <td>Core Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420113</th>\n",
       "      <td>cg27665648</td>\n",
       "      <td>30116343</td>\n",
       "      <td>CABP7</td>\n",
       "      <td>-3940.0</td>\n",
       "      <td>cg27665648</td>\n",
       "      <td>0.680242</td>\n",
       "      <td>0.683700</td>\n",
       "      <td>0.662226</td>\n",
       "      <td>0.711880</td>\n",
       "      <td>0.699262</td>\n",
       "      <td>...</td>\n",
       "      <td>0.606769</td>\n",
       "      <td>0.694081</td>\n",
       "      <td>0.700518</td>\n",
       "      <td>0.710467</td>\n",
       "      <td>0.697145</td>\n",
       "      <td>0.675443</td>\n",
       "      <td>0.671625</td>\n",
       "      <td>0.682511</td>\n",
       "      <td>0.718816</td>\n",
       "      <td>Upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420120</th>\n",
       "      <td>ch.22.439136F</td>\n",
       "      <td>32113276</td>\n",
       "      <td>PRR14L</td>\n",
       "      <td>1202.0</td>\n",
       "      <td>ch.22.439136F</td>\n",
       "      <td>0.066196</td>\n",
       "      <td>0.074446</td>\n",
       "      <td>0.083344</td>\n",
       "      <td>0.051879</td>\n",
       "      <td>0.046465</td>\n",
       "      <td>...</td>\n",
       "      <td>0.059595</td>\n",
       "      <td>0.075262</td>\n",
       "      <td>0.083866</td>\n",
       "      <td>0.065719</td>\n",
       "      <td>0.092159</td>\n",
       "      <td>0.064988</td>\n",
       "      <td>0.074380</td>\n",
       "      <td>0.047593</td>\n",
       "      <td>0.058276</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249506 rows × 746 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ID Closest_TSS Closest_TSS_gene_name  Distance_closest_TSS  \\\n",
       "3          cg00001349   166958518                  MAEL                 -79.0   \n",
       "5          cg00001446    43833698                ELOVL1                2658.0   \n",
       "7          cg00001583   200011716                 NR5A2                  70.0   \n",
       "8          cg00001593   170489886              AK096329                -547.0   \n",
       "9          cg00002028    20959947                 PINK1                  63.0   \n",
       "...               ...         ...                   ...                   ...   \n",
       "420108     cg27652464    48885287               FAM19A5                1488.0   \n",
       "420111     cg27657537    20862338                 MED15                1424.0   \n",
       "420112     cg27662611    38599026                  MAFF                 -45.0   \n",
       "420113     cg27665648    30116343                 CABP7               -3940.0   \n",
       "420120  ch.22.439136F    32113276                PRR14L                1202.0   \n",
       "\n",
       "             TargetID  TBI-AUTO73325-PT-3149   PT-BZHL   PT-BZCH   PT-BY9H  \\\n",
       "3          cg00001349               0.865488  0.919570  0.882256  0.902912   \n",
       "5          cg00001446               0.842025  0.831457  0.835075  0.834157   \n",
       "7          cg00001583               0.099888  0.076240  0.075861  0.101837   \n",
       "8          cg00001593               0.918081  0.955305  0.896318  0.941423   \n",
       "9          cg00002028               0.026821  0.035839  0.020016  0.014101   \n",
       "...               ...                    ...       ...       ...       ...   \n",
       "420108     cg27652464               0.077791  0.081391  0.066050  0.076101   \n",
       "420111     cg27657537               0.246730  0.273854  0.250933  0.297172   \n",
       "420112     cg27662611               0.091869  0.098040  0.066290  0.094694   \n",
       "420113     cg27665648               0.680242  0.683700  0.662226  0.711880   \n",
       "420120  ch.22.439136F               0.066196  0.074446  0.083344  0.051879   \n",
       "\n",
       "        TBI-AUTO73307-PT-314I  ...  TBI-AUTO73291-PT-35OD   PT-BZD7   PT-BZG8  \\\n",
       "3                    0.907610  ...               0.900684  0.906257  0.756173   \n",
       "5                    0.839395  ...               0.847344  0.825179  0.887574   \n",
       "7                    0.071947  ...               0.084641  0.092234  0.066651   \n",
       "8                    0.940628  ...               0.918949  0.929937  0.952355   \n",
       "9                    0.031390  ...               0.021462  0.033637  0.024296   \n",
       "...                       ...  ...                    ...       ...       ...   \n",
       "420108               0.081930  ...               0.082855  0.078940  0.033035   \n",
       "420111               0.256656  ...               0.296850  0.295459  0.266729   \n",
       "420112               0.069012  ...               0.080595  0.067712  0.072670   \n",
       "420113               0.699262  ...               0.606769  0.694081  0.700518   \n",
       "420120               0.046465  ...               0.059595  0.075262  0.083866   \n",
       "\n",
       "         PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  TBI-AUTO73257-PT-35OE  \\\n",
       "3       0.948341  0.904996  0.927201  0.856637               0.917827   \n",
       "5       0.881636  0.852642  0.833056  0.842108               0.841747   \n",
       "7       0.040554  0.059607  0.116837  0.069391               0.061016   \n",
       "8       0.968486  0.972349  0.944855  0.924879               0.885508   \n",
       "9       0.033715  0.019535  0.011352  0.020888               0.035047   \n",
       "...          ...       ...       ...       ...                    ...   \n",
       "420108  0.029362  0.041915  0.059740  0.060660               0.040806   \n",
       "420111  0.301941  0.265106  0.263730  0.268427               0.252280   \n",
       "420112  0.069177  0.084736  0.071235  0.072670               0.083982   \n",
       "420113  0.710467  0.697145  0.675443  0.671625               0.682511   \n",
       "420120  0.065719  0.092159  0.064988  0.074380               0.047593   \n",
       "\n",
       "         PT-BZD3             Region  \n",
       "3       0.943320  Proximal Promoter  \n",
       "5       0.833336         Downstream  \n",
       "7       0.035535         Downstream  \n",
       "8       0.939249    Distal Promoter  \n",
       "9       0.026995         Downstream  \n",
       "...          ...                ...  \n",
       "420108  0.044582         Downstream  \n",
       "420111  0.243976         Downstream  \n",
       "420112  0.062013      Core Promoter  \n",
       "420113  0.718816           Upstream  \n",
       "420120  0.058276         Downstream  \n",
       "\n",
       "[249506 rows x 746 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylation_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(                   ID Closest_TSS Closest_TSS_gene_name  Distance_closest_TSS  \\\n",
       " 3          cg00001349   166958518                  MAEL                 -79.0   \n",
       " 5          cg00001446    43833698                ELOVL1                2658.0   \n",
       " 7          cg00001583   200011716                 NR5A2                  70.0   \n",
       " 8          cg00001593   170489886              AK096329                -547.0   \n",
       " 9          cg00002028    20959947                 PINK1                  63.0   \n",
       " ...               ...         ...                   ...                   ...   \n",
       " 420108     cg27652464    48885287               FAM19A5                1488.0   \n",
       " 420111     cg27657537    20862338                 MED15                1424.0   \n",
       " 420112     cg27662611    38599026                  MAFF                 -45.0   \n",
       " 420113     cg27665648    30116343                 CABP7               -3940.0   \n",
       " 420120  ch.22.439136F    32113276                PRR14L                1202.0   \n",
       " \n",
       "              TargetID  TBI-AUTO73325-PT-3149   PT-BZHL   PT-BZCH   PT-BY9H  \\\n",
       " 3          cg00001349               0.865488  0.919570  0.882256  0.902912   \n",
       " 5          cg00001446               0.842025  0.831457  0.835075  0.834157   \n",
       " 7          cg00001583               0.099888  0.076240  0.075861  0.101837   \n",
       " 8          cg00001593               0.918081  0.955305  0.896318  0.941423   \n",
       " 9          cg00002028               0.026821  0.035839  0.020016  0.014101   \n",
       " ...               ...                    ...       ...       ...       ...   \n",
       " 420108     cg27652464               0.077791  0.081391  0.066050  0.076101   \n",
       " 420111     cg27657537               0.246730  0.273854  0.250933  0.297172   \n",
       " 420112     cg27662611               0.091869  0.098040  0.066290  0.094694   \n",
       " 420113     cg27665648               0.680242  0.683700  0.662226  0.711880   \n",
       " 420120  ch.22.439136F               0.066196  0.074446  0.083344  0.051879   \n",
       " \n",
       "         TBI-AUTO73307-PT-314I  ...  TBI-AUTO73291-PT-35OD   PT-BZD7   PT-BZG8  \\\n",
       " 3                    0.907610  ...               0.900684  0.906257  0.756173   \n",
       " 5                    0.839395  ...               0.847344  0.825179  0.887574   \n",
       " 7                    0.071947  ...               0.084641  0.092234  0.066651   \n",
       " 8                    0.940628  ...               0.918949  0.929937  0.952355   \n",
       " 9                    0.031390  ...               0.021462  0.033637  0.024296   \n",
       " ...                       ...  ...                    ...       ...       ...   \n",
       " 420108               0.081930  ...               0.082855  0.078940  0.033035   \n",
       " 420111               0.256656  ...               0.296850  0.295459  0.266729   \n",
       " 420112               0.069012  ...               0.080595  0.067712  0.072670   \n",
       " 420113               0.699262  ...               0.606769  0.694081  0.700518   \n",
       " 420120               0.046465  ...               0.059595  0.075262  0.083866   \n",
       " \n",
       "          PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  TBI-AUTO73257-PT-35OE  \\\n",
       " 3       0.948341  0.904996  0.927201  0.856637               0.917827   \n",
       " 5       0.881636  0.852642  0.833056  0.842108               0.841747   \n",
       " 7       0.040554  0.059607  0.116837  0.069391               0.061016   \n",
       " 8       0.968486  0.972349  0.944855  0.924879               0.885508   \n",
       " 9       0.033715  0.019535  0.011352  0.020888               0.035047   \n",
       " ...          ...       ...       ...       ...                    ...   \n",
       " 420108  0.029362  0.041915  0.059740  0.060660               0.040806   \n",
       " 420111  0.301941  0.265106  0.263730  0.268427               0.252280   \n",
       " 420112  0.069177  0.084736  0.071235  0.072670               0.083982   \n",
       " 420113  0.710467  0.697145  0.675443  0.671625               0.682511   \n",
       " 420120  0.065719  0.092159  0.064988  0.074380               0.047593   \n",
       " \n",
       "          PT-BZD3             Region  \n",
       " 3       0.943320  Proximal Promoter  \n",
       " 5       0.833336         Downstream  \n",
       " 7       0.035535         Downstream  \n",
       " 8       0.939249    Distal Promoter  \n",
       " 9       0.026995         Downstream  \n",
       " ...          ...                ...  \n",
       " 420108  0.044582         Downstream  \n",
       " 420111  0.243976         Downstream  \n",
       " 420112  0.062013      Core Promoter  \n",
       " 420113  0.718816           Upstream  \n",
       " 420120  0.058276         Downstream  \n",
       " \n",
       " [249506 rows x 746 columns],\n",
       " 65)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "original_gene_names = methylation_merged_df['Closest_TSS_gene_name'].copy()\n",
    "\n",
    "# Use a regular expression to remove the period and any characters following it\n",
    "methylation_merged_df['Closest_TSS_gene_name'] = methylation_merged_df['Closest_TSS_gene_name'].str.replace(r'\\..*', '', regex=True)\n",
    "\n",
    "# Determine how many rows were changed by comparing the new values to the original ones\n",
    "rows_changed = (original_gene_names != methylation_merged_df['Closest_TSS_gene_name']).sum()\n",
    "\n",
    "# Output the updated DataFrame and the number of rows that were changed\n",
    "methylation_merged_df, rows_changed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Delete the 'sample' column\n",
    "methylation_merged_df = methylation_merged_df.drop('TargetID', axis=1)\n",
    "# Delete the 'ID' column\n",
    "methylation_merged_df = methylation_merged_df.drop('ID', axis=1)\n",
    "# Delete the 'Distance_closest_TSS' column\n",
    "methylation_merged_df = methylation_merged_df.drop('Distance_closest_TSS', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>Closest_TSS_gene_name</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "      <th>Region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>166958518</td>\n",
       "      <td>MAEL</td>\n",
       "      <td>0.865488</td>\n",
       "      <td>0.919570</td>\n",
       "      <td>0.882256</td>\n",
       "      <td>0.902912</td>\n",
       "      <td>0.907610</td>\n",
       "      <td>0.897496</td>\n",
       "      <td>0.939212</td>\n",
       "      <td>0.919514</td>\n",
       "      <td>...</td>\n",
       "      <td>0.900684</td>\n",
       "      <td>0.906257</td>\n",
       "      <td>0.756173</td>\n",
       "      <td>0.948341</td>\n",
       "      <td>0.904996</td>\n",
       "      <td>0.927201</td>\n",
       "      <td>0.856637</td>\n",
       "      <td>0.917827</td>\n",
       "      <td>0.943320</td>\n",
       "      <td>Proximal Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>43833698</td>\n",
       "      <td>ELOVL1</td>\n",
       "      <td>0.842025</td>\n",
       "      <td>0.831457</td>\n",
       "      <td>0.835075</td>\n",
       "      <td>0.834157</td>\n",
       "      <td>0.839395</td>\n",
       "      <td>0.842799</td>\n",
       "      <td>0.850811</td>\n",
       "      <td>0.842074</td>\n",
       "      <td>...</td>\n",
       "      <td>0.847344</td>\n",
       "      <td>0.825179</td>\n",
       "      <td>0.887574</td>\n",
       "      <td>0.881636</td>\n",
       "      <td>0.852642</td>\n",
       "      <td>0.833056</td>\n",
       "      <td>0.842108</td>\n",
       "      <td>0.841747</td>\n",
       "      <td>0.833336</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>200011716</td>\n",
       "      <td>NR5A2</td>\n",
       "      <td>0.099888</td>\n",
       "      <td>0.076240</td>\n",
       "      <td>0.075861</td>\n",
       "      <td>0.101837</td>\n",
       "      <td>0.071947</td>\n",
       "      <td>0.036712</td>\n",
       "      <td>0.041233</td>\n",
       "      <td>0.083998</td>\n",
       "      <td>...</td>\n",
       "      <td>0.084641</td>\n",
       "      <td>0.092234</td>\n",
       "      <td>0.066651</td>\n",
       "      <td>0.040554</td>\n",
       "      <td>0.059607</td>\n",
       "      <td>0.116837</td>\n",
       "      <td>0.069391</td>\n",
       "      <td>0.061016</td>\n",
       "      <td>0.035535</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>170489886</td>\n",
       "      <td>AK096329</td>\n",
       "      <td>0.918081</td>\n",
       "      <td>0.955305</td>\n",
       "      <td>0.896318</td>\n",
       "      <td>0.941423</td>\n",
       "      <td>0.940628</td>\n",
       "      <td>0.930346</td>\n",
       "      <td>0.944459</td>\n",
       "      <td>0.977162</td>\n",
       "      <td>...</td>\n",
       "      <td>0.918949</td>\n",
       "      <td>0.929937</td>\n",
       "      <td>0.952355</td>\n",
       "      <td>0.968486</td>\n",
       "      <td>0.972349</td>\n",
       "      <td>0.944855</td>\n",
       "      <td>0.924879</td>\n",
       "      <td>0.885508</td>\n",
       "      <td>0.939249</td>\n",
       "      <td>Distal Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>20959947</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>0.026821</td>\n",
       "      <td>0.035839</td>\n",
       "      <td>0.020016</td>\n",
       "      <td>0.014101</td>\n",
       "      <td>0.031390</td>\n",
       "      <td>0.022777</td>\n",
       "      <td>0.031833</td>\n",
       "      <td>0.014578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.021462</td>\n",
       "      <td>0.033637</td>\n",
       "      <td>0.024296</td>\n",
       "      <td>0.033715</td>\n",
       "      <td>0.019535</td>\n",
       "      <td>0.011352</td>\n",
       "      <td>0.020888</td>\n",
       "      <td>0.035047</td>\n",
       "      <td>0.026995</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420108</th>\n",
       "      <td>48885287</td>\n",
       "      <td>FAM19A5</td>\n",
       "      <td>0.077791</td>\n",
       "      <td>0.081391</td>\n",
       "      <td>0.066050</td>\n",
       "      <td>0.076101</td>\n",
       "      <td>0.081930</td>\n",
       "      <td>0.035654</td>\n",
       "      <td>0.070314</td>\n",
       "      <td>0.065970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.082855</td>\n",
       "      <td>0.078940</td>\n",
       "      <td>0.033035</td>\n",
       "      <td>0.029362</td>\n",
       "      <td>0.041915</td>\n",
       "      <td>0.059740</td>\n",
       "      <td>0.060660</td>\n",
       "      <td>0.040806</td>\n",
       "      <td>0.044582</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420111</th>\n",
       "      <td>20862338</td>\n",
       "      <td>MED15</td>\n",
       "      <td>0.246730</td>\n",
       "      <td>0.273854</td>\n",
       "      <td>0.250933</td>\n",
       "      <td>0.297172</td>\n",
       "      <td>0.256656</td>\n",
       "      <td>0.273096</td>\n",
       "      <td>0.278133</td>\n",
       "      <td>0.267989</td>\n",
       "      <td>...</td>\n",
       "      <td>0.296850</td>\n",
       "      <td>0.295459</td>\n",
       "      <td>0.266729</td>\n",
       "      <td>0.301941</td>\n",
       "      <td>0.265106</td>\n",
       "      <td>0.263730</td>\n",
       "      <td>0.268427</td>\n",
       "      <td>0.252280</td>\n",
       "      <td>0.243976</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420112</th>\n",
       "      <td>38599026</td>\n",
       "      <td>MAFF</td>\n",
       "      <td>0.091869</td>\n",
       "      <td>0.098040</td>\n",
       "      <td>0.066290</td>\n",
       "      <td>0.094694</td>\n",
       "      <td>0.069012</td>\n",
       "      <td>0.100561</td>\n",
       "      <td>0.103787</td>\n",
       "      <td>0.059369</td>\n",
       "      <td>...</td>\n",
       "      <td>0.080595</td>\n",
       "      <td>0.067712</td>\n",
       "      <td>0.072670</td>\n",
       "      <td>0.069177</td>\n",
       "      <td>0.084736</td>\n",
       "      <td>0.071235</td>\n",
       "      <td>0.072670</td>\n",
       "      <td>0.083982</td>\n",
       "      <td>0.062013</td>\n",
       "      <td>Core Promoter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420113</th>\n",
       "      <td>30116343</td>\n",
       "      <td>CABP7</td>\n",
       "      <td>0.680242</td>\n",
       "      <td>0.683700</td>\n",
       "      <td>0.662226</td>\n",
       "      <td>0.711880</td>\n",
       "      <td>0.699262</td>\n",
       "      <td>0.738775</td>\n",
       "      <td>0.720822</td>\n",
       "      <td>0.667222</td>\n",
       "      <td>...</td>\n",
       "      <td>0.606769</td>\n",
       "      <td>0.694081</td>\n",
       "      <td>0.700518</td>\n",
       "      <td>0.710467</td>\n",
       "      <td>0.697145</td>\n",
       "      <td>0.675443</td>\n",
       "      <td>0.671625</td>\n",
       "      <td>0.682511</td>\n",
       "      <td>0.718816</td>\n",
       "      <td>Upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420120</th>\n",
       "      <td>32113276</td>\n",
       "      <td>PRR14L</td>\n",
       "      <td>0.066196</td>\n",
       "      <td>0.074446</td>\n",
       "      <td>0.083344</td>\n",
       "      <td>0.051879</td>\n",
       "      <td>0.046465</td>\n",
       "      <td>0.075939</td>\n",
       "      <td>0.091995</td>\n",
       "      <td>0.039107</td>\n",
       "      <td>...</td>\n",
       "      <td>0.059595</td>\n",
       "      <td>0.075262</td>\n",
       "      <td>0.083866</td>\n",
       "      <td>0.065719</td>\n",
       "      <td>0.092159</td>\n",
       "      <td>0.064988</td>\n",
       "      <td>0.074380</td>\n",
       "      <td>0.047593</td>\n",
       "      <td>0.058276</td>\n",
       "      <td>Downstream</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249506 rows × 743 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Closest_TSS Closest_TSS_gene_name  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "3        166958518                  MAEL               0.865488  0.919570   \n",
       "5         43833698                ELOVL1               0.842025  0.831457   \n",
       "7        200011716                 NR5A2               0.099888  0.076240   \n",
       "8        170489886              AK096329               0.918081  0.955305   \n",
       "9         20959947                 PINK1               0.026821  0.035839   \n",
       "...            ...                   ...                    ...       ...   \n",
       "420108    48885287               FAM19A5               0.077791  0.081391   \n",
       "420111    20862338                 MED15               0.246730  0.273854   \n",
       "420112    38599026                  MAFF               0.091869  0.098040   \n",
       "420113    30116343                 CABP7               0.680242  0.683700   \n",
       "420120    32113276                PRR14L               0.066196  0.074446   \n",
       "\n",
       "         PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "3       0.882256  0.902912               0.907610               0.897496   \n",
       "5       0.835075  0.834157               0.839395               0.842799   \n",
       "7       0.075861  0.101837               0.071947               0.036712   \n",
       "8       0.896318  0.941423               0.940628               0.930346   \n",
       "9       0.020016  0.014101               0.031390               0.022777   \n",
       "...          ...       ...                    ...                    ...   \n",
       "420108  0.066050  0.076101               0.081930               0.035654   \n",
       "420111  0.250933  0.297172               0.256656               0.273096   \n",
       "420112  0.066290  0.094694               0.069012               0.100561   \n",
       "420113  0.662226  0.711880               0.699262               0.738775   \n",
       "420120  0.083344  0.051879               0.046465               0.075939   \n",
       "\n",
       "         PT-BZI5   PT-BZ1A  ...  TBI-AUTO73291-PT-35OD   PT-BZD7   PT-BZG8  \\\n",
       "3       0.939212  0.919514  ...               0.900684  0.906257  0.756173   \n",
       "5       0.850811  0.842074  ...               0.847344  0.825179  0.887574   \n",
       "7       0.041233  0.083998  ...               0.084641  0.092234  0.066651   \n",
       "8       0.944459  0.977162  ...               0.918949  0.929937  0.952355   \n",
       "9       0.031833  0.014578  ...               0.021462  0.033637  0.024296   \n",
       "...          ...       ...  ...                    ...       ...       ...   \n",
       "420108  0.070314  0.065970  ...               0.082855  0.078940  0.033035   \n",
       "420111  0.278133  0.267989  ...               0.296850  0.295459  0.266729   \n",
       "420112  0.103787  0.059369  ...               0.080595  0.067712  0.072670   \n",
       "420113  0.720822  0.667222  ...               0.606769  0.694081  0.700518   \n",
       "420120  0.091995  0.039107  ...               0.059595  0.075262  0.083866   \n",
       "\n",
       "         PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  TBI-AUTO73257-PT-35OE  \\\n",
       "3       0.948341  0.904996  0.927201  0.856637               0.917827   \n",
       "5       0.881636  0.852642  0.833056  0.842108               0.841747   \n",
       "7       0.040554  0.059607  0.116837  0.069391               0.061016   \n",
       "8       0.968486  0.972349  0.944855  0.924879               0.885508   \n",
       "9       0.033715  0.019535  0.011352  0.020888               0.035047   \n",
       "...          ...       ...       ...       ...                    ...   \n",
       "420108  0.029362  0.041915  0.059740  0.060660               0.040806   \n",
       "420111  0.301941  0.265106  0.263730  0.268427               0.252280   \n",
       "420112  0.069177  0.084736  0.071235  0.072670               0.083982   \n",
       "420113  0.710467  0.697145  0.675443  0.671625               0.682511   \n",
       "420120  0.065719  0.092159  0.064988  0.074380               0.047593   \n",
       "\n",
       "         PT-BZD3             Region  \n",
       "3       0.943320  Proximal Promoter  \n",
       "5       0.833336         Downstream  \n",
       "7       0.035535         Downstream  \n",
       "8       0.939249    Distal Promoter  \n",
       "9       0.026995         Downstream  \n",
       "...          ...                ...  \n",
       "420108  0.044582         Downstream  \n",
       "420111  0.243976         Downstream  \n",
       "420112  0.062013      Core Promoter  \n",
       "420113  0.718816           Upstream  \n",
       "420120  0.058276         Downstream  \n",
       "\n",
       "[249506 rows x 743 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylation_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylation_merged_df['Closest_TSS'] = methylation_merged_df['Closest_TSS'].astype(int)\n",
    "methylation_merged_df['Closest_TSS_gene_name'] = methylation_merged_df['Closest_TSS_gene_name'].astype(str)\n",
    "methylation_merged_df['Region'] = methylation_merged_df['Region'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Closest_TSS               int32\n",
      "Closest_TSS_gene_name    object\n",
      "Region                   object\n",
      "dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(methylation_merged_df[['Closest_TSS', 'Closest_TSS_gene_name', 'Region']].dtypes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.3 Calculate the average methylation value of five regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Proximal Promoter', 'Downstream', 'Distal Promoter',\n",
       "       'Core Promoter', 'Upstream'], dtype=object)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# obtain all regions\n",
    "regions = methylation_merged_df['Region'].unique()\n",
    "regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of Proximal Promoter: (15812, 742)\n",
      "Shape of Downstream: (20577, 742)\n",
      "Shape of Distal Promoter: (18238, 742)\n",
      "Shape of Core Promoter: (9871, 742)\n",
      "Shape of Upstream: (7128, 742)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Initialize empty DataFrames for each region\n",
    "Upstream_df = pd.DataFrame()\n",
    "Distal_Promoter_df = pd.DataFrame()\n",
    "Proximal_Promoter_df = pd.DataFrame()\n",
    "Core_Promoter_df = pd.DataFrame()\n",
    "Downstream_df = pd.DataFrame()\n",
    "\n",
    "# Operate on each region\n",
    "for region in regions:\n",
    "    # Get all data for this region\n",
    "    region_data = methylation_merged_df[methylation_merged_df['Region'] == region]\n",
    "    \n",
    "    # Group and calculate the average for each (TSS, Region) combination\n",
    "    grouped = region_data.groupby(['Closest_TSS_gene_name', 'Region'], as_index=False).mean()\n",
    "    \n",
    "    # Since we split the data into different files based on Region, we can delete this column\n",
    "    grouped = grouped.drop(columns=['Region'])\n",
    "    \n",
    "    # Print the shape of the grouped data\n",
    "    print(f\"Shape of {region}: {grouped.shape}\")\n",
    "    \n",
    "    # Assign the grouped data to the respective DataFrame\n",
    "    if region == 'Upstream':\n",
    "        Upstream_df = grouped\n",
    "    elif region == 'Distal Promoter':\n",
    "        Distal_Promoter_df = grouped\n",
    "    elif region == 'Proximal Promoter':\n",
    "        Proximal_Promoter_df = grouped\n",
    "    elif region == 'Core Promoter':\n",
    "        Core_Promoter_df = grouped\n",
    "    elif region == 'Downstream':\n",
    "        Downstream_df = grouped\n",
    "\n",
    "    # Optionally, save the data for this region to a new csv file\n",
    "    # grouped.to_csv(f\"{region}_averaged_tss_data.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.4 Unify gene and TSS for five methylation value files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# from methylation files above DataFrame \n",
    "dfs = [Upstream_df, Distal_Promoter_df, Proximal_Promoter_df, Core_Promoter_df, Downstream_df]\n",
    "\n",
    "# merge those files to find all combos \n",
    "all_genes_tss = pd.concat(dfs)['Closest_TSS_gene_name'].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0            1/2-SBSRNA4\n",
       "1                  40969\n",
       "2                  40970\n",
       "3                  40971\n",
       "4                  40972\n",
       "              ...       \n",
       "20459             ZNF791\n",
       "20497             ZNF862\n",
       "20506              ZNF91\n",
       "20528             ZRANB2\n",
       "20572    hsa-miR-3194-3p\n",
       "Name: Closest_TSS_gene_name, Length: 23280, dtype: object"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes_tss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of Upstream_df: (23280, 742)\n",
      "Shape of Distal_Promoter_df: (23280, 742)\n",
      "Shape of Proximal_Promoter_df: (23280, 742)\n",
      "Shape of Core_Promoter_df: (23280, 742)\n",
      "Shape of Downstream_df: (23280, 742)\n"
     ]
    }
   ],
   "source": [
    "# Merge unique combinations back into each DataFrame and fill NaN values with 0\n",
    "# Upstream\n",
    "Upstream_df = pd.merge(all_genes_tss, Upstream_df, on=['Closest_TSS_gene_name'], how='outer').fillna(0)\n",
    "print(f\"Shape of Upstream_df: {Upstream_df.shape}\")\n",
    "\n",
    "# Distal Promoter\n",
    "Distal_Promoter_df = pd.merge(all_genes_tss, Distal_Promoter_df, on=['Closest_TSS_gene_name'], how='outer').fillna(0)\n",
    "print(f\"Shape of Distal_Promoter_df: {Distal_Promoter_df.shape}\")\n",
    "\n",
    "# Proximal Promoter\n",
    "Proximal_Promoter_df = pd.merge(all_genes_tss, Proximal_Promoter_df, on=['Closest_TSS_gene_name'], how='outer').fillna(0)\n",
    "print(f\"Shape of Proximal_Promoter_df: {Proximal_Promoter_df.shape}\")\n",
    "\n",
    "# Core Promoter\n",
    "Core_Promoter_df = pd.merge(all_genes_tss, Core_Promoter_df, on=['Closest_TSS_gene_name'], how='outer').fillna(0)\n",
    "print(f\"Shape of Core_Promoter_df: {Core_Promoter_df.shape}\")\n",
    "\n",
    "# Downstream\n",
    "Downstream_df = pd.merge(all_genes_tss, Downstream_df, on=['Closest_TSS_gene_name'], how='outer').fillna(0)\n",
    "print(f\"Shape of Downstream_df: {Downstream_df.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1/2-SBSRNA4</td>\n",
       "      <td>110354972.0</td>\n",
       "      <td>0.931928</td>\n",
       "      <td>0.922927</td>\n",
       "      <td>0.921658</td>\n",
       "      <td>0.934680</td>\n",
       "      <td>0.950280</td>\n",
       "      <td>0.915567</td>\n",
       "      <td>0.895290</td>\n",
       "      <td>0.942072</td>\n",
       "      <td>...</td>\n",
       "      <td>0.929474</td>\n",
       "      <td>0.932108</td>\n",
       "      <td>0.909341</td>\n",
       "      <td>0.921341</td>\n",
       "      <td>0.930176</td>\n",
       "      <td>0.916491</td>\n",
       "      <td>0.913546</td>\n",
       "      <td>0.936391</td>\n",
       "      <td>0.905927</td>\n",
       "      <td>0.942917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40969</td>\n",
       "      <td>220960038.0</td>\n",
       "      <td>0.718230</td>\n",
       "      <td>0.698030</td>\n",
       "      <td>0.712573</td>\n",
       "      <td>0.705048</td>\n",
       "      <td>0.734309</td>\n",
       "      <td>0.771367</td>\n",
       "      <td>0.782897</td>\n",
       "      <td>0.704986</td>\n",
       "      <td>...</td>\n",
       "      <td>0.766211</td>\n",
       "      <td>0.675562</td>\n",
       "      <td>0.720371</td>\n",
       "      <td>0.689035</td>\n",
       "      <td>0.618610</td>\n",
       "      <td>0.764157</td>\n",
       "      <td>0.576607</td>\n",
       "      <td>0.754224</td>\n",
       "      <td>0.745078</td>\n",
       "      <td>0.710471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>40970</td>\n",
       "      <td>220921675.0</td>\n",
       "      <td>0.868322</td>\n",
       "      <td>0.845299</td>\n",
       "      <td>0.849537</td>\n",
       "      <td>0.892048</td>\n",
       "      <td>0.881078</td>\n",
       "      <td>0.867949</td>\n",
       "      <td>0.886519</td>\n",
       "      <td>0.902767</td>\n",
       "      <td>...</td>\n",
       "      <td>0.874877</td>\n",
       "      <td>0.851768</td>\n",
       "      <td>0.870165</td>\n",
       "      <td>0.878390</td>\n",
       "      <td>0.893037</td>\n",
       "      <td>0.901645</td>\n",
       "      <td>0.876906</td>\n",
       "      <td>0.870745</td>\n",
       "      <td>0.852633</td>\n",
       "      <td>0.877366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40971</td>\n",
       "      <td>126366439.0</td>\n",
       "      <td>0.914631</td>\n",
       "      <td>0.890200</td>\n",
       "      <td>0.893712</td>\n",
       "      <td>0.904816</td>\n",
       "      <td>0.934809</td>\n",
       "      <td>0.901929</td>\n",
       "      <td>0.918363</td>\n",
       "      <td>0.889949</td>\n",
       "      <td>...</td>\n",
       "      <td>0.907466</td>\n",
       "      <td>0.905140</td>\n",
       "      <td>0.898941</td>\n",
       "      <td>0.857999</td>\n",
       "      <td>0.960562</td>\n",
       "      <td>0.964220</td>\n",
       "      <td>0.896651</td>\n",
       "      <td>0.901460</td>\n",
       "      <td>0.907702</td>\n",
       "      <td>0.936499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>40972</td>\n",
       "      <td>217236749.0</td>\n",
       "      <td>0.689192</td>\n",
       "      <td>0.726004</td>\n",
       "      <td>0.691986</td>\n",
       "      <td>0.669473</td>\n",
       "      <td>0.690793</td>\n",
       "      <td>0.725609</td>\n",
       "      <td>0.711018</td>\n",
       "      <td>0.717188</td>\n",
       "      <td>...</td>\n",
       "      <td>0.822901</td>\n",
       "      <td>0.727588</td>\n",
       "      <td>0.706725</td>\n",
       "      <td>0.726323</td>\n",
       "      <td>0.695048</td>\n",
       "      <td>0.800269</td>\n",
       "      <td>0.685743</td>\n",
       "      <td>0.758808</td>\n",
       "      <td>0.726784</td>\n",
       "      <td>0.716149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23275</th>\n",
       "      <td>ZNF791</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23276</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>ZNF91</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23279</th>\n",
       "      <td>hsa-miR-3194-3p</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23280 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name  Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "0          1/2-SBSRNA4  110354972.0               0.931928  0.922927   \n",
       "1                40969  220960038.0               0.718230  0.698030   \n",
       "2                40970  220921675.0               0.868322  0.845299   \n",
       "3                40971  126366439.0               0.914631  0.890200   \n",
       "4                40972  217236749.0               0.689192  0.726004   \n",
       "...                ...          ...                    ...       ...   \n",
       "23275           ZNF791          0.0               0.000000  0.000000   \n",
       "23276           ZNF862          0.0               0.000000  0.000000   \n",
       "23277            ZNF91          0.0               0.000000  0.000000   \n",
       "23278           ZRANB2          0.0               0.000000  0.000000   \n",
       "23279  hsa-miR-3194-3p          0.0               0.000000  0.000000   \n",
       "\n",
       "        PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "0      0.921658  0.934680               0.950280               0.915567   \n",
       "1      0.712573  0.705048               0.734309               0.771367   \n",
       "2      0.849537  0.892048               0.881078               0.867949   \n",
       "3      0.893712  0.904816               0.934809               0.901929   \n",
       "4      0.691986  0.669473               0.690793               0.725609   \n",
       "...         ...       ...                    ...                    ...   \n",
       "23275  0.000000  0.000000               0.000000               0.000000   \n",
       "23276  0.000000  0.000000               0.000000               0.000000   \n",
       "23277  0.000000  0.000000               0.000000               0.000000   \n",
       "23278  0.000000  0.000000               0.000000               0.000000   \n",
       "23279  0.000000  0.000000               0.000000               0.000000   \n",
       "\n",
       "        PT-BZI5   PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD  \\\n",
       "0      0.895290  0.942072  ...               0.929474               0.932108   \n",
       "1      0.782897  0.704986  ...               0.766211               0.675562   \n",
       "2      0.886519  0.902767  ...               0.874877               0.851768   \n",
       "3      0.918363  0.889949  ...               0.907466               0.905140   \n",
       "4      0.711018  0.717188  ...               0.822901               0.727588   \n",
       "...         ...       ...  ...                    ...                    ...   \n",
       "23275  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23276  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23277  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23278  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23279  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "\n",
       "        PT-BZD7   PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0      0.909341  0.921341  0.930176  0.916491  0.913546  0.936391   \n",
       "1      0.720371  0.689035  0.618610  0.764157  0.576607  0.754224   \n",
       "2      0.870165  0.878390  0.893037  0.901645  0.876906  0.870745   \n",
       "3      0.898941  0.857999  0.960562  0.964220  0.896651  0.901460   \n",
       "4      0.706725  0.726323  0.695048  0.800269  0.685743  0.758808   \n",
       "...         ...       ...       ...       ...       ...       ...   \n",
       "23275  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23276  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23277  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23278  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23279  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "       TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                   0.905927  0.942917  \n",
       "1                   0.745078  0.710471  \n",
       "2                   0.852633  0.877366  \n",
       "3                   0.907702  0.936499  \n",
       "4                   0.726784  0.716149  \n",
       "...                      ...       ...  \n",
       "23275               0.000000  0.000000  \n",
       "23276               0.000000  0.000000  \n",
       "23277               0.000000  0.000000  \n",
       "23278               0.000000  0.000000  \n",
       "23279               0.000000  0.000000  \n",
       "\n",
       "[23280 rows x 742 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1/2-SBSRNA4</td>\n",
       "      <td>1.103550e+08</td>\n",
       "      <td>0.050086</td>\n",
       "      <td>0.046152</td>\n",
       "      <td>0.047936</td>\n",
       "      <td>0.047600</td>\n",
       "      <td>0.044529</td>\n",
       "      <td>0.042937</td>\n",
       "      <td>0.046128</td>\n",
       "      <td>0.036349</td>\n",
       "      <td>...</td>\n",
       "      <td>0.046718</td>\n",
       "      <td>0.050942</td>\n",
       "      <td>0.057558</td>\n",
       "      <td>0.042817</td>\n",
       "      <td>0.034843</td>\n",
       "      <td>0.049129</td>\n",
       "      <td>0.050484</td>\n",
       "      <td>0.048231</td>\n",
       "      <td>0.041461</td>\n",
       "      <td>0.035273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40969</td>\n",
       "      <td>1.823108e+08</td>\n",
       "      <td>0.392057</td>\n",
       "      <td>0.393938</td>\n",
       "      <td>0.374393</td>\n",
       "      <td>0.394217</td>\n",
       "      <td>0.387216</td>\n",
       "      <td>0.384722</td>\n",
       "      <td>0.399945</td>\n",
       "      <td>0.400160</td>\n",
       "      <td>...</td>\n",
       "      <td>0.387394</td>\n",
       "      <td>0.387438</td>\n",
       "      <td>0.388211</td>\n",
       "      <td>0.399827</td>\n",
       "      <td>0.391225</td>\n",
       "      <td>0.393465</td>\n",
       "      <td>0.399843</td>\n",
       "      <td>0.385534</td>\n",
       "      <td>0.397759</td>\n",
       "      <td>0.397958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>40970</td>\n",
       "      <td>1.359443e+08</td>\n",
       "      <td>0.310090</td>\n",
       "      <td>0.305279</td>\n",
       "      <td>0.298020</td>\n",
       "      <td>0.309443</td>\n",
       "      <td>0.307445</td>\n",
       "      <td>0.316317</td>\n",
       "      <td>0.325516</td>\n",
       "      <td>0.289902</td>\n",
       "      <td>...</td>\n",
       "      <td>0.314234</td>\n",
       "      <td>0.318619</td>\n",
       "      <td>0.322369</td>\n",
       "      <td>0.331648</td>\n",
       "      <td>0.313204</td>\n",
       "      <td>0.322797</td>\n",
       "      <td>0.317953</td>\n",
       "      <td>0.311885</td>\n",
       "      <td>0.323359</td>\n",
       "      <td>0.317395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40971</td>\n",
       "      <td>1.263664e+08</td>\n",
       "      <td>0.405201</td>\n",
       "      <td>0.427466</td>\n",
       "      <td>0.397301</td>\n",
       "      <td>0.469572</td>\n",
       "      <td>0.417824</td>\n",
       "      <td>0.425026</td>\n",
       "      <td>0.451523</td>\n",
       "      <td>0.430546</td>\n",
       "      <td>...</td>\n",
       "      <td>0.401051</td>\n",
       "      <td>0.387067</td>\n",
       "      <td>0.409743</td>\n",
       "      <td>0.477469</td>\n",
       "      <td>0.452182</td>\n",
       "      <td>0.450699</td>\n",
       "      <td>0.406353</td>\n",
       "      <td>0.405420</td>\n",
       "      <td>0.429750</td>\n",
       "      <td>0.438600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>40972</td>\n",
       "      <td>2.172367e+08</td>\n",
       "      <td>0.121688</td>\n",
       "      <td>0.126594</td>\n",
       "      <td>0.126133</td>\n",
       "      <td>0.134649</td>\n",
       "      <td>0.099313</td>\n",
       "      <td>0.099806</td>\n",
       "      <td>0.132725</td>\n",
       "      <td>0.135713</td>\n",
       "      <td>...</td>\n",
       "      <td>0.108077</td>\n",
       "      <td>0.124255</td>\n",
       "      <td>0.136012</td>\n",
       "      <td>0.122335</td>\n",
       "      <td>0.115328</td>\n",
       "      <td>0.124249</td>\n",
       "      <td>0.118947</td>\n",
       "      <td>0.132945</td>\n",
       "      <td>0.122179</td>\n",
       "      <td>0.122315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23275</th>\n",
       "      <td>ZNF791</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23276</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>ZNF91</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23279</th>\n",
       "      <td>hsa-miR-3194-3p</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23280 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name   Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "0          1/2-SBSRNA4  1.103550e+08               0.050086  0.046152   \n",
       "1                40969  1.823108e+08               0.392057  0.393938   \n",
       "2                40970  1.359443e+08               0.310090  0.305279   \n",
       "3                40971  1.263664e+08               0.405201  0.427466   \n",
       "4                40972  2.172367e+08               0.121688  0.126594   \n",
       "...                ...           ...                    ...       ...   \n",
       "23275           ZNF791  0.000000e+00               0.000000  0.000000   \n",
       "23276           ZNF862  0.000000e+00               0.000000  0.000000   \n",
       "23277            ZNF91  0.000000e+00               0.000000  0.000000   \n",
       "23278           ZRANB2  0.000000e+00               0.000000  0.000000   \n",
       "23279  hsa-miR-3194-3p  0.000000e+00               0.000000  0.000000   \n",
       "\n",
       "        PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "0      0.047936  0.047600               0.044529               0.042937   \n",
       "1      0.374393  0.394217               0.387216               0.384722   \n",
       "2      0.298020  0.309443               0.307445               0.316317   \n",
       "3      0.397301  0.469572               0.417824               0.425026   \n",
       "4      0.126133  0.134649               0.099313               0.099806   \n",
       "...         ...       ...                    ...                    ...   \n",
       "23275  0.000000  0.000000               0.000000               0.000000   \n",
       "23276  0.000000  0.000000               0.000000               0.000000   \n",
       "23277  0.000000  0.000000               0.000000               0.000000   \n",
       "23278  0.000000  0.000000               0.000000               0.000000   \n",
       "23279  0.000000  0.000000               0.000000               0.000000   \n",
       "\n",
       "        PT-BZI5   PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD  \\\n",
       "0      0.046128  0.036349  ...               0.046718               0.050942   \n",
       "1      0.399945  0.400160  ...               0.387394               0.387438   \n",
       "2      0.325516  0.289902  ...               0.314234               0.318619   \n",
       "3      0.451523  0.430546  ...               0.401051               0.387067   \n",
       "4      0.132725  0.135713  ...               0.108077               0.124255   \n",
       "...         ...       ...  ...                    ...                    ...   \n",
       "23275  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23276  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23277  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23278  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23279  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "\n",
       "        PT-BZD7   PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0      0.057558  0.042817  0.034843  0.049129  0.050484  0.048231   \n",
       "1      0.388211  0.399827  0.391225  0.393465  0.399843  0.385534   \n",
       "2      0.322369  0.331648  0.313204  0.322797  0.317953  0.311885   \n",
       "3      0.409743  0.477469  0.452182  0.450699  0.406353  0.405420   \n",
       "4      0.136012  0.122335  0.115328  0.124249  0.118947  0.132945   \n",
       "...         ...       ...       ...       ...       ...       ...   \n",
       "23275  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23276  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23277  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23278  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23279  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "       TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                   0.041461  0.035273  \n",
       "1                   0.397759  0.397958  \n",
       "2                   0.323359  0.317395  \n",
       "3                   0.429750  0.438600  \n",
       "4                   0.122179  0.122315  \n",
       "...                      ...       ...  \n",
       "23275               0.000000  0.000000  \n",
       "23276               0.000000  0.000000  \n",
       "23277               0.000000  0.000000  \n",
       "23278               0.000000  0.000000  \n",
       "23279               0.000000  0.000000  \n",
       "\n",
       "[23280 rows x 742 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1/2-SBSRNA4</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40969</td>\n",
       "      <td>1.745803e+08</td>\n",
       "      <td>0.062457</td>\n",
       "      <td>0.062992</td>\n",
       "      <td>0.060792</td>\n",
       "      <td>0.072635</td>\n",
       "      <td>0.053607</td>\n",
       "      <td>0.052409</td>\n",
       "      <td>0.078899</td>\n",
       "      <td>0.054821</td>\n",
       "      <td>...</td>\n",
       "      <td>0.056719</td>\n",
       "      <td>0.058685</td>\n",
       "      <td>0.062915</td>\n",
       "      <td>0.054353</td>\n",
       "      <td>0.047246</td>\n",
       "      <td>0.058168</td>\n",
       "      <td>0.051281</td>\n",
       "      <td>0.059223</td>\n",
       "      <td>0.063899</td>\n",
       "      <td>0.051598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>40970</td>\n",
       "      <td>1.146999e+08</td>\n",
       "      <td>0.019823</td>\n",
       "      <td>0.016219</td>\n",
       "      <td>0.025095</td>\n",
       "      <td>0.004104</td>\n",
       "      <td>0.018158</td>\n",
       "      <td>0.023213</td>\n",
       "      <td>0.011316</td>\n",
       "      <td>0.006909</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023464</td>\n",
       "      <td>0.018230</td>\n",
       "      <td>0.026458</td>\n",
       "      <td>0.016108</td>\n",
       "      <td>0.007415</td>\n",
       "      <td>0.018649</td>\n",
       "      <td>0.013601</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>0.016403</td>\n",
       "      <td>0.013799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40971</td>\n",
       "      <td>1.263664e+08</td>\n",
       "      <td>0.053635</td>\n",
       "      <td>0.043674</td>\n",
       "      <td>0.056735</td>\n",
       "      <td>0.039288</td>\n",
       "      <td>0.056034</td>\n",
       "      <td>0.049312</td>\n",
       "      <td>0.037346</td>\n",
       "      <td>0.035949</td>\n",
       "      <td>...</td>\n",
       "      <td>0.053628</td>\n",
       "      <td>0.046762</td>\n",
       "      <td>0.060741</td>\n",
       "      <td>0.037930</td>\n",
       "      <td>0.031851</td>\n",
       "      <td>0.047490</td>\n",
       "      <td>0.048975</td>\n",
       "      <td>0.049245</td>\n",
       "      <td>0.071294</td>\n",
       "      <td>0.042729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>40972</td>\n",
       "      <td>2.172367e+08</td>\n",
       "      <td>0.041339</td>\n",
       "      <td>0.040678</td>\n",
       "      <td>0.044696</td>\n",
       "      <td>0.046232</td>\n",
       "      <td>0.038569</td>\n",
       "      <td>0.040817</td>\n",
       "      <td>0.037882</td>\n",
       "      <td>0.040225</td>\n",
       "      <td>...</td>\n",
       "      <td>0.042183</td>\n",
       "      <td>0.044143</td>\n",
       "      <td>0.051431</td>\n",
       "      <td>0.036785</td>\n",
       "      <td>0.036056</td>\n",
       "      <td>0.040026</td>\n",
       "      <td>0.041975</td>\n",
       "      <td>0.051319</td>\n",
       "      <td>0.045344</td>\n",
       "      <td>0.036178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23275</th>\n",
       "      <td>ZNF791</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23276</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>ZNF91</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23279</th>\n",
       "      <td>hsa-miR-3194-3p</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23280 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name   Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "0          1/2-SBSRNA4  0.000000e+00               0.000000  0.000000   \n",
       "1                40969  1.745803e+08               0.062457  0.062992   \n",
       "2                40970  1.146999e+08               0.019823  0.016219   \n",
       "3                40971  1.263664e+08               0.053635  0.043674   \n",
       "4                40972  2.172367e+08               0.041339  0.040678   \n",
       "...                ...           ...                    ...       ...   \n",
       "23275           ZNF791  0.000000e+00               0.000000  0.000000   \n",
       "23276           ZNF862  0.000000e+00               0.000000  0.000000   \n",
       "23277            ZNF91  0.000000e+00               0.000000  0.000000   \n",
       "23278           ZRANB2  0.000000e+00               0.000000  0.000000   \n",
       "23279  hsa-miR-3194-3p  0.000000e+00               0.000000  0.000000   \n",
       "\n",
       "        PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "0      0.000000  0.000000               0.000000               0.000000   \n",
       "1      0.060792  0.072635               0.053607               0.052409   \n",
       "2      0.025095  0.004104               0.018158               0.023213   \n",
       "3      0.056735  0.039288               0.056034               0.049312   \n",
       "4      0.044696  0.046232               0.038569               0.040817   \n",
       "...         ...       ...                    ...                    ...   \n",
       "23275  0.000000  0.000000               0.000000               0.000000   \n",
       "23276  0.000000  0.000000               0.000000               0.000000   \n",
       "23277  0.000000  0.000000               0.000000               0.000000   \n",
       "23278  0.000000  0.000000               0.000000               0.000000   \n",
       "23279  0.000000  0.000000               0.000000               0.000000   \n",
       "\n",
       "        PT-BZI5   PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD  \\\n",
       "0      0.000000  0.000000  ...               0.000000               0.000000   \n",
       "1      0.078899  0.054821  ...               0.056719               0.058685   \n",
       "2      0.011316  0.006909  ...               0.023464               0.018230   \n",
       "3      0.037346  0.035949  ...               0.053628               0.046762   \n",
       "4      0.037882  0.040225  ...               0.042183               0.044143   \n",
       "...         ...       ...  ...                    ...                    ...   \n",
       "23275  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23276  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23277  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23278  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23279  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "\n",
       "        PT-BZD7   PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0      0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "1      0.062915  0.054353  0.047246  0.058168  0.051281  0.059223   \n",
       "2      0.026458  0.016108  0.007415  0.018649  0.013601  0.015024   \n",
       "3      0.060741  0.037930  0.031851  0.047490  0.048975  0.049245   \n",
       "4      0.051431  0.036785  0.036056  0.040026  0.041975  0.051319   \n",
       "...         ...       ...       ...       ...       ...       ...   \n",
       "23275  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23276  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23277  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23278  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23279  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "       TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                   0.000000  0.000000  \n",
       "1                   0.063899  0.051598  \n",
       "2                   0.016403  0.013799  \n",
       "3                   0.071294  0.042729  \n",
       "4                   0.045344  0.036178  \n",
       "...                      ...       ...  \n",
       "23275               0.000000  0.000000  \n",
       "23276               0.000000  0.000000  \n",
       "23277               0.000000  0.000000  \n",
       "23278               0.000000  0.000000  \n",
       "23279               0.000000  0.000000  \n",
       "\n",
       "[23280 rows x 742 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1/2-SBSRNA4</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40969</td>\n",
       "      <td>1.931322e+08</td>\n",
       "      <td>0.032889</td>\n",
       "      <td>0.022269</td>\n",
       "      <td>0.038779</td>\n",
       "      <td>0.027230</td>\n",
       "      <td>0.020263</td>\n",
       "      <td>0.024609</td>\n",
       "      <td>0.019103</td>\n",
       "      <td>0.027352</td>\n",
       "      <td>...</td>\n",
       "      <td>0.019140</td>\n",
       "      <td>0.020243</td>\n",
       "      <td>0.031096</td>\n",
       "      <td>0.014201</td>\n",
       "      <td>0.001955</td>\n",
       "      <td>0.021315</td>\n",
       "      <td>0.013171</td>\n",
       "      <td>0.026491</td>\n",
       "      <td>0.021192</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>40970</td>\n",
       "      <td>4.388543e+07</td>\n",
       "      <td>0.010026</td>\n",
       "      <td>0.005568</td>\n",
       "      <td>0.009481</td>\n",
       "      <td>0.001321</td>\n",
       "      <td>0.005410</td>\n",
       "      <td>0.012487</td>\n",
       "      <td>0.007074</td>\n",
       "      <td>0.007051</td>\n",
       "      <td>...</td>\n",
       "      <td>0.011333</td>\n",
       "      <td>0.009509</td>\n",
       "      <td>0.009776</td>\n",
       "      <td>0.007986</td>\n",
       "      <td>0.004716</td>\n",
       "      <td>0.003443</td>\n",
       "      <td>0.005855</td>\n",
       "      <td>0.007794</td>\n",
       "      <td>0.009484</td>\n",
       "      <td>0.008447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40971</td>\n",
       "      <td>1.263664e+08</td>\n",
       "      <td>0.002757</td>\n",
       "      <td>0.000874</td>\n",
       "      <td>0.005387</td>\n",
       "      <td>0.000785</td>\n",
       "      <td>0.000619</td>\n",
       "      <td>0.011755</td>\n",
       "      <td>0.007571</td>\n",
       "      <td>0.002990</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006142</td>\n",
       "      <td>0.004279</td>\n",
       "      <td>0.005550</td>\n",
       "      <td>0.011236</td>\n",
       "      <td>0.010861</td>\n",
       "      <td>0.005055</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008307</td>\n",
       "      <td>0.009602</td>\n",
       "      <td>0.003405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>40972</td>\n",
       "      <td>2.172367e+08</td>\n",
       "      <td>0.064524</td>\n",
       "      <td>0.056894</td>\n",
       "      <td>0.055694</td>\n",
       "      <td>0.057410</td>\n",
       "      <td>0.049283</td>\n",
       "      <td>0.048565</td>\n",
       "      <td>0.048002</td>\n",
       "      <td>0.048425</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049171</td>\n",
       "      <td>0.051556</td>\n",
       "      <td>0.055586</td>\n",
       "      <td>0.049027</td>\n",
       "      <td>0.026494</td>\n",
       "      <td>0.052469</td>\n",
       "      <td>0.056334</td>\n",
       "      <td>0.053705</td>\n",
       "      <td>0.049452</td>\n",
       "      <td>0.038607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23275</th>\n",
       "      <td>ZNF791</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23276</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>ZNF91</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23279</th>\n",
       "      <td>hsa-miR-3194-3p</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23280 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name   Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "0          1/2-SBSRNA4  0.000000e+00               0.000000  0.000000   \n",
       "1                40969  1.931322e+08               0.032889  0.022269   \n",
       "2                40970  4.388543e+07               0.010026  0.005568   \n",
       "3                40971  1.263664e+08               0.002757  0.000874   \n",
       "4                40972  2.172367e+08               0.064524  0.056894   \n",
       "...                ...           ...                    ...       ...   \n",
       "23275           ZNF791  0.000000e+00               0.000000  0.000000   \n",
       "23276           ZNF862  0.000000e+00               0.000000  0.000000   \n",
       "23277            ZNF91  0.000000e+00               0.000000  0.000000   \n",
       "23278           ZRANB2  0.000000e+00               0.000000  0.000000   \n",
       "23279  hsa-miR-3194-3p  0.000000e+00               0.000000  0.000000   \n",
       "\n",
       "        PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "0      0.000000  0.000000               0.000000               0.000000   \n",
       "1      0.038779  0.027230               0.020263               0.024609   \n",
       "2      0.009481  0.001321               0.005410               0.012487   \n",
       "3      0.005387  0.000785               0.000619               0.011755   \n",
       "4      0.055694  0.057410               0.049283               0.048565   \n",
       "...         ...       ...                    ...                    ...   \n",
       "23275  0.000000  0.000000               0.000000               0.000000   \n",
       "23276  0.000000  0.000000               0.000000               0.000000   \n",
       "23277  0.000000  0.000000               0.000000               0.000000   \n",
       "23278  0.000000  0.000000               0.000000               0.000000   \n",
       "23279  0.000000  0.000000               0.000000               0.000000   \n",
       "\n",
       "        PT-BZI5   PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD  \\\n",
       "0      0.000000  0.000000  ...               0.000000               0.000000   \n",
       "1      0.019103  0.027352  ...               0.019140               0.020243   \n",
       "2      0.007074  0.007051  ...               0.011333               0.009509   \n",
       "3      0.007571  0.002990  ...               0.006142               0.004279   \n",
       "4      0.048002  0.048425  ...               0.049171               0.051556   \n",
       "...         ...       ...  ...                    ...                    ...   \n",
       "23275  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23276  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23277  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23278  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23279  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "\n",
       "        PT-BZD7   PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0      0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "1      0.031096  0.014201  0.001955  0.021315  0.013171  0.026491   \n",
       "2      0.009776  0.007986  0.004716  0.003443  0.005855  0.007794   \n",
       "3      0.005550  0.011236  0.010861  0.005055  0.000000  0.008307   \n",
       "4      0.055586  0.049027  0.026494  0.052469  0.056334  0.053705   \n",
       "...         ...       ...       ...       ...       ...       ...   \n",
       "23275  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23276  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23277  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23278  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23279  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "       TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                   0.000000  0.000000  \n",
       "1                   0.021192  0.000000  \n",
       "2                   0.009484  0.008447  \n",
       "3                   0.009602  0.003405  \n",
       "4                   0.049452  0.038607  \n",
       "...                      ...       ...  \n",
       "23275               0.000000  0.000000  \n",
       "23276               0.000000  0.000000  \n",
       "23277               0.000000  0.000000  \n",
       "23278               0.000000  0.000000  \n",
       "23279               0.000000  0.000000  \n",
       "\n",
       "[23280 rows x 742 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1/2-SBSRNA4</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40969</td>\n",
       "      <td>1.928757e+08</td>\n",
       "      <td>0.455367</td>\n",
       "      <td>0.445996</td>\n",
       "      <td>0.431180</td>\n",
       "      <td>0.449385</td>\n",
       "      <td>0.443984</td>\n",
       "      <td>0.461588</td>\n",
       "      <td>0.450010</td>\n",
       "      <td>0.451255</td>\n",
       "      <td>...</td>\n",
       "      <td>0.451212</td>\n",
       "      <td>0.474548</td>\n",
       "      <td>0.450920</td>\n",
       "      <td>0.464083</td>\n",
       "      <td>0.455649</td>\n",
       "      <td>0.469531</td>\n",
       "      <td>0.459546</td>\n",
       "      <td>0.445075</td>\n",
       "      <td>0.440247</td>\n",
       "      <td>0.473129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>40970</td>\n",
       "      <td>1.855144e+08</td>\n",
       "      <td>0.053927</td>\n",
       "      <td>0.049527</td>\n",
       "      <td>0.048195</td>\n",
       "      <td>0.042055</td>\n",
       "      <td>0.051275</td>\n",
       "      <td>0.046544</td>\n",
       "      <td>0.043220</td>\n",
       "      <td>0.038363</td>\n",
       "      <td>...</td>\n",
       "      <td>0.047662</td>\n",
       "      <td>0.045282</td>\n",
       "      <td>0.048769</td>\n",
       "      <td>0.044007</td>\n",
       "      <td>0.044640</td>\n",
       "      <td>0.044352</td>\n",
       "      <td>0.052840</td>\n",
       "      <td>0.057617</td>\n",
       "      <td>0.054666</td>\n",
       "      <td>0.044574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40971</td>\n",
       "      <td>1.263664e+08</td>\n",
       "      <td>0.065878</td>\n",
       "      <td>0.071462</td>\n",
       "      <td>0.068046</td>\n",
       "      <td>0.064942</td>\n",
       "      <td>0.068278</td>\n",
       "      <td>0.069006</td>\n",
       "      <td>0.063565</td>\n",
       "      <td>0.058064</td>\n",
       "      <td>...</td>\n",
       "      <td>0.071960</td>\n",
       "      <td>0.068659</td>\n",
       "      <td>0.073089</td>\n",
       "      <td>0.057048</td>\n",
       "      <td>0.047540</td>\n",
       "      <td>0.063163</td>\n",
       "      <td>0.073969</td>\n",
       "      <td>0.060095</td>\n",
       "      <td>0.087613</td>\n",
       "      <td>0.044895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>40972</td>\n",
       "      <td>2.172367e+08</td>\n",
       "      <td>0.155312</td>\n",
       "      <td>0.137769</td>\n",
       "      <td>0.140559</td>\n",
       "      <td>0.157943</td>\n",
       "      <td>0.141373</td>\n",
       "      <td>0.130886</td>\n",
       "      <td>0.142527</td>\n",
       "      <td>0.136348</td>\n",
       "      <td>...</td>\n",
       "      <td>0.128640</td>\n",
       "      <td>0.144662</td>\n",
       "      <td>0.143587</td>\n",
       "      <td>0.163197</td>\n",
       "      <td>0.133078</td>\n",
       "      <td>0.143944</td>\n",
       "      <td>0.151770</td>\n",
       "      <td>0.139862</td>\n",
       "      <td>0.138751</td>\n",
       "      <td>0.149246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23275</th>\n",
       "      <td>ZNF791</td>\n",
       "      <td>1.272173e+07</td>\n",
       "      <td>0.051714</td>\n",
       "      <td>0.037007</td>\n",
       "      <td>0.041184</td>\n",
       "      <td>0.046509</td>\n",
       "      <td>0.044618</td>\n",
       "      <td>0.041508</td>\n",
       "      <td>0.038769</td>\n",
       "      <td>0.036825</td>\n",
       "      <td>...</td>\n",
       "      <td>0.040514</td>\n",
       "      <td>0.050522</td>\n",
       "      <td>0.051470</td>\n",
       "      <td>0.045516</td>\n",
       "      <td>0.037322</td>\n",
       "      <td>0.042542</td>\n",
       "      <td>0.038704</td>\n",
       "      <td>0.040140</td>\n",
       "      <td>0.026948</td>\n",
       "      <td>0.037751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23276</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>1.495355e+08</td>\n",
       "      <td>0.034747</td>\n",
       "      <td>0.021135</td>\n",
       "      <td>0.023729</td>\n",
       "      <td>0.012815</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>0.009904</td>\n",
       "      <td>0.009447</td>\n",
       "      <td>0.014334</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024416</td>\n",
       "      <td>0.025586</td>\n",
       "      <td>0.031914</td>\n",
       "      <td>0.014402</td>\n",
       "      <td>0.002733</td>\n",
       "      <td>0.016691</td>\n",
       "      <td>0.017068</td>\n",
       "      <td>0.033253</td>\n",
       "      <td>0.011848</td>\n",
       "      <td>0.008686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>ZNF91</td>\n",
       "      <td>2.357827e+07</td>\n",
       "      <td>0.057134</td>\n",
       "      <td>0.043614</td>\n",
       "      <td>0.053993</td>\n",
       "      <td>0.036108</td>\n",
       "      <td>0.046052</td>\n",
       "      <td>0.039620</td>\n",
       "      <td>0.041359</td>\n",
       "      <td>0.043246</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034475</td>\n",
       "      <td>0.046898</td>\n",
       "      <td>0.046136</td>\n",
       "      <td>0.045172</td>\n",
       "      <td>0.012000</td>\n",
       "      <td>0.046121</td>\n",
       "      <td>0.047162</td>\n",
       "      <td>0.051674</td>\n",
       "      <td>0.027724</td>\n",
       "      <td>0.044179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>7.154697e+07</td>\n",
       "      <td>0.046917</td>\n",
       "      <td>0.042107</td>\n",
       "      <td>0.043511</td>\n",
       "      <td>0.037403</td>\n",
       "      <td>0.040875</td>\n",
       "      <td>0.054482</td>\n",
       "      <td>0.041680</td>\n",
       "      <td>0.030635</td>\n",
       "      <td>...</td>\n",
       "      <td>0.039016</td>\n",
       "      <td>0.039753</td>\n",
       "      <td>0.041944</td>\n",
       "      <td>0.044347</td>\n",
       "      <td>0.030421</td>\n",
       "      <td>0.046929</td>\n",
       "      <td>0.039487</td>\n",
       "      <td>0.047963</td>\n",
       "      <td>0.041057</td>\n",
       "      <td>0.043777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23279</th>\n",
       "      <td>hsa-miR-3194-3p</td>\n",
       "      <td>3.137928e+07</td>\n",
       "      <td>0.557829</td>\n",
       "      <td>0.645971</td>\n",
       "      <td>0.555485</td>\n",
       "      <td>0.625114</td>\n",
       "      <td>0.647693</td>\n",
       "      <td>0.589112</td>\n",
       "      <td>0.564723</td>\n",
       "      <td>0.616729</td>\n",
       "      <td>...</td>\n",
       "      <td>0.615957</td>\n",
       "      <td>0.652565</td>\n",
       "      <td>0.648132</td>\n",
       "      <td>0.632591</td>\n",
       "      <td>0.618641</td>\n",
       "      <td>0.640838</td>\n",
       "      <td>0.619501</td>\n",
       "      <td>0.586397</td>\n",
       "      <td>0.617261</td>\n",
       "      <td>0.617818</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23280 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name   Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "0          1/2-SBSRNA4  0.000000e+00               0.000000  0.000000   \n",
       "1                40969  1.928757e+08               0.455367  0.445996   \n",
       "2                40970  1.855144e+08               0.053927  0.049527   \n",
       "3                40971  1.263664e+08               0.065878  0.071462   \n",
       "4                40972  2.172367e+08               0.155312  0.137769   \n",
       "...                ...           ...                    ...       ...   \n",
       "23275           ZNF791  1.272173e+07               0.051714  0.037007   \n",
       "23276           ZNF862  1.495355e+08               0.034747  0.021135   \n",
       "23277            ZNF91  2.357827e+07               0.057134  0.043614   \n",
       "23278           ZRANB2  7.154697e+07               0.046917  0.042107   \n",
       "23279  hsa-miR-3194-3p  3.137928e+07               0.557829  0.645971   \n",
       "\n",
       "        PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "0      0.000000  0.000000               0.000000               0.000000   \n",
       "1      0.431180  0.449385               0.443984               0.461588   \n",
       "2      0.048195  0.042055               0.051275               0.046544   \n",
       "3      0.068046  0.064942               0.068278               0.069006   \n",
       "4      0.140559  0.157943               0.141373               0.130886   \n",
       "...         ...       ...                    ...                    ...   \n",
       "23275  0.041184  0.046509               0.044618               0.041508   \n",
       "23276  0.023729  0.012815               0.031746               0.009904   \n",
       "23277  0.053993  0.036108               0.046052               0.039620   \n",
       "23278  0.043511  0.037403               0.040875               0.054482   \n",
       "23279  0.555485  0.625114               0.647693               0.589112   \n",
       "\n",
       "        PT-BZI5   PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD  \\\n",
       "0      0.000000  0.000000  ...               0.000000               0.000000   \n",
       "1      0.450010  0.451255  ...               0.451212               0.474548   \n",
       "2      0.043220  0.038363  ...               0.047662               0.045282   \n",
       "3      0.063565  0.058064  ...               0.071960               0.068659   \n",
       "4      0.142527  0.136348  ...               0.128640               0.144662   \n",
       "...         ...       ...  ...                    ...                    ...   \n",
       "23275  0.038769  0.036825  ...               0.040514               0.050522   \n",
       "23276  0.009447  0.014334  ...               0.024416               0.025586   \n",
       "23277  0.041359  0.043246  ...               0.034475               0.046898   \n",
       "23278  0.041680  0.030635  ...               0.039016               0.039753   \n",
       "23279  0.564723  0.616729  ...               0.615957               0.652565   \n",
       "\n",
       "        PT-BZD7   PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0      0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "1      0.450920  0.464083  0.455649  0.469531  0.459546  0.445075   \n",
       "2      0.048769  0.044007  0.044640  0.044352  0.052840  0.057617   \n",
       "3      0.073089  0.057048  0.047540  0.063163  0.073969  0.060095   \n",
       "4      0.143587  0.163197  0.133078  0.143944  0.151770  0.139862   \n",
       "...         ...       ...       ...       ...       ...       ...   \n",
       "23275  0.051470  0.045516  0.037322  0.042542  0.038704  0.040140   \n",
       "23276  0.031914  0.014402  0.002733  0.016691  0.017068  0.033253   \n",
       "23277  0.046136  0.045172  0.012000  0.046121  0.047162  0.051674   \n",
       "23278  0.041944  0.044347  0.030421  0.046929  0.039487  0.047963   \n",
       "23279  0.648132  0.632591  0.618641  0.640838  0.619501  0.586397   \n",
       "\n",
       "       TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                   0.000000  0.000000  \n",
       "1                   0.440247  0.473129  \n",
       "2                   0.054666  0.044574  \n",
       "3                   0.087613  0.044895  \n",
       "4                   0.138751  0.149246  \n",
       "...                      ...       ...  \n",
       "23275               0.026948  0.037751  \n",
       "23276               0.011848  0.008686  \n",
       "23277               0.027724  0.044179  \n",
       "23278               0.041057  0.043777  \n",
       "23279               0.617261  0.617818  \n",
       "\n",
       "[23280 rows x 742 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Upstream_df.rename(columns={'Closest_TSS_gene_name': 'gene_name'}, inplace=True)\n",
    "Distal_Promoter_df.rename(columns={'Closest_TSS_gene_name': 'gene_name'}, inplace=True)\n",
    "Proximal_Promoter_df.rename(columns={'Closest_TSS_gene_name': 'gene_name'}, inplace=True)\n",
    "Core_Promoter_df.rename(columns={'Closest_TSS_gene_name': 'gene_name'}, inplace=True)\n",
    "Downstream_df.rename(columns={'Closest_TSS_gene_name': 'gene_name'}, inplace=True)\n",
    "\n",
    "display(Upstream_df)\n",
    "display(Distal_Promoter_df)\n",
    "display(Proximal_Promoter_df)\n",
    "display(Core_Promoter_df)\n",
    "display(Downstream_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.Unify genes and patient samples within datasets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4.1 Unify gene and TSS for methylation, copynumer, and gene expression data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>416_120503_0</th>\n",
       "      <th>...</th>\n",
       "      <th>831_130725_8</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7SK</td>\n",
       "      <td>233.11</td>\n",
       "      <td>148.79</td>\n",
       "      <td>177.12</td>\n",
       "      <td>197.38</td>\n",
       "      <td>238.15</td>\n",
       "      <td>156.24</td>\n",
       "      <td>340.67</td>\n",
       "      <td>111.16</td>\n",
       "      <td>212.84</td>\n",
       "      <td>...</td>\n",
       "      <td>63.46</td>\n",
       "      <td>70.21</td>\n",
       "      <td>89.59</td>\n",
       "      <td>85.90</td>\n",
       "      <td>140.91</td>\n",
       "      <td>132.21</td>\n",
       "      <td>125.26</td>\n",
       "      <td>67.56</td>\n",
       "      <td>111.18</td>\n",
       "      <td>89.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>2.93</td>\n",
       "      <td>2.44</td>\n",
       "      <td>3.78</td>\n",
       "      <td>2.82</td>\n",
       "      <td>3.26</td>\n",
       "      <td>3.19</td>\n",
       "      <td>3.40</td>\n",
       "      <td>3.62</td>\n",
       "      <td>2.82</td>\n",
       "      <td>...</td>\n",
       "      <td>1.23</td>\n",
       "      <td>1.86</td>\n",
       "      <td>1.66</td>\n",
       "      <td>1.59</td>\n",
       "      <td>1.32</td>\n",
       "      <td>1.59</td>\n",
       "      <td>1.03</td>\n",
       "      <td>1.95</td>\n",
       "      <td>3.23</td>\n",
       "      <td>1.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>15.37</td>\n",
       "      <td>16.15</td>\n",
       "      <td>31.83</td>\n",
       "      <td>32.41</td>\n",
       "      <td>29.42</td>\n",
       "      <td>33.33</td>\n",
       "      <td>46.22</td>\n",
       "      <td>30.92</td>\n",
       "      <td>42.70</td>\n",
       "      <td>...</td>\n",
       "      <td>26.35</td>\n",
       "      <td>33.72</td>\n",
       "      <td>46.68</td>\n",
       "      <td>34.92</td>\n",
       "      <td>32.48</td>\n",
       "      <td>26.25</td>\n",
       "      <td>62.94</td>\n",
       "      <td>25.71</td>\n",
       "      <td>21.72</td>\n",
       "      <td>34.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.45</td>\n",
       "      <td>...</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.76</td>\n",
       "      <td>0.45</td>\n",
       "      <td>1.40</td>\n",
       "      <td>0.75</td>\n",
       "      <td>1.09</td>\n",
       "      <td>1.52</td>\n",
       "      <td>0.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37893</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>10.24</td>\n",
       "      <td>7.94</td>\n",
       "      <td>6.20</td>\n",
       "      <td>6.08</td>\n",
       "      <td>5.98</td>\n",
       "      <td>17.90</td>\n",
       "      <td>5.65</td>\n",
       "      <td>10.92</td>\n",
       "      <td>7.60</td>\n",
       "      <td>...</td>\n",
       "      <td>17.64</td>\n",
       "      <td>16.46</td>\n",
       "      <td>14.33</td>\n",
       "      <td>13.96</td>\n",
       "      <td>11.23</td>\n",
       "      <td>14.36</td>\n",
       "      <td>8.10</td>\n",
       "      <td>12.67</td>\n",
       "      <td>14.02</td>\n",
       "      <td>11.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37894</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>55.86</td>\n",
       "      <td>62.33</td>\n",
       "      <td>77.69</td>\n",
       "      <td>54.95</td>\n",
       "      <td>55.05</td>\n",
       "      <td>49.34</td>\n",
       "      <td>62.83</td>\n",
       "      <td>86.67</td>\n",
       "      <td>69.39</td>\n",
       "      <td>...</td>\n",
       "      <td>42.19</td>\n",
       "      <td>50.30</td>\n",
       "      <td>47.13</td>\n",
       "      <td>36.20</td>\n",
       "      <td>32.47</td>\n",
       "      <td>41.14</td>\n",
       "      <td>33.11</td>\n",
       "      <td>55.77</td>\n",
       "      <td>35.80</td>\n",
       "      <td>44.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37895</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>5.57</td>\n",
       "      <td>6.09</td>\n",
       "      <td>5.19</td>\n",
       "      <td>5.36</td>\n",
       "      <td>5.53</td>\n",
       "      <td>8.74</td>\n",
       "      <td>9.64</td>\n",
       "      <td>9.37</td>\n",
       "      <td>6.14</td>\n",
       "      <td>...</td>\n",
       "      <td>5.91</td>\n",
       "      <td>8.76</td>\n",
       "      <td>6.18</td>\n",
       "      <td>5.91</td>\n",
       "      <td>6.18</td>\n",
       "      <td>5.03</td>\n",
       "      <td>4.97</td>\n",
       "      <td>5.45</td>\n",
       "      <td>6.13</td>\n",
       "      <td>6.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37896</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>4.73</td>\n",
       "      <td>5.13</td>\n",
       "      <td>3.41</td>\n",
       "      <td>4.87</td>\n",
       "      <td>5.21</td>\n",
       "      <td>6.66</td>\n",
       "      <td>5.09</td>\n",
       "      <td>5.12</td>\n",
       "      <td>4.76</td>\n",
       "      <td>...</td>\n",
       "      <td>8.89</td>\n",
       "      <td>6.25</td>\n",
       "      <td>8.17</td>\n",
       "      <td>8.35</td>\n",
       "      <td>8.74</td>\n",
       "      <td>7.51</td>\n",
       "      <td>6.41</td>\n",
       "      <td>6.73</td>\n",
       "      <td>9.76</td>\n",
       "      <td>8.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37897</th>\n",
       "      <td>hsa-mir-423</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37898 rows × 641 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene_name  525_120515_0  383_120503_0  93_120417_0  610_120523_0  \\\n",
       "0              7SK        233.11        148.79       177.12        197.38   \n",
       "1             A1BG          2.93          2.44         3.78          2.82   \n",
       "2             A1CF          0.00          0.00         0.00          0.00   \n",
       "3              A2M         15.37         16.15        31.83         32.41   \n",
       "4          A2M-AS1          0.59          0.79         0.71          0.63   \n",
       "...            ...           ...           ...          ...           ...   \n",
       "37893       ZYG11B         10.24          7.94         6.20          6.08   \n",
       "37894          ZYX         55.86         62.33        77.69         54.95   \n",
       "37895        ZZEF1          5.57          6.09         5.19          5.36   \n",
       "37896         ZZZ3          4.73          5.13         3.41          4.87   \n",
       "37897  hsa-mir-423          0.00          0.00         0.00          0.00   \n",
       "\n",
       "       560_120517_0  492_120515_0  576_120521_0  150_120419_0  416_120503_0  \\\n",
       "0            238.15        156.24        340.67        111.16        212.84   \n",
       "1              3.26          3.19          3.40          3.62          2.82   \n",
       "2              0.00          0.00          0.00          0.00          0.00   \n",
       "3             29.42         33.33         46.22         30.92         42.70   \n",
       "4              0.39          0.77          0.34          0.57          0.45   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "37893          5.98         17.90          5.65         10.92          7.60   \n",
       "37894         55.05         49.34         62.83         86.67         69.39   \n",
       "37895          5.53          8.74          9.64          9.37          6.14   \n",
       "37896          5.21          6.66          5.09          5.12          4.76   \n",
       "37897          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       ...  831_130725_8  901_131010_8  894_130923_8  938_131101_8  \\\n",
       "0      ...         63.46         70.21         89.59         85.90   \n",
       "1      ...          1.23          1.86          1.66          1.59   \n",
       "2      ...          0.00          0.00          0.00          0.00   \n",
       "3      ...         26.35         33.72         46.68         34.92   \n",
       "4      ...          1.18          0.95          0.57          0.76   \n",
       "...    ...           ...           ...           ...           ...   \n",
       "37893  ...         17.64         16.46         14.33         13.96   \n",
       "37894  ...         42.19         50.30         47.13         36.20   \n",
       "37895  ...          5.91          8.76          6.18          5.91   \n",
       "37896  ...          8.89          6.25          8.17          8.35   \n",
       "37897  ...          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       942_131101_8  939_131101_8  895_130923_8  829_130725_8  944_131107_8  \\\n",
       "0            140.91        132.21        125.26         67.56        111.18   \n",
       "1              1.32          1.59          1.03          1.95          3.23   \n",
       "2              0.00          0.00          0.00          0.00          0.00   \n",
       "3             32.48         26.25         62.94         25.71         21.72   \n",
       "4              0.45          1.40          0.75          1.09          1.52   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "37893         11.23         14.36          8.10         12.67         14.02   \n",
       "37894         32.47         41.14         33.11         55.77         35.80   \n",
       "37895          6.18          5.03          4.97          5.45          6.13   \n",
       "37896          8.74          7.51          6.41          6.73          9.76   \n",
       "37897          0.00          0.00          0.00          0.00          0.00   \n",
       "\n",
       "       775_130528_8  \n",
       "0             89.51  \n",
       "1              1.98  \n",
       "2              0.00  \n",
       "3             34.44  \n",
       "4              0.92  \n",
       "...             ...  \n",
       "37893         11.08  \n",
       "37894         44.74  \n",
       "37895          6.53  \n",
       "37896          8.06  \n",
       "37897          0.00  \n",
       "\n",
       "[37898 rows x 641 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_expression = gene_expression_new.copy()\n",
    "gene_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1/2-SBSRNA4</td>\n",
       "      <td>110354972.0</td>\n",
       "      <td>0.931928</td>\n",
       "      <td>0.922927</td>\n",
       "      <td>0.921658</td>\n",
       "      <td>0.934680</td>\n",
       "      <td>0.950280</td>\n",
       "      <td>0.915567</td>\n",
       "      <td>0.895290</td>\n",
       "      <td>0.942072</td>\n",
       "      <td>...</td>\n",
       "      <td>0.929474</td>\n",
       "      <td>0.932108</td>\n",
       "      <td>0.909341</td>\n",
       "      <td>0.921341</td>\n",
       "      <td>0.930176</td>\n",
       "      <td>0.916491</td>\n",
       "      <td>0.913546</td>\n",
       "      <td>0.936391</td>\n",
       "      <td>0.905927</td>\n",
       "      <td>0.942917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40969</td>\n",
       "      <td>220960038.0</td>\n",
       "      <td>0.718230</td>\n",
       "      <td>0.698030</td>\n",
       "      <td>0.712573</td>\n",
       "      <td>0.705048</td>\n",
       "      <td>0.734309</td>\n",
       "      <td>0.771367</td>\n",
       "      <td>0.782897</td>\n",
       "      <td>0.704986</td>\n",
       "      <td>...</td>\n",
       "      <td>0.766211</td>\n",
       "      <td>0.675562</td>\n",
       "      <td>0.720371</td>\n",
       "      <td>0.689035</td>\n",
       "      <td>0.618610</td>\n",
       "      <td>0.764157</td>\n",
       "      <td>0.576607</td>\n",
       "      <td>0.754224</td>\n",
       "      <td>0.745078</td>\n",
       "      <td>0.710471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>40970</td>\n",
       "      <td>220921675.0</td>\n",
       "      <td>0.868322</td>\n",
       "      <td>0.845299</td>\n",
       "      <td>0.849537</td>\n",
       "      <td>0.892048</td>\n",
       "      <td>0.881078</td>\n",
       "      <td>0.867949</td>\n",
       "      <td>0.886519</td>\n",
       "      <td>0.902767</td>\n",
       "      <td>...</td>\n",
       "      <td>0.874877</td>\n",
       "      <td>0.851768</td>\n",
       "      <td>0.870165</td>\n",
       "      <td>0.878390</td>\n",
       "      <td>0.893037</td>\n",
       "      <td>0.901645</td>\n",
       "      <td>0.876906</td>\n",
       "      <td>0.870745</td>\n",
       "      <td>0.852633</td>\n",
       "      <td>0.877366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40971</td>\n",
       "      <td>126366439.0</td>\n",
       "      <td>0.914631</td>\n",
       "      <td>0.890200</td>\n",
       "      <td>0.893712</td>\n",
       "      <td>0.904816</td>\n",
       "      <td>0.934809</td>\n",
       "      <td>0.901929</td>\n",
       "      <td>0.918363</td>\n",
       "      <td>0.889949</td>\n",
       "      <td>...</td>\n",
       "      <td>0.907466</td>\n",
       "      <td>0.905140</td>\n",
       "      <td>0.898941</td>\n",
       "      <td>0.857999</td>\n",
       "      <td>0.960562</td>\n",
       "      <td>0.964220</td>\n",
       "      <td>0.896651</td>\n",
       "      <td>0.901460</td>\n",
       "      <td>0.907702</td>\n",
       "      <td>0.936499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>40972</td>\n",
       "      <td>217236749.0</td>\n",
       "      <td>0.689192</td>\n",
       "      <td>0.726004</td>\n",
       "      <td>0.691986</td>\n",
       "      <td>0.669473</td>\n",
       "      <td>0.690793</td>\n",
       "      <td>0.725609</td>\n",
       "      <td>0.711018</td>\n",
       "      <td>0.717188</td>\n",
       "      <td>...</td>\n",
       "      <td>0.822901</td>\n",
       "      <td>0.727588</td>\n",
       "      <td>0.706725</td>\n",
       "      <td>0.726323</td>\n",
       "      <td>0.695048</td>\n",
       "      <td>0.800269</td>\n",
       "      <td>0.685743</td>\n",
       "      <td>0.758808</td>\n",
       "      <td>0.726784</td>\n",
       "      <td>0.716149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23275</th>\n",
       "      <td>ZNF791</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23276</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>ZNF91</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23279</th>\n",
       "      <td>hsa-miR-3194-3p</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23280 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name  Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL  \\\n",
       "0          1/2-SBSRNA4  110354972.0               0.931928  0.922927   \n",
       "1                40969  220960038.0               0.718230  0.698030   \n",
       "2                40970  220921675.0               0.868322  0.845299   \n",
       "3                40971  126366439.0               0.914631  0.890200   \n",
       "4                40972  217236749.0               0.689192  0.726004   \n",
       "...                ...          ...                    ...       ...   \n",
       "23275           ZNF791          0.0               0.000000  0.000000   \n",
       "23276           ZNF862          0.0               0.000000  0.000000   \n",
       "23277            ZNF91          0.0               0.000000  0.000000   \n",
       "23278           ZRANB2          0.0               0.000000  0.000000   \n",
       "23279  hsa-miR-3194-3p          0.0               0.000000  0.000000   \n",
       "\n",
       "        PT-BZCH   PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD  \\\n",
       "0      0.921658  0.934680               0.950280               0.915567   \n",
       "1      0.712573  0.705048               0.734309               0.771367   \n",
       "2      0.849537  0.892048               0.881078               0.867949   \n",
       "3      0.893712  0.904816               0.934809               0.901929   \n",
       "4      0.691986  0.669473               0.690793               0.725609   \n",
       "...         ...       ...                    ...                    ...   \n",
       "23275  0.000000  0.000000               0.000000               0.000000   \n",
       "23276  0.000000  0.000000               0.000000               0.000000   \n",
       "23277  0.000000  0.000000               0.000000               0.000000   \n",
       "23278  0.000000  0.000000               0.000000               0.000000   \n",
       "23279  0.000000  0.000000               0.000000               0.000000   \n",
       "\n",
       "        PT-BZI5   PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD  \\\n",
       "0      0.895290  0.942072  ...               0.929474               0.932108   \n",
       "1      0.782897  0.704986  ...               0.766211               0.675562   \n",
       "2      0.886519  0.902767  ...               0.874877               0.851768   \n",
       "3      0.918363  0.889949  ...               0.907466               0.905140   \n",
       "4      0.711018  0.717188  ...               0.822901               0.727588   \n",
       "...         ...       ...  ...                    ...                    ...   \n",
       "23275  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23276  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23277  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23278  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "23279  0.000000  0.000000  ...               0.000000               0.000000   \n",
       "\n",
       "        PT-BZD7   PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  \\\n",
       "0      0.909341  0.921341  0.930176  0.916491  0.913546  0.936391   \n",
       "1      0.720371  0.689035  0.618610  0.764157  0.576607  0.754224   \n",
       "2      0.870165  0.878390  0.893037  0.901645  0.876906  0.870745   \n",
       "3      0.898941  0.857999  0.960562  0.964220  0.896651  0.901460   \n",
       "4      0.706725  0.726323  0.695048  0.800269  0.685743  0.758808   \n",
       "...         ...       ...       ...       ...       ...       ...   \n",
       "23275  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23276  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23277  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23278  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "23279  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "       TBI-AUTO73257-PT-35OE   PT-BZD3  \n",
       "0                   0.905927  0.942917  \n",
       "1                   0.745078  0.710471  \n",
       "2                   0.852633  0.877366  \n",
       "3                   0.907702  0.936499  \n",
       "4                   0.726784  0.716149  \n",
       "...                      ...       ...  \n",
       "23275               0.000000  0.000000  \n",
       "23276               0.000000  0.000000  \n",
       "23277               0.000000  0.000000  \n",
       "23278               0.000000  0.000000  \n",
       "23279               0.000000  0.000000  \n",
       "\n",
       "[23280 rows x 742 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Upstream_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48105</th>\n",
       "      <td>ZNF593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48106</th>\n",
       "      <td>CNKSR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48107</th>\n",
       "      <td>ZPLD2P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48108</th>\n",
       "      <td>DDX11L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48109</th>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48110 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene name\n",
       "0         MT-TF\n",
       "1       MT-RNR1\n",
       "2         MT-TV\n",
       "3       MT-RNR2\n",
       "4        MT-TL1\n",
       "...         ...\n",
       "48105    ZNF593\n",
       "48106    CNKSR1\n",
       "48107    ZPLD2P\n",
       "48108   DDX11L2\n",
       "48109    WASH7P\n",
       "\n",
       "[48110 rows x 1 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl_data_unique_gene = pd.read_csv(\"./ROSMAP-raw/Meta-Data/mart_export_unique_gene.txt\")\n",
    "ensembl_data_unique_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "#filter the genes not in Ensembl Dataset\n",
    "gene_expression = gene_expression[gene_expression['gene_name'].isin(ensembl_data_unique_gene['Gene name'])]\n",
    "Upstream_df = Upstream_df[Upstream_df['gene_name'].isin(ensembl_data_unique_gene['Gene name'])]\n",
    "protein = protein[protein['gene_name'].isin(ensembl_data_unique_gene['Gene name'])]\n",
    "cnv_aggregated_del = cnv_aggregated_del[cnv_aggregated_del['gene_name'].isin(ensembl_data_unique_gene['Gene name'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes in gene_expression: 37898\n",
      "Number of genes in methylation: 17606\n",
      "Number of genes in protein: 7817\n",
      "Number of genes in cnv_aggregated: 6419\n",
      "Number of common genes: 38686\n"
     ]
    }
   ],
   "source": [
    "# Convert the gene name columns from each DataFrame to sets\n",
    "gene_expression_genes = set(gene_expression['gene_name'])\n",
    "print(f\"Number of genes in gene_expression: {len(gene_expression_genes)}\")\n",
    "methylation_genes = set(Upstream_df['gene_name'])\n",
    "print(f\"Number of genes in methylation: {len(methylation_genes)}\")\n",
    "protein_genes = set(protein['gene_name'])\n",
    "print(f\"Number of genes in protein: {len(protein_genes)}\")\n",
    "cnv_aggregated_genes = set(cnv_aggregated_del['gene_name'])\n",
    "print(f\"Number of genes in cnv_aggregated: {len(cnv_aggregated_genes)}\")\n",
    "\n",
    "# Find the intersection of these sets\n",
    "common_genes =  gene_expression_genes | methylation_genes | protein_genes | cnv_aggregated_genes\n",
    "\n",
    "# Convert the intersection back to a list, if needed\n",
    "common_genes_list = list(common_genes)\n",
    "\n",
    "# Print the number of common genes\n",
    "print(f\"Number of common genes: {len(common_genes)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows with \"protein_coding\": 15387\n",
      "Number of rows with \"protein_coding\": 12959\n",
      "Number of rows with \"protein_coding\": 6246\n",
      "Number of rows with \"protein_coding\": 3268\n"
     ]
    }
   ],
   "source": [
    "#count the Transcript type\n",
    "ensembl_data_type = pd.read_csv(\"./ROSMAP-raw/Meta-Data/mart_export.txt\")\n",
    "ensembl_data_type= ensembl_data_type.rename(columns={'Gene name': 'gene_name'})\n",
    "ensembl_data_type = ensembl_data_type.drop_duplicates(subset='gene_name', keep='first')\n",
    "ensembl_data_type\n",
    "# Now, merge the two dataframes on the 'gene_name' column\n",
    "gene_expression_transcript = pd.merge(gene_expression, ensembl_data_type, on='gene_name', how='left')\n",
    "protein_coding_count = (gene_expression_transcript['Transcript type'] == 'protein_coding').sum()\n",
    "print(f'Number of rows with \"protein_coding\": {protein_coding_count}')\n",
    "\n",
    "Upstream_df_transcript = pd.merge(Upstream_df, ensembl_data_type, on='gene_name', how='left')\n",
    "protein_coding_count = (Upstream_df_transcript['Transcript type'] == 'protein_coding').sum()\n",
    "print(f'Number of rows with \"protein_coding\": {protein_coding_count}')\n",
    "\n",
    "protein_transcript = pd.merge(protein, ensembl_data_type, on='gene_name', how='left')\n",
    "protein_coding_count = (protein_transcript['Transcript type'] == 'protein_coding').sum()\n",
    "print(f'Number of rows with \"protein_coding\": {protein_coding_count}')\n",
    "\n",
    "cnv_aggregated_del_transcript = pd.merge(cnv_aggregated_del, ensembl_data_type, on='gene_name', how='left')\n",
    "protein_coding_count = (cnv_aggregated_del_transcript['Transcript type'] == 'protein_coding').sum()\n",
    "print(f'Number of rows with \"protein_coding\": {protein_coding_count}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "# common_genes_left = common_genes - common_genes_cnv_with_others\n",
    "# print(f\"Number of common genes: {len(common_genes_left)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cnv_aggregated_filtered = pd.DataFrame(columns=cnv_aggregated.columns)\n",
    "\n",
    "# for gene in common_genes_cnv_with_others:\n",
    "#     if gene in cnv_aggregated['Gene_Names'].values:\n",
    "#         gene_row = cnv_aggregated[cnv_aggregated['Gene_Names'] == gene]\n",
    "#         cnv_aggregated_filtered = pd.concat([cnv_aggregated_filtered, gene_row])\n",
    "\n",
    "# for gene in common_genes_left:\n",
    "#     zero_data = [0] * (len(cnv_aggregated.columns) - 1)\n",
    "#     zero_row = pd.DataFrame([[gene] + zero_data], columns=cnv_aggregated.columns)\n",
    "#     cnv_aggregated_filtered = pd.concat([cnv_aggregated_filtered, zero_row])\n",
    "\n",
    "# cnv_aggregated_filtered.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cnv_aggregated_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.1 Make the gene expression, mutation, copy number and proteomics data inputted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RGS13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SDC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BTBD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IQSEC3-AS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SPX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38681</th>\n",
       "      <td>DEFB134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38682</th>\n",
       "      <td>STARD13-AS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38683</th>\n",
       "      <td>ETHE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38684</th>\n",
       "      <td>SSC4D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38685</th>\n",
       "      <td>MAP7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38686 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_name\n",
       "0           RGS13\n",
       "1            SDC4\n",
       "2           BTBD1\n",
       "3      IQSEC3-AS2\n",
       "4             SPX\n",
       "...           ...\n",
       "38681     DEFB134\n",
       "38682  STARD13-AS\n",
       "38683       ETHE1\n",
       "38684       SSC4D\n",
       "38685        MAP7\n",
       "\n",
       "[38686 rows x 1 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_genes_df = pd.DataFrame(common_genes, columns=['gene_name'])\n",
    "common_genes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows with \"protein_coding\": 15620\n"
     ]
    }
   ],
   "source": [
    "merged_data_transcript_Upstream_df = pd.merge(common_genes_df, ensembl_data_type, on='gene_name', how='left')\n",
    "protein_coding_count = (merged_data_transcript_Upstream_df['Transcript type'] == 'protein_coding').sum()\n",
    "print(f'Number of rows with \"protein_coding\": {protein_coding_count}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Transcript type\n",
       "protein_coding                        15620\n",
       "processed_pseudogene                   6268\n",
       "lncRNA                                 5057\n",
       "snRNA                                  1794\n",
       "protein_coding_CDS_not_defined         1416\n",
       "unprocessed_pseudogene                 1412\n",
       "miRNA                                  1398\n",
       "nonsense_mediated_decay                1090\n",
       "retained_intron                         954\n",
       "misc_RNA                                954\n",
       "transcribed_unprocessed_pseudogene      617\n",
       "rRNA_pseudogene                         480\n",
       "snoRNA                                  434\n",
       "transcribed_processed_pseudogene        299\n",
       "IG_V_pseudogene                         165\n",
       "IG_V_gene                               133\n",
       "transcribed_unitary_pseudogene          118\n",
       "TR_V_gene                                91\n",
       "TR_J_gene                                71\n",
       "unitary_pseudogene                       67\n",
       "TR_V_pseudogene                          33\n",
       "protein_coding_LoF                       31\n",
       "IG_D_gene                                27\n",
       "rRNA                                     25\n",
       "Mt_tRNA                                  22\n",
       "scaRNA                                   19\n",
       "IG_J_gene                                18\n",
       "IG_C_gene                                14\n",
       "TEC                                       9\n",
       "processed_transcript                      8\n",
       "IG_C_pseudogene                           8\n",
       "TR_C_gene                                 6\n",
       "non_stop_decay                            5\n",
       "ribozyme                                  5\n",
       "TR_J_pseudogene                           4\n",
       "vault_RNA                                 4\n",
       "TR_D_gene                                 3\n",
       "IG_J_pseudogene                           3\n",
       "Mt_rRNA                                   2\n",
       "translated_processed_pseudogene           1\n",
       "scRNA                                     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_values = merged_data_transcript_Upstream_df['Transcript type'].value_counts()\n",
    "unique_values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>416_120503_0</th>\n",
       "      <th>...</th>\n",
       "      <th>831_130725_8</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RGS13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SDC4</td>\n",
       "      <td>77.47</td>\n",
       "      <td>20.86</td>\n",
       "      <td>25.46</td>\n",
       "      <td>42.39</td>\n",
       "      <td>43.32</td>\n",
       "      <td>16.80</td>\n",
       "      <td>125.23</td>\n",
       "      <td>14.50</td>\n",
       "      <td>43.16</td>\n",
       "      <td>...</td>\n",
       "      <td>27.94</td>\n",
       "      <td>38.05</td>\n",
       "      <td>20.78</td>\n",
       "      <td>23.12</td>\n",
       "      <td>29.53</td>\n",
       "      <td>29.20</td>\n",
       "      <td>42.21</td>\n",
       "      <td>24.51</td>\n",
       "      <td>48.05</td>\n",
       "      <td>21.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BTBD1</td>\n",
       "      <td>23.58</td>\n",
       "      <td>18.11</td>\n",
       "      <td>13.62</td>\n",
       "      <td>13.03</td>\n",
       "      <td>15.78</td>\n",
       "      <td>31.76</td>\n",
       "      <td>17.69</td>\n",
       "      <td>19.41</td>\n",
       "      <td>14.72</td>\n",
       "      <td>...</td>\n",
       "      <td>41.48</td>\n",
       "      <td>36.14</td>\n",
       "      <td>36.19</td>\n",
       "      <td>33.56</td>\n",
       "      <td>33.50</td>\n",
       "      <td>27.84</td>\n",
       "      <td>17.54</td>\n",
       "      <td>23.78</td>\n",
       "      <td>24.01</td>\n",
       "      <td>20.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IQSEC3-AS2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SPX</td>\n",
       "      <td>11.50</td>\n",
       "      <td>8.68</td>\n",
       "      <td>7.13</td>\n",
       "      <td>9.82</td>\n",
       "      <td>8.93</td>\n",
       "      <td>7.99</td>\n",
       "      <td>19.25</td>\n",
       "      <td>6.26</td>\n",
       "      <td>5.37</td>\n",
       "      <td>...</td>\n",
       "      <td>14.47</td>\n",
       "      <td>18.78</td>\n",
       "      <td>20.83</td>\n",
       "      <td>9.98</td>\n",
       "      <td>11.66</td>\n",
       "      <td>15.62</td>\n",
       "      <td>19.74</td>\n",
       "      <td>6.65</td>\n",
       "      <td>18.57</td>\n",
       "      <td>8.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38681</th>\n",
       "      <td>DEFB134</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.11</td>\n",
       "      <td>...</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38682</th>\n",
       "      <td>STARD13-AS</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>...</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38683</th>\n",
       "      <td>ETHE1</td>\n",
       "      <td>14.43</td>\n",
       "      <td>11.98</td>\n",
       "      <td>9.46</td>\n",
       "      <td>9.86</td>\n",
       "      <td>8.80</td>\n",
       "      <td>10.28</td>\n",
       "      <td>13.97</td>\n",
       "      <td>15.44</td>\n",
       "      <td>11.87</td>\n",
       "      <td>...</td>\n",
       "      <td>9.62</td>\n",
       "      <td>8.77</td>\n",
       "      <td>5.54</td>\n",
       "      <td>7.63</td>\n",
       "      <td>6.57</td>\n",
       "      <td>9.51</td>\n",
       "      <td>8.98</td>\n",
       "      <td>9.01</td>\n",
       "      <td>12.42</td>\n",
       "      <td>6.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38684</th>\n",
       "      <td>SSC4D</td>\n",
       "      <td>1.10</td>\n",
       "      <td>1.60</td>\n",
       "      <td>1.53</td>\n",
       "      <td>1.41</td>\n",
       "      <td>1.25</td>\n",
       "      <td>1.23</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.73</td>\n",
       "      <td>1.46</td>\n",
       "      <td>...</td>\n",
       "      <td>0.73</td>\n",
       "      <td>1.16</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.76</td>\n",
       "      <td>0.48</td>\n",
       "      <td>1.31</td>\n",
       "      <td>0.56</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.88</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38685</th>\n",
       "      <td>MAP7</td>\n",
       "      <td>11.25</td>\n",
       "      <td>15.27</td>\n",
       "      <td>12.95</td>\n",
       "      <td>12.79</td>\n",
       "      <td>13.93</td>\n",
       "      <td>19.49</td>\n",
       "      <td>15.67</td>\n",
       "      <td>17.02</td>\n",
       "      <td>13.10</td>\n",
       "      <td>...</td>\n",
       "      <td>23.80</td>\n",
       "      <td>20.07</td>\n",
       "      <td>25.41</td>\n",
       "      <td>21.12</td>\n",
       "      <td>23.42</td>\n",
       "      <td>14.52</td>\n",
       "      <td>18.27</td>\n",
       "      <td>16.75</td>\n",
       "      <td>24.87</td>\n",
       "      <td>27.37</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38686 rows × 641 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_name  525_120515_0  383_120503_0  93_120417_0  610_120523_0  \\\n",
       "0           RGS13          0.00          0.02         0.00          0.00   \n",
       "1            SDC4         77.47         20.86        25.46         42.39   \n",
       "2           BTBD1         23.58         18.11        13.62         13.03   \n",
       "3      IQSEC3-AS2          0.00          0.07         0.00          0.00   \n",
       "4             SPX         11.50          8.68         7.13          9.82   \n",
       "...           ...           ...           ...          ...           ...   \n",
       "38681     DEFB134          0.02          0.07         0.03          0.11   \n",
       "38682  STARD13-AS          0.02          0.00         0.16          0.00   \n",
       "38683       ETHE1         14.43         11.98         9.46          9.86   \n",
       "38684       SSC4D          1.10          1.60         1.53          1.41   \n",
       "38685        MAP7         11.25         15.27        12.95         12.79   \n",
       "\n",
       "       560_120517_0  492_120515_0  576_120521_0  150_120419_0  416_120503_0  \\\n",
       "0              0.00          0.00          0.01          0.01          0.02   \n",
       "1             43.32         16.80        125.23         14.50         43.16   \n",
       "2             15.78         31.76         17.69         19.41         14.72   \n",
       "3              0.00          0.00          0.00          0.00          0.04   \n",
       "4              8.93          7.99         19.25          6.26          5.37   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "38681          0.06          0.03          0.00          0.11          0.11   \n",
       "38682          0.03          0.10          0.00          0.00          0.05   \n",
       "38683          8.80         10.28         13.97         15.44         11.87   \n",
       "38684          1.25          1.23          0.89          0.73          1.46   \n",
       "38685         13.93         19.49         15.67         17.02         13.10   \n",
       "\n",
       "       ...  831_130725_8  901_131010_8  894_130923_8  938_131101_8  \\\n",
       "0      ...          0.00          0.05          0.00          0.00   \n",
       "1      ...         27.94         38.05         20.78         23.12   \n",
       "2      ...         41.48         36.14         36.19         33.56   \n",
       "3      ...          0.00          0.00          0.00          0.00   \n",
       "4      ...         14.47         18.78         20.83          9.98   \n",
       "...    ...           ...           ...           ...           ...   \n",
       "38681  ...          0.20          0.00          0.00          0.07   \n",
       "38682  ...          0.16          0.00          0.00          0.00   \n",
       "38683  ...          9.62          8.77          5.54          7.63   \n",
       "38684  ...          0.73          1.16          0.81          0.76   \n",
       "38685  ...         23.80         20.07         25.41         21.12   \n",
       "\n",
       "       942_131101_8  939_131101_8  895_130923_8  829_130725_8  944_131107_8  \\\n",
       "0              0.00          0.04          0.05          0.00          0.04   \n",
       "1             29.53         29.20         42.21         24.51         48.05   \n",
       "2             33.50         27.84         17.54         23.78         24.01   \n",
       "3              0.00          0.00          0.00          0.00          0.00   \n",
       "4             11.66         15.62         19.74          6.65         18.57   \n",
       "...             ...           ...           ...           ...           ...   \n",
       "38681          0.13          0.00          0.07          0.07          0.00   \n",
       "38682          0.00          0.39          0.00          0.00          0.00   \n",
       "38683          6.57          9.51          8.98          9.01         12.42   \n",
       "38684          0.48          1.31          0.56          1.21          1.88   \n",
       "38685         23.42         14.52         18.27         16.75         24.87   \n",
       "\n",
       "       775_130528_8  \n",
       "0              0.00  \n",
       "1             21.50  \n",
       "2             20.48  \n",
       "3              0.00  \n",
       "4              8.52  \n",
       "...             ...  \n",
       "38681          0.00  \n",
       "38682          0.00  \n",
       "38683          6.72  \n",
       "38684          0.90  \n",
       "38685         27.37  \n",
       "\n",
       "[38686 rows x 641 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_expression_inputted = pd.merge(common_genes_df ,gene_expression,on='gene_name',how='outer').fillna(0)\n",
    "protein_inputted = pd.merge(common_genes_df, protein, on='gene_name', how='outer').fillna(0)\n",
    "\n",
    "Upstream_df_inputted = pd.merge(common_genes_df, Upstream_df,on='gene_name',how='outer').fillna(0)\n",
    "Distal_Promoter_df_inputted = pd.merge(common_genes_df, Distal_Promoter_df,on='gene_name',how='outer').fillna(0)\n",
    "Proximal_Promoter_df_inputted = pd.merge(common_genes_df, Proximal_Promoter_df,on='gene_name',how='outer').fillna(0)\n",
    "Core_Promoter_df_inputted = pd.merge(common_genes_df, Core_Promoter_df,on='gene_name',how='outer').fillna(0)\n",
    "Downstream_df_inputted = pd.merge(common_genes_df, Downstream_df,on='gene_name',how='outer').fillna(0)\n",
    "\n",
    "cnv_aggregated_inputted_del = pd.merge(common_genes_df, cnv_aggregated_del, on='gene_name', how='outer').fillna(-1)\n",
    "cnv_aggregated_inputted_dup = pd.merge(common_genes_df, cnv_aggregated_dup, on='gene_name', how='outer').fillna(-1)\n",
    "cnv_aggregated_inputted_mcnv = pd.merge(common_genes_df, cnv_aggregated_mcnv, on='gene_name', how='outer').fillna(-1)\n",
    "\n",
    "gene_expression_inputted"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.2  Intersecting genes with various databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pathway_name\n",
      "PI3K-Akt signaling pathway                                  3992\n",
      "JAK-STAT signaling pathway                                  3280\n",
      "Chemokine signaling pathway                                 2766\n",
      "MAPK signaling pathway                                      2002\n",
      "cAMP signaling pathway                                      1680\n",
      "Ras signaling pathway                                       1640\n",
      "Rap1 signaling pathway                                      1304\n",
      "Calcium signaling pathway                                   1260\n",
      "Apelin signaling pathway                                    1011\n",
      "Wnt signaling pathway                                        935\n",
      "mTOR signaling pathway                                       836\n",
      "Hippo signaling pathway                                      830\n",
      "Insulin signaling pathway                                    721\n",
      "Glucagon signaling pathway                                   707\n",
      "Relaxin signaling pathway                                    684\n",
      "Phospholipase D signaling pathway                            676\n",
      "Thyroid hormone signaling pathway                            650\n",
      "FoxO signaling pathway                                       632\n",
      "AGE-RAGE signaling pathway in diabetic complications         627\n",
      "Oxytocin signaling pathway                                   609\n",
      "AMPK signaling pathway                                       564\n",
      "cGMP-PKG signaling pathway                                   560\n",
      "Neurotrophin signaling pathway                               549\n",
      "Estrogen signaling pathway                                   515\n",
      "Signaling pathways regulating pluripotency of stem cells     500\n",
      "IL-17 signaling pathway                                      500\n",
      "C-type lectin receptor signaling pathway                     427\n",
      "HIF-1 signaling pathway                                      407\n",
      "Sphingolipid signaling pathway                               395\n",
      "T cell receptor signaling pathway                            376\n",
      "TGF-beta signaling pathway                                   352\n",
      "NOD-like receptor signaling pathway                          352\n",
      "Adipocytokine signaling pathway                              311\n",
      "PPAR signaling pathway                                       305\n",
      "ErbB signaling pathway                                       264\n",
      "Toll-like receptor signaling pathway                         263\n",
      "GnRH signaling pathway                                       248\n",
      "NF-kappa B signaling pathway                                 238\n",
      "Notch signaling pathway                                      238\n",
      "VEGF signaling pathway                                       236\n",
      "Hedgehog signaling pathway                                   227\n",
      "Prolactin signaling pathway                                  219\n",
      "RIG-I-like receptor signaling pathway                        191\n",
      "Fc epsilon RI signaling pathway                              182\n",
      "B cell receptor signaling pathway                            177\n",
      "TNF signaling pathway                                        167\n",
      "p53 signaling pathway                                        111\n",
      "Hippo signaling pathway - multiple species                    96\n",
      "Name: count, dtype: int64\n",
      "----- NUMBER OF GENES IN KEGG: 2241 -----\n",
      "(21041, 2)\n",
      "----- NUMBER OF GENES IN KEGG PATH: 2241 -----\n",
      "(30733, 3)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Add the gene names from databases like [KEGG / BioGRID] to intersect with the common genes\n",
    "# KEGG\n",
    "kegg_pathway_df = pd.read_csv('./Regulatory-network-data/KEGG/full_kegg_pathway_list.csv')\n",
    "kegg_pathway_df = kegg_pathway_df[['source', 'target', 'pathway_name']]\n",
    "kegg_df = kegg_pathway_df[kegg_pathway_df['pathway_name'].str.contains('signaling pathway|signaling pathways', case=False)]\n",
    "print(kegg_df['pathway_name'].value_counts())\n",
    "kegg_df = kegg_df.rename(columns={'source': 'src', 'target': 'dest'})\n",
    "src_list = list(kegg_df['src'])\n",
    "dest_list = list(kegg_df['dest'])\n",
    "path_list = list(kegg_df['pathway_name'])\n",
    "# ADJUST ALL GENES TO UPPERCASE\n",
    "up_src_list = []\n",
    "for src in src_list:\n",
    "    up_src = src.upper()\n",
    "    up_src_list.append(up_src)\n",
    "up_dest_list = []\n",
    "for dest in dest_list:\n",
    "    up_dest = dest.upper()\n",
    "    up_dest_list.append(up_dest)\n",
    "up_kegg_conn_dict = {'src': up_src_list, 'dest': up_dest_list}\n",
    "up_kegg_df = pd.DataFrame(up_kegg_conn_dict)\n",
    "up_kegg_df = up_kegg_df.drop_duplicates()\n",
    "up_kegg_df.to_csv('./Regulatory-network-data/KEGG/up_kegg.csv', index=False, header=True)\n",
    "kegg_gene_list = list(set(list(up_kegg_df['src']) + list(up_kegg_df['dest'])))\n",
    "print('----- NUMBER OF GENES IN KEGG: ' + str(len(kegg_gene_list)) + ' -----')\n",
    "print(up_kegg_df.shape)\n",
    "\n",
    "up_kegg_path_conn_dict = {'src': up_src_list, 'dest': up_dest_list, 'path': path_list}\n",
    "up_kegg_path_df = pd.DataFrame(up_kegg_path_conn_dict)\n",
    "up_kegg_path_df = up_kegg_path_df.drop_duplicates()\n",
    "up_kegg_path_df.to_csv('./Regulatory-network-data/KEGG/up_kegg_path.csv', index=False, header=True)\n",
    "kegg_path_gene_list = list(set(list(up_kegg_path_df['src']) + list(up_kegg_path_df['dest'])))\n",
    "print('----- NUMBER OF GENES IN KEGG PATH: ' + str(len(kegg_path_gene_list)) + ' -----')\n",
    "print(up_kegg_path_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           src    dest\n",
      "0       MAP2K4    FLNC\n",
      "1         MYPN   ACTN2\n",
      "2        ACVR1    FNTA\n",
      "3        GATA2     PML\n",
      "4         RPA2   STAT3\n",
      "...        ...     ...\n",
      "472638   USP18  SAMHD1\n",
      "472639   USP18    SKP2\n",
      "472640  SAMHD1   USP18\n",
      "472641  SAMHD1   CCNA2\n",
      "472642  SAMHD1    CDK1\n",
      "\n",
      "[472643 rows x 2 columns]\n",
      "(472643, 2)\n",
      "----- NUMBER OF GENES IN BioGRID: 19349 -----\n"
     ]
    }
   ],
   "source": [
    "# BioGRID\n",
    "biogrid_df = pd.read_table('./Regulatory-network-data/BioGrid/BIOGRID-ALL-3.5.174.mitab.Symbol.txt', delimiter = '\\t')\n",
    "eh_list = list(biogrid_df['e_h'])\n",
    "et_list = list(biogrid_df['e_t'])\n",
    "# ADJUST ALL GENES TO UPPERCASE\n",
    "up_eh_list = []\n",
    "for eh in eh_list:\n",
    "    up_eh = eh.upper()\n",
    "    up_eh_list.append(up_eh)\n",
    "up_et_list = []\n",
    "for et in et_list:\n",
    "    up_et = et.upper()\n",
    "    up_et_list.append(up_et)\n",
    "up_biogrid_conn_dict = {'src': up_eh_list, 'dest': up_et_list}\n",
    "up_biogrid_df = pd.DataFrame(up_biogrid_conn_dict)\n",
    "print(up_biogrid_df)\n",
    "print(up_biogrid_df.shape)\n",
    "up_biogrid_df.to_csv('./Regulatory-network-data/BioGrid/up_biogrid.csv', index = False, header = True)\n",
    "up_biogrid_gene_list = list(set(list(up_biogrid_df['src']) + list(up_biogrid_df['dest'])))\n",
    "print('----- NUMBER OF GENES IN BioGRID: ' + str(len(up_biogrid_gene_list)) + ' -----')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          src    dest\n",
      "0        ARF5  SPTBN2\n",
      "1        ARF5  KIF13B\n",
      "2        ARF5   AP1B1\n",
      "3        ARF5  KIF21A\n",
      "4        ARF5   TMED7\n",
      "...       ...     ...\n",
      "841063  OR6Q1   REEP1\n",
      "841064  OR6Q1   REEP4\n",
      "841065  OR6Q1    GNB1\n",
      "841066  OR6Q1    RTP3\n",
      "841067  OR6Q1   REEP2\n",
      "\n",
      "[841068 rows x 2 columns]\n",
      "----- NUMBER OF GENES IN STRING: 17179 -----\n"
     ]
    }
   ],
   "source": [
    "# STRING\n",
    "string_df = pd.read_csv('./Regulatory-network-data/STRING/9606.protein.links.detailed.v11.0_sym.csv', low_memory=False)\n",
    "src_list = list(string_df['Source'])\n",
    "tar_list = list(string_df['Target'])\n",
    "# ADJUST ALL GENES TO UPPERCASE\n",
    "up_src_list = []\n",
    "for src in src_list:\n",
    "    up_src = src.upper()\n",
    "    up_src_list.append(up_src)\n",
    "up_tar_list = []\n",
    "for tar in tar_list:\n",
    "    up_tar = tar.upper()\n",
    "    up_tar_list.append(up_tar)\n",
    "up_string_conn_dict = {'src': up_src_list, 'dest': up_tar_list}\n",
    "up_string_df = pd.DataFrame(up_string_conn_dict)\n",
    "print(up_string_df)\n",
    "up_string_df.to_csv('./Regulatory-network-data/STRING/up_string.csv', index = False, header = True)\n",
    "up_string_gene_list = list(set(list(up_string_df['src']) + list(up_string_df['dest'])))\n",
    "print('----- NUMBER OF GENES IN STRING: ' + str(len(up_string_gene_list)) + ' -----')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----- NUMBER OF INTERSECTED GENES IN KEGG: 2144 -----\n",
      "----- NEW KEGG EDGE CONNECTIONS: 19856 -----\n",
      "----- NEW KEGG PATHWAY CONNECTIONS: 28843 -----\n"
     ]
    }
   ],
   "source": [
    "# intersect the [common genes] with the genes in the different databases [KEGG / BioGRID / STRING]\n",
    "selected_database = 'KEGG'\n",
    "# selected_database = 'BioGRID'\n",
    "# selected_database = 'STRING'\n",
    "if selected_database == 'KEGG':\n",
    "    edge_common_genes = list(set(common_genes) & set(kegg_gene_list))\n",
    "    print('----- NUMBER OF INTERSECTED GENES IN KEGG: ' + str(len(edge_common_genes)) + ' -----')\n",
    "elif selected_database == 'BioGRID':\n",
    "    edge_common_genes = list(set(common_genes) & set(up_biogrid_gene_list))\n",
    "    print('----- NUMBER OF INTERSECTED GENES IN BioGRID: ' + str(len(edge_common_genes)) + ' -----')\n",
    "elif selected_database == 'STRING':\n",
    "    edge_common_genes = list(set(common_genes) & set(up_string_gene_list))\n",
    "    print('----- NUMBER OF INTERSECTED GENES IN STRING: ' + str(len(edge_common_genes)) + ' -----')\n",
    "\n",
    "# filter the genes in the different databases [KEGG / BioGRID / STRING] with the [common genes]\n",
    "if selected_database == 'KEGG':\n",
    "    filtered_up_kegg_df = up_kegg_df[up_kegg_df['src'].isin(edge_common_genes) & up_kegg_df['dest'].isin(edge_common_genes)]\n",
    "    filtered_up_kegg_df = filtered_up_kegg_df.drop_duplicates()\n",
    "    filtered_up_kegg_df = filtered_up_kegg_df.sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "    print('----- NEW KEGG EDGE CONNECTIONS: ' + str(len(filtered_up_kegg_df)) + ' -----')\n",
    "    filtered_up_kegg_path_df = up_kegg_path_df[up_kegg_path_df['src'].isin(edge_common_genes) & up_kegg_path_df['dest'].isin(edge_common_genes)]\n",
    "    filtered_up_kegg_path_df = filtered_up_kegg_path_df.drop_duplicates()\n",
    "    filtered_up_kegg_path_df = filtered_up_kegg_path_df.sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "    print('----- NEW KEGG PATHWAY CONNECTIONS: ' + str(len(filtered_up_kegg_path_df)) + ' -----')\n",
    "elif selected_database == 'BioGRID':\n",
    "    filtered_up_biogrid_df = up_biogrid_df[up_biogrid_df['src'].isin(edge_common_genes) & up_biogrid_df['dest'].isin(edge_common_genes)]\n",
    "    filtered_up_biogrid_df = filtered_up_biogrid_df.drop_duplicates()\n",
    "    filtered_up_biogrid_df = filtered_up_biogrid_df.sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "    print('----- NEW BioGRID EDGE CONNECTIONS: ' + str(len(filtered_up_biogrid_df)) + ' -----')\n",
    "elif selected_database == 'STRING':\n",
    "    filtered_up_string_df = up_string_df[up_string_df['src'].isin(edge_common_genes) & up_string_df['dest'].isin(edge_common_genes)]\n",
    "    filtered_up_string_df = filtered_up_string_df.drop_duplicates()\n",
    "    filtered_up_string_df = filtered_up_string_df.sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "    print('----- NEW STRING EDGE CONNECTIONS: ' + str(len(filtered_up_string_df)) + ' -----')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>MYL6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>MYL6B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>MYL9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>STK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>MYL6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19851</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19852</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19853</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19854</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19855</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19856 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         src   dest\n",
       "0       ACTB   MYL6\n",
       "1       ACTB  MYL6B\n",
       "2       ACTB   MYL9\n",
       "3       ACTB   STK3\n",
       "4      ACTG1   MYL6\n",
       "...      ...    ...\n",
       "19851  ZNRF3   FZD5\n",
       "19852  ZNRF3   FZD6\n",
       "19853  ZNRF3   FZD7\n",
       "19854  ZNRF3   FZD8\n",
       "19855  ZNRF3   FZD9\n",
       "\n",
       "[19856 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>MYL6</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>MYL6B</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>MYL9</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>STK3</td>\n",
       "      <td>Hippo signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>MYL6</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28838</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD5</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28839</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD6</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28840</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD7</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28841</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD8</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28842</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>FZD9</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28843 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         src   dest                        path\n",
       "0       ACTB   MYL6  Oxytocin signaling pathway\n",
       "1       ACTB  MYL6B  Oxytocin signaling pathway\n",
       "2       ACTB   MYL9  Oxytocin signaling pathway\n",
       "3       ACTB   STK3     Hippo signaling pathway\n",
       "4      ACTG1   MYL6  Oxytocin signaling pathway\n",
       "...      ...    ...                         ...\n",
       "28838  ZNRF3   FZD5       Wnt signaling pathway\n",
       "28839  ZNRF3   FZD6       Wnt signaling pathway\n",
       "28840  ZNRF3   FZD7       Wnt signaling pathway\n",
       "28841  ZNRF3   FZD8       Wnt signaling pathway\n",
       "28842  ZNRF3   FZD9       Wnt signaling pathway\n",
       "\n",
       "[28843 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if selected_database == 'KEGG':\n",
    "    display(filtered_up_kegg_df)\n",
    "    display(filtered_up_kegg_path_df)\n",
    "elif selected_database == 'BioGRID':\n",
    "    display(filtered_up_biogrid_df)\n",
    "elif selected_database == 'STRING':\n",
    "    display(filtered_up_string_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4.1.3 Filtering the gene names across the gene expression, cnv, proteomics and methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>525_120515_0</th>\n",
       "      <th>383_120503_0</th>\n",
       "      <th>93_120417_0</th>\n",
       "      <th>610_120523_0</th>\n",
       "      <th>560_120517_0</th>\n",
       "      <th>492_120515_0</th>\n",
       "      <th>576_120521_0</th>\n",
       "      <th>150_120419_0</th>\n",
       "      <th>416_120503_0</th>\n",
       "      <th>...</th>\n",
       "      <th>831_130725_8</th>\n",
       "      <th>901_131010_8</th>\n",
       "      <th>894_130923_8</th>\n",
       "      <th>938_131101_8</th>\n",
       "      <th>942_131101_8</th>\n",
       "      <th>939_131101_8</th>\n",
       "      <th>895_130923_8</th>\n",
       "      <th>829_130725_8</th>\n",
       "      <th>944_131107_8</th>\n",
       "      <th>775_130528_8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>8.84</td>\n",
       "      <td>8.55</td>\n",
       "      <td>7.78</td>\n",
       "      <td>10.04</td>\n",
       "      <td>11.92</td>\n",
       "      <td>7.72</td>\n",
       "      <td>11.81</td>\n",
       "      <td>9.18</td>\n",
       "      <td>8.44</td>\n",
       "      <td>...</td>\n",
       "      <td>7.36</td>\n",
       "      <td>6.76</td>\n",
       "      <td>6.97</td>\n",
       "      <td>7.13</td>\n",
       "      <td>9.99</td>\n",
       "      <td>6.59</td>\n",
       "      <td>8.27</td>\n",
       "      <td>6.87</td>\n",
       "      <td>8.75</td>\n",
       "      <td>10.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>5.48</td>\n",
       "      <td>4.72</td>\n",
       "      <td>3.95</td>\n",
       "      <td>4.09</td>\n",
       "      <td>4.91</td>\n",
       "      <td>4.87</td>\n",
       "      <td>4.78</td>\n",
       "      <td>5.79</td>\n",
       "      <td>5.07</td>\n",
       "      <td>...</td>\n",
       "      <td>6.69</td>\n",
       "      <td>6.70</td>\n",
       "      <td>6.26</td>\n",
       "      <td>7.27</td>\n",
       "      <td>7.20</td>\n",
       "      <td>5.98</td>\n",
       "      <td>5.14</td>\n",
       "      <td>6.71</td>\n",
       "      <td>8.11</td>\n",
       "      <td>6.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>47.72</td>\n",
       "      <td>51.73</td>\n",
       "      <td>53.82</td>\n",
       "      <td>55.76</td>\n",
       "      <td>53.33</td>\n",
       "      <td>36.55</td>\n",
       "      <td>48.76</td>\n",
       "      <td>51.45</td>\n",
       "      <td>56.51</td>\n",
       "      <td>...</td>\n",
       "      <td>35.83</td>\n",
       "      <td>40.62</td>\n",
       "      <td>38.63</td>\n",
       "      <td>41.68</td>\n",
       "      <td>36.64</td>\n",
       "      <td>50.03</td>\n",
       "      <td>47.65</td>\n",
       "      <td>43.69</td>\n",
       "      <td>49.13</td>\n",
       "      <td>46.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>9.99</td>\n",
       "      <td>4.14</td>\n",
       "      <td>5.25</td>\n",
       "      <td>16.37</td>\n",
       "      <td>10.72</td>\n",
       "      <td>6.32</td>\n",
       "      <td>10.99</td>\n",
       "      <td>6.19</td>\n",
       "      <td>8.42</td>\n",
       "      <td>...</td>\n",
       "      <td>8.07</td>\n",
       "      <td>4.93</td>\n",
       "      <td>6.69</td>\n",
       "      <td>7.33</td>\n",
       "      <td>11.94</td>\n",
       "      <td>7.54</td>\n",
       "      <td>11.21</td>\n",
       "      <td>6.79</td>\n",
       "      <td>4.97</td>\n",
       "      <td>5.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>6.19</td>\n",
       "      <td>7.05</td>\n",
       "      <td>5.68</td>\n",
       "      <td>6.38</td>\n",
       "      <td>5.10</td>\n",
       "      <td>10.13</td>\n",
       "      <td>10.06</td>\n",
       "      <td>9.68</td>\n",
       "      <td>7.07</td>\n",
       "      <td>...</td>\n",
       "      <td>14.34</td>\n",
       "      <td>12.17</td>\n",
       "      <td>16.64</td>\n",
       "      <td>11.56</td>\n",
       "      <td>10.34</td>\n",
       "      <td>10.75</td>\n",
       "      <td>9.41</td>\n",
       "      <td>8.79</td>\n",
       "      <td>17.05</td>\n",
       "      <td>15.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>15.82</td>\n",
       "      <td>15.60</td>\n",
       "      <td>10.21</td>\n",
       "      <td>9.83</td>\n",
       "      <td>9.53</td>\n",
       "      <td>23.22</td>\n",
       "      <td>9.06</td>\n",
       "      <td>15.90</td>\n",
       "      <td>14.09</td>\n",
       "      <td>...</td>\n",
       "      <td>22.38</td>\n",
       "      <td>19.88</td>\n",
       "      <td>14.51</td>\n",
       "      <td>18.18</td>\n",
       "      <td>14.28</td>\n",
       "      <td>17.75</td>\n",
       "      <td>10.73</td>\n",
       "      <td>16.04</td>\n",
       "      <td>12.01</td>\n",
       "      <td>12.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>17.50</td>\n",
       "      <td>13.84</td>\n",
       "      <td>8.48</td>\n",
       "      <td>9.11</td>\n",
       "      <td>11.44</td>\n",
       "      <td>18.72</td>\n",
       "      <td>14.66</td>\n",
       "      <td>14.55</td>\n",
       "      <td>13.01</td>\n",
       "      <td>...</td>\n",
       "      <td>26.78</td>\n",
       "      <td>20.76</td>\n",
       "      <td>21.36</td>\n",
       "      <td>20.10</td>\n",
       "      <td>24.53</td>\n",
       "      <td>20.34</td>\n",
       "      <td>15.88</td>\n",
       "      <td>19.24</td>\n",
       "      <td>16.10</td>\n",
       "      <td>15.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>3.55</td>\n",
       "      <td>3.07</td>\n",
       "      <td>2.36</td>\n",
       "      <td>3.75</td>\n",
       "      <td>2.11</td>\n",
       "      <td>2.81</td>\n",
       "      <td>9.79</td>\n",
       "      <td>4.56</td>\n",
       "      <td>2.88</td>\n",
       "      <td>...</td>\n",
       "      <td>3.97</td>\n",
       "      <td>5.85</td>\n",
       "      <td>3.75</td>\n",
       "      <td>4.09</td>\n",
       "      <td>3.59</td>\n",
       "      <td>4.57</td>\n",
       "      <td>3.67</td>\n",
       "      <td>3.85</td>\n",
       "      <td>3.38</td>\n",
       "      <td>3.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>4.77</td>\n",
       "      <td>4.34</td>\n",
       "      <td>3.34</td>\n",
       "      <td>5.86</td>\n",
       "      <td>6.28</td>\n",
       "      <td>4.77</td>\n",
       "      <td>16.58</td>\n",
       "      <td>4.59</td>\n",
       "      <td>5.61</td>\n",
       "      <td>...</td>\n",
       "      <td>7.96</td>\n",
       "      <td>4.48</td>\n",
       "      <td>6.10</td>\n",
       "      <td>7.10</td>\n",
       "      <td>8.03</td>\n",
       "      <td>3.82</td>\n",
       "      <td>8.42</td>\n",
       "      <td>5.20</td>\n",
       "      <td>8.03</td>\n",
       "      <td>6.03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 641 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  525_120515_0  383_120503_0  93_120417_0  610_120523_0  \\\n",
       "0         ABL1          8.84          8.55         7.78         10.04   \n",
       "1         ABL2          5.48          4.72         3.95          4.09   \n",
       "2        ACAA1         47.72         51.73        53.82         55.76   \n",
       "3        ACACA          0.00          0.00         0.00          0.00   \n",
       "4        ACACB          9.99          4.14         5.25         16.37   \n",
       "...        ...           ...           ...          ...           ...   \n",
       "2139   ZFYVE16          6.19          7.05         5.68          6.38   \n",
       "2140    ZFYVE9         15.82         15.60        10.21          9.83   \n",
       "2141     ZMAT3         17.50         13.84         8.48          9.11   \n",
       "2142    ZNF274          3.55          3.07         2.36          3.75   \n",
       "2143     ZNRF3          4.77          4.34         3.34          5.86   \n",
       "\n",
       "      560_120517_0  492_120515_0  576_120521_0  150_120419_0  416_120503_0  \\\n",
       "0            11.92          7.72         11.81          9.18          8.44   \n",
       "1             4.91          4.87          4.78          5.79          5.07   \n",
       "2            53.33         36.55         48.76         51.45         56.51   \n",
       "3             0.00          0.00          0.00          0.00          0.00   \n",
       "4            10.72          6.32         10.99          6.19          8.42   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "2139          5.10         10.13         10.06          9.68          7.07   \n",
       "2140          9.53         23.22          9.06         15.90         14.09   \n",
       "2141         11.44         18.72         14.66         14.55         13.01   \n",
       "2142          2.11          2.81          9.79          4.56          2.88   \n",
       "2143          6.28          4.77         16.58          4.59          5.61   \n",
       "\n",
       "      ...  831_130725_8  901_131010_8  894_130923_8  938_131101_8  \\\n",
       "0     ...          7.36          6.76          6.97          7.13   \n",
       "1     ...          6.69          6.70          6.26          7.27   \n",
       "2     ...         35.83         40.62         38.63         41.68   \n",
       "3     ...          0.00          0.00          0.00          0.00   \n",
       "4     ...          8.07          4.93          6.69          7.33   \n",
       "...   ...           ...           ...           ...           ...   \n",
       "2139  ...         14.34         12.17         16.64         11.56   \n",
       "2140  ...         22.38         19.88         14.51         18.18   \n",
       "2141  ...         26.78         20.76         21.36         20.10   \n",
       "2142  ...          3.97          5.85          3.75          4.09   \n",
       "2143  ...          7.96          4.48          6.10          7.10   \n",
       "\n",
       "      942_131101_8  939_131101_8  895_130923_8  829_130725_8  944_131107_8  \\\n",
       "0             9.99          6.59          8.27          6.87          8.75   \n",
       "1             7.20          5.98          5.14          6.71          8.11   \n",
       "2            36.64         50.03         47.65         43.69         49.13   \n",
       "3             0.00          0.00          0.00          0.00          0.00   \n",
       "4            11.94          7.54         11.21          6.79          4.97   \n",
       "...            ...           ...           ...           ...           ...   \n",
       "2139         10.34         10.75          9.41          8.79         17.05   \n",
       "2140         14.28         17.75         10.73         16.04         12.01   \n",
       "2141         24.53         20.34         15.88         19.24         16.10   \n",
       "2142          3.59          4.57          3.67          3.85          3.38   \n",
       "2143          8.03          3.82          8.42          5.20          8.03   \n",
       "\n",
       "      775_130528_8  \n",
       "0            10.48  \n",
       "1             6.38  \n",
       "2            46.26  \n",
       "3             0.00  \n",
       "4             5.59  \n",
       "...            ...  \n",
       "2139         15.82  \n",
       "2140         12.59  \n",
       "2141         15.79  \n",
       "2142          3.91  \n",
       "2143          6.03  \n",
       "\n",
       "[2144 rows x 641 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# select common genes in gene expression data\n",
    "gene_expression_filtered = gene_expression_inputted.loc[gene_expression_inputted['gene_name'].isin(edge_common_genes)]\n",
    "gene_expression_filtered = gene_expression_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "gene_expression_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>b01.127C</th>\n",
       "      <th>b01.127N</th>\n",
       "      <th>b01.128C</th>\n",
       "      <th>b01.128N</th>\n",
       "      <th>b01.129C</th>\n",
       "      <th>b01.129N</th>\n",
       "      <th>b01.130C</th>\n",
       "      <th>b01.130N</th>\n",
       "      <th>b02.127C</th>\n",
       "      <th>...</th>\n",
       "      <th>b49.130C</th>\n",
       "      <th>b49.130N</th>\n",
       "      <th>b50.127C</th>\n",
       "      <th>b50.127N</th>\n",
       "      <th>b50.128C</th>\n",
       "      <th>b50.128N</th>\n",
       "      <th>b50.129C</th>\n",
       "      <th>b50.129N</th>\n",
       "      <th>b50.130C</th>\n",
       "      <th>b50.130N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-0.103582</td>\n",
       "      <td>0.015336</td>\n",
       "      <td>-0.060367</td>\n",
       "      <td>-0.301662</td>\n",
       "      <td>-0.176597</td>\n",
       "      <td>-0.168287</td>\n",
       "      <td>-0.083194</td>\n",
       "      <td>-0.200645</td>\n",
       "      <td>-0.648434</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.286807</td>\n",
       "      <td>-0.162889</td>\n",
       "      <td>-0.183929</td>\n",
       "      <td>-0.135078</td>\n",
       "      <td>0.057625</td>\n",
       "      <td>-0.115102</td>\n",
       "      <td>-0.273137</td>\n",
       "      <td>-0.154150</td>\n",
       "      <td>-0.212377</td>\n",
       "      <td>-0.080477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.013649</td>\n",
       "      <td>0.120403</td>\n",
       "      <td>-0.160178</td>\n",
       "      <td>0.060566</td>\n",
       "      <td>0.062877</td>\n",
       "      <td>0.029663</td>\n",
       "      <td>0.029067</td>\n",
       "      <td>0.176011</td>\n",
       "      <td>0.115946</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.054777</td>\n",
       "      <td>0.024013</td>\n",
       "      <td>-0.099587</td>\n",
       "      <td>0.306420</td>\n",
       "      <td>-0.147124</td>\n",
       "      <td>0.145804</td>\n",
       "      <td>0.025654</td>\n",
       "      <td>0.047217</td>\n",
       "      <td>-0.050547</td>\n",
       "      <td>0.074370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-0.200711</td>\n",
       "      <td>-0.201481</td>\n",
       "      <td>-0.243198</td>\n",
       "      <td>-0.030152</td>\n",
       "      <td>-0.051683</td>\n",
       "      <td>-0.362033</td>\n",
       "      <td>-0.040971</td>\n",
       "      <td>-0.223601</td>\n",
       "      <td>-0.031392</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.363212</td>\n",
       "      <td>-0.072299</td>\n",
       "      <td>-0.024155</td>\n",
       "      <td>-0.232771</td>\n",
       "      <td>-0.205745</td>\n",
       "      <td>-0.336676</td>\n",
       "      <td>-0.135752</td>\n",
       "      <td>-0.333979</td>\n",
       "      <td>-0.213806</td>\n",
       "      <td>-0.128683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>-0.045958</td>\n",
       "      <td>0.237595</td>\n",
       "      <td>-0.010871</td>\n",
       "      <td>-0.001613</td>\n",
       "      <td>0.007412</td>\n",
       "      <td>0.115163</td>\n",
       "      <td>0.092460</td>\n",
       "      <td>0.038121</td>\n",
       "      <td>-0.098167</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003519</td>\n",
       "      <td>0.126243</td>\n",
       "      <td>-0.037437</td>\n",
       "      <td>-0.096267</td>\n",
       "      <td>0.065899</td>\n",
       "      <td>-0.051790</td>\n",
       "      <td>0.024523</td>\n",
       "      <td>0.003781</td>\n",
       "      <td>0.024393</td>\n",
       "      <td>0.040647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0.239321</td>\n",
       "      <td>-0.171347</td>\n",
       "      <td>-0.020115</td>\n",
       "      <td>0.072067</td>\n",
       "      <td>0.024964</td>\n",
       "      <td>0.027992</td>\n",
       "      <td>0.084575</td>\n",
       "      <td>0.015419</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.473441</td>\n",
       "      <td>0.229008</td>\n",
       "      <td>0.078061</td>\n",
       "      <td>0.159639</td>\n",
       "      <td>0.092709</td>\n",
       "      <td>0.033288</td>\n",
       "      <td>-0.112379</td>\n",
       "      <td>-0.070789</td>\n",
       "      <td>-0.068971</td>\n",
       "      <td>0.002309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-0.023356</td>\n",
       "      <td>0.318564</td>\n",
       "      <td>0.248660</td>\n",
       "      <td>0.036006</td>\n",
       "      <td>0.319589</td>\n",
       "      <td>0.168960</td>\n",
       "      <td>0.145361</td>\n",
       "      <td>0.072773</td>\n",
       "      <td>0.304339</td>\n",
       "      <td>...</td>\n",
       "      <td>0.107408</td>\n",
       "      <td>0.262709</td>\n",
       "      <td>0.139146</td>\n",
       "      <td>0.168117</td>\n",
       "      <td>0.053220</td>\n",
       "      <td>0.234324</td>\n",
       "      <td>0.129981</td>\n",
       "      <td>0.247600</td>\n",
       "      <td>0.157816</td>\n",
       "      <td>0.132918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 401 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  b01.127C  b01.127N  b01.128C  b01.128N  b01.129C  b01.129N  \\\n",
       "0         ABL1 -0.103582  0.015336 -0.060367 -0.301662 -0.176597 -0.168287   \n",
       "1         ABL2  0.013649  0.120403 -0.160178  0.060566  0.062877  0.029663   \n",
       "2        ACAA1 -0.200711 -0.201481 -0.243198 -0.030152 -0.051683 -0.362033   \n",
       "3        ACACA -0.045958  0.237595 -0.010871 -0.001613  0.007412  0.115163   \n",
       "4        ACACB  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16  0.239321 -0.171347 -0.020115  0.072067  0.024964  0.027992   \n",
       "2140    ZFYVE9 -0.023356  0.318564  0.248660  0.036006  0.319589  0.168960   \n",
       "2141     ZMAT3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "      b01.130C  b01.130N  b02.127C  ...  b49.130C  b49.130N  b50.127C  \\\n",
       "0    -0.083194 -0.200645 -0.648434  ... -0.286807 -0.162889 -0.183929   \n",
       "1     0.029067  0.176011  0.115946  ... -0.054777  0.024013 -0.099587   \n",
       "2    -0.040971 -0.223601 -0.031392  ... -0.363212 -0.072299 -0.024155   \n",
       "3     0.092460  0.038121 -0.098167  ...  0.003519  0.126243 -0.037437   \n",
       "4     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.084575  0.015419  0.000000  ...  0.473441  0.229008  0.078061   \n",
       "2140  0.145361  0.072773  0.304339  ...  0.107408  0.262709  0.139146   \n",
       "2141  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "\n",
       "      b50.127N  b50.128C  b50.128N  b50.129C  b50.129N  b50.130C  b50.130N  \n",
       "0    -0.135078  0.057625 -0.115102 -0.273137 -0.154150 -0.212377 -0.080477  \n",
       "1     0.306420 -0.147124  0.145804  0.025654  0.047217 -0.050547  0.074370  \n",
       "2    -0.232771 -0.205745 -0.336676 -0.135752 -0.333979 -0.213806 -0.128683  \n",
       "3    -0.096267  0.065899 -0.051790  0.024523  0.003781  0.024393  0.040647  \n",
       "4     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.159639  0.092709  0.033288 -0.112379 -0.070789 -0.068971  0.002309  \n",
       "2140  0.168117  0.053220  0.234324  0.129981  0.247600  0.157816  0.132918  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "\n",
       "[2144 rows x 401 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_filtered = protein_inputted.loc[protein_inputted['gene_name'].isin(edge_common_genes)]\n",
    "protein_filtered = protein_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "protein_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>TBI-AUTO73325-PT-3149</th>\n",
       "      <th>PT-BZHL</th>\n",
       "      <th>PT-BZCH</th>\n",
       "      <th>PT-BY9H</th>\n",
       "      <th>TBI-AUTO73307-PT-314I</th>\n",
       "      <th>TBI-AUTO73043-PT-35BD</th>\n",
       "      <th>PT-BZI5</th>\n",
       "      <th>PT-BZ1A</th>\n",
       "      <th>...</th>\n",
       "      <th>TBI-AUTO72955-PT-35OC</th>\n",
       "      <th>TBI-AUTO73291-PT-35OD</th>\n",
       "      <th>PT-BZD7</th>\n",
       "      <th>PT-BZG8</th>\n",
       "      <th>PT-C1N8</th>\n",
       "      <th>PT-BYJP</th>\n",
       "      <th>PT-BZHV</th>\n",
       "      <th>PT-BZI2</th>\n",
       "      <th>TBI-AUTO73257-PT-35OE</th>\n",
       "      <th>PT-BZD3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>133710830.0</td>\n",
       "      <td>0.866411</td>\n",
       "      <td>0.850319</td>\n",
       "      <td>0.850696</td>\n",
       "      <td>0.875926</td>\n",
       "      <td>0.853794</td>\n",
       "      <td>0.848104</td>\n",
       "      <td>0.876568</td>\n",
       "      <td>0.876418</td>\n",
       "      <td>...</td>\n",
       "      <td>0.872650</td>\n",
       "      <td>0.885689</td>\n",
       "      <td>0.877116</td>\n",
       "      <td>0.848954</td>\n",
       "      <td>0.873008</td>\n",
       "      <td>0.867169</td>\n",
       "      <td>0.880158</td>\n",
       "      <td>0.845706</td>\n",
       "      <td>0.833943</td>\n",
       "      <td>0.896733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>109554399.0</td>\n",
       "      <td>0.253519</td>\n",
       "      <td>0.310569</td>\n",
       "      <td>0.235067</td>\n",
       "      <td>0.317705</td>\n",
       "      <td>0.298272</td>\n",
       "      <td>0.284207</td>\n",
       "      <td>0.274299</td>\n",
       "      <td>0.276363</td>\n",
       "      <td>...</td>\n",
       "      <td>0.285576</td>\n",
       "      <td>0.295447</td>\n",
       "      <td>0.310517</td>\n",
       "      <td>0.277536</td>\n",
       "      <td>0.297944</td>\n",
       "      <td>0.324885</td>\n",
       "      <td>0.273978</td>\n",
       "      <td>0.250956</td>\n",
       "      <td>0.297658</td>\n",
       "      <td>0.290459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>79703837.0</td>\n",
       "      <td>0.691740</td>\n",
       "      <td>0.711987</td>\n",
       "      <td>0.683392</td>\n",
       "      <td>0.735226</td>\n",
       "      <td>0.668339</td>\n",
       "      <td>0.788046</td>\n",
       "      <td>0.752924</td>\n",
       "      <td>0.665839</td>\n",
       "      <td>...</td>\n",
       "      <td>0.785049</td>\n",
       "      <td>0.644080</td>\n",
       "      <td>0.676521</td>\n",
       "      <td>0.770764</td>\n",
       "      <td>0.703474</td>\n",
       "      <td>0.779585</td>\n",
       "      <td>0.701459</td>\n",
       "      <td>0.692296</td>\n",
       "      <td>0.719752</td>\n",
       "      <td>0.729142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>52608045.0</td>\n",
       "      <td>0.876294</td>\n",
       "      <td>0.852107</td>\n",
       "      <td>0.832202</td>\n",
       "      <td>0.834502</td>\n",
       "      <td>0.851029</td>\n",
       "      <td>0.873013</td>\n",
       "      <td>0.872753</td>\n",
       "      <td>0.856148</td>\n",
       "      <td>...</td>\n",
       "      <td>0.875879</td>\n",
       "      <td>0.847689</td>\n",
       "      <td>0.826881</td>\n",
       "      <td>0.913076</td>\n",
       "      <td>0.883889</td>\n",
       "      <td>0.873290</td>\n",
       "      <td>0.863916</td>\n",
       "      <td>0.845064</td>\n",
       "      <td>0.862344</td>\n",
       "      <td>0.886732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>29279754.0</td>\n",
       "      <td>0.888925</td>\n",
       "      <td>0.900024</td>\n",
       "      <td>0.884002</td>\n",
       "      <td>0.925396</td>\n",
       "      <td>0.941288</td>\n",
       "      <td>0.877461</td>\n",
       "      <td>0.908885</td>\n",
       "      <td>0.908793</td>\n",
       "      <td>...</td>\n",
       "      <td>0.851395</td>\n",
       "      <td>0.921918</td>\n",
       "      <td>0.863043</td>\n",
       "      <td>0.871368</td>\n",
       "      <td>0.941071</td>\n",
       "      <td>0.885557</td>\n",
       "      <td>0.894746</td>\n",
       "      <td>0.855646</td>\n",
       "      <td>0.807761</td>\n",
       "      <td>0.899658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  Closest_TSS  TBI-AUTO73325-PT-3149   PT-BZHL   PT-BZCH  \\\n",
       "0         ABL1  133710830.0               0.866411  0.850319  0.850696   \n",
       "1         ABL2          0.0               0.000000  0.000000  0.000000   \n",
       "2        ACAA1          0.0               0.000000  0.000000  0.000000   \n",
       "3        ACACA          0.0               0.000000  0.000000  0.000000   \n",
       "4        ACACB  109554399.0               0.253519  0.310569  0.235067   \n",
       "...        ...          ...                    ...       ...       ...   \n",
       "2139   ZFYVE16   79703837.0               0.691740  0.711987  0.683392   \n",
       "2140    ZFYVE9   52608045.0               0.876294  0.852107  0.832202   \n",
       "2141     ZMAT3          0.0               0.000000  0.000000  0.000000   \n",
       "2142    ZNF274          0.0               0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3   29279754.0               0.888925  0.900024  0.884002   \n",
       "\n",
       "       PT-BY9H  TBI-AUTO73307-PT-314I  TBI-AUTO73043-PT-35BD   PT-BZI5  \\\n",
       "0     0.875926               0.853794               0.848104  0.876568   \n",
       "1     0.000000               0.000000               0.000000  0.000000   \n",
       "2     0.000000               0.000000               0.000000  0.000000   \n",
       "3     0.000000               0.000000               0.000000  0.000000   \n",
       "4     0.317705               0.298272               0.284207  0.274299   \n",
       "...        ...                    ...                    ...       ...   \n",
       "2139  0.735226               0.668339               0.788046  0.752924   \n",
       "2140  0.834502               0.851029               0.873013  0.872753   \n",
       "2141  0.000000               0.000000               0.000000  0.000000   \n",
       "2142  0.000000               0.000000               0.000000  0.000000   \n",
       "2143  0.925396               0.941288               0.877461  0.908885   \n",
       "\n",
       "       PT-BZ1A  ...  TBI-AUTO72955-PT-35OC  TBI-AUTO73291-PT-35OD   PT-BZD7  \\\n",
       "0     0.876418  ...               0.872650               0.885689  0.877116   \n",
       "1     0.000000  ...               0.000000               0.000000  0.000000   \n",
       "2     0.000000  ...               0.000000               0.000000  0.000000   \n",
       "3     0.000000  ...               0.000000               0.000000  0.000000   \n",
       "4     0.276363  ...               0.285576               0.295447  0.310517   \n",
       "...        ...  ...                    ...                    ...       ...   \n",
       "2139  0.665839  ...               0.785049               0.644080  0.676521   \n",
       "2140  0.856148  ...               0.875879               0.847689  0.826881   \n",
       "2141  0.000000  ...               0.000000               0.000000  0.000000   \n",
       "2142  0.000000  ...               0.000000               0.000000  0.000000   \n",
       "2143  0.908793  ...               0.851395               0.921918  0.863043   \n",
       "\n",
       "       PT-BZG8   PT-C1N8   PT-BYJP   PT-BZHV   PT-BZI2  TBI-AUTO73257-PT-35OE  \\\n",
       "0     0.848954  0.873008  0.867169  0.880158  0.845706               0.833943   \n",
       "1     0.000000  0.000000  0.000000  0.000000  0.000000               0.000000   \n",
       "2     0.000000  0.000000  0.000000  0.000000  0.000000               0.000000   \n",
       "3     0.000000  0.000000  0.000000  0.000000  0.000000               0.000000   \n",
       "4     0.277536  0.297944  0.324885  0.273978  0.250956               0.297658   \n",
       "...        ...       ...       ...       ...       ...                    ...   \n",
       "2139  0.770764  0.703474  0.779585  0.701459  0.692296               0.719752   \n",
       "2140  0.913076  0.883889  0.873290  0.863916  0.845064               0.862344   \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000               0.000000   \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000               0.000000   \n",
       "2143  0.871368  0.941071  0.885557  0.894746  0.855646               0.807761   \n",
       "\n",
       "       PT-BZD3  \n",
       "0     0.896733  \n",
       "1     0.000000  \n",
       "2     0.000000  \n",
       "3     0.000000  \n",
       "4     0.290459  \n",
       "...        ...  \n",
       "2139  0.729142  \n",
       "2140  0.886732  \n",
       "2141  0.000000  \n",
       "2142  0.000000  \n",
       "2143  0.899658  \n",
       "\n",
       "[2144 rows x 742 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# select common genes in methylation data\n",
    "Upstream_df_filtered = Upstream_df_inputted.loc[Upstream_df_inputted['gene_name'].isin(edge_common_genes)]\n",
    "Upstream_df_filtered = Upstream_df_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "Distal_Promoter_df_filtered = Distal_Promoter_df_inputted.loc[Distal_Promoter_df_inputted['gene_name'].isin(edge_common_genes)]\n",
    "Distal_Promoter_df_filtered = Distal_Promoter_df_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "Proximal_Promoter_df_filtered = Proximal_Promoter_df_inputted.loc[Proximal_Promoter_df_inputted['gene_name'].isin(edge_common_genes)]\n",
    "Proximal_Promoter_df_filtered = Proximal_Promoter_df_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "Core_Promoter_df_filtered = Core_Promoter_df_inputted.loc[Core_Promoter_df_inputted['gene_name'].isin(edge_common_genes)]\n",
    "Core_Promoter_df_filtered = Core_Promoter_df_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "Downstream_df_filtered = Downstream_df_inputted.loc[Downstream_df_inputted['gene_name'].isin(edge_common_genes)]\n",
    "Downstream_df_filtered = Downstream_df_filtered.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "\n",
    "Upstream_df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>SM-CTDSC</th>\n",
       "      <th>SM-CJGMZ</th>\n",
       "      <th>SM-CJFM3</th>\n",
       "      <th>SM-CTEET</th>\n",
       "      <th>SM-CJK4L</th>\n",
       "      <th>SM-CJIZ4</th>\n",
       "      <th>SM-CJK3I</th>\n",
       "      <th>SM-CJGN4</th>\n",
       "      <th>SM-CTDSM</th>\n",
       "      <th>...</th>\n",
       "      <th>SM-CTDVK</th>\n",
       "      <th>SM-CTED3</th>\n",
       "      <th>SM-CJK4V</th>\n",
       "      <th>SM-CJK5F</th>\n",
       "      <th>SM-CJIY9</th>\n",
       "      <th>SM-CTDQZ</th>\n",
       "      <th>SM-CJIY3</th>\n",
       "      <th>SM-CTEN7</th>\n",
       "      <th>SM-CJGH9</th>\n",
       "      <th>SM-CTEFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 1128 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  SM-CTDSC  SM-CJGMZ  SM-CJFM3  SM-CTEET  SM-CJK4L  SM-CJIZ4  \\\n",
       "0         ABL1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "1         ABL2      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2        ACAA1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "3        ACACA       2.0       2.0       2.0       2.0       1.0       2.0   \n",
       "4        ACACB      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2140    ZFYVE9      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2141     ZMAT3      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2142    ZNF274      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2143     ZNRF3       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "\n",
       "      SM-CJK3I  SM-CJGN4  SM-CTDSM  ...  SM-CTDVK  SM-CTED3  SM-CJK4V  \\\n",
       "0         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "1         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "3          2.0       2.0       2.0  ...       2.0       2.0       1.0   \n",
       "4         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2140      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2141      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2142      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2143       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "\n",
       "      SM-CJK5F  SM-CJIY9  SM-CTDQZ  SM-CJIY3  SM-CTEN7  SM-CJGH9  SM-CTEFP  \n",
       "0         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "1         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "3          2.0       2.0       1.0       2.0       2.0       2.0       2.0  \n",
       "4         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2140      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2141      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2142      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2143       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "\n",
       "[2144 rows x 1128 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# select common genes in cnv data\n",
    "cnv_aggregated_filtered_del = cnv_aggregated_inputted_del.loc[cnv_aggregated_inputted_del['gene_name'].isin(edge_common_genes)]\n",
    "cnv_aggregated_filtered_del = cnv_aggregated_filtered_del.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "cnv_aggregated_filtered_dup = cnv_aggregated_inputted_dup.loc[cnv_aggregated_inputted_dup['gene_name'].isin(edge_common_genes)]\n",
    "cnv_aggregated_filtered_dup = cnv_aggregated_filtered_dup.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "cnv_aggregated_filtered_mcnv = cnv_aggregated_inputted_mcnv.loc[cnv_aggregated_inputted_mcnv['gene_name'].isin(edge_common_genes)]\n",
    "cnv_aggregated_filtered_mcnv = cnv_aggregated_filtered_mcnv.sort_values(by=['gene_name']).reset_index(drop=True)\n",
    "\n",
    "cnv_aggregated_filtered_del"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4.2 Make all data use a unified patient ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>specimenID</th>\n",
       "      <th>specimenIdSource</th>\n",
       "      <th>organ</th>\n",
       "      <th>tissue</th>\n",
       "      <th>BrodmannArea</th>\n",
       "      <th>sampleStatus</th>\n",
       "      <th>tissueWeight</th>\n",
       "      <th>tissueVolume</th>\n",
       "      <th>nucleicAcidSource</th>\n",
       "      <th>cellType</th>\n",
       "      <th>fastingState</th>\n",
       "      <th>isPostMortem</th>\n",
       "      <th>samplingAge</th>\n",
       "      <th>samplingAgeUnits</th>\n",
       "      <th>visitNumber</th>\n",
       "      <th>assay</th>\n",
       "      <th>exclude</th>\n",
       "      <th>excludeReason</th>\n",
       "      <th>samplingDate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R1743384</td>\n",
       "      <td>190403-B4-A_R1743384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>scrnaSeq</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R2670295</td>\n",
       "      <td>190403-B4-A_R2670295</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>scrnaSeq</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R4119160</td>\n",
       "      <td>190403-B4-A_R4119160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>scrnaSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>RNA genotype discordant with WGS</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R4641987</td>\n",
       "      <td>190403-B4-A_R4641987</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>scrnaSeq</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R5693901</td>\n",
       "      <td>190403-B4-A_R5693901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>scrnaSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>Duplicated donor</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13230</th>\n",
       "      <td>R2757104</td>\n",
       "      <td>DUKE-08135</td>\n",
       "      <td>Human</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13231</th>\n",
       "      <td>R9354381</td>\n",
       "      <td>DUKE-08136</td>\n",
       "      <td>Human</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13232</th>\n",
       "      <td>R2711188</td>\n",
       "      <td>DUKE-08137</td>\n",
       "      <td>Human</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13233</th>\n",
       "      <td>R9047934</td>\n",
       "      <td>DUKE-08138</td>\n",
       "      <td>Human</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13234</th>\n",
       "      <td>R2385937</td>\n",
       "      <td>DUKE-08139</td>\n",
       "      <td>Human</td>\n",
       "      <td>brain</td>\n",
       "      <td>dorsolateral prefrontal cortex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13235 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      individualID            specimenID specimenIdSource  organ  \\\n",
       "0         R1743384  190403-B4-A_R1743384              NaN  brain   \n",
       "1         R2670295  190403-B4-A_R2670295              NaN  brain   \n",
       "2         R4119160  190403-B4-A_R4119160              NaN  brain   \n",
       "3         R4641987  190403-B4-A_R4641987              NaN  brain   \n",
       "4         R5693901  190403-B4-A_R5693901              NaN  brain   \n",
       "...            ...                   ...              ...    ...   \n",
       "13230     R2757104            DUKE-08135            Human  brain   \n",
       "13231     R9354381            DUKE-08136            Human  brain   \n",
       "13232     R2711188            DUKE-08137            Human  brain   \n",
       "13233     R9047934            DUKE-08138            Human  brain   \n",
       "13234     R2385937            DUKE-08139            Human  brain   \n",
       "\n",
       "                               tissue BrodmannArea sampleStatus  tissueWeight  \\\n",
       "0      dorsolateral prefrontal cortex          NaN       frozen           NaN   \n",
       "1      dorsolateral prefrontal cortex          NaN       frozen           NaN   \n",
       "2      dorsolateral prefrontal cortex          NaN       frozen           NaN   \n",
       "3      dorsolateral prefrontal cortex          NaN       frozen           NaN   \n",
       "4      dorsolateral prefrontal cortex          NaN       frozen           NaN   \n",
       "...                               ...          ...          ...           ...   \n",
       "13230  dorsolateral prefrontal cortex          NaN          NaN           NaN   \n",
       "13231  dorsolateral prefrontal cortex          NaN          NaN           NaN   \n",
       "13232  dorsolateral prefrontal cortex          NaN          NaN           NaN   \n",
       "13233  dorsolateral prefrontal cortex          NaN          NaN           NaN   \n",
       "13234  dorsolateral prefrontal cortex          NaN          NaN           NaN   \n",
       "\n",
       "       tissueVolume nucleicAcidSource cellType fastingState isPostMortem  \\\n",
       "0               NaN    single nucleus      NaN          NaN         True   \n",
       "1               NaN    single nucleus      NaN          NaN         True   \n",
       "2               NaN    single nucleus      NaN          NaN         True   \n",
       "3               NaN    single nucleus      NaN          NaN         True   \n",
       "4               NaN    single nucleus      NaN          NaN         True   \n",
       "...             ...               ...      ...          ...          ...   \n",
       "13230          10.0               NaN      NaN        False         True   \n",
       "13231          10.0               NaN      NaN        False         True   \n",
       "13232          10.0               NaN      NaN        False         True   \n",
       "13233          10.0               NaN      NaN        False         True   \n",
       "13234          10.0               NaN      NaN        False         True   \n",
       "\n",
       "      samplingAge  samplingAgeUnits  visitNumber      assay exclude  \\\n",
       "0             NaN               NaN          NaN   scrnaSeq   False   \n",
       "1             NaN               NaN          NaN   scrnaSeq   False   \n",
       "2             NaN               NaN          NaN   scrnaSeq    True   \n",
       "3             NaN               NaN          NaN   scrnaSeq   False   \n",
       "4             NaN               NaN          NaN   scrnaSeq    True   \n",
       "...           ...               ...          ...        ...     ...   \n",
       "13230         NaN               NaN          NaN  Metabolon     NaN   \n",
       "13231         NaN               NaN          NaN  Metabolon     NaN   \n",
       "13232         NaN               NaN          NaN  Metabolon     NaN   \n",
       "13233         NaN               NaN          NaN  Metabolon     NaN   \n",
       "13234         NaN               NaN          NaN  Metabolon     NaN   \n",
       "\n",
       "                          excludeReason  samplingDate  \n",
       "0                                   NaN           NaN  \n",
       "1                                   NaN           NaN  \n",
       "2      RNA genotype discordant with WGS           NaN  \n",
       "3                                   NaN           NaN  \n",
       "4                      Duplicated donor           NaN  \n",
       "...                                 ...           ...  \n",
       "13230                               NaN           NaN  \n",
       "13231                               NaN           NaN  \n",
       "13232                               NaN           NaN  \n",
       "13233                               NaN           NaN  \n",
       "13234                               NaN           NaN  \n",
       "\n",
       "[13235 rows x 20 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ROSMAP_biospecimen = pd.read_csv('./ROSMAP-raw/Meta-Data/ROSMAP_biospecimen_metadata.csv', sep=',')\n",
    "ROSMAP_biospecimen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R8316516</th>\n",
       "      <th>R1822146</th>\n",
       "      <th>R3512949</th>\n",
       "      <th>R3143439</th>\n",
       "      <th>R4590536</th>\n",
       "      <th>R9679238</th>\n",
       "      <th>R6879714</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R4182458</th>\n",
       "      <th>...</th>\n",
       "      <th>R9680160</th>\n",
       "      <th>R7015250</th>\n",
       "      <th>R5676537</th>\n",
       "      <th>R2087569</th>\n",
       "      <th>R2421650</th>\n",
       "      <th>R1498848</th>\n",
       "      <th>R9794121</th>\n",
       "      <th>R7298955</th>\n",
       "      <th>R6084846</th>\n",
       "      <th>R9798367</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-0.103582</td>\n",
       "      <td>0.015336</td>\n",
       "      <td>-0.060367</td>\n",
       "      <td>-0.301662</td>\n",
       "      <td>-0.176597</td>\n",
       "      <td>-0.168287</td>\n",
       "      <td>-0.083194</td>\n",
       "      <td>-0.200645</td>\n",
       "      <td>-0.648434</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.286807</td>\n",
       "      <td>-0.162889</td>\n",
       "      <td>-0.183929</td>\n",
       "      <td>-0.135078</td>\n",
       "      <td>0.057625</td>\n",
       "      <td>-0.115102</td>\n",
       "      <td>-0.273137</td>\n",
       "      <td>-0.154150</td>\n",
       "      <td>-0.212377</td>\n",
       "      <td>-0.080477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.013649</td>\n",
       "      <td>0.120403</td>\n",
       "      <td>-0.160178</td>\n",
       "      <td>0.060566</td>\n",
       "      <td>0.062877</td>\n",
       "      <td>0.029663</td>\n",
       "      <td>0.029067</td>\n",
       "      <td>0.176011</td>\n",
       "      <td>0.115946</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.054777</td>\n",
       "      <td>0.024013</td>\n",
       "      <td>-0.099587</td>\n",
       "      <td>0.306420</td>\n",
       "      <td>-0.147124</td>\n",
       "      <td>0.145804</td>\n",
       "      <td>0.025654</td>\n",
       "      <td>0.047217</td>\n",
       "      <td>-0.050547</td>\n",
       "      <td>0.074370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-0.200711</td>\n",
       "      <td>-0.201481</td>\n",
       "      <td>-0.243198</td>\n",
       "      <td>-0.030152</td>\n",
       "      <td>-0.051683</td>\n",
       "      <td>-0.362033</td>\n",
       "      <td>-0.040971</td>\n",
       "      <td>-0.223601</td>\n",
       "      <td>-0.031392</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.363212</td>\n",
       "      <td>-0.072299</td>\n",
       "      <td>-0.024155</td>\n",
       "      <td>-0.232771</td>\n",
       "      <td>-0.205745</td>\n",
       "      <td>-0.336676</td>\n",
       "      <td>-0.135752</td>\n",
       "      <td>-0.333979</td>\n",
       "      <td>-0.213806</td>\n",
       "      <td>-0.128683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>-0.045958</td>\n",
       "      <td>0.237595</td>\n",
       "      <td>-0.010871</td>\n",
       "      <td>-0.001613</td>\n",
       "      <td>0.007412</td>\n",
       "      <td>0.115163</td>\n",
       "      <td>0.092460</td>\n",
       "      <td>0.038121</td>\n",
       "      <td>-0.098167</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003519</td>\n",
       "      <td>0.126243</td>\n",
       "      <td>-0.037437</td>\n",
       "      <td>-0.096267</td>\n",
       "      <td>0.065899</td>\n",
       "      <td>-0.051790</td>\n",
       "      <td>0.024523</td>\n",
       "      <td>0.003781</td>\n",
       "      <td>0.024393</td>\n",
       "      <td>0.040647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0.239321</td>\n",
       "      <td>-0.171347</td>\n",
       "      <td>-0.020115</td>\n",
       "      <td>0.072067</td>\n",
       "      <td>0.024964</td>\n",
       "      <td>0.027992</td>\n",
       "      <td>0.084575</td>\n",
       "      <td>0.015419</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.473441</td>\n",
       "      <td>0.229008</td>\n",
       "      <td>0.078061</td>\n",
       "      <td>0.159639</td>\n",
       "      <td>0.092709</td>\n",
       "      <td>0.033288</td>\n",
       "      <td>-0.112379</td>\n",
       "      <td>-0.070789</td>\n",
       "      <td>-0.068971</td>\n",
       "      <td>0.002309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-0.023356</td>\n",
       "      <td>0.318564</td>\n",
       "      <td>0.248660</td>\n",
       "      <td>0.036006</td>\n",
       "      <td>0.319589</td>\n",
       "      <td>0.168960</td>\n",
       "      <td>0.145361</td>\n",
       "      <td>0.072773</td>\n",
       "      <td>0.304339</td>\n",
       "      <td>...</td>\n",
       "      <td>0.107408</td>\n",
       "      <td>0.262709</td>\n",
       "      <td>0.139146</td>\n",
       "      <td>0.168117</td>\n",
       "      <td>0.053220</td>\n",
       "      <td>0.234324</td>\n",
       "      <td>0.129981</td>\n",
       "      <td>0.247600</td>\n",
       "      <td>0.157816</td>\n",
       "      <td>0.132918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 401 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R8316516  R1822146  R3512949  R3143439  R4590536  R9679238  \\\n",
       "0         ABL1 -0.103582  0.015336 -0.060367 -0.301662 -0.176597 -0.168287   \n",
       "1         ABL2  0.013649  0.120403 -0.160178  0.060566  0.062877  0.029663   \n",
       "2        ACAA1 -0.200711 -0.201481 -0.243198 -0.030152 -0.051683 -0.362033   \n",
       "3        ACACA -0.045958  0.237595 -0.010871 -0.001613  0.007412  0.115163   \n",
       "4        ACACB  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16  0.239321 -0.171347 -0.020115  0.072067  0.024964  0.027992   \n",
       "2140    ZFYVE9 -0.023356  0.318564  0.248660  0.036006  0.319589  0.168960   \n",
       "2141     ZMAT3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "      R6879714  R8963331  R4182458  ...  R9680160  R7015250  R5676537  \\\n",
       "0    -0.083194 -0.200645 -0.648434  ... -0.286807 -0.162889 -0.183929   \n",
       "1     0.029067  0.176011  0.115946  ... -0.054777  0.024013 -0.099587   \n",
       "2    -0.040971 -0.223601 -0.031392  ... -0.363212 -0.072299 -0.024155   \n",
       "3     0.092460  0.038121 -0.098167  ...  0.003519  0.126243 -0.037437   \n",
       "4     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.084575  0.015419  0.000000  ...  0.473441  0.229008  0.078061   \n",
       "2140  0.145361  0.072773  0.304339  ...  0.107408  0.262709  0.139146   \n",
       "2141  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "\n",
       "      R2087569  R2421650  R1498848  R9794121  R7298955  R6084846  R9798367  \n",
       "0    -0.135078  0.057625 -0.115102 -0.273137 -0.154150 -0.212377 -0.080477  \n",
       "1     0.306420 -0.147124  0.145804  0.025654  0.047217 -0.050547  0.074370  \n",
       "2    -0.232771 -0.205745 -0.336676 -0.135752 -0.333979 -0.213806 -0.128683  \n",
       "3    -0.096267  0.065899 -0.051790  0.024523  0.003781  0.024393  0.040647  \n",
       "4     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.159639  0.092709  0.033288 -0.112379 -0.070789 -0.068971  0.002309  \n",
       "2140  0.168117  0.053220  0.234324  0.129981  0.247600  0.157816  0.132918  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "\n",
       "[2144 rows x 401 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split the 'specimenID' and construct the matching ID\n",
    "ROSMAP_biospecimen['matching_id'] = ROSMAP_biospecimen['specimenID'].apply(lambda x: '.'.join(x.split('.')[2:4]))\n",
    "\n",
    "# Update the dictionary mapping with the new matching IDs\n",
    "matching_id_to_individual_protein = dict(zip(ROSMAP_biospecimen['matching_id'], ROSMAP_biospecimen['individualID']))\n",
    "\n",
    "# Replace the column names in the protein DataFrame if they are in the matching_id_to_individual_protein mapping\n",
    "protein_filtered.columns = [matching_id_to_individual_protein.get(col, col) for col in protein_filtered.columns]\n",
    "protein_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>mirna_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R8316516</td>\n",
       "      <td>b01.127C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R1822146</td>\n",
       "      <td>b01.127N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R3512949</td>\n",
       "      <td>b01.128C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R3143439</td>\n",
       "      <td>b01.128N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R4590536</td>\n",
       "      <td>b01.129C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>R1498848</td>\n",
       "      <td>b50.128N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>R9794121</td>\n",
       "      <td>b50.129C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>R7298955</td>\n",
       "      <td>b50.129N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>R6084846</td>\n",
       "      <td>b50.130C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>R9798367</td>\n",
       "      <td>b50.130N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>400 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID  mirna_id\n",
       "0       R8316516  b01.127C\n",
       "1       R1822146  b01.127N\n",
       "2       R3512949  b01.128C\n",
       "3       R3143439  b01.128N\n",
       "4       R4590536  b01.129C\n",
       "..           ...       ...\n",
       "395     R1498848  b50.128N\n",
       "396     R9794121  b50.129C\n",
       "397     R7298955  b50.129N\n",
       "398     R6084846  b50.130C\n",
       "399     R9798367  b50.130N\n",
       "\n",
       "[400 rows x 2 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mirna_ids = protein.columns.tolist()[1:]\n",
    "individual_ids = protein_filtered.columns.tolist()[1:]\n",
    "\n",
    "protein_map = pd.DataFrame({\n",
    "    'individualID': individual_ids,\n",
    "    'mirna_id': mirna_ids\n",
    "})\n",
    "\n",
    "protein_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Closest_TSS</th>\n",
       "      <th>R3978789</th>\n",
       "      <th>R8140052</th>\n",
       "      <th>R7881801</th>\n",
       "      <th>R6108690</th>\n",
       "      <th>R9662437</th>\n",
       "      <th>R9936070</th>\n",
       "      <th>R4119160</th>\n",
       "      <th>R6284795</th>\n",
       "      <th>...</th>\n",
       "      <th>R5508487</th>\n",
       "      <th>R7503784</th>\n",
       "      <th>R4887967</th>\n",
       "      <th>R5130901</th>\n",
       "      <th>R9380629</th>\n",
       "      <th>R6016948</th>\n",
       "      <th>R5259690</th>\n",
       "      <th>R1977848</th>\n",
       "      <th>R6536689</th>\n",
       "      <th>R6759986</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>133710830.0</td>\n",
       "      <td>0.866411</td>\n",
       "      <td>0.850319</td>\n",
       "      <td>0.850696</td>\n",
       "      <td>0.875926</td>\n",
       "      <td>0.853794</td>\n",
       "      <td>0.848104</td>\n",
       "      <td>0.876568</td>\n",
       "      <td>0.876418</td>\n",
       "      <td>...</td>\n",
       "      <td>0.872650</td>\n",
       "      <td>0.885689</td>\n",
       "      <td>0.877116</td>\n",
       "      <td>0.848954</td>\n",
       "      <td>0.873008</td>\n",
       "      <td>0.867169</td>\n",
       "      <td>0.880158</td>\n",
       "      <td>0.845706</td>\n",
       "      <td>0.833943</td>\n",
       "      <td>0.896733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>109554399.0</td>\n",
       "      <td>0.253519</td>\n",
       "      <td>0.310569</td>\n",
       "      <td>0.235067</td>\n",
       "      <td>0.317705</td>\n",
       "      <td>0.298272</td>\n",
       "      <td>0.284207</td>\n",
       "      <td>0.274299</td>\n",
       "      <td>0.276363</td>\n",
       "      <td>...</td>\n",
       "      <td>0.285576</td>\n",
       "      <td>0.295447</td>\n",
       "      <td>0.310517</td>\n",
       "      <td>0.277536</td>\n",
       "      <td>0.297944</td>\n",
       "      <td>0.324885</td>\n",
       "      <td>0.273978</td>\n",
       "      <td>0.250956</td>\n",
       "      <td>0.297658</td>\n",
       "      <td>0.290459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>79703837.0</td>\n",
       "      <td>0.691740</td>\n",
       "      <td>0.711987</td>\n",
       "      <td>0.683392</td>\n",
       "      <td>0.735226</td>\n",
       "      <td>0.668339</td>\n",
       "      <td>0.788046</td>\n",
       "      <td>0.752924</td>\n",
       "      <td>0.665839</td>\n",
       "      <td>...</td>\n",
       "      <td>0.785049</td>\n",
       "      <td>0.644080</td>\n",
       "      <td>0.676521</td>\n",
       "      <td>0.770764</td>\n",
       "      <td>0.703474</td>\n",
       "      <td>0.779585</td>\n",
       "      <td>0.701459</td>\n",
       "      <td>0.692296</td>\n",
       "      <td>0.719752</td>\n",
       "      <td>0.729142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>52608045.0</td>\n",
       "      <td>0.876294</td>\n",
       "      <td>0.852107</td>\n",
       "      <td>0.832202</td>\n",
       "      <td>0.834502</td>\n",
       "      <td>0.851029</td>\n",
       "      <td>0.873013</td>\n",
       "      <td>0.872753</td>\n",
       "      <td>0.856148</td>\n",
       "      <td>...</td>\n",
       "      <td>0.875879</td>\n",
       "      <td>0.847689</td>\n",
       "      <td>0.826881</td>\n",
       "      <td>0.913076</td>\n",
       "      <td>0.883889</td>\n",
       "      <td>0.873290</td>\n",
       "      <td>0.863916</td>\n",
       "      <td>0.845064</td>\n",
       "      <td>0.862344</td>\n",
       "      <td>0.886732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>29279754.0</td>\n",
       "      <td>0.888925</td>\n",
       "      <td>0.900024</td>\n",
       "      <td>0.884002</td>\n",
       "      <td>0.925396</td>\n",
       "      <td>0.941288</td>\n",
       "      <td>0.877461</td>\n",
       "      <td>0.908885</td>\n",
       "      <td>0.908793</td>\n",
       "      <td>...</td>\n",
       "      <td>0.851395</td>\n",
       "      <td>0.921918</td>\n",
       "      <td>0.863043</td>\n",
       "      <td>0.871368</td>\n",
       "      <td>0.941071</td>\n",
       "      <td>0.885557</td>\n",
       "      <td>0.894746</td>\n",
       "      <td>0.855646</td>\n",
       "      <td>0.807761</td>\n",
       "      <td>0.899658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 742 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  Closest_TSS  R3978789  R8140052  R7881801  R6108690  R9662437  \\\n",
       "0         ABL1  133710830.0  0.866411  0.850319  0.850696  0.875926  0.853794   \n",
       "1         ABL2          0.0  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2        ACAA1          0.0  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "3        ACACA          0.0  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "4        ACACB  109554399.0  0.253519  0.310569  0.235067  0.317705  0.298272   \n",
       "...        ...          ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16   79703837.0  0.691740  0.711987  0.683392  0.735226  0.668339   \n",
       "2140    ZFYVE9   52608045.0  0.876294  0.852107  0.832202  0.834502  0.851029   \n",
       "2141     ZMAT3          0.0  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274          0.0  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3   29279754.0  0.888925  0.900024  0.884002  0.925396  0.941288   \n",
       "\n",
       "      R9936070  R4119160  R6284795  ...  R5508487  R7503784  R4887967  \\\n",
       "0     0.848104  0.876568  0.876418  ...  0.872650  0.885689  0.877116   \n",
       "1     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "3     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "4     0.284207  0.274299  0.276363  ...  0.285576  0.295447  0.310517   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.788046  0.752924  0.665839  ...  0.785049  0.644080  0.676521   \n",
       "2140  0.873013  0.872753  0.856148  ...  0.875879  0.847689  0.826881   \n",
       "2141  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.877461  0.908885  0.908793  ...  0.851395  0.921918  0.863043   \n",
       "\n",
       "      R5130901  R9380629  R6016948  R5259690  R1977848  R6536689  R6759986  \n",
       "0     0.848954  0.873008  0.867169  0.880158  0.845706  0.833943  0.896733  \n",
       "1     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "3     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "4     0.277536  0.297944  0.324885  0.273978  0.250956  0.297658  0.290459  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.770764  0.703474  0.779585  0.701459  0.692296  0.719752  0.729142  \n",
       "2140  0.913076  0.883889  0.873290  0.863916  0.845064  0.862344  0.886732  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.871368  0.941071  0.885557  0.894746  0.855646  0.807761  0.899658  \n",
       "\n",
       "[2144 rows x 742 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matching_id_to_individual = dict(zip(ROSMAP_biospecimen['specimenID'], ROSMAP_biospecimen['individualID']))\n",
    "Upstream_df_filtered.columns = [matching_id_to_individual.get(col, col) for col in Upstream_df_filtered.columns]\n",
    "Distal_Promoter_df_filtered.columns = [matching_id_to_individual.get(col, col) for col in Distal_Promoter_df_filtered.columns]\n",
    "Proximal_Promoter_df_filtered.columns = [matching_id_to_individual.get(col, col) for col in Proximal_Promoter_df_filtered.columns]\n",
    "Core_Promoter_df_filtered.columns = [matching_id_to_individual.get(col, col) for col in Core_Promoter_df_filtered.columns]\n",
    "Downstream_df_filtered.columns = [matching_id_to_individual.get(col, col) for col in Downstream_df_filtered.columns]\n",
    "Upstream_df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>mwas_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R3978789</td>\n",
       "      <td>TBI-AUTO73325-PT-3149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R8140052</td>\n",
       "      <td>PT-BZHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R7881801</td>\n",
       "      <td>PT-BZCH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R6108690</td>\n",
       "      <td>PT-BY9H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R9662437</td>\n",
       "      <td>TBI-AUTO73307-PT-314I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>R6016948</td>\n",
       "      <td>PT-BYJP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>736</th>\n",
       "      <td>R5259690</td>\n",
       "      <td>PT-BZHV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737</th>\n",
       "      <td>R1977848</td>\n",
       "      <td>PT-BZI2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>R6536689</td>\n",
       "      <td>TBI-AUTO73257-PT-35OE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>739</th>\n",
       "      <td>R6759986</td>\n",
       "      <td>PT-BZD3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>740 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID                mwas_id\n",
       "0       R3978789  TBI-AUTO73325-PT-3149\n",
       "1       R8140052                PT-BZHL\n",
       "2       R7881801                PT-BZCH\n",
       "3       R6108690                PT-BY9H\n",
       "4       R9662437  TBI-AUTO73307-PT-314I\n",
       "..           ...                    ...\n",
       "735     R6016948                PT-BYJP\n",
       "736     R5259690                PT-BZHV\n",
       "737     R1977848                PT-BZI2\n",
       "738     R6536689  TBI-AUTO73257-PT-35OE\n",
       "739     R6759986                PT-BZD3\n",
       "\n",
       "[740 rows x 2 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mwas_id = methylation_value.columns.tolist()[1:]\n",
    "individual_ids = Upstream_df_filtered.columns.tolist()[2:]\n",
    "\n",
    "methylation_map = pd.DataFrame({\n",
    "    'individualID': individual_ids,\n",
    "    'mwas_id': mwas_id\n",
    "})\n",
    "\n",
    "methylation_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R1743384</th>\n",
       "      <th>R6862468</th>\n",
       "      <th>R5415701</th>\n",
       "      <th>R1407047</th>\n",
       "      <th>R2197944</th>\n",
       "      <th>R5693901</th>\n",
       "      <th>R9210731</th>\n",
       "      <th>R6911631</th>\n",
       "      <th>R5636935</th>\n",
       "      <th>...</th>\n",
       "      <th>R7868788</th>\n",
       "      <th>R2901804</th>\n",
       "      <th>R4956716</th>\n",
       "      <th>R5924065</th>\n",
       "      <th>R3914030</th>\n",
       "      <th>R1133959</th>\n",
       "      <th>R2975126</th>\n",
       "      <th>R1224782</th>\n",
       "      <th>R3211474</th>\n",
       "      <th>R3341095</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>8.84</td>\n",
       "      <td>8.55</td>\n",
       "      <td>7.78</td>\n",
       "      <td>10.04</td>\n",
       "      <td>11.92</td>\n",
       "      <td>7.72</td>\n",
       "      <td>11.81</td>\n",
       "      <td>9.18</td>\n",
       "      <td>8.44</td>\n",
       "      <td>...</td>\n",
       "      <td>7.36</td>\n",
       "      <td>6.76</td>\n",
       "      <td>6.97</td>\n",
       "      <td>7.13</td>\n",
       "      <td>9.99</td>\n",
       "      <td>6.59</td>\n",
       "      <td>8.27</td>\n",
       "      <td>6.87</td>\n",
       "      <td>8.75</td>\n",
       "      <td>10.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>5.48</td>\n",
       "      <td>4.72</td>\n",
       "      <td>3.95</td>\n",
       "      <td>4.09</td>\n",
       "      <td>4.91</td>\n",
       "      <td>4.87</td>\n",
       "      <td>4.78</td>\n",
       "      <td>5.79</td>\n",
       "      <td>5.07</td>\n",
       "      <td>...</td>\n",
       "      <td>6.69</td>\n",
       "      <td>6.70</td>\n",
       "      <td>6.26</td>\n",
       "      <td>7.27</td>\n",
       "      <td>7.20</td>\n",
       "      <td>5.98</td>\n",
       "      <td>5.14</td>\n",
       "      <td>6.71</td>\n",
       "      <td>8.11</td>\n",
       "      <td>6.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>47.72</td>\n",
       "      <td>51.73</td>\n",
       "      <td>53.82</td>\n",
       "      <td>55.76</td>\n",
       "      <td>53.33</td>\n",
       "      <td>36.55</td>\n",
       "      <td>48.76</td>\n",
       "      <td>51.45</td>\n",
       "      <td>56.51</td>\n",
       "      <td>...</td>\n",
       "      <td>35.83</td>\n",
       "      <td>40.62</td>\n",
       "      <td>38.63</td>\n",
       "      <td>41.68</td>\n",
       "      <td>36.64</td>\n",
       "      <td>50.03</td>\n",
       "      <td>47.65</td>\n",
       "      <td>43.69</td>\n",
       "      <td>49.13</td>\n",
       "      <td>46.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>9.99</td>\n",
       "      <td>4.14</td>\n",
       "      <td>5.25</td>\n",
       "      <td>16.37</td>\n",
       "      <td>10.72</td>\n",
       "      <td>6.32</td>\n",
       "      <td>10.99</td>\n",
       "      <td>6.19</td>\n",
       "      <td>8.42</td>\n",
       "      <td>...</td>\n",
       "      <td>8.07</td>\n",
       "      <td>4.93</td>\n",
       "      <td>6.69</td>\n",
       "      <td>7.33</td>\n",
       "      <td>11.94</td>\n",
       "      <td>7.54</td>\n",
       "      <td>11.21</td>\n",
       "      <td>6.79</td>\n",
       "      <td>4.97</td>\n",
       "      <td>5.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>6.19</td>\n",
       "      <td>7.05</td>\n",
       "      <td>5.68</td>\n",
       "      <td>6.38</td>\n",
       "      <td>5.10</td>\n",
       "      <td>10.13</td>\n",
       "      <td>10.06</td>\n",
       "      <td>9.68</td>\n",
       "      <td>7.07</td>\n",
       "      <td>...</td>\n",
       "      <td>14.34</td>\n",
       "      <td>12.17</td>\n",
       "      <td>16.64</td>\n",
       "      <td>11.56</td>\n",
       "      <td>10.34</td>\n",
       "      <td>10.75</td>\n",
       "      <td>9.41</td>\n",
       "      <td>8.79</td>\n",
       "      <td>17.05</td>\n",
       "      <td>15.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>15.82</td>\n",
       "      <td>15.60</td>\n",
       "      <td>10.21</td>\n",
       "      <td>9.83</td>\n",
       "      <td>9.53</td>\n",
       "      <td>23.22</td>\n",
       "      <td>9.06</td>\n",
       "      <td>15.90</td>\n",
       "      <td>14.09</td>\n",
       "      <td>...</td>\n",
       "      <td>22.38</td>\n",
       "      <td>19.88</td>\n",
       "      <td>14.51</td>\n",
       "      <td>18.18</td>\n",
       "      <td>14.28</td>\n",
       "      <td>17.75</td>\n",
       "      <td>10.73</td>\n",
       "      <td>16.04</td>\n",
       "      <td>12.01</td>\n",
       "      <td>12.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>17.50</td>\n",
       "      <td>13.84</td>\n",
       "      <td>8.48</td>\n",
       "      <td>9.11</td>\n",
       "      <td>11.44</td>\n",
       "      <td>18.72</td>\n",
       "      <td>14.66</td>\n",
       "      <td>14.55</td>\n",
       "      <td>13.01</td>\n",
       "      <td>...</td>\n",
       "      <td>26.78</td>\n",
       "      <td>20.76</td>\n",
       "      <td>21.36</td>\n",
       "      <td>20.10</td>\n",
       "      <td>24.53</td>\n",
       "      <td>20.34</td>\n",
       "      <td>15.88</td>\n",
       "      <td>19.24</td>\n",
       "      <td>16.10</td>\n",
       "      <td>15.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>3.55</td>\n",
       "      <td>3.07</td>\n",
       "      <td>2.36</td>\n",
       "      <td>3.75</td>\n",
       "      <td>2.11</td>\n",
       "      <td>2.81</td>\n",
       "      <td>9.79</td>\n",
       "      <td>4.56</td>\n",
       "      <td>2.88</td>\n",
       "      <td>...</td>\n",
       "      <td>3.97</td>\n",
       "      <td>5.85</td>\n",
       "      <td>3.75</td>\n",
       "      <td>4.09</td>\n",
       "      <td>3.59</td>\n",
       "      <td>4.57</td>\n",
       "      <td>3.67</td>\n",
       "      <td>3.85</td>\n",
       "      <td>3.38</td>\n",
       "      <td>3.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>4.77</td>\n",
       "      <td>4.34</td>\n",
       "      <td>3.34</td>\n",
       "      <td>5.86</td>\n",
       "      <td>6.28</td>\n",
       "      <td>4.77</td>\n",
       "      <td>16.58</td>\n",
       "      <td>4.59</td>\n",
       "      <td>5.61</td>\n",
       "      <td>...</td>\n",
       "      <td>7.96</td>\n",
       "      <td>4.48</td>\n",
       "      <td>6.10</td>\n",
       "      <td>7.10</td>\n",
       "      <td>8.03</td>\n",
       "      <td>3.82</td>\n",
       "      <td>8.42</td>\n",
       "      <td>5.20</td>\n",
       "      <td>8.03</td>\n",
       "      <td>6.03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 641 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R1743384  R6862468  R5415701  R1407047  R2197944  R5693901  \\\n",
       "0         ABL1      8.84      8.55      7.78     10.04     11.92      7.72   \n",
       "1         ABL2      5.48      4.72      3.95      4.09      4.91      4.87   \n",
       "2        ACAA1     47.72     51.73     53.82     55.76     53.33     36.55   \n",
       "3        ACACA      0.00      0.00      0.00      0.00      0.00      0.00   \n",
       "4        ACACB      9.99      4.14      5.25     16.37     10.72      6.32   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      6.19      7.05      5.68      6.38      5.10     10.13   \n",
       "2140    ZFYVE9     15.82     15.60     10.21      9.83      9.53     23.22   \n",
       "2141     ZMAT3     17.50     13.84      8.48      9.11     11.44     18.72   \n",
       "2142    ZNF274      3.55      3.07      2.36      3.75      2.11      2.81   \n",
       "2143     ZNRF3      4.77      4.34      3.34      5.86      6.28      4.77   \n",
       "\n",
       "      R9210731  R6911631  R5636935  ...  R7868788  R2901804  R4956716  \\\n",
       "0        11.81      9.18      8.44  ...      7.36      6.76      6.97   \n",
       "1         4.78      5.79      5.07  ...      6.69      6.70      6.26   \n",
       "2        48.76     51.45     56.51  ...     35.83     40.62     38.63   \n",
       "3         0.00      0.00      0.00  ...      0.00      0.00      0.00   \n",
       "4        10.99      6.19      8.42  ...      8.07      4.93      6.69   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139     10.06      9.68      7.07  ...     14.34     12.17     16.64   \n",
       "2140      9.06     15.90     14.09  ...     22.38     19.88     14.51   \n",
       "2141     14.66     14.55     13.01  ...     26.78     20.76     21.36   \n",
       "2142      9.79      4.56      2.88  ...      3.97      5.85      3.75   \n",
       "2143     16.58      4.59      5.61  ...      7.96      4.48      6.10   \n",
       "\n",
       "      R5924065  R3914030  R1133959  R2975126  R1224782  R3211474  R3341095  \n",
       "0         7.13      9.99      6.59      8.27      6.87      8.75     10.48  \n",
       "1         7.27      7.20      5.98      5.14      6.71      8.11      6.38  \n",
       "2        41.68     36.64     50.03     47.65     43.69     49.13     46.26  \n",
       "3         0.00      0.00      0.00      0.00      0.00      0.00      0.00  \n",
       "4         7.33     11.94      7.54     11.21      6.79      4.97      5.59  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139     11.56     10.34     10.75      9.41      8.79     17.05     15.82  \n",
       "2140     18.18     14.28     17.75     10.73     16.04     12.01     12.59  \n",
       "2141     20.10     24.53     20.34     15.88     19.24     16.10     15.79  \n",
       "2142      4.09      3.59      4.57      3.67      3.85      3.38      3.91  \n",
       "2143      7.10      8.03      3.82      8.42      5.20      8.03      6.03  \n",
       "\n",
       "[2144 rows x 641 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def process_gene_column_name(col_name):\n",
    "    parts = col_name.split('_')\n",
    "    return '_'.join(parts[:2]) if len(parts) > 1 else col_name\n",
    "gene_expression_filtered.columns = [process_gene_column_name(col) for col in gene_expression_filtered.columns]\n",
    "matching_id_to_individual_gene = dict(zip(ROSMAP_biospecimen['specimenID'], ROSMAP_biospecimen['individualID']))\n",
    "gene_expression_filtered.columns = [matching_id_to_individual_gene.get(col, col) for col in gene_expression_filtered.columns]\n",
    "gene_expression_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>mrna_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R1743384</td>\n",
       "      <td>525_120515_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R6862468</td>\n",
       "      <td>383_120503_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R5415701</td>\n",
       "      <td>93_120417_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R1407047</td>\n",
       "      <td>610_120523_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R2197944</td>\n",
       "      <td>560_120517_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>R1133959</td>\n",
       "      <td>939_131101_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636</th>\n",
       "      <td>R2975126</td>\n",
       "      <td>895_130923_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>R1224782</td>\n",
       "      <td>829_130725_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>R3211474</td>\n",
       "      <td>944_131107_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>R3341095</td>\n",
       "      <td>775_130528_8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>640 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID       mrna_id\n",
       "0       R1743384  525_120515_0\n",
       "1       R6862468  383_120503_0\n",
       "2       R5415701   93_120417_0\n",
       "3       R1407047  610_120523_0\n",
       "4       R2197944  560_120517_0\n",
       "..           ...           ...\n",
       "635     R1133959  939_131101_8\n",
       "636     R2975126  895_130923_8\n",
       "637     R1224782  829_130725_8\n",
       "638     R3211474  944_131107_8\n",
       "639     R3341095  775_130528_8\n",
       "\n",
       "[640 rows x 2 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mrna_id = gene_expression.columns.tolist()[1:]\n",
    "individual_ids = gene_expression_filtered.columns.tolist()[1:]\n",
    "\n",
    "gene_expression_map = pd.DataFrame({\n",
    "    'individualID': individual_ids,\n",
    "    'mrna_id': mrna_id\n",
    "})\n",
    "\n",
    "gene_expression_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R3978789</th>\n",
       "      <th>R6108690</th>\n",
       "      <th>R6013281</th>\n",
       "      <th>R2367199</th>\n",
       "      <th>R4323608</th>\n",
       "      <th>R2577726</th>\n",
       "      <th>R9489952</th>\n",
       "      <th>R8125311</th>\n",
       "      <th>R1234575</th>\n",
       "      <th>...</th>\n",
       "      <th>R1724172</th>\n",
       "      <th>R9323873</th>\n",
       "      <th>R4859661</th>\n",
       "      <th>R1074668</th>\n",
       "      <th>R6739300</th>\n",
       "      <th>R3008520</th>\n",
       "      <th>R3151599</th>\n",
       "      <th>R7993799</th>\n",
       "      <th>R9907075</th>\n",
       "      <th>R5026720</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 1128 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R3978789  R6108690  R6013281  R2367199  R4323608  R2577726  \\\n",
       "0         ABL1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "1         ABL2      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2        ACAA1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "3        ACACA       2.0       2.0       2.0       2.0       1.0       2.0   \n",
       "4        ACACB      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2140    ZFYVE9      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2141     ZMAT3      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2142    ZNF274      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2143     ZNRF3       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "\n",
       "      R9489952  R8125311  R1234575  ...  R1724172  R9323873  R4859661  \\\n",
       "0         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "1         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "3          2.0       2.0       2.0  ...       2.0       2.0       1.0   \n",
       "4         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2140      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2141      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2142      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2143       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "\n",
       "      R1074668  R6739300  R3008520  R3151599  R7993799  R9907075  R5026720  \n",
       "0         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "1         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "3          2.0       2.0       1.0       2.0       2.0       2.0       2.0  \n",
       "4         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2140      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2141      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2142      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2143       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "\n",
       "[2144 rows x 1128 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_aggregated_filtered_del.columns = [matching_id_to_individual.get(col, col) for col in cnv_aggregated_filtered_del.columns]\n",
    "cnv_aggregated_filtered_dup.columns = [matching_id_to_individual.get(col, col) for col in cnv_aggregated_filtered_dup.columns]\n",
    "cnv_aggregated_filtered_mcnv.columns = [matching_id_to_individual.get(col, col) for col in cnv_aggregated_filtered_mcnv.columns]\n",
    "cnv_aggregated_filtered_del"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>cnvdata_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R3978789</td>\n",
       "      <td>SM-CTDSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R6108690</td>\n",
       "      <td>SM-CJGMZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R6013281</td>\n",
       "      <td>SM-CJFM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R2367199</td>\n",
       "      <td>SM-CTEET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R4323608</td>\n",
       "      <td>SM-CJK4L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>R3008520</td>\n",
       "      <td>SM-CTDQZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>R3151599</td>\n",
       "      <td>SM-CJIY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1124</th>\n",
       "      <td>R7993799</td>\n",
       "      <td>SM-CTEN7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1125</th>\n",
       "      <td>R9907075</td>\n",
       "      <td>SM-CJGH9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1126</th>\n",
       "      <td>R5026720</td>\n",
       "      <td>SM-CTEFP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1127 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     individualID cnvdata_id\n",
       "0        R3978789   SM-CTDSC\n",
       "1        R6108690   SM-CJGMZ\n",
       "2        R6013281   SM-CJFM3\n",
       "3        R2367199   SM-CTEET\n",
       "4        R4323608   SM-CJK4L\n",
       "...           ...        ...\n",
       "1122     R3008520   SM-CTDQZ\n",
       "1123     R3151599   SM-CJIY3\n",
       "1124     R7993799   SM-CTEN7\n",
       "1125     R9907075   SM-CJGH9\n",
       "1126     R5026720   SM-CTEFP\n",
       "\n",
       "[1127 rows x 2 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnvdata_id = cnv_aggregated_del.columns.tolist()[1:]\n",
    "individual_ids = cnv_aggregated_filtered_del.columns.tolist()[1:]\n",
    "\n",
    "cnv_aggregated_map = pd.DataFrame({\n",
    "    'individualID': individual_ids,\n",
    "    'cnvdata_id': cnvdata_id\n",
    "})\n",
    "\n",
    "cnv_aggregated_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>mirna_id</th>\n",
       "      <th>mwas_id</th>\n",
       "      <th>mrna_id</th>\n",
       "      <th>cnvdata_id</th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R8316516</td>\n",
       "      <td>b01.127C</td>\n",
       "      <td>TBI-AUTO73025-PT-319J</td>\n",
       "      <td>279_120430_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33477756.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R1822146</td>\n",
       "      <td>b01.127N</td>\n",
       "      <td>PT-M5AF</td>\n",
       "      <td>575_120521_2</td>\n",
       "      <td>SM-CTDQQ</td>\n",
       "      <td>20271359.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R3512949</td>\n",
       "      <td>b01.128C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SM-CTDVR</td>\n",
       "      <td>10100736.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R3143439</td>\n",
       "      <td>b01.128N</td>\n",
       "      <td>PT-BZBT</td>\n",
       "      <td>502_120515_1</td>\n",
       "      <td>SM-CTEMJ</td>\n",
       "      <td>38967303.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R4590536</td>\n",
       "      <td>b01.129C</td>\n",
       "      <td>PT-BZLT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SM-CJFNQ</td>\n",
       "      <td>46291609.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3628</th>\n",
       "      <td>R2268751</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75861964.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3629</th>\n",
       "      <td>R5306025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22207815.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3630</th>\n",
       "      <td>R6142763</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22207941.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3631</th>\n",
       "      <td>R4468842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>49333806.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3632</th>\n",
       "      <td>R9446033</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59720188.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3633 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     individualID  mirna_id                mwas_id       mrna_id cnvdata_id  \\\n",
       "0        R8316516  b01.127C  TBI-AUTO73025-PT-319J  279_120430_3        NaN   \n",
       "1        R1822146  b01.127N                PT-M5AF  575_120521_2   SM-CTDQQ   \n",
       "2        R3512949  b01.128C                    NaN           NaN   SM-CTDVR   \n",
       "3        R3143439  b01.128N                PT-BZBT  502_120515_1   SM-CTEMJ   \n",
       "4        R4590536  b01.129C                PT-BZLT           NaN   SM-CJFNQ   \n",
       "...           ...       ...                    ...           ...        ...   \n",
       "3628     R2268751       NaN                    NaN           NaN        NaN   \n",
       "3629     R5306025       NaN                    NaN           NaN        NaN   \n",
       "3630     R6142763       NaN                    NaN           NaN        NaN   \n",
       "3631     R4468842       NaN                    NaN           NaN        NaN   \n",
       "3632     R9446033       NaN                    NaN           NaN        NaN   \n",
       "\n",
       "          projid Study  \n",
       "0     33477756.0   MAP  \n",
       "1     20271359.0   ROS  \n",
       "2     10100736.0   ROS  \n",
       "3     38967303.0   MAP  \n",
       "4     46291609.0   MAP  \n",
       "...          ...   ...  \n",
       "3628  75861964.0   MAP  \n",
       "3629  22207815.0   ROS  \n",
       "3630  22207941.0   ROS  \n",
       "3631  49333806.0   MAP  \n",
       "3632  59720188.0   MAP  \n",
       "\n",
       "[3633 rows x 7 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_individualID = pd.concat([\n",
    "    protein_map['individualID'],\n",
    "    methylation_map['individualID'],\n",
    "    gene_expression_map['individualID'],\n",
    "    # snphead_updated_withoutNA['individualID'],\n",
    "    cnv_aggregated_map['individualID'],\n",
    "    survival['individualID']\n",
    "]).unique()\n",
    "\n",
    "# Now we create a new DataFrame that contains all unique 'individualID' and their corresponding values\n",
    "# from the other columns in the original DataFrames.\n",
    "# We perform an outer merge to ensure all unique 'individualID' are included.\n",
    "union_map = pd.DataFrame(combined_individualID, columns=['individualID'])\n",
    "\n",
    "# Merge with each original DataFrame\n",
    "union_map = union_map.merge(protein_map, on='individualID', how='outer')\n",
    "union_map = union_map.merge(methylation_map, on='individualID', how='outer')\n",
    "union_map = union_map.merge(gene_expression_map, on='individualID', how='outer')\n",
    "union_map = union_map.merge(cnv_aggregated_map, on='individualID', how='outer')\n",
    "union_map = union_map.merge(survival[['individualID', 'projid', 'Study']], on='individualID', how='outer')\n",
    "\n",
    "# Display the merged DataFrame\n",
    "union_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>mirna_id</th>\n",
       "      <th>mwas_id</th>\n",
       "      <th>mrna_id</th>\n",
       "      <th>cnvdata_id</th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R1822146</td>\n",
       "      <td>b01.127N</td>\n",
       "      <td>PT-M5AF</td>\n",
       "      <td>575_120521_2</td>\n",
       "      <td>SM-CTDQQ</td>\n",
       "      <td>20271359.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R3143439</td>\n",
       "      <td>b01.128N</td>\n",
       "      <td>PT-BZBT</td>\n",
       "      <td>502_120515_1</td>\n",
       "      <td>SM-CTEMJ</td>\n",
       "      <td>38967303.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R6879714</td>\n",
       "      <td>b01.130C</td>\n",
       "      <td>PT-3PTN</td>\n",
       "      <td>660_120530_1</td>\n",
       "      <td>SM-CTEEU</td>\n",
       "      <td>65736039.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R8963331</td>\n",
       "      <td>b01.130N</td>\n",
       "      <td>PT-M5GT</td>\n",
       "      <td>607_120523_2</td>\n",
       "      <td>SM-CJEKL</td>\n",
       "      <td>20197364.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R5656511</td>\n",
       "      <td>b02.128C</td>\n",
       "      <td>PT-BYE5</td>\n",
       "      <td>551_120517_2</td>\n",
       "      <td>SM-CTEFZ</td>\n",
       "      <td>67185070.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>R9781891</td>\n",
       "      <td>b48.130N</td>\n",
       "      <td>PT-BZGE</td>\n",
       "      <td>522_120515_1</td>\n",
       "      <td>SM-CTEFJ</td>\n",
       "      <td>83034844.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>R6759986</td>\n",
       "      <td>b49.127N</td>\n",
       "      <td>PT-BZD3</td>\n",
       "      <td>489_120515_1</td>\n",
       "      <td>SM-CTDRI</td>\n",
       "      <td>98953007.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>R2516394</td>\n",
       "      <td>b49.128C</td>\n",
       "      <td>PT-BZBN</td>\n",
       "      <td>398_120503_0</td>\n",
       "      <td>SM-CJEJG</td>\n",
       "      <td>50403446.0</td>\n",
       "      <td>MAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>R9286126</td>\n",
       "      <td>b49.128N</td>\n",
       "      <td>PT-M5Q5</td>\n",
       "      <td>589_120522_6</td>\n",
       "      <td>SM-CJEK7</td>\n",
       "      <td>20584923.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>R9680160</td>\n",
       "      <td>b49.130C</td>\n",
       "      <td>PT-M5H5</td>\n",
       "      <td>602_120523_2</td>\n",
       "      <td>SM-CJIXP</td>\n",
       "      <td>20501668.0</td>\n",
       "      <td>ROS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID  mirna_id  mwas_id       mrna_id cnvdata_id      projid Study\n",
       "0       R1822146  b01.127N  PT-M5AF  575_120521_2   SM-CTDQQ  20271359.0   ROS\n",
       "1       R3143439  b01.128N  PT-BZBT  502_120515_1   SM-CTEMJ  38967303.0   MAP\n",
       "2       R6879714  b01.130C  PT-3PTN  660_120530_1   SM-CTEEU  65736039.0   MAP\n",
       "3       R8963331  b01.130N  PT-M5GT  607_120523_2   SM-CJEKL  20197364.0   ROS\n",
       "4       R5656511  b02.128C  PT-BYE5  551_120517_2   SM-CTEFZ  67185070.0   MAP\n",
       "..           ...       ...      ...           ...        ...         ...   ...\n",
       "133     R9781891  b48.130N  PT-BZGE  522_120515_1   SM-CTEFJ  83034844.0   MAP\n",
       "134     R6759986  b49.127N  PT-BZD3  489_120515_1   SM-CTDRI  98953007.0   MAP\n",
       "135     R2516394  b49.128C  PT-BZBN  398_120503_0   SM-CJEJG  50403446.0   MAP\n",
       "136     R9286126  b49.128N  PT-M5Q5  589_120522_6   SM-CJEK7  20584923.0   ROS\n",
       "137     R9680160  b49.130C  PT-M5H5  602_120523_2   SM-CJIXP  20501668.0   ROS\n",
       "\n",
       "[138 rows x 7 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "union_map_cleaned = union_map.dropna()\n",
    "union_map_cleaned.reset_index(drop=True, inplace=True)\n",
    "union_map_cleaned"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4.3 Unify patient samples within methylation, copynumer,  gene expression, clinical, proteomics, molecular subtype, and sample type, primary disease datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_first_ad_dx</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_first_ad_dx</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "      <th>individualID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10101589</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R6939144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>86767530</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R3893503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9650662</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.850000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R8937093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>50402855</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R7139444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20544321</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R4971237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3579</th>\n",
       "      <td>22207815</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>57.653661875427787</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R5306025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3580</th>\n",
       "      <td>22207941</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>56.651608487337441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R6142763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3581</th>\n",
       "      <td>49333806</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.599589322381931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R4468842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3582</th>\n",
       "      <td>59720188</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.622861054072551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R9446033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3583</th>\n",
       "      <td>90214403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R5615105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3584 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0     10101589   ROS   1.0  20.0   1.0      2.0           34.0   \n",
       "1     86767530   MAP   0.0  10.0   1.0      2.0           33.0   \n",
       "2      9650662   MAP   0.0  15.0   1.0      2.0           23.0   \n",
       "3     50402855   MAP   0.0  21.0   1.0      2.0           33.0   \n",
       "4     20544321   ROS   0.0  16.0   1.0      2.0           23.0   \n",
       "...        ...   ...   ...   ...   ...      ...            ...   \n",
       "3579  22207815   ROS   0.0  18.0   2.0      2.0           23.0   \n",
       "3580  22207941   ROS   0.0  16.0   2.0      2.0           34.0   \n",
       "3581  49333806   MAP   0.0  12.0   2.0      2.0            NaN   \n",
       "3582  59720188   MAP   0.0  13.0   1.0      1.0            NaN   \n",
       "3583  90214403   NaN   NaN   NaN   NaN      NaN            NaN   \n",
       "\n",
       "        age_at_visit_max age_first_ad_dx age_death  cts_mmse30_first_ad_dx  \\\n",
       "0                    90+             90+       90+                    18.0   \n",
       "1                    90+             90+       90+                    18.0   \n",
       "2                    90+             90+       90+                     0.0   \n",
       "3                    90+             NaN       NaN                     NaN   \n",
       "4                    90+             90+       NaN                    13.0   \n",
       "...                  ...             ...       ...                     ...   \n",
       "3579  57.653661875427787             NaN       NaN                     NaN   \n",
       "3580  56.651608487337441             NaN       NaN                     NaN   \n",
       "3581  56.599589322381931             NaN       NaN                     NaN   \n",
       "3582  54.622861054072551             NaN       NaN                     NaN   \n",
       "3583                 NaN             NaN       NaN                     NaN   \n",
       "\n",
       "      cts_mmse30_lv       pmi  braaksc  ceradsc  cogdx  dcfdx_lv individualID  \n",
       "0               5.0  9.916667      4.0      2.0    4.0       4.0     R6939144  \n",
       "1              10.0  6.500000      4.0      2.0    4.0       4.0     R3893503  \n",
       "2               0.0  3.850000      3.0      2.0    4.0       4.0     R8937093  \n",
       "3              27.0       NaN      NaN      NaN    NaN       1.0     R7139444  \n",
       "4              14.0       NaN      NaN      NaN    NaN       4.0     R4971237  \n",
       "...             ...       ...      ...      ...    ...       ...          ...  \n",
       "3579           29.0       NaN      NaN      NaN    NaN       1.0     R5306025  \n",
       "3580           27.0       NaN      NaN      NaN    NaN       1.0     R6142763  \n",
       "3581           30.0       NaN      NaN      NaN    NaN       1.0     R4468842  \n",
       "3582           29.0       NaN      NaN      NaN    NaN       1.0     R9446033  \n",
       "3583            NaN       NaN      NaN      NaN    NaN       NaN     R5615105  \n",
       "\n",
       "[3584 rows x 18 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#clinical data\n",
    "survival"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R8316516</th>\n",
       "      <th>R1822146</th>\n",
       "      <th>R3512949</th>\n",
       "      <th>R3143439</th>\n",
       "      <th>R4590536</th>\n",
       "      <th>R9679238</th>\n",
       "      <th>R6879714</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R4182458</th>\n",
       "      <th>...</th>\n",
       "      <th>R9680160</th>\n",
       "      <th>R7015250</th>\n",
       "      <th>R5676537</th>\n",
       "      <th>R2087569</th>\n",
       "      <th>R2421650</th>\n",
       "      <th>R1498848</th>\n",
       "      <th>R9794121</th>\n",
       "      <th>R7298955</th>\n",
       "      <th>R6084846</th>\n",
       "      <th>R9798367</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-0.103582</td>\n",
       "      <td>0.015336</td>\n",
       "      <td>-0.060367</td>\n",
       "      <td>-0.301662</td>\n",
       "      <td>-0.176597</td>\n",
       "      <td>-0.168287</td>\n",
       "      <td>-0.083194</td>\n",
       "      <td>-0.200645</td>\n",
       "      <td>-0.648434</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.286807</td>\n",
       "      <td>-0.162889</td>\n",
       "      <td>-0.183929</td>\n",
       "      <td>-0.135078</td>\n",
       "      <td>0.057625</td>\n",
       "      <td>-0.115102</td>\n",
       "      <td>-0.273137</td>\n",
       "      <td>-0.154150</td>\n",
       "      <td>-0.212377</td>\n",
       "      <td>-0.080477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.013649</td>\n",
       "      <td>0.120403</td>\n",
       "      <td>-0.160178</td>\n",
       "      <td>0.060566</td>\n",
       "      <td>0.062877</td>\n",
       "      <td>0.029663</td>\n",
       "      <td>0.029067</td>\n",
       "      <td>0.176011</td>\n",
       "      <td>0.115946</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.054777</td>\n",
       "      <td>0.024013</td>\n",
       "      <td>-0.099587</td>\n",
       "      <td>0.306420</td>\n",
       "      <td>-0.147124</td>\n",
       "      <td>0.145804</td>\n",
       "      <td>0.025654</td>\n",
       "      <td>0.047217</td>\n",
       "      <td>-0.050547</td>\n",
       "      <td>0.074370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-0.200711</td>\n",
       "      <td>-0.201481</td>\n",
       "      <td>-0.243198</td>\n",
       "      <td>-0.030152</td>\n",
       "      <td>-0.051683</td>\n",
       "      <td>-0.362033</td>\n",
       "      <td>-0.040971</td>\n",
       "      <td>-0.223601</td>\n",
       "      <td>-0.031392</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.363212</td>\n",
       "      <td>-0.072299</td>\n",
       "      <td>-0.024155</td>\n",
       "      <td>-0.232771</td>\n",
       "      <td>-0.205745</td>\n",
       "      <td>-0.336676</td>\n",
       "      <td>-0.135752</td>\n",
       "      <td>-0.333979</td>\n",
       "      <td>-0.213806</td>\n",
       "      <td>-0.128683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>-0.045958</td>\n",
       "      <td>0.237595</td>\n",
       "      <td>-0.010871</td>\n",
       "      <td>-0.001613</td>\n",
       "      <td>0.007412</td>\n",
       "      <td>0.115163</td>\n",
       "      <td>0.092460</td>\n",
       "      <td>0.038121</td>\n",
       "      <td>-0.098167</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003519</td>\n",
       "      <td>0.126243</td>\n",
       "      <td>-0.037437</td>\n",
       "      <td>-0.096267</td>\n",
       "      <td>0.065899</td>\n",
       "      <td>-0.051790</td>\n",
       "      <td>0.024523</td>\n",
       "      <td>0.003781</td>\n",
       "      <td>0.024393</td>\n",
       "      <td>0.040647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0.239321</td>\n",
       "      <td>-0.171347</td>\n",
       "      <td>-0.020115</td>\n",
       "      <td>0.072067</td>\n",
       "      <td>0.024964</td>\n",
       "      <td>0.027992</td>\n",
       "      <td>0.084575</td>\n",
       "      <td>0.015419</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.473441</td>\n",
       "      <td>0.229008</td>\n",
       "      <td>0.078061</td>\n",
       "      <td>0.159639</td>\n",
       "      <td>0.092709</td>\n",
       "      <td>0.033288</td>\n",
       "      <td>-0.112379</td>\n",
       "      <td>-0.070789</td>\n",
       "      <td>-0.068971</td>\n",
       "      <td>0.002309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-0.023356</td>\n",
       "      <td>0.318564</td>\n",
       "      <td>0.248660</td>\n",
       "      <td>0.036006</td>\n",
       "      <td>0.319589</td>\n",
       "      <td>0.168960</td>\n",
       "      <td>0.145361</td>\n",
       "      <td>0.072773</td>\n",
       "      <td>0.304339</td>\n",
       "      <td>...</td>\n",
       "      <td>0.107408</td>\n",
       "      <td>0.262709</td>\n",
       "      <td>0.139146</td>\n",
       "      <td>0.168117</td>\n",
       "      <td>0.053220</td>\n",
       "      <td>0.234324</td>\n",
       "      <td>0.129981</td>\n",
       "      <td>0.247600</td>\n",
       "      <td>0.157816</td>\n",
       "      <td>0.132918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 401 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R8316516  R1822146  R3512949  R3143439  R4590536  R9679238  \\\n",
       "0         ABL1 -0.103582  0.015336 -0.060367 -0.301662 -0.176597 -0.168287   \n",
       "1         ABL2  0.013649  0.120403 -0.160178  0.060566  0.062877  0.029663   \n",
       "2        ACAA1 -0.200711 -0.201481 -0.243198 -0.030152 -0.051683 -0.362033   \n",
       "3        ACACA -0.045958  0.237595 -0.010871 -0.001613  0.007412  0.115163   \n",
       "4        ACACB  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16  0.239321 -0.171347 -0.020115  0.072067  0.024964  0.027992   \n",
       "2140    ZFYVE9 -0.023356  0.318564  0.248660  0.036006  0.319589  0.168960   \n",
       "2141     ZMAT3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "      R6879714  R8963331  R4182458  ...  R9680160  R7015250  R5676537  \\\n",
       "0    -0.083194 -0.200645 -0.648434  ... -0.286807 -0.162889 -0.183929   \n",
       "1     0.029067  0.176011  0.115946  ... -0.054777  0.024013 -0.099587   \n",
       "2    -0.040971 -0.223601 -0.031392  ... -0.363212 -0.072299 -0.024155   \n",
       "3     0.092460  0.038121 -0.098167  ...  0.003519  0.126243 -0.037437   \n",
       "4     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.084575  0.015419  0.000000  ...  0.473441  0.229008  0.078061   \n",
       "2140  0.145361  0.072773  0.304339  ...  0.107408  0.262709  0.139146   \n",
       "2141  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "\n",
       "      R2087569  R2421650  R1498848  R9794121  R7298955  R6084846  R9798367  \n",
       "0    -0.135078  0.057625 -0.115102 -0.273137 -0.154150 -0.212377 -0.080477  \n",
       "1     0.306420 -0.147124  0.145804  0.025654  0.047217 -0.050547  0.074370  \n",
       "2    -0.232771 -0.205745 -0.336676 -0.135752 -0.333979 -0.213806 -0.128683  \n",
       "3    -0.096267  0.065899 -0.051790  0.024523  0.003781  0.024393  0.040647  \n",
       "4     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.159639  0.092709  0.033288 -0.112379 -0.070789 -0.068971  0.002309  \n",
       "2140  0.168117  0.053220  0.234324  0.129981  0.247600  0.157816  0.132918  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "\n",
       "[2144 rows x 401 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#proteomics data\n",
    "protein_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common R columns: 437\n"
     ]
    }
   ],
   "source": [
    "# Extract column names starting with 'R' from methylation datasets\n",
    "R_columns_upstream = [col for col in Upstream_df_filtered.columns if col.startswith('R')]\n",
    "R_columns_distal = [col for col in Distal_Promoter_df_filtered.columns if col.startswith('R')]\n",
    "R_columns_proximal = [col for col in Proximal_Promoter_df_filtered.columns if col.startswith('R')]\n",
    "R_columns_core = [col for col in Core_Promoter_df_filtered.columns if col.startswith('R')]\n",
    "R_columns_downstream = [col for col in Downstream_df_filtered.columns if col.startswith('R')]\n",
    "R_columns_cnv =[col for col in cnv_aggregated_filtered_del.columns if col.startswith('R')]\n",
    "# Extract 'R' columns from other datasets\n",
    "R_columns_gene_expression = [col for col in gene_expression_filtered.columns if col != 'gene_name']\n",
    "R_columns_survival = [col for col in survival['individualID'] if col.startswith('R')]\n",
    "R_columns_protein = [col for col in protein_filtered.columns if col.startswith('R')]\n",
    "# Find the intersection of R column names across all DataFrames\n",
    "\n",
    "# TODO\n",
    "# common_R_columns = set(R_columns_upstream) & set(R_columns_distal) & set(R_columns_proximal) & set(R_columns_core) & set(R_columns_downstream) &set(R_columns_gene_expression) &set(R_columns_survival) &set(R_columns_protein) & set(R_columns_cnv)\n",
    "common_R_columns = set(R_columns_upstream) & set(R_columns_distal) & set(R_columns_proximal) & set(R_columns_core) & set(R_columns_downstream) &set(R_columns_gene_expression) &set(R_columns_survival) & set(R_columns_cnv)\n",
    "\n",
    "# Convert the intersection back to a list, if needed\n",
    "common_R_columns_list = list(common_R_columns)\n",
    "\n",
    "# Print the number and the list of common R columns\n",
    "print(f\"Number of common R columns: {len(common_R_columns)}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R3978789\n",
      "Methylation Dataset has: 740\n",
      "After intersection with survival: 740\n",
      "Copy Number Variations Dataset has: 1102\n",
      "After intersection with survival: 1082\n",
      "RNASeq Dataset has: 640\n",
      "After intersection with survival 636\n"
     ]
    }
   ],
   "source": [
    "print(R_columns_downstream[0])\n",
    "print(f\"Methylation Dataset has: {len(R_columns_downstream)}\")\n",
    "R_columns_downstream_withsurvival = set(R_columns_downstream) & set(R_columns_survival)\n",
    "print(f\"After intersection with survival: {len(R_columns_downstream_withsurvival)}\")\n",
    "\n",
    "print(f\"Copy Number Variations Dataset has: {len(R_columns_cnv)}\")\n",
    "R_columns_cnv_withsurvival = set(R_columns_cnv) & set(R_columns_survival)\n",
    "print(f\"After intersection with survival: {len(R_columns_cnv_withsurvival)}\")\n",
    "\n",
    "print(f\"RNASeq Dataset has: {len(R_columns_gene_expression)}\")\n",
    "R_columns_gene_expression_withsurvival = set(R_columns_gene_expression) & set(R_columns_survival)\n",
    "print(f\"After intersection with survival {len(R_columns_gene_expression_withsurvival)}\")\n",
    "\n",
    "# print(f\"Proteomics Dataset has: {len(R_columns_protein)}\")\n",
    "# R_columns_protein_withsurvival = set(R_columns_protein) & set(R_columns_survival)\n",
    "# print(f\"After intersection with survival: {len(R_columns_protein_withsurvival)}\")\n",
    "# R_columns_cnv, \n",
    "# R_columns_gene_expression, \n",
    "# R_columns_survival, \n",
    "# R_columns_protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import venn\n",
    "# labels = venn.get_labels([\n",
    "#     R_columns_cnv_withsurvival,\n",
    "#     R_columns_gene_expression_withsurvival,\n",
    "#     # R_columns_protein_withsurvival,\n",
    "#     R_columns_downstream_withsurvival\n",
    "# ], fill=['number'])\n",
    "# fig, ax = venn.venn4(labels, names=['Copy Number Variations Dataset', 'RNASeq Dataset', 'Proteomics Dataset','Methylation Dataset'])\n",
    "# fig.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "# R_columns_cnv_withsurvival\n",
    "# R_columns_gene_expression_withsurvival\n",
    "# R_columns_protein_withsurvival\n",
    "# R_columns_downstream_withsurvival\n",
    "\n",
    "# intersections = {\n",
    "#     'CNV∩RNASeq': R_columns_cnv_withsurvival & R_columns_gene_expression_withsurvival,\n",
    "#     'CNV∩Proteomics': R_columns_cnv_withsurvival & R_columns_protein_withsurvival,\n",
    "#     'CNV∩Methylation': R_columns_cnv_withsurvival & R_columns_downstream_withsurvival,\n",
    "#     'RNASeq∩Proteomics': R_columns_gene_expression_withsurvival & R_columns_protein_withsurvival,\n",
    "#     'RNASeq∩Methylation': R_columns_gene_expression_withsurvival & R_columns_downstream_withsurvival,\n",
    "#     'Proteomics∩Methylation': R_columns_protein_withsurvival & R_columns_downstream_withsurvival,\n",
    "#     'CNV∩RNASeq∩Proteomics': R_columns_cnv_withsurvival & R_columns_gene_expression_withsurvival & R_columns_protein_withsurvival,\n",
    "#     'CNV∩RNASeq∩Methylation': R_columns_cnv_withsurvival & R_columns_gene_expression_withsurvival & R_columns_downstream_withsurvival,\n",
    "#     'CNV∩Proteomics∩Methylation': R_columns_cnv_withsurvival & R_columns_protein_withsurvival & R_columns_downstream_withsurvival,\n",
    "#     'RNASeq∩Proteomics∩Methylation': R_columns_gene_expression_withsurvival & R_columns_protein_withsurvival & R_columns_downstream_withsurvival,\n",
    "# }\n",
    "\n",
    "# for name, intersection in intersections.items():\n",
    "#     print(f\"{name}: {len(intersection)} elements\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from upsetplot import plot\n",
    "# import pandas as pd\n",
    "# import matplotlib.pyplot as plt\n",
    "\n",
    "# # Sample data: replace these with your actual sets\n",
    "# sets = {\n",
    "#     'Copy Number Variations Dataset': R_columns_cnv_withsurvival,\n",
    "#     'RNASeq Dataset': R_columns_gene_expression_withsurvival,\n",
    "#     'Proteomics Dataset': R_columns_protein_withsurvival,\n",
    "#     'Methylation Dataset': R_columns_downstream_withsurvival\n",
    "# }\n",
    "\n",
    "# # Identify all unique elements across all sets\n",
    "# all_elements = set.union(*sets.values())\n",
    "\n",
    "# # Transform the data for UpSetPlot\n",
    "# data = []\n",
    "# for element in all_elements:\n",
    "#     data.append([\n",
    "#         element in sets['Copy Number Variations Dataset'],\n",
    "#         element in sets['RNASeq Dataset'],\n",
    "#         element in sets['Proteomics Dataset'],\n",
    "#         element in sets['Methylation Dataset']\n",
    "#     ])\n",
    "\n",
    "# # Create a DataFrame\n",
    "# df = pd.DataFrame(data, columns=['Copy Number Variations Dataset', 'RNASeq Dataset', 'Proteomics Dataset', 'Methylation Dataset'])\n",
    "\n",
    "# # Convert boolean values to integers\n",
    "# df = df.astype(int)\n",
    "\n",
    "# # Prepare the DataFrame for UpSetPlot\n",
    "# df_upset = df.groupby(list(df.columns)).size()\n",
    "\n",
    "# # Plot using UpSetPlot\n",
    "# upset_axes = plot(df_upset, orientation=\"vertical\", show_counts=True,element_size=40)\n",
    "# # totals_ax = upset_axes['totals']\n",
    "# # totals_ax.tick_params(labelleft=False, left=False)\n",
    "# # totals_ax.tick_params(bottom=False, labelbottom=False)\n",
    "# # for spine in totals_ax.spines.values():\n",
    "# #     spine.set_visible(False)\n",
    "# # plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define columns to keep along with common R columns\n",
    "additional_cols_methylation = ['gene_name']\n",
    "\n",
    "# Filter each methylation DataFrame\n",
    "Upstream_df_filtered = Upstream_df_filtered[additional_cols_methylation + common_R_columns_list]\n",
    "Distal_Promoter_df_filtered = Distal_Promoter_df_filtered[additional_cols_methylation + common_R_columns_list]\n",
    "Proximal_Promoter_df_filtered = Proximal_Promoter_df_filtered[additional_cols_methylation + common_R_columns_list]\n",
    "Core_Promoter_df_filtered = Core_Promoter_df_filtered[additional_cols_methylation + common_R_columns_list]\n",
    "Downstream_df_filtered = Downstream_df_filtered[additional_cols_methylation + common_R_columns_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R7096488</th>\n",
       "      <th>R3077672</th>\n",
       "      <th>R3877498</th>\n",
       "      <th>R5079107</th>\n",
       "      <th>R4570065</th>\n",
       "      <th>R2880377</th>\n",
       "      <th>R9072353</th>\n",
       "      <th>R9489952</th>\n",
       "      <th>R1133844</th>\n",
       "      <th>...</th>\n",
       "      <th>R1980623</th>\n",
       "      <th>R6132310</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R1620679</th>\n",
       "      <th>R8457289</th>\n",
       "      <th>R3910349</th>\n",
       "      <th>R8553415</th>\n",
       "      <th>R2731324</th>\n",
       "      <th>R5883172</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>0.077372</td>\n",
       "      <td>0.080380</td>\n",
       "      <td>0.080127</td>\n",
       "      <td>0.090941</td>\n",
       "      <td>0.095504</td>\n",
       "      <td>0.079865</td>\n",
       "      <td>0.071615</td>\n",
       "      <td>0.06277</td>\n",
       "      <td>0.094777</td>\n",
       "      <td>...</td>\n",
       "      <td>0.079385</td>\n",
       "      <td>0.067713</td>\n",
       "      <td>0.063793</td>\n",
       "      <td>0.089984</td>\n",
       "      <td>0.083661</td>\n",
       "      <td>0.082494</td>\n",
       "      <td>0.077761</td>\n",
       "      <td>0.068745</td>\n",
       "      <td>0.078200</td>\n",
       "      <td>0.069772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036919</td>\n",
       "      <td>0.064893</td>\n",
       "      <td>0.075258</td>\n",
       "      <td>0.063847</td>\n",
       "      <td>0.080252</td>\n",
       "      <td>0.052336</td>\n",
       "      <td>0.02243</td>\n",
       "      <td>0.076898</td>\n",
       "      <td>...</td>\n",
       "      <td>0.044564</td>\n",
       "      <td>0.053097</td>\n",
       "      <td>0.041645</td>\n",
       "      <td>0.043139</td>\n",
       "      <td>0.056224</td>\n",
       "      <td>0.053140</td>\n",
       "      <td>0.057161</td>\n",
       "      <td>0.049193</td>\n",
       "      <td>0.052876</td>\n",
       "      <td>0.063180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R7096488  R3077672  R3877498  R5079107  R4570065  R2880377  \\\n",
       "0         ABL1  0.077372  0.080380  0.080127  0.090941  0.095504  0.079865   \n",
       "1         ABL2  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2        ACAA1  0.000000  0.036919  0.064893  0.075258  0.063847  0.080252   \n",
       "3        ACACA  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "4        ACACB  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2140    ZFYVE9  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2141     ZMAT3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "      R9072353  R9489952  R1133844  ...  R1980623  R6132310  R8963331  \\\n",
       "0     0.071615   0.06277  0.094777  ...  0.079385  0.067713  0.063793   \n",
       "1     0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2     0.052336   0.02243  0.076898  ...  0.044564  0.053097  0.041645   \n",
       "3     0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "4     0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2140  0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2141  0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.000000   0.00000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "\n",
       "      R9817056  R1620679  R8457289  R3910349  R8553415  R2731324  R5883172  \n",
       "0     0.089984  0.083661  0.082494  0.077761  0.068745  0.078200  0.069772  \n",
       "1     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2     0.043139  0.056224  0.053140  0.057161  0.049193  0.052876  0.063180  \n",
       "3     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "4     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2140  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Core_Promoter_df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R7096488</th>\n",
       "      <th>R3077672</th>\n",
       "      <th>R3877498</th>\n",
       "      <th>R5079107</th>\n",
       "      <th>R4570065</th>\n",
       "      <th>R2880377</th>\n",
       "      <th>R9072353</th>\n",
       "      <th>R9489952</th>\n",
       "      <th>R1133844</th>\n",
       "      <th>...</th>\n",
       "      <th>R1980623</th>\n",
       "      <th>R6132310</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R1620679</th>\n",
       "      <th>R8457289</th>\n",
       "      <th>R3910349</th>\n",
       "      <th>R8553415</th>\n",
       "      <th>R2731324</th>\n",
       "      <th>R5883172</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>0.859431</td>\n",
       "      <td>0.859822</td>\n",
       "      <td>0.882562</td>\n",
       "      <td>0.890605</td>\n",
       "      <td>0.892634</td>\n",
       "      <td>0.960634</td>\n",
       "      <td>0.859633</td>\n",
       "      <td>0.888154</td>\n",
       "      <td>0.860513</td>\n",
       "      <td>...</td>\n",
       "      <td>0.809126</td>\n",
       "      <td>0.907031</td>\n",
       "      <td>0.901879</td>\n",
       "      <td>0.872847</td>\n",
       "      <td>0.809050</td>\n",
       "      <td>0.877340</td>\n",
       "      <td>0.893433</td>\n",
       "      <td>0.904675</td>\n",
       "      <td>0.836060</td>\n",
       "      <td>0.888120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.361222</td>\n",
       "      <td>0.305814</td>\n",
       "      <td>0.349697</td>\n",
       "      <td>0.253362</td>\n",
       "      <td>0.226666</td>\n",
       "      <td>0.281904</td>\n",
       "      <td>0.258209</td>\n",
       "      <td>0.287913</td>\n",
       "      <td>0.273697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.238627</td>\n",
       "      <td>0.261479</td>\n",
       "      <td>0.276413</td>\n",
       "      <td>0.244424</td>\n",
       "      <td>0.241519</td>\n",
       "      <td>0.280974</td>\n",
       "      <td>0.220202</td>\n",
       "      <td>0.232087</td>\n",
       "      <td>0.282009</td>\n",
       "      <td>0.309036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0.786661</td>\n",
       "      <td>0.685183</td>\n",
       "      <td>0.695781</td>\n",
       "      <td>0.751262</td>\n",
       "      <td>0.629690</td>\n",
       "      <td>0.729570</td>\n",
       "      <td>0.709658</td>\n",
       "      <td>0.772072</td>\n",
       "      <td>0.708518</td>\n",
       "      <td>...</td>\n",
       "      <td>0.726929</td>\n",
       "      <td>0.726180</td>\n",
       "      <td>0.725093</td>\n",
       "      <td>0.648021</td>\n",
       "      <td>0.714814</td>\n",
       "      <td>0.715148</td>\n",
       "      <td>0.659504</td>\n",
       "      <td>0.746394</td>\n",
       "      <td>0.763735</td>\n",
       "      <td>0.742810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>0.884775</td>\n",
       "      <td>0.841959</td>\n",
       "      <td>0.878475</td>\n",
       "      <td>0.842841</td>\n",
       "      <td>0.830034</td>\n",
       "      <td>0.844307</td>\n",
       "      <td>0.856236</td>\n",
       "      <td>0.865060</td>\n",
       "      <td>0.841633</td>\n",
       "      <td>...</td>\n",
       "      <td>0.847574</td>\n",
       "      <td>0.848256</td>\n",
       "      <td>0.874510</td>\n",
       "      <td>0.830433</td>\n",
       "      <td>0.840246</td>\n",
       "      <td>0.847951</td>\n",
       "      <td>0.863596</td>\n",
       "      <td>0.845876</td>\n",
       "      <td>0.869048</td>\n",
       "      <td>0.879700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0.886397</td>\n",
       "      <td>0.913772</td>\n",
       "      <td>0.902243</td>\n",
       "      <td>0.895470</td>\n",
       "      <td>0.915496</td>\n",
       "      <td>0.865307</td>\n",
       "      <td>0.906011</td>\n",
       "      <td>0.899601</td>\n",
       "      <td>0.860599</td>\n",
       "      <td>...</td>\n",
       "      <td>0.863926</td>\n",
       "      <td>0.902570</td>\n",
       "      <td>0.911511</td>\n",
       "      <td>0.891111</td>\n",
       "      <td>0.848410</td>\n",
       "      <td>0.872636</td>\n",
       "      <td>0.868855</td>\n",
       "      <td>0.879771</td>\n",
       "      <td>0.899417</td>\n",
       "      <td>0.934926</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R7096488  R3077672  R3877498  R5079107  R4570065  R2880377  \\\n",
       "0         ABL1  0.859431  0.859822  0.882562  0.890605  0.892634  0.960634   \n",
       "1         ABL2  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2        ACAA1  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "3        ACACA  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "4        ACACB  0.361222  0.305814  0.349697  0.253362  0.226666  0.281904   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16  0.786661  0.685183  0.695781  0.751262  0.629690  0.729570   \n",
       "2140    ZFYVE9  0.884775  0.841959  0.878475  0.842841  0.830034  0.844307   \n",
       "2141     ZMAT3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3  0.886397  0.913772  0.902243  0.895470  0.915496  0.865307   \n",
       "\n",
       "      R9072353  R9489952  R1133844  ...  R1980623  R6132310  R8963331  \\\n",
       "0     0.859633  0.888154  0.860513  ...  0.809126  0.907031  0.901879   \n",
       "1     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "3     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "4     0.258209  0.287913  0.273697  ...  0.238627  0.261479  0.276413   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.709658  0.772072  0.708518  ...  0.726929  0.726180  0.725093   \n",
       "2140  0.856236  0.865060  0.841633  ...  0.847574  0.848256  0.874510   \n",
       "2141  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.906011  0.899601  0.860599  ...  0.863926  0.902570  0.911511   \n",
       "\n",
       "      R9817056  R1620679  R8457289  R3910349  R8553415  R2731324  R5883172  \n",
       "0     0.872847  0.809050  0.877340  0.893433  0.904675  0.836060  0.888120  \n",
       "1     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "3     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "4     0.244424  0.241519  0.280974  0.220202  0.232087  0.282009  0.309036  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.648021  0.714814  0.715148  0.659504  0.746394  0.763735  0.742810  \n",
       "2140  0.830433  0.840246  0.847951  0.863596  0.845876  0.869048  0.879700  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.891111  0.848410  0.872636  0.868855  0.879771  0.899417  0.934926  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Upstream_df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R7096488</th>\n",
       "      <th>R3077672</th>\n",
       "      <th>R3877498</th>\n",
       "      <th>R5079107</th>\n",
       "      <th>R4570065</th>\n",
       "      <th>R2880377</th>\n",
       "      <th>R9072353</th>\n",
       "      <th>R9489952</th>\n",
       "      <th>R1133844</th>\n",
       "      <th>...</th>\n",
       "      <th>R1980623</th>\n",
       "      <th>R6132310</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R1620679</th>\n",
       "      <th>R8457289</th>\n",
       "      <th>R3910349</th>\n",
       "      <th>R8553415</th>\n",
       "      <th>R2731324</th>\n",
       "      <th>R5883172</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>10.90</td>\n",
       "      <td>8.91</td>\n",
       "      <td>9.73</td>\n",
       "      <td>8.55</td>\n",
       "      <td>9.66</td>\n",
       "      <td>11.71</td>\n",
       "      <td>10.14</td>\n",
       "      <td>9.52</td>\n",
       "      <td>8.97</td>\n",
       "      <td>...</td>\n",
       "      <td>9.61</td>\n",
       "      <td>9.36</td>\n",
       "      <td>9.31</td>\n",
       "      <td>10.43</td>\n",
       "      <td>7.02</td>\n",
       "      <td>11.15</td>\n",
       "      <td>5.44</td>\n",
       "      <td>9.60</td>\n",
       "      <td>9.13</td>\n",
       "      <td>9.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>2.43</td>\n",
       "      <td>5.27</td>\n",
       "      <td>3.07</td>\n",
       "      <td>4.52</td>\n",
       "      <td>4.44</td>\n",
       "      <td>5.46</td>\n",
       "      <td>5.44</td>\n",
       "      <td>4.87</td>\n",
       "      <td>5.26</td>\n",
       "      <td>...</td>\n",
       "      <td>5.05</td>\n",
       "      <td>4.31</td>\n",
       "      <td>5.29</td>\n",
       "      <td>3.80</td>\n",
       "      <td>3.38</td>\n",
       "      <td>4.50</td>\n",
       "      <td>4.07</td>\n",
       "      <td>4.54</td>\n",
       "      <td>3.85</td>\n",
       "      <td>5.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>37.03</td>\n",
       "      <td>48.11</td>\n",
       "      <td>29.13</td>\n",
       "      <td>40.61</td>\n",
       "      <td>58.54</td>\n",
       "      <td>43.43</td>\n",
       "      <td>43.13</td>\n",
       "      <td>61.92</td>\n",
       "      <td>57.68</td>\n",
       "      <td>...</td>\n",
       "      <td>44.80</td>\n",
       "      <td>52.56</td>\n",
       "      <td>46.22</td>\n",
       "      <td>55.52</td>\n",
       "      <td>26.33</td>\n",
       "      <td>45.76</td>\n",
       "      <td>57.02</td>\n",
       "      <td>56.28</td>\n",
       "      <td>61.66</td>\n",
       "      <td>46.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>11.89</td>\n",
       "      <td>5.33</td>\n",
       "      <td>5.24</td>\n",
       "      <td>5.88</td>\n",
       "      <td>8.82</td>\n",
       "      <td>13.80</td>\n",
       "      <td>5.22</td>\n",
       "      <td>6.82</td>\n",
       "      <td>8.62</td>\n",
       "      <td>...</td>\n",
       "      <td>5.30</td>\n",
       "      <td>7.17</td>\n",
       "      <td>6.12</td>\n",
       "      <td>10.01</td>\n",
       "      <td>6.70</td>\n",
       "      <td>8.08</td>\n",
       "      <td>1.91</td>\n",
       "      <td>5.59</td>\n",
       "      <td>7.07</td>\n",
       "      <td>10.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>6.78</td>\n",
       "      <td>9.38</td>\n",
       "      <td>3.45</td>\n",
       "      <td>5.56</td>\n",
       "      <td>6.21</td>\n",
       "      <td>10.64</td>\n",
       "      <td>7.40</td>\n",
       "      <td>6.32</td>\n",
       "      <td>7.77</td>\n",
       "      <td>...</td>\n",
       "      <td>7.79</td>\n",
       "      <td>4.06</td>\n",
       "      <td>4.88</td>\n",
       "      <td>8.11</td>\n",
       "      <td>3.17</td>\n",
       "      <td>6.61</td>\n",
       "      <td>6.06</td>\n",
       "      <td>6.92</td>\n",
       "      <td>6.11</td>\n",
       "      <td>4.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>4.90</td>\n",
       "      <td>12.26</td>\n",
       "      <td>6.36</td>\n",
       "      <td>13.29</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.90</td>\n",
       "      <td>15.80</td>\n",
       "      <td>14.49</td>\n",
       "      <td>14.99</td>\n",
       "      <td>...</td>\n",
       "      <td>13.86</td>\n",
       "      <td>10.20</td>\n",
       "      <td>10.54</td>\n",
       "      <td>8.19</td>\n",
       "      <td>3.80</td>\n",
       "      <td>7.76</td>\n",
       "      <td>20.49</td>\n",
       "      <td>10.76</td>\n",
       "      <td>8.20</td>\n",
       "      <td>7.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>9.04</td>\n",
       "      <td>13.85</td>\n",
       "      <td>11.83</td>\n",
       "      <td>14.57</td>\n",
       "      <td>12.26</td>\n",
       "      <td>15.10</td>\n",
       "      <td>17.08</td>\n",
       "      <td>14.22</td>\n",
       "      <td>13.72</td>\n",
       "      <td>...</td>\n",
       "      <td>12.92</td>\n",
       "      <td>9.84</td>\n",
       "      <td>12.74</td>\n",
       "      <td>9.50</td>\n",
       "      <td>7.91</td>\n",
       "      <td>8.74</td>\n",
       "      <td>15.44</td>\n",
       "      <td>12.09</td>\n",
       "      <td>8.53</td>\n",
       "      <td>9.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>2.74</td>\n",
       "      <td>3.18</td>\n",
       "      <td>2.16</td>\n",
       "      <td>2.71</td>\n",
       "      <td>2.75</td>\n",
       "      <td>3.49</td>\n",
       "      <td>2.65</td>\n",
       "      <td>2.89</td>\n",
       "      <td>3.06</td>\n",
       "      <td>...</td>\n",
       "      <td>2.74</td>\n",
       "      <td>2.79</td>\n",
       "      <td>2.51</td>\n",
       "      <td>4.16</td>\n",
       "      <td>1.13</td>\n",
       "      <td>2.69</td>\n",
       "      <td>2.09</td>\n",
       "      <td>3.33</td>\n",
       "      <td>2.56</td>\n",
       "      <td>2.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>8.87</td>\n",
       "      <td>3.33</td>\n",
       "      <td>4.06</td>\n",
       "      <td>6.85</td>\n",
       "      <td>6.76</td>\n",
       "      <td>8.73</td>\n",
       "      <td>5.32</td>\n",
       "      <td>5.32</td>\n",
       "      <td>4.13</td>\n",
       "      <td>...</td>\n",
       "      <td>4.66</td>\n",
       "      <td>4.37</td>\n",
       "      <td>3.95</td>\n",
       "      <td>6.22</td>\n",
       "      <td>2.98</td>\n",
       "      <td>5.14</td>\n",
       "      <td>0.84</td>\n",
       "      <td>5.19</td>\n",
       "      <td>1.95</td>\n",
       "      <td>6.64</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R7096488  R3077672  R3877498  R5079107  R4570065  R2880377  \\\n",
       "0         ABL1     10.90      8.91      9.73      8.55      9.66     11.71   \n",
       "1         ABL2      2.43      5.27      3.07      4.52      4.44      5.46   \n",
       "2        ACAA1     37.03     48.11     29.13     40.61     58.54     43.43   \n",
       "3        ACACA      0.00      0.00      0.00      0.00      0.00      0.00   \n",
       "4        ACACB     11.89      5.33      5.24      5.88      8.82     13.80   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      6.78      9.38      3.45      5.56      6.21     10.64   \n",
       "2140    ZFYVE9      4.90     12.26      6.36     13.29     13.27     13.90   \n",
       "2141     ZMAT3      9.04     13.85     11.83     14.57     12.26     15.10   \n",
       "2142    ZNF274      2.74      3.18      2.16      2.71      2.75      3.49   \n",
       "2143     ZNRF3      8.87      3.33      4.06      6.85      6.76      8.73   \n",
       "\n",
       "      R9072353  R9489952  R1133844  ...  R1980623  R6132310  R8963331  \\\n",
       "0        10.14      9.52      8.97  ...      9.61      9.36      9.31   \n",
       "1         5.44      4.87      5.26  ...      5.05      4.31      5.29   \n",
       "2        43.13     61.92     57.68  ...     44.80     52.56     46.22   \n",
       "3         0.00      0.00      0.00  ...      0.00      0.00      0.00   \n",
       "4         5.22      6.82      8.62  ...      5.30      7.17      6.12   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139      7.40      6.32      7.77  ...      7.79      4.06      4.88   \n",
       "2140     15.80     14.49     14.99  ...     13.86     10.20     10.54   \n",
       "2141     17.08     14.22     13.72  ...     12.92      9.84     12.74   \n",
       "2142      2.65      2.89      3.06  ...      2.74      2.79      2.51   \n",
       "2143      5.32      5.32      4.13  ...      4.66      4.37      3.95   \n",
       "\n",
       "      R9817056  R1620679  R8457289  R3910349  R8553415  R2731324  R5883172  \n",
       "0        10.43      7.02     11.15      5.44      9.60      9.13      9.38  \n",
       "1         3.80      3.38      4.50      4.07      4.54      3.85      5.10  \n",
       "2        55.52     26.33     45.76     57.02     56.28     61.66     46.77  \n",
       "3         0.00      0.00      0.00      0.00      0.00      0.00      0.00  \n",
       "4        10.01      6.70      8.08      1.91      5.59      7.07     10.16  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139      8.11      3.17      6.61      6.06      6.92      6.11      4.59  \n",
       "2140      8.19      3.80      7.76     20.49     10.76      8.20      7.11  \n",
       "2141      9.50      7.91      8.74     15.44     12.09      8.53      9.43  \n",
       "2142      4.16      1.13      2.69      2.09      3.33      2.56      2.55  \n",
       "2143      6.22      2.98      5.14      0.84      5.19      1.95      6.64  \n",
       "\n",
       "[2144 rows x 440 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define columns to keep along with common R columns\n",
    "additional_cols_gene_expression = ['gene_name']\n",
    "\n",
    "# Filter the gene expression DataFrame\n",
    "gene_expression_filtered = gene_expression_filtered[additional_cols_gene_expression + common_R_columns_list]\n",
    "gene_expression_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_44640\\2279412070.py:11: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  protein_filtered[col] = 0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R7096488</th>\n",
       "      <th>R3077672</th>\n",
       "      <th>R3877498</th>\n",
       "      <th>R5079107</th>\n",
       "      <th>R4570065</th>\n",
       "      <th>R2880377</th>\n",
       "      <th>R9072353</th>\n",
       "      <th>R9489952</th>\n",
       "      <th>R1133844</th>\n",
       "      <th>...</th>\n",
       "      <th>R1980623</th>\n",
       "      <th>R6132310</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R1620679</th>\n",
       "      <th>R8457289</th>\n",
       "      <th>R3910349</th>\n",
       "      <th>R8553415</th>\n",
       "      <th>R2731324</th>\n",
       "      <th>R5883172</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R7096488  R3077672  R3877498  R5079107  R4570065  R2880377  \\\n",
       "0         ABL1         0         0         0         0         0         0   \n",
       "1         ABL2         0         0         0         0         0         0   \n",
       "2        ACAA1         0         0         0         0         0         0   \n",
       "3        ACACA         0         0         0         0         0         0   \n",
       "4        ACACB         0         0         0         0         0         0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16         0         0         0         0         0         0   \n",
       "2140    ZFYVE9         0         0         0         0         0         0   \n",
       "2141     ZMAT3         0         0         0         0         0         0   \n",
       "2142    ZNF274         0         0         0         0         0         0   \n",
       "2143     ZNRF3         0         0         0         0         0         0   \n",
       "\n",
       "      R9072353  R9489952  R1133844  ...  R1980623  R6132310  R8963331  \\\n",
       "0            0         0         0  ...         0         0         0   \n",
       "1            0         0         0  ...         0         0         0   \n",
       "2            0         0         0  ...         0         0         0   \n",
       "3            0         0         0  ...         0         0         0   \n",
       "4            0         0         0  ...         0         0         0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139         0         0         0  ...         0         0         0   \n",
       "2140         0         0         0  ...         0         0         0   \n",
       "2141         0         0         0  ...         0         0         0   \n",
       "2142         0         0         0  ...         0         0         0   \n",
       "2143         0         0         0  ...         0         0         0   \n",
       "\n",
       "      R9817056  R1620679  R8457289  R3910349  R8553415  R2731324  R5883172  \n",
       "0            0         0         0         0         0         0         0  \n",
       "1            0         0         0         0         0         0         0  \n",
       "2            0         0         0         0         0         0         0  \n",
       "3            0         0         0         0         0         0         0  \n",
       "4            0         0         0         0         0         0         0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139         0         0         0         0         0         0         0  \n",
       "2140         0         0         0         0         0         0         0  \n",
       "2141         0         0         0         0         0         0         0  \n",
       "2142         0         0         0         0         0         0         0  \n",
       "2143         0         0         0         0         0         0         0  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define columns to keep along with common R columns\n",
    "additional_cols_protein = ['gene_name']\n",
    "\n",
    "# # Filter the protein DataFrame\n",
    "# protein_filtered = protein_filtered[additional_cols_protein + common_R_columns_list]\n",
    "# protein_filtered\n",
    "\n",
    "protein_filtered = protein_filtered[additional_cols_protein].copy()\n",
    "for col in common_R_columns_list:\n",
    "    if col not in protein_filtered.columns:\n",
    "        protein_filtered[col] = 0\n",
    "protein_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R7096488</th>\n",
       "      <th>R3077672</th>\n",
       "      <th>R3877498</th>\n",
       "      <th>R5079107</th>\n",
       "      <th>R4570065</th>\n",
       "      <th>R2880377</th>\n",
       "      <th>R9072353</th>\n",
       "      <th>R9489952</th>\n",
       "      <th>R1133844</th>\n",
       "      <th>...</th>\n",
       "      <th>R1980623</th>\n",
       "      <th>R6132310</th>\n",
       "      <th>R8963331</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R1620679</th>\n",
       "      <th>R8457289</th>\n",
       "      <th>R3910349</th>\n",
       "      <th>R8553415</th>\n",
       "      <th>R2731324</th>\n",
       "      <th>R5883172</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R7096488  R3077672  R3877498  R5079107  R4570065  R2880377  \\\n",
       "0         ABL1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "1         ABL2      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2        ACAA1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "3        ACACA       1.0       2.0       2.0       2.0       1.0       2.0   \n",
       "4        ACACB      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2140    ZFYVE9      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2141     ZMAT3      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2142    ZNF274      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2143     ZNRF3       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "\n",
       "      R9072353  R9489952  R1133844  ...  R1980623  R6132310  R8963331  \\\n",
       "0         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "1         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "3          2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "4         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2140      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2141      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2142      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2143       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "\n",
       "      R9817056  R1620679  R8457289  R3910349  R8553415  R2731324  R5883172  \n",
       "0         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "1         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "3          1.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "4         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2140      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2141      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2142      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2143       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define columns to keep along with common R columns\n",
    "additional_cols_protein = ['gene_name']\n",
    "\n",
    "# Filter the cnv DataFrame\n",
    "cnv_aggregated_filtered_del = cnv_aggregated_filtered_del[additional_cols_protein + common_R_columns_list]\n",
    "cnv_aggregated_filtered_del.reset_index(drop=True, inplace=True)\n",
    "\n",
    "cnv_aggregated_filtered_dup = cnv_aggregated_filtered_dup[additional_cols_protein + common_R_columns_list]\n",
    "cnv_aggregated_filtered_dup.reset_index(drop=True, inplace=True)\n",
    "\n",
    "cnv_aggregated_filtered_mcnv = cnv_aggregated_filtered_mcnv[additional_cols_protein + common_R_columns_list]\n",
    "cnv_aggregated_filtered_mcnv.reset_index(drop=True, inplace=True)\n",
    "\n",
    "cnv_aggregated_filtered_del"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_first_ad_dx</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_first_ad_dx</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "      <th>individualID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10101589</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R6939144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9650662</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.850000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R8937093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>20630946</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.900000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R4859661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10433070</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R1158585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>20969088</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>R2397348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3398</th>\n",
       "      <td>10850171</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.482546201232026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71.624914442162904</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>4.083333</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R8935140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3410</th>\n",
       "      <td>20656964</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.227926078028744</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71.310061601642715</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.800000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>R8704458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3456</th>\n",
       "      <td>11413170</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>69.733059548254616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70.269678302532512</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.250000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>R3769291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3498</th>\n",
       "      <td>24039289</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>68.440793976728273</td>\n",
       "      <td>NaN</td>\n",
       "      <td>72.147843942505133</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>5.833333</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>R7334123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3528</th>\n",
       "      <td>11631558</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>66.959616700889796</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67.370294318959623</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R7347788</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0     10101589   ROS   1.0  20.0   1.0      2.0           34.0   \n",
       "2      9650662   MAP   0.0  15.0   1.0      2.0           23.0   \n",
       "5     20630946   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "9     10433070   ROS   1.0  13.0   1.0      2.0           23.0   \n",
       "13    20969088   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "...        ...   ...   ...   ...   ...      ...            ...   \n",
       "3398  10850171   ROS   1.0  26.0   1.0      2.0           33.0   \n",
       "3410  20656964   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "3456  11413170   ROS   1.0  18.0   1.0      2.0           23.0   \n",
       "3498  24039289   MAP   1.0  18.0   1.0      2.0           23.0   \n",
       "3528  11631558   ROS   1.0  12.0   1.0      2.0           23.0   \n",
       "\n",
       "        age_at_visit_max age_first_ad_dx           age_death  \\\n",
       "0                    90+             90+                 90+   \n",
       "2                    90+             90+                 90+   \n",
       "5                    90+             90+                 90+   \n",
       "9                    90+             90+                 90+   \n",
       "13                   90+             NaN                 90+   \n",
       "...                  ...             ...                 ...   \n",
       "3398  71.482546201232026             NaN  71.624914442162904   \n",
       "3410  71.227926078028744             NaN  71.310061601642715   \n",
       "3456  69.733059548254616             NaN  70.269678302532512   \n",
       "3498  68.440793976728273             NaN  72.147843942505133   \n",
       "3528  66.959616700889796             NaN  67.370294318959623   \n",
       "\n",
       "      cts_mmse30_first_ad_dx  cts_mmse30_lv       pmi  braaksc  ceradsc  \\\n",
       "0                       18.0            5.0  9.916667      4.0      2.0   \n",
       "2                        0.0            0.0  3.850000      3.0      2.0   \n",
       "5                        NaN            0.0  2.900000      3.0      4.0   \n",
       "9                       18.0            8.0  3.500000      4.0      2.0   \n",
       "13                       NaN           23.0  9.916667      4.0      2.0   \n",
       "...                      ...            ...       ...      ...      ...   \n",
       "3398                     NaN           30.0  4.083333      1.0      4.0   \n",
       "3410                     NaN           29.0  7.800000      3.0      2.0   \n",
       "3456                     NaN            5.0  8.250000      3.0      4.0   \n",
       "3498                     NaN           26.0  5.833333      2.0      4.0   \n",
       "3528                     NaN           30.0  5.000000      1.0      4.0   \n",
       "\n",
       "      cogdx  dcfdx_lv individualID  \n",
       "0       4.0       4.0     R6939144  \n",
       "2       4.0       4.0     R8937093  \n",
       "5       4.0       4.0     R4859661  \n",
       "9       5.0       4.0     R1158585  \n",
       "13      2.0       2.0     R2397348  \n",
       "...     ...       ...          ...  \n",
       "3398    1.0       1.0     R8935140  \n",
       "3410    1.0       2.0     R8704458  \n",
       "3456    5.0       5.0     R3769291  \n",
       "3498    2.0       2.0     R7334123  \n",
       "3528    1.0       1.0     R7347788  \n",
       "\n",
       "[437 rows x 18 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "survival_filtered = survival[survival['individualID'].isin(common_R_columns_list)]\n",
    "survival_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "projid                    0.000000\n",
      "Study                     0.000000\n",
      "msex                      0.000000\n",
      "educ                      0.000000\n",
      "race                      0.000000\n",
      "spanish                   0.000000\n",
      "apoe_genotype             0.000000\n",
      "age_at_visit_max          0.000000\n",
      "age_first_ad_dx           0.672769\n",
      "age_death                 0.000000\n",
      "cts_mmse30_first_ad_dx    0.681922\n",
      "cts_mmse30_lv             0.000000\n",
      "pmi                       0.004577\n",
      "braaksc                   0.000000\n",
      "ceradsc                   0.000000\n",
      "cogdx                     0.000000\n",
      "dcfdx_lv                  0.000000\n",
      "individualID              0.000000\n",
      "dtype: float64\n"
     ]
    }
   ],
   "source": [
    "survival_nan_column_proportions = survival_filtered.isna().mean()\n",
    "\n",
    "# Display the results\n",
    "print(survival_nan_column_proportions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns dropped: ['age_first_ad_dx', 'cts_mmse30_first_ad_dx']\n"
     ]
    }
   ],
   "source": [
    "# Calculate the proportion of NaN values in each column\n",
    "survival_nan_column_proportions = survival_filtered.isna().mean()\n",
    "\n",
    "# Identify columns to be dropped (where proportion of NaN values is greater than 1/3)\n",
    "columns_to_drop = survival_nan_column_proportions[survival_nan_column_proportions > 1/3].index.tolist()\n",
    "\n",
    "# Drop these columns from the DataFrame\n",
    "survival_filtered = survival_filtered.drop(columns=columns_to_drop)\n",
    "\n",
    "# List of columns that were dropped\n",
    "print(\"Columns dropped:\", columns_to_drop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "      <th>individualID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10101589</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R6939144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9650662</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.850000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R8937093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>20630946</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.900000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R4859661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10433070</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R1158585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>20969088</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>23.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>R2397348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3398</th>\n",
       "      <td>10850171</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.482546201232026</td>\n",
       "      <td>71.624914442162904</td>\n",
       "      <td>30.0</td>\n",
       "      <td>4.083333</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R8935140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3410</th>\n",
       "      <td>20656964</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.227926078028744</td>\n",
       "      <td>71.310061601642715</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.800000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>R8704458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3456</th>\n",
       "      <td>11413170</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>69.733059548254616</td>\n",
       "      <td>70.269678302532512</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.250000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>R3769291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3498</th>\n",
       "      <td>24039289</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>68.440793976728273</td>\n",
       "      <td>72.147843942505133</td>\n",
       "      <td>26.0</td>\n",
       "      <td>5.833333</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>R7334123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3528</th>\n",
       "      <td>11631558</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>66.959616700889796</td>\n",
       "      <td>67.370294318959623</td>\n",
       "      <td>30.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R7347788</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0     10101589   ROS   1.0  20.0   1.0      2.0           34.0   \n",
       "2      9650662   MAP   0.0  15.0   1.0      2.0           23.0   \n",
       "5     20630946   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "9     10433070   ROS   1.0  13.0   1.0      2.0           23.0   \n",
       "13    20969088   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "...        ...   ...   ...   ...   ...      ...            ...   \n",
       "3398  10850171   ROS   1.0  26.0   1.0      2.0           33.0   \n",
       "3410  20656964   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "3456  11413170   ROS   1.0  18.0   1.0      2.0           23.0   \n",
       "3498  24039289   MAP   1.0  18.0   1.0      2.0           23.0   \n",
       "3528  11631558   ROS   1.0  12.0   1.0      2.0           23.0   \n",
       "\n",
       "        age_at_visit_max           age_death  cts_mmse30_lv       pmi  \\\n",
       "0                    90+                 90+            5.0  9.916667   \n",
       "2                    90+                 90+            0.0  3.850000   \n",
       "5                    90+                 90+            0.0  2.900000   \n",
       "9                    90+                 90+            8.0  3.500000   \n",
       "13                   90+                 90+           23.0  9.916667   \n",
       "...                  ...                 ...            ...       ...   \n",
       "3398  71.482546201232026  71.624914442162904           30.0  4.083333   \n",
       "3410  71.227926078028744  71.310061601642715           29.0  7.800000   \n",
       "3456  69.733059548254616  70.269678302532512            5.0  8.250000   \n",
       "3498  68.440793976728273  72.147843942505133           26.0  5.833333   \n",
       "3528  66.959616700889796  67.370294318959623           30.0  5.000000   \n",
       "\n",
       "      braaksc  ceradsc  cogdx  dcfdx_lv individualID  \n",
       "0         4.0      2.0    4.0       4.0     R6939144  \n",
       "2         3.0      2.0    4.0       4.0     R8937093  \n",
       "5         3.0      4.0    4.0       4.0     R4859661  \n",
       "9         4.0      2.0    5.0       4.0     R1158585  \n",
       "13        4.0      2.0    2.0       2.0     R2397348  \n",
       "...       ...      ...    ...       ...          ...  \n",
       "3398      1.0      4.0    1.0       1.0     R8935140  \n",
       "3410      3.0      2.0    1.0       2.0     R8704458  \n",
       "3456      3.0      4.0    5.0       5.0     R3769291  \n",
       "3498      2.0      4.0    2.0       2.0     R7334123  \n",
       "3528      1.0      4.0    1.0       1.0     R7347788  \n",
       "\n",
       "[437 rows x 16 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "survival_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R6939144</td>\n",
       "      <td>10101589</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R8937093</td>\n",
       "      <td>9650662</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.850000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R4859661</td>\n",
       "      <td>20630946</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.900000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R1158585</td>\n",
       "      <td>10433070</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R2397348</td>\n",
       "      <td>20969088</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>23.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>R8935140</td>\n",
       "      <td>10850171</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.482546201232026</td>\n",
       "      <td>71.624914442162904</td>\n",
       "      <td>30.0</td>\n",
       "      <td>4.083333</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>R8704458</td>\n",
       "      <td>20656964</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>71.227926078028744</td>\n",
       "      <td>71.310061601642715</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.800000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>R3769291</td>\n",
       "      <td>11413170</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>69.733059548254616</td>\n",
       "      <td>70.269678302532512</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.250000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>R7334123</td>\n",
       "      <td>24039289</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>68.440793976728273</td>\n",
       "      <td>72.147843942505133</td>\n",
       "      <td>26.0</td>\n",
       "      <td>5.833333</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>R7347788</td>\n",
       "      <td>11631558</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>66.959616700889796</td>\n",
       "      <td>67.370294318959623</td>\n",
       "      <td>30.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID    projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0       R6939144  10101589   ROS   1.0  20.0   1.0      2.0           34.0   \n",
       "1       R8937093   9650662   MAP   0.0  15.0   1.0      2.0           23.0   \n",
       "2       R4859661  20630946   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "3       R1158585  10433070   ROS   1.0  13.0   1.0      2.0           23.0   \n",
       "4       R2397348  20969088   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "..           ...       ...   ...   ...   ...   ...      ...            ...   \n",
       "432     R8935140  10850171   ROS   1.0  26.0   1.0      2.0           33.0   \n",
       "433     R8704458  20656964   ROS   0.0  18.0   1.0      2.0           33.0   \n",
       "434     R3769291  11413170   ROS   1.0  18.0   1.0      2.0           23.0   \n",
       "435     R7334123  24039289   MAP   1.0  18.0   1.0      2.0           23.0   \n",
       "436     R7347788  11631558   ROS   1.0  12.0   1.0      2.0           23.0   \n",
       "\n",
       "       age_at_visit_max           age_death  cts_mmse30_lv       pmi  braaksc  \\\n",
       "0                   90+                 90+            5.0  9.916667      4.0   \n",
       "1                   90+                 90+            0.0  3.850000      3.0   \n",
       "2                   90+                 90+            0.0  2.900000      3.0   \n",
       "3                   90+                 90+            8.0  3.500000      4.0   \n",
       "4                   90+                 90+           23.0  9.916667      4.0   \n",
       "..                  ...                 ...            ...       ...      ...   \n",
       "432  71.482546201232026  71.624914442162904           30.0  4.083333      1.0   \n",
       "433  71.227926078028744  71.310061601642715           29.0  7.800000      3.0   \n",
       "434  69.733059548254616  70.269678302532512            5.0  8.250000      3.0   \n",
       "435  68.440793976728273  72.147843942505133           26.0  5.833333      2.0   \n",
       "436  66.959616700889796  67.370294318959623           30.0  5.000000      1.0   \n",
       "\n",
       "     ceradsc  cogdx  dcfdx_lv  \n",
       "0        2.0    4.0       4.0  \n",
       "1        2.0    4.0       4.0  \n",
       "2        4.0    4.0       4.0  \n",
       "3        2.0    5.0       4.0  \n",
       "4        2.0    2.0       2.0  \n",
       "..       ...    ...       ...  \n",
       "432      4.0    1.0       1.0  \n",
       "433      2.0    1.0       2.0  \n",
       "434      4.0    5.0       5.0  \n",
       "435      4.0    2.0       2.0  \n",
       "436      4.0    1.0       1.0  \n",
       "\n",
       "[437 rows x 16 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = ['individualID'] + [col for col in survival_filtered.columns if col != 'individualID']\n",
    "survival_filtered = survival_filtered[cols]\n",
    "survival_filtered.reset_index(drop=True, inplace=True)\n",
    "survival_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5.Gene name/patient samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          ABL1\n",
       "1          ABL2\n",
       "2         ACAA1\n",
       "3         ACACA\n",
       "4         ACACB\n",
       "         ...   \n",
       "2139    ZFYVE16\n",
       "2140     ZFYVE9\n",
       "2141      ZMAT3\n",
       "2142     ZNF274\n",
       "2143      ZNRF3\n",
       "Name: gene_name, Length: 2144, dtype: object"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list = gene_expression_filtered['gene_name']\n",
    "gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R7096488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R3077672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R3877498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R5079107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R4570065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>R8457289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>R3910349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>R8553415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>R2731324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>R5883172</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample\n",
       "0    R7096488\n",
       "1    R3077672\n",
       "2    R3877498\n",
       "3    R5079107\n",
       "4    R4570065\n",
       "..        ...\n",
       "432  R8457289\n",
       "433  R3910349\n",
       "434  R8553415\n",
       "435  R2731324\n",
       "436  R5883172\n",
       "\n",
       "[437 rows x 1 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_sample_list = pd.DataFrame(common_R_columns,columns=['sample'])\n",
    "patient_sample_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6.Save processed datasets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6.1 Keep the consistency for dataframes on genes and samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3-TRAN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene\n",
       "0        ABL1-TRAN\n",
       "1        ABL2-TRAN\n",
       "2       ACAA1-TRAN\n",
       "3       ACACA-TRAN\n",
       "4       ACACB-TRAN\n",
       "...            ...\n",
       "2139  ZFYVE16-TRAN\n",
       "2140   ZFYVE9-TRAN\n",
       "2141    ZMAT3-TRAN\n",
       "2142   ZNF274-TRAN\n",
       "2143    ZNRF3-TRAN\n",
       "\n",
       "[2144 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3-METH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene\n",
       "0        ABL1-METH\n",
       "1        ABL2-METH\n",
       "2       ACAA1-METH\n",
       "3       ACACA-METH\n",
       "4       ACACB-METH\n",
       "...            ...\n",
       "2139  ZFYVE16-METH\n",
       "2140   ZFYVE9-METH\n",
       "2141    ZMAT3-METH\n",
       "2142   ZNF274-METH\n",
       "2143    ZNRF3-METH\n",
       "\n",
       "[2144 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene\n",
       "0        ABL1-PROT\n",
       "1        ABL2-PROT\n",
       "2       ACAA1-PROT\n",
       "3       ACACA-PROT\n",
       "4       ACACB-PROT\n",
       "...            ...\n",
       "2139  ZFYVE16-PROT\n",
       "2140   ZFYVE9-PROT\n",
       "2141    ZMAT3-PROT\n",
       "2142   ZNF274-PROT\n",
       "2143    ZNRF3-PROT\n",
       "\n",
       "[2144 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>ZFYVE16-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6428</th>\n",
       "      <td>ZFYVE9-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6429</th>\n",
       "      <td>ZMAT3-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6430</th>\n",
       "      <td>ZNF274-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6431</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6432 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene\n",
       "0        ABL1-TRAN\n",
       "1        ABL2-TRAN\n",
       "2       ACAA1-TRAN\n",
       "3       ACACA-TRAN\n",
       "4       ACACB-TRAN\n",
       "...            ...\n",
       "6427  ZFYVE16-PROT\n",
       "6428   ZFYVE9-PROT\n",
       "6429    ZMAT3-PROT\n",
       "6430   ZNF274-PROT\n",
       "6431    ZNRF3-PROT\n",
       "\n",
       "[6432 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# [gene_list]\n",
    "# gene-tran\n",
    "sorted_gene_list = gene_list.sort_values()\n",
    "sorted_gene = sorted_gene_list.tolist()\n",
    "sorted_gene_tran = [gene + '-TRAN' for gene in sorted_gene]\n",
    "sorted_gene_tran_df = pd.DataFrame(sorted_gene_tran, columns=['Gene'])\n",
    "display(sorted_gene_tran_df)\n",
    "# gene-meth\n",
    "sorted_gene_methy = [gene + '-METH' for gene in sorted_gene]\n",
    "sorted_gene_methy_df = pd.DataFrame(sorted_gene_methy, columns=['Gene'])\n",
    "display(sorted_gene_methy_df)\n",
    "# gene-protein\n",
    "sorted_gene_protein = [gene + '-PROT' for gene in sorted_gene]\n",
    "sorted_gene_protein_df = pd.DataFrame(sorted_gene_protein, columns=['Gene'])\n",
    "display(sorted_gene_protein_df)\n",
    "# all-gene\n",
    "sorted_gene_all = sorted_gene_tran + sorted_gene_methy + sorted_gene_protein\n",
    "sorted_all_gene_df = pd.DataFrame(sorted_gene_all, columns=['Gene'])\n",
    "display(sorted_all_gene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['R1015854', 'R1022980', 'R1028639', 'R1039781', 'R1042011', 'R1042566', 'R1074668', 'R1103580', 'R1105988', 'R1133844', 'R1154454', 'R1158585', 'R1173111', 'R1213305', 'R1218460', 'R1236313', 'R1262106', 'R1287407', 'R1344510', 'R1407047', 'R1476204', 'R1489314', 'R1521786', 'R1531359', 'R1547566', 'R1583702', 'R1588427', 'R1617674', 'R1620679', 'R1658200', 'R1660726', 'R1664857', 'R1672797', 'R1687970', 'R1710143', 'R1721075', 'R1743384', 'R1748114', 'R1777884', 'R1822146', 'R1830086', 'R1845714', 'R1888413', 'R1924801', 'R1944372', 'R1969233', 'R1977848', 'R1980623', 'R2006886', 'R2045909', 'R2079629', 'R2110067', 'R2111091', 'R2124216', 'R2140101', 'R2167020', 'R2197944', 'R2237803', 'R2264515', 'R2367199', 'R2372487', 'R2373966', 'R2386521', 'R2397348', 'R2455930', 'R2474257', 'R2488731', 'R2494273', 'R2516394', 'R2543886', 'R2554598', 'R2565200', 'R2575548', 'R2582129', 'R2616766', 'R2624931', 'R2645096', 'R2645101', 'R2649907', 'R2670295', 'R2678902', 'R2697774', 'R2703808', 'R2709337', 'R2716798', 'R2721311', 'R2731324', 'R2731764', 'R2752714', 'R2803490', 'R2859649', 'R2880377', 'R2881301', 'R2895885', 'R2992930', 'R3008520', 'R3035452', 'R3077672', 'R3081601', 'R3111222', 'R3128463', 'R3143439', 'R3160138', 'R3176125', 'R3185659', 'R3201576', 'R3257830', 'R3292822', 'R3322663', 'R3328752', 'R3328867', 'R3368249', 'R3391715', 'R3400726', 'R3408213', 'R3426726', 'R3442506', 'R3444124', 'R3452739', 'R3477250', 'R3485645', 'R3535957', 'R3553915', 'R3579460', 'R3589148', 'R3596834', 'R3607578', 'R3631183', 'R3722356', 'R3739042', 'R3740754', 'R3741788', 'R3744330', 'R3757880', 'R3769291', 'R3811781', 'R3825485', 'R3837901', 'R3841825', 'R3844037', 'R3857147', 'R3857927', 'R3874626', 'R3877498', 'R3884524', 'R3898113', 'R3900996', 'R3910349', 'R3917022', 'R3922205', 'R3927700', 'R3978789', 'R3984996', 'R3987433', 'R4014413', 'R4021549', 'R4022468', 'R4042599', 'R4053976', 'R4077358', 'R4078277', 'R4087141', 'R4087581', 'R4119160', 'R4147686', 'R4174623', 'R4242008', 'R4242773', 'R4249236', 'R4260171', 'R4276053', 'R4289838', 'R4332132', 'R4345802', 'R4361022', 'R4379962', 'R4415805', 'R4527133', 'R4531942', 'R4567280', 'R4570065', 'R4581557', 'R4619330', 'R4641987', 'R4762074', 'R4790045', 'R4817881', 'R4859661', 'R4900012', 'R4903779', 'R4917253', 'R4935546', 'R4937504', 'R4976287', 'R4986745', 'R4990009', 'R5026720', 'R5031238', 'R5053212', 'R5079107', 'R5079327', 'R5131375', 'R5158294', 'R5177066', 'R5184427', 'R5188003', 'R5196723', 'R5211056', 'R5221394', 'R5234179', 'R5256488', 'R5259690', 'R5334541', 'R5356410', 'R5357999', 'R5369295', 'R5374583', 'R5385855', 'R5405023', 'R5415701', 'R5447358', 'R5449861', 'R5454379', 'R5483154', 'R5489358', 'R5499256', 'R5508047', 'R5525186', 'R5541746', 'R5546461', 'R5557408', 'R5574987', 'R5636935', 'R5656511', 'R5683778', 'R5693901', 'R5694375', 'R5739959', 'R5766881', 'R5789564', 'R5850046', 'R5883172', 'R5907586', 'R5927382', 'R5935442', 'R5955028', 'R5965031', 'R5973191', 'R5992638', 'R6001210', 'R6057469', 'R6058053', 'R6108690', 'R6114572', 'R6132310', 'R6171213', 'R6226355', 'R6231758', 'R6249698', 'R6265038', 'R6267541', 'R6268135', 'R6292415', 'R6312023', 'R6319811', 'R6337324', 'R6346298', 'R6352950', 'R6392007', 'R6411801', 'R6415047', 'R6476704', 'R6487961', 'R6531744', 'R6536134', 'R6557859', 'R6604064', 'R6622577', 'R6636386', 'R6665276', 'R6681836', 'R6691514', 'R6698302', 'R6728038', 'R6739300', 'R6743664', 'R6751504', 'R6759986', 'R6765868', 'R6799253', 'R6820248', 'R6847727', 'R6849345', 'R6862468', 'R6879714', 'R6887989', 'R6911631', 'R6934314', 'R6939144', 'R7025378', 'R7043881', 'R7066784', 'R7090624', 'R7095349', 'R7096488', 'R7137826', 'R7141740', 'R7143693', 'R7157957', 'R7208968', 'R7210604', 'R7272395', 'R7295303', 'R7325259', 'R7334123', 'R7347788', 'R7384738', 'R7412731', 'R7438391', 'R7462671', 'R7483736', 'R7516349', 'R7551006', 'R7594705', 'R7596878', 'R7641350', 'R7696135', 'R7738727', 'R7767837', 'R7791442', 'R7796947', 'R7840275', 'R7851092', 'R7874995', 'R7876628', 'R7881801', 'R7903817', 'R7911202', 'R7912121', 'R7915228', 'R7959406', 'R7978618', 'R7993799', 'R7995082', 'R8044591', 'R8054504', 'R8059994', 'R8075334', 'R8155560', 'R8180874', 'R8209563', 'R8293796', 'R8329066', 'R8330118', 'R8351613', 'R8375980', 'R8410459', 'R8415954', 'R8432213', 'R8444624', 'R8451530', 'R8457289', 'R8466373', 'R8553415', 'R8553520', 'R8572172', 'R8573871', 'R8608442', 'R8615683', 'R8626395', 'R8629052', 'R8635714', 'R8647890', 'R8704458', 'R8724814', 'R8725848', 'R8741068', 'R8744945', 'R8759568', 'R8774534', 'R8781115', 'R8783288', 'R8799615', 'R8845676', 'R8850524', 'R8860977', 'R8882846', 'R8935140', 'R8937093', 'R8963331', 'R8971831', 'R9033345', 'R9047934', 'R9053141', 'R9072353', 'R9088015', 'R9101835', 'R9101940', 'R9113571', 'R9116123', 'R9127940', 'R9137173', 'R9145673', 'R9210731', 'R9225354', 'R9245150', 'R9253870', 'R9286126', 'R9328823', 'R9329742', 'R9330569', 'R9354381', 'R9393519', 'R9400562', 'R9419876', 'R9426782', 'R9473965', 'R9489952', 'R9506463', 'R9566976', 'R9582641', 'R9585968', 'R9596785', 'R9598418', 'R9649089', 'R9680160', 'R9771008', 'R9781891', 'R9809441', 'R9817056', 'R9817161', 'R9818080', 'R9904978', 'R9905342', 'R9907075', 'R9936070', 'R9976232']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R1015854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R1022980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R1028639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R1039781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R1042011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>R9904978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>R9905342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>R9907075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>R9936070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>R9976232</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample\n",
       "0    R1015854\n",
       "1    R1022980\n",
       "2    R1028639\n",
       "3    R1039781\n",
       "4    R1042011\n",
       "..        ...\n",
       "432  R9904978\n",
       "433  R9905342\n",
       "434  R9907075\n",
       "435  R9936070\n",
       "436  R9976232\n",
       "\n",
       "[437 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# [patient-sample-list]\n",
    "sorted_patient_sample_list = patient_sample_list.sort_values(by='sample')['sample'].tolist()\n",
    "print(sorted_patient_sample_list)\n",
    "sorted_patient_sample_df = patient_sample_list.sort_values(by='sample').reset_index(drop=True)\n",
    "display(sorted_patient_sample_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R1015854</th>\n",
       "      <th>R1022980</th>\n",
       "      <th>R1028639</th>\n",
       "      <th>R1039781</th>\n",
       "      <th>R1042011</th>\n",
       "      <th>R1042566</th>\n",
       "      <th>R1074668</th>\n",
       "      <th>R1103580</th>\n",
       "      <th>R1105988</th>\n",
       "      <th>...</th>\n",
       "      <th>R9781891</th>\n",
       "      <th>R9809441</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R9817161</th>\n",
       "      <th>R9818080</th>\n",
       "      <th>R9904978</th>\n",
       "      <th>R9905342</th>\n",
       "      <th>R9907075</th>\n",
       "      <th>R9936070</th>\n",
       "      <th>R9976232</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>0.826545</td>\n",
       "      <td>0.830970</td>\n",
       "      <td>0.916602</td>\n",
       "      <td>0.968890</td>\n",
       "      <td>0.840924</td>\n",
       "      <td>0.905204</td>\n",
       "      <td>0.864399</td>\n",
       "      <td>0.915589</td>\n",
       "      <td>0.828570</td>\n",
       "      <td>...</td>\n",
       "      <td>0.890607</td>\n",
       "      <td>0.895296</td>\n",
       "      <td>0.872847</td>\n",
       "      <td>0.847487</td>\n",
       "      <td>0.877945</td>\n",
       "      <td>0.857498</td>\n",
       "      <td>0.824360</td>\n",
       "      <td>0.771866</td>\n",
       "      <td>0.848104</td>\n",
       "      <td>0.853716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.283792</td>\n",
       "      <td>0.313389</td>\n",
       "      <td>0.280595</td>\n",
       "      <td>0.269576</td>\n",
       "      <td>0.268002</td>\n",
       "      <td>0.278162</td>\n",
       "      <td>0.273308</td>\n",
       "      <td>0.285687</td>\n",
       "      <td>0.225393</td>\n",
       "      <td>...</td>\n",
       "      <td>0.249913</td>\n",
       "      <td>0.304661</td>\n",
       "      <td>0.244424</td>\n",
       "      <td>0.286038</td>\n",
       "      <td>0.247225</td>\n",
       "      <td>0.294531</td>\n",
       "      <td>0.297333</td>\n",
       "      <td>0.295928</td>\n",
       "      <td>0.284207</td>\n",
       "      <td>0.279544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0.666868</td>\n",
       "      <td>0.741976</td>\n",
       "      <td>0.702788</td>\n",
       "      <td>0.703254</td>\n",
       "      <td>0.708944</td>\n",
       "      <td>0.802478</td>\n",
       "      <td>0.728516</td>\n",
       "      <td>0.679412</td>\n",
       "      <td>0.532274</td>\n",
       "      <td>...</td>\n",
       "      <td>0.669224</td>\n",
       "      <td>0.689250</td>\n",
       "      <td>0.648021</td>\n",
       "      <td>0.683317</td>\n",
       "      <td>0.720930</td>\n",
       "      <td>0.693378</td>\n",
       "      <td>0.741921</td>\n",
       "      <td>0.718989</td>\n",
       "      <td>0.788046</td>\n",
       "      <td>0.753159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>0.862777</td>\n",
       "      <td>0.848637</td>\n",
       "      <td>0.880283</td>\n",
       "      <td>0.870562</td>\n",
       "      <td>0.857823</td>\n",
       "      <td>0.866217</td>\n",
       "      <td>0.862784</td>\n",
       "      <td>0.848932</td>\n",
       "      <td>0.846581</td>\n",
       "      <td>...</td>\n",
       "      <td>0.859689</td>\n",
       "      <td>0.879186</td>\n",
       "      <td>0.830433</td>\n",
       "      <td>0.879998</td>\n",
       "      <td>0.864329</td>\n",
       "      <td>0.841689</td>\n",
       "      <td>0.854218</td>\n",
       "      <td>0.876408</td>\n",
       "      <td>0.873013</td>\n",
       "      <td>0.859260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0.856126</td>\n",
       "      <td>0.923398</td>\n",
       "      <td>0.889105</td>\n",
       "      <td>0.928353</td>\n",
       "      <td>0.852230</td>\n",
       "      <td>0.908245</td>\n",
       "      <td>0.879845</td>\n",
       "      <td>0.880875</td>\n",
       "      <td>0.904502</td>\n",
       "      <td>...</td>\n",
       "      <td>0.862413</td>\n",
       "      <td>0.889804</td>\n",
       "      <td>0.891111</td>\n",
       "      <td>0.940654</td>\n",
       "      <td>0.868550</td>\n",
       "      <td>0.893651</td>\n",
       "      <td>0.912273</td>\n",
       "      <td>0.914288</td>\n",
       "      <td>0.877461</td>\n",
       "      <td>0.885304</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R1015854  R1022980  R1028639  R1039781  R1042011  R1042566  \\\n",
       "0         ABL1  0.826545  0.830970  0.916602  0.968890  0.840924  0.905204   \n",
       "1         ABL2  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2        ACAA1  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "3        ACACA  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "4        ACACB  0.283792  0.313389  0.280595  0.269576  0.268002  0.278162   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16  0.666868  0.741976  0.702788  0.703254  0.708944  0.802478   \n",
       "2140    ZFYVE9  0.862777  0.848637  0.880283  0.870562  0.857823  0.866217   \n",
       "2141     ZMAT3  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2142    ZNF274  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "2143     ZNRF3  0.856126  0.923398  0.889105  0.928353  0.852230  0.908245   \n",
       "\n",
       "      R1074668  R1103580  R1105988  ...  R9781891  R9809441  R9817056  \\\n",
       "0     0.864399  0.915589  0.828570  ...  0.890607  0.895296  0.872847   \n",
       "1     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "3     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "4     0.273308  0.285687  0.225393  ...  0.249913  0.304661  0.244424   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139  0.728516  0.679412  0.532274  ...  0.669224  0.689250  0.648021   \n",
       "2140  0.862784  0.848932  0.846581  ...  0.859689  0.879186  0.830433   \n",
       "2141  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2142  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "2143  0.879845  0.880875  0.904502  ...  0.862413  0.889804  0.891111   \n",
       "\n",
       "      R9817161  R9818080  R9904978  R9905342  R9907075  R9936070  R9976232  \n",
       "0     0.847487  0.877945  0.857498  0.824360  0.771866  0.848104  0.853716  \n",
       "1     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "3     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "4     0.286038  0.247225  0.294531  0.297333  0.295928  0.284207  0.279544  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139  0.683317  0.720930  0.693378  0.741921  0.718989  0.788046  0.753159  \n",
       "2140  0.879998  0.864329  0.841689  0.854218  0.876408  0.873013  0.859260  \n",
       "2141  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2142  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "2143  0.940654  0.868550  0.893651  0.912273  0.914288  0.877461  0.885304  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Upstream_df_filtered = Upstream_df_filtered[['gene_name'] + sorted_patient_sample_list]\n",
    "Distal_Promoter_df_filtered = Distal_Promoter_df_filtered[['gene_name'] + sorted_patient_sample_list]\n",
    "Proximal_Promoter_df_filtered = Proximal_Promoter_df_filtered[['gene_name'] + sorted_patient_sample_list]\n",
    "Core_Promoter_df_filtered = Core_Promoter_df_filtered[['gene_name'] + sorted_patient_sample_list]\n",
    "Downstream_df_filtered = Downstream_df_filtered[['gene_name'] + sorted_patient_sample_list]\n",
    "\n",
    "Upstream_df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R1015854</th>\n",
       "      <th>R1022980</th>\n",
       "      <th>R1028639</th>\n",
       "      <th>R1039781</th>\n",
       "      <th>R1042011</th>\n",
       "      <th>R1042566</th>\n",
       "      <th>R1074668</th>\n",
       "      <th>R1103580</th>\n",
       "      <th>R1105988</th>\n",
       "      <th>...</th>\n",
       "      <th>R9781891</th>\n",
       "      <th>R9809441</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R9817161</th>\n",
       "      <th>R9818080</th>\n",
       "      <th>R9904978</th>\n",
       "      <th>R9905342</th>\n",
       "      <th>R9907075</th>\n",
       "      <th>R9936070</th>\n",
       "      <th>R9976232</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R1015854  R1022980  R1028639  R1039781  R1042011  R1042566  \\\n",
       "0         ABL1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "1         ABL2      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2        ACAA1      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "3        ACACA       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "4        ACACB      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2140    ZFYVE9      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2141     ZMAT3      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2142    ZNF274      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0   \n",
       "2143     ZNRF3       2.0       2.0       2.0       2.0       2.0       2.0   \n",
       "\n",
       "      R1074668  R1103580  R1105988  ...  R9781891  R9809441  R9817056  \\\n",
       "0         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "1         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "3          2.0       2.0       2.0  ...       2.0       2.0       1.0   \n",
       "4         -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2140      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2141      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2142      -1.0      -1.0      -1.0  ...      -1.0      -1.0      -1.0   \n",
       "2143       2.0       2.0       2.0  ...       2.0       2.0       2.0   \n",
       "\n",
       "      R9817161  R9818080  R9904978  R9905342  R9907075  R9936070  R9976232  \n",
       "0         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "1         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "3          2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "4         -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2140      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2141      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2142      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0      -1.0  \n",
       "2143       2.0       2.0       2.0       2.0       2.0       2.0       2.0  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv_aggregated_filtered_del = cnv_aggregated_filtered_del[['gene_name'] + sorted_patient_sample_list].sort_values(by='gene_name').reset_index(drop=True)\n",
    "cnv_aggregated_filtered_dup = cnv_aggregated_filtered_dup[['gene_name'] + sorted_patient_sample_list].sort_values(by='gene_name').reset_index(drop=True)\n",
    "cnv_aggregated_filtered_mcnv = cnv_aggregated_filtered_mcnv[['gene_name'] + sorted_patient_sample_list].sort_values(by='gene_name').reset_index(drop=True)\n",
    "\n",
    "cnv_aggregated_filtered_del"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R1015854</th>\n",
       "      <th>R1022980</th>\n",
       "      <th>R1028639</th>\n",
       "      <th>R1039781</th>\n",
       "      <th>R1042011</th>\n",
       "      <th>R1042566</th>\n",
       "      <th>R1074668</th>\n",
       "      <th>R1103580</th>\n",
       "      <th>R1105988</th>\n",
       "      <th>...</th>\n",
       "      <th>R9781891</th>\n",
       "      <th>R9809441</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R9817161</th>\n",
       "      <th>R9818080</th>\n",
       "      <th>R9904978</th>\n",
       "      <th>R9905342</th>\n",
       "      <th>R9907075</th>\n",
       "      <th>R9936070</th>\n",
       "      <th>R9976232</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>10.45</td>\n",
       "      <td>13.03</td>\n",
       "      <td>8.60</td>\n",
       "      <td>10.57</td>\n",
       "      <td>9.55</td>\n",
       "      <td>9.24</td>\n",
       "      <td>8.25</td>\n",
       "      <td>8.90</td>\n",
       "      <td>8.59</td>\n",
       "      <td>...</td>\n",
       "      <td>8.69</td>\n",
       "      <td>12.01</td>\n",
       "      <td>10.43</td>\n",
       "      <td>8.79</td>\n",
       "      <td>8.19</td>\n",
       "      <td>7.80</td>\n",
       "      <td>9.61</td>\n",
       "      <td>9.42</td>\n",
       "      <td>9.17</td>\n",
       "      <td>11.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>4.82</td>\n",
       "      <td>3.38</td>\n",
       "      <td>4.48</td>\n",
       "      <td>4.83</td>\n",
       "      <td>4.46</td>\n",
       "      <td>4.69</td>\n",
       "      <td>5.52</td>\n",
       "      <td>4.70</td>\n",
       "      <td>4.92</td>\n",
       "      <td>...</td>\n",
       "      <td>6.52</td>\n",
       "      <td>4.91</td>\n",
       "      <td>3.80</td>\n",
       "      <td>5.25</td>\n",
       "      <td>5.25</td>\n",
       "      <td>4.77</td>\n",
       "      <td>4.41</td>\n",
       "      <td>4.82</td>\n",
       "      <td>3.71</td>\n",
       "      <td>4.62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>47.55</td>\n",
       "      <td>48.40</td>\n",
       "      <td>60.64</td>\n",
       "      <td>62.86</td>\n",
       "      <td>44.27</td>\n",
       "      <td>45.67</td>\n",
       "      <td>60.64</td>\n",
       "      <td>36.94</td>\n",
       "      <td>41.21</td>\n",
       "      <td>...</td>\n",
       "      <td>36.82</td>\n",
       "      <td>46.60</td>\n",
       "      <td>55.52</td>\n",
       "      <td>49.10</td>\n",
       "      <td>45.94</td>\n",
       "      <td>43.55</td>\n",
       "      <td>57.80</td>\n",
       "      <td>55.79</td>\n",
       "      <td>43.84</td>\n",
       "      <td>52.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>7.02</td>\n",
       "      <td>15.44</td>\n",
       "      <td>6.66</td>\n",
       "      <td>8.80</td>\n",
       "      <td>12.00</td>\n",
       "      <td>12.09</td>\n",
       "      <td>4.09</td>\n",
       "      <td>4.52</td>\n",
       "      <td>4.34</td>\n",
       "      <td>...</td>\n",
       "      <td>2.11</td>\n",
       "      <td>10.73</td>\n",
       "      <td>10.01</td>\n",
       "      <td>8.70</td>\n",
       "      <td>4.99</td>\n",
       "      <td>3.00</td>\n",
       "      <td>15.50</td>\n",
       "      <td>9.68</td>\n",
       "      <td>5.78</td>\n",
       "      <td>11.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>9.55</td>\n",
       "      <td>6.38</td>\n",
       "      <td>7.60</td>\n",
       "      <td>4.16</td>\n",
       "      <td>9.82</td>\n",
       "      <td>5.04</td>\n",
       "      <td>6.44</td>\n",
       "      <td>2.93</td>\n",
       "      <td>4.34</td>\n",
       "      <td>...</td>\n",
       "      <td>6.67</td>\n",
       "      <td>8.47</td>\n",
       "      <td>8.11</td>\n",
       "      <td>6.43</td>\n",
       "      <td>8.17</td>\n",
       "      <td>4.09</td>\n",
       "      <td>7.73</td>\n",
       "      <td>4.67</td>\n",
       "      <td>5.24</td>\n",
       "      <td>8.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>13.34</td>\n",
       "      <td>7.98</td>\n",
       "      <td>11.19</td>\n",
       "      <td>9.64</td>\n",
       "      <td>13.66</td>\n",
       "      <td>11.13</td>\n",
       "      <td>13.59</td>\n",
       "      <td>10.32</td>\n",
       "      <td>9.06</td>\n",
       "      <td>...</td>\n",
       "      <td>20.95</td>\n",
       "      <td>7.14</td>\n",
       "      <td>8.19</td>\n",
       "      <td>14.90</td>\n",
       "      <td>18.90</td>\n",
       "      <td>8.21</td>\n",
       "      <td>11.05</td>\n",
       "      <td>9.97</td>\n",
       "      <td>7.60</td>\n",
       "      <td>10.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>14.06</td>\n",
       "      <td>9.81</td>\n",
       "      <td>12.04</td>\n",
       "      <td>10.95</td>\n",
       "      <td>13.06</td>\n",
       "      <td>11.83</td>\n",
       "      <td>10.97</td>\n",
       "      <td>11.90</td>\n",
       "      <td>9.47</td>\n",
       "      <td>...</td>\n",
       "      <td>18.97</td>\n",
       "      <td>9.78</td>\n",
       "      <td>9.50</td>\n",
       "      <td>11.65</td>\n",
       "      <td>14.25</td>\n",
       "      <td>10.48</td>\n",
       "      <td>10.00</td>\n",
       "      <td>10.84</td>\n",
       "      <td>8.48</td>\n",
       "      <td>12.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>4.23</td>\n",
       "      <td>2.98</td>\n",
       "      <td>3.03</td>\n",
       "      <td>2.50</td>\n",
       "      <td>3.59</td>\n",
       "      <td>2.16</td>\n",
       "      <td>3.56</td>\n",
       "      <td>4.21</td>\n",
       "      <td>1.80</td>\n",
       "      <td>...</td>\n",
       "      <td>4.02</td>\n",
       "      <td>3.11</td>\n",
       "      <td>4.16</td>\n",
       "      <td>2.58</td>\n",
       "      <td>2.41</td>\n",
       "      <td>2.06</td>\n",
       "      <td>4.26</td>\n",
       "      <td>2.37</td>\n",
       "      <td>2.98</td>\n",
       "      <td>4.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>6.45</td>\n",
       "      <td>10.04</td>\n",
       "      <td>4.25</td>\n",
       "      <td>5.95</td>\n",
       "      <td>6.83</td>\n",
       "      <td>5.05</td>\n",
       "      <td>3.01</td>\n",
       "      <td>3.07</td>\n",
       "      <td>2.12</td>\n",
       "      <td>...</td>\n",
       "      <td>3.33</td>\n",
       "      <td>7.09</td>\n",
       "      <td>6.22</td>\n",
       "      <td>3.88</td>\n",
       "      <td>3.80</td>\n",
       "      <td>3.51</td>\n",
       "      <td>7.08</td>\n",
       "      <td>6.07</td>\n",
       "      <td>3.79</td>\n",
       "      <td>6.86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R1015854  R1022980  R1028639  R1039781  R1042011  R1042566  \\\n",
       "0         ABL1     10.45     13.03      8.60     10.57      9.55      9.24   \n",
       "1         ABL2      4.82      3.38      4.48      4.83      4.46      4.69   \n",
       "2        ACAA1     47.55     48.40     60.64     62.86     44.27     45.67   \n",
       "3        ACACA      0.00      0.00      0.00      0.00      0.00      0.00   \n",
       "4        ACACB      7.02     15.44      6.66      8.80     12.00     12.09   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16      9.55      6.38      7.60      4.16      9.82      5.04   \n",
       "2140    ZFYVE9     13.34      7.98     11.19      9.64     13.66     11.13   \n",
       "2141     ZMAT3     14.06      9.81     12.04     10.95     13.06     11.83   \n",
       "2142    ZNF274      4.23      2.98      3.03      2.50      3.59      2.16   \n",
       "2143     ZNRF3      6.45     10.04      4.25      5.95      6.83      5.05   \n",
       "\n",
       "      R1074668  R1103580  R1105988  ...  R9781891  R9809441  R9817056  \\\n",
       "0         8.25      8.90      8.59  ...      8.69     12.01     10.43   \n",
       "1         5.52      4.70      4.92  ...      6.52      4.91      3.80   \n",
       "2        60.64     36.94     41.21  ...     36.82     46.60     55.52   \n",
       "3         0.00      0.00      0.00  ...      0.00      0.00      0.00   \n",
       "4         4.09      4.52      4.34  ...      2.11     10.73     10.01   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139      6.44      2.93      4.34  ...      6.67      8.47      8.11   \n",
       "2140     13.59     10.32      9.06  ...     20.95      7.14      8.19   \n",
       "2141     10.97     11.90      9.47  ...     18.97      9.78      9.50   \n",
       "2142      3.56      4.21      1.80  ...      4.02      3.11      4.16   \n",
       "2143      3.01      3.07      2.12  ...      3.33      7.09      6.22   \n",
       "\n",
       "      R9817161  R9818080  R9904978  R9905342  R9907075  R9936070  R9976232  \n",
       "0         8.79      8.19      7.80      9.61      9.42      9.17     11.57  \n",
       "1         5.25      5.25      4.77      4.41      4.82      3.71      4.62  \n",
       "2        49.10     45.94     43.55     57.80     55.79     43.84     52.56  \n",
       "3         0.00      0.00      0.00      0.00      0.00      0.00      0.00  \n",
       "4         8.70      4.99      3.00     15.50      9.68      5.78     11.74  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139      6.43      8.17      4.09      7.73      4.67      5.24      8.55  \n",
       "2140     14.90     18.90      8.21     11.05      9.97      7.60     10.30  \n",
       "2141     11.65     14.25     10.48     10.00     10.84      8.48     12.61  \n",
       "2142      2.58      2.41      2.06      4.26      2.37      2.98      4.01  \n",
       "2143      3.88      3.80      3.51      7.08      6.07      3.79      6.86  \n",
       "\n",
       "[2144 rows x 440 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_expression_filtered = gene_expression_filtered[['gene_name'] + sorted_patient_sample_list].sort_values(by='gene_name').reset_index(drop=True)\n",
    "gene_expression_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>R1015854</th>\n",
       "      <th>R1022980</th>\n",
       "      <th>R1028639</th>\n",
       "      <th>R1039781</th>\n",
       "      <th>R1042011</th>\n",
       "      <th>R1042566</th>\n",
       "      <th>R1074668</th>\n",
       "      <th>R1103580</th>\n",
       "      <th>R1105988</th>\n",
       "      <th>...</th>\n",
       "      <th>R9781891</th>\n",
       "      <th>R9809441</th>\n",
       "      <th>R9817056</th>\n",
       "      <th>R9817161</th>\n",
       "      <th>R9818080</th>\n",
       "      <th>R9904978</th>\n",
       "      <th>R9905342</th>\n",
       "      <th>R9907075</th>\n",
       "      <th>R9936070</th>\n",
       "      <th>R9976232</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 438 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_name  R1015854  R1022980  R1028639  R1039781  R1042011  R1042566  \\\n",
       "0         ABL1         0         0         0         0         0         0   \n",
       "1         ABL2         0         0         0         0         0         0   \n",
       "2        ACAA1         0         0         0         0         0         0   \n",
       "3        ACACA         0         0         0         0         0         0   \n",
       "4        ACACB         0         0         0         0         0         0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "2139   ZFYVE16         0         0         0         0         0         0   \n",
       "2140    ZFYVE9         0         0         0         0         0         0   \n",
       "2141     ZMAT3         0         0         0         0         0         0   \n",
       "2142    ZNF274         0         0         0         0         0         0   \n",
       "2143     ZNRF3         0         0         0         0         0         0   \n",
       "\n",
       "      R1074668  R1103580  R1105988  ...  R9781891  R9809441  R9817056  \\\n",
       "0            0         0         0  ...         0         0         0   \n",
       "1            0         0         0  ...         0         0         0   \n",
       "2            0         0         0  ...         0         0         0   \n",
       "3            0         0         0  ...         0         0         0   \n",
       "4            0         0         0  ...         0         0         0   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2139         0         0         0  ...         0         0         0   \n",
       "2140         0         0         0  ...         0         0         0   \n",
       "2141         0         0         0  ...         0         0         0   \n",
       "2142         0         0         0  ...         0         0         0   \n",
       "2143         0         0         0  ...         0         0         0   \n",
       "\n",
       "      R9817161  R9818080  R9904978  R9905342  R9907075  R9936070  R9976232  \n",
       "0            0         0         0         0         0         0         0  \n",
       "1            0         0         0         0         0         0         0  \n",
       "2            0         0         0         0         0         0         0  \n",
       "3            0         0         0         0         0         0         0  \n",
       "4            0         0         0         0         0         0         0  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2139         0         0         0         0         0         0         0  \n",
       "2140         0         0         0         0         0         0         0  \n",
       "2141         0         0         0         0         0         0         0  \n",
       "2142         0         0         0         0         0         0         0  \n",
       "2143         0         0         0         0         0         0         0  \n",
       "\n",
       "[2144 rows x 438 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_filtered = protein_filtered[['gene_name'] + sorted_patient_sample_list].sort_values(by='gene_name').reset_index(drop=True)\n",
    "protein_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R1015854</td>\n",
       "      <td>45115248</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>86.811772758384663</td>\n",
       "      <td>87.115674195756327</td>\n",
       "      <td>26.000000</td>\n",
       "      <td>4.633333</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R1022980</td>\n",
       "      <td>30819298</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>89.111567419575636</td>\n",
       "      <td>89.900068446269685</td>\n",
       "      <td>8.000000</td>\n",
       "      <td>8.083333</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R1028639</td>\n",
       "      <td>46000440</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>88.703627652292951</td>\n",
       "      <td>89.642710472279262</td>\n",
       "      <td>19.000000</td>\n",
       "      <td>6.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R1039781</td>\n",
       "      <td>50302428</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>84.312114989733061</td>\n",
       "      <td>84.665297741273108</td>\n",
       "      <td>18.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R1042011</td>\n",
       "      <td>20532115</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>89.505817932922653</td>\n",
       "      <td>89.727583846680361</td>\n",
       "      <td>23.793103</td>\n",
       "      <td>5.266667</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>R9904978</td>\n",
       "      <td>51668135</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>72.093086926762496</td>\n",
       "      <td>72.720054757015745</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>12.433333</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>R9905342</td>\n",
       "      <td>21293107</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>80.558521560574945</td>\n",
       "      <td>80.687200547570157</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>6.750000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>R9907075</td>\n",
       "      <td>11616707</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>72.933607118412041</td>\n",
       "      <td>73.790554414784395</td>\n",
       "      <td>14.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>R9936070</td>\n",
       "      <td>2899847</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>73.489390828199859</td>\n",
       "      <td>74.450376454483234</td>\n",
       "      <td>27.000000</td>\n",
       "      <td>7.016667</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>R9976232</td>\n",
       "      <td>21183160</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>89.338809034907598</td>\n",
       "      <td>89.478439425051334</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>4.333333</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID    projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0       R1015854  45115248   MAP   0.0  16.0   1.0      2.0           33.0   \n",
       "1       R1022980  30819298   MAP   0.0  10.0   1.0      2.0           33.0   \n",
       "2       R1028639  46000440   MAP   1.0  16.0   1.0      2.0           34.0   \n",
       "3       R1039781  50302428   MAP   0.0  12.0   1.0      2.0           33.0   \n",
       "4       R1042011  20532115   ROS   0.0  17.0   1.0      2.0           23.0   \n",
       "..           ...       ...   ...   ...   ...   ...      ...            ...   \n",
       "432     R9904978  51668135   MAP   1.0  22.0   1.0      2.0           33.0   \n",
       "433     R9905342  21293107   ROS   0.0  21.0   1.0      2.0           34.0   \n",
       "434     R9907075  11616707   ROS   1.0  19.0   1.0      2.0           34.0   \n",
       "435     R9936070   2899847   MAP   1.0  14.0   1.0      2.0           33.0   \n",
       "436     R9976232  21183160   ROS   0.0  16.0   1.0      2.0           33.0   \n",
       "\n",
       "       age_at_visit_max           age_death  cts_mmse30_lv        pmi  \\\n",
       "0    86.811772758384663  87.115674195756327      26.000000   4.633333   \n",
       "1    89.111567419575636  89.900068446269685       8.000000   8.083333   \n",
       "2    88.703627652292951  89.642710472279262      19.000000   6.916667   \n",
       "3    84.312114989733061  84.665297741273108      18.000000   3.000000   \n",
       "4    89.505817932922653  89.727583846680361      23.793103   5.266667   \n",
       "..                  ...                 ...            ...        ...   \n",
       "432  72.093086926762496  72.720054757015745      30.000000  12.433333   \n",
       "433  80.558521560574945  80.687200547570157      28.000000   6.750000   \n",
       "434  72.933607118412041  73.790554414784395      14.000000   3.000000   \n",
       "435  73.489390828199859  74.450376454483234      27.000000   7.016667   \n",
       "436  89.338809034907598  89.478439425051334      10.000000   4.333333   \n",
       "\n",
       "     braaksc  ceradsc  cogdx  dcfdx_lv  \n",
       "0        2.0      4.0    2.0       2.0  \n",
       "1        3.0      3.0    4.0       4.0  \n",
       "2        4.0      1.0    4.0       4.0  \n",
       "3        5.0      1.0    4.0       4.0  \n",
       "4        2.0      4.0    4.0       2.0  \n",
       "..       ...      ...    ...       ...  \n",
       "432      1.0      3.0    1.0       1.0  \n",
       "433      3.0      1.0    4.0       4.0  \n",
       "434      2.0      3.0    5.0       4.0  \n",
       "435      2.0      2.0    3.0       3.0  \n",
       "436      4.0      4.0    4.0       4.0  \n",
       "\n",
       "[437 rows x 16 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "survival_filtered = survival_filtered.sort_values(by='individualID').reset_index(drop=True)\n",
    "survival_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6.2 create output folder and save processed datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# outputfile name\n",
    "output_folder = 'ROSMAP-process'\n",
    "# create folder if not exist\n",
    "if not os.path.exists(output_folder):\n",
    "    os.makedirs(output_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DataFrame needed to be saved\n",
    "dataframes = {\n",
    "    'gene-tran-list.csv': sorted_gene_tran_df,\n",
    "    'gene-methy-list.csv': sorted_gene_methy_df,\n",
    "    'gene-protein-list.csv': sorted_gene_protein_df,\n",
    "    'gene-all-list.csv': sorted_all_gene_df,\n",
    "    'gene-kegg-edge-list.csv': filtered_up_kegg_df,\n",
    "    'gene-kegg-path-edge-list.csv': filtered_up_kegg_path_df,\n",
    "    # 'gene-biogrid-edge-list.csv': filtered_up_biogrid_df,\n",
    "    # 'gene-string-edge-list.csv': filtered_up_string_df,\n",
    "    'patient-sample-list.csv': sorted_patient_sample_df,\n",
    "    # 'phenotype-lists.csv': phenotype_lists,\n",
    "    'processed-genotype-methy-Upstream.csv': Upstream_df_filtered,\n",
    "    'processed-genotype-methy-Distal-Promoter.csv': Distal_Promoter_df_filtered,\n",
    "    'processed-genotype-methy-Proximal-Promoter.csv': Proximal_Promoter_df_filtered,\n",
    "    'processed-genotype-methy-Core-Promoter.csv': Core_Promoter_df_filtered,\n",
    "    'processed-genotype-methy-Downstream.csv': Downstream_df_filtered,\n",
    "    'processed-genotype-cnv_del.csv': cnv_aggregated_filtered_del,\n",
    "    'processed-genotype-cnv_dup.csv': cnv_aggregated_filtered_dup,\n",
    "    'processed-genotype-cnv_mcnv.csv': cnv_aggregated_filtered_mcnv,\n",
    "    'processed-genotype-gene-expression.csv': gene_expression_filtered,\n",
    "    'processed-genotype-proteomics.csv': protein_filtered,\n",
    "    # 'processed-phenotype-immune-subtype-transposed.csv': immune_subtype_filtered,\n",
    "    'processed-phenotype-survival-transposed.csv': survival_filtered,\n",
    "    # 'processed-phenotype-dense-transposed.csv': dense_filtered,\n",
    "    # 'processed-phenotype-cellsub-transposed.csv': cellsub_filtered\n",
    "    'processed-mapping-idmap.csv': union_map,\n",
    "    'processed-mapping-idmap-withoutnull.csv': union_map_cleaned\n",
    "}\n",
    "\n",
    "# save to output folder\n",
    "for file_name, df in dataframes.items():\n",
    "    df.to_csv(os.path.join(output_folder, file_name), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7.Convert the processed data into node dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load processed data\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# read the file names under the folder\n",
    "# Define the path to the output folder where CSV files are stored\n",
    "output_folder = 'ROSMAP-process'\n",
    "\n",
    "# List of file names you saved earlier\n",
    "file_names = [\n",
    "    'gene-tran-list', 'gene-methy-list', 'gene-protein-list', 'gene-all-list', \n",
    "    'gene-kegg-edge-list', 'gene-kegg-path-edge-list', \n",
    "    # 'gene-biogrid-edge-list', \n",
    "    # 'gene-string-edge-list',\n",
    "    'patient-sample-list', 'processed-genotype-methy-Upstream', \n",
    "    'processed-genotype-methy-Distal-Promoter', \n",
    "    'processed-genotype-methy-Proximal-Promoter', \n",
    "    'processed-genotype-methy-Core-Promoter', 'processed-genotype-methy-Downstream', \n",
    "    'processed-genotype-cnv_del', 'processed-genotype-cnv_dup', 'processed-genotype-cnv_mcnv', \n",
    "    'processed-genotype-gene-expression', \n",
    "    'processed-genotype-proteomics',\n",
    "    'processed-phenotype-survival-transposed'\n",
    "]\n",
    "\n",
    "# Dictionary to hold the dataframes\n",
    "dataframes = {}\n",
    "\n",
    "# Read each file and assign to a dataframe\n",
    "for file_name in file_names:\n",
    "    full_path = os.path.join(output_folder, file_name + '.csv')\n",
    "    dataframes[file_name] = pd.read_csv(full_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assign each dataframe to a variable\n",
    "sorted_gene_tran_df = dataframes['gene-tran-list']\n",
    "sorted_gene_methy_df = dataframes['gene-methy-list']\n",
    "sorted_gene_protein_df = dataframes['gene-protein-list']\n",
    "sorted_all_gene_df = dataframes['gene-all-list']\n",
    "filtered_up_kegg_df = dataframes['gene-kegg-edge-list']\n",
    "filtered_up_kegg_path_df = dataframes['gene-kegg-path-edge-list']\n",
    "# filtered_up_biogrid_df = dataframes['gene-biogrid-edge-list']\n",
    "# filtered_up_string_df = dataframes['gene-string-edge-list']\n",
    "sorted_patient_sample_df = dataframes['patient-sample-list']\n",
    "# phenotype_lists = dataframes['phenotype-lists']\n",
    "Upstream_df_filtered = dataframes['processed-genotype-methy-Upstream']\n",
    "Distal_Promoter_df_filtered = dataframes['processed-genotype-methy-Distal-Promoter']\n",
    "Proximal_Promoter_df_filtered = dataframes['processed-genotype-methy-Proximal-Promoter']\n",
    "Core_Promoter_df_filtered = dataframes['processed-genotype-methy-Core-Promoter']\n",
    "Downstream_df_filtered = dataframes['processed-genotype-methy-Downstream']\n",
    "copynumber_filtered_del = dataframes['processed-genotype-cnv_del']\n",
    "copynumber_filtered_dup = dataframes['processed-genotype-cnv_dup']\n",
    "copynumber_filtered_mcnv = dataframes['processed-genotype-cnv_mcnv']\n",
    "gene_expression_filtered = dataframes['processed-genotype-gene-expression']\n",
    "protein_filtered = dataframes['processed-genotype-proteomics']\n",
    "# immune_subtype_filtered = dataframes['processed-phenotype-immune-subtype-transposed']\n",
    "survival_filtered = dataframes['processed-phenotype-survival-transposed']\n",
    "# dense_filtered = dataframes['processed-phenotype-dense-transposed']\n",
    "# cellsub_filtered = dataframes['processed-phenotype-cellsub-transposed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "# outputfile name\n",
    "graph_output_folder = 'ROSMAP-graph-data'\n",
    "# create folder if not exist\n",
    "if not os.path.exists(graph_output_folder):\n",
    "    os.makedirs(graph_output_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.1 Make nodes dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_num</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>NodeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ABL1-TRAN</td>\n",
       "      <td>Gene-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ABL2-TRAN</td>\n",
       "      <td>Gene-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "      <td>Gene-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>ACACA-TRAN</td>\n",
       "      <td>Gene-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>ACACB-TRAN</td>\n",
       "      <td>Gene-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>6427</td>\n",
       "      <td>ZFYVE16-PROT</td>\n",
       "      <td>Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6428</th>\n",
       "      <td>6428</td>\n",
       "      <td>ZFYVE9-PROT</td>\n",
       "      <td>Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6429</th>\n",
       "      <td>6429</td>\n",
       "      <td>ZMAT3-PROT</td>\n",
       "      <td>Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6430</th>\n",
       "      <td>6430</td>\n",
       "      <td>ZNF274-PROT</td>\n",
       "      <td>Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6431</th>\n",
       "      <td>6431</td>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6432 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene_num     Gene_name   NodeType\n",
       "0            0     ABL1-TRAN  Gene-TRAN\n",
       "1            1     ABL2-TRAN  Gene-TRAN\n",
       "2            2    ACAA1-TRAN  Gene-TRAN\n",
       "3            3    ACACA-TRAN  Gene-TRAN\n",
       "4            4    ACACB-TRAN  Gene-TRAN\n",
       "...        ...           ...        ...\n",
       "6427      6427  ZFYVE16-PROT  Gene-PROT\n",
       "6428      6428   ZFYVE9-PROT  Gene-PROT\n",
       "6429      6429    ZMAT3-PROT  Gene-PROT\n",
       "6430      6430   ZNF274-PROT  Gene-PROT\n",
       "6431      6431    ZNRF3-PROT  Gene-PROT\n",
       "\n",
       "[6432 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sorted_all_gene_dict = sorted_all_gene_df['Gene'].to_dict()\n",
    "sorted_all_gene_name_dict = {value: key for key, value in sorted_all_gene_dict.items()}\n",
    "num_gene = sorted_gene_tran_df.shape[0]\n",
    "num_gene_protein = sorted_gene_protein_df.shape[0]\n",
    "nodetype_list = ['Gene-TRAN'] * num_gene + ['Gene-METH'] * num_gene + ['Gene-PROT'] * num_gene_protein\n",
    "map_all_gene_df = pd.DataFrame({'Gene_num': sorted_all_gene_dict.keys(), 'Gene_name': sorted_all_gene_dict.values(), 'NodeType': nodetype_list})\n",
    "display(map_all_gene_df)\n",
    "map_all_gene_df.to_csv(os.path.join(graph_output_folder, 'map-all-gene.csv'), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.2 Create the edges connection between promoter methylations and proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-METH</td>\n",
       "      <td>ABL1-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-METH</td>\n",
       "      <td>ABL2-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-METH</td>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-METH</td>\n",
       "      <td>ACACA-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-METH</td>\n",
       "      <td>ACACB-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16-METH</td>\n",
       "      <td>ZFYVE16-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9-METH</td>\n",
       "      <td>ZFYVE9-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3-METH</td>\n",
       "      <td>ZMAT3-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274-METH</td>\n",
       "      <td>ZNF274-TRAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3-METH</td>\n",
       "      <td>ZNRF3-TRAN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               src          dest\n",
       "0        ABL1-METH     ABL1-TRAN\n",
       "1        ABL2-METH     ABL2-TRAN\n",
       "2       ACAA1-METH    ACAA1-TRAN\n",
       "3       ACACA-METH    ACACA-TRAN\n",
       "4       ACACB-METH    ACACB-TRAN\n",
       "...            ...           ...\n",
       "2139  ZFYVE16-METH  ZFYVE16-TRAN\n",
       "2140   ZFYVE9-METH   ZFYVE9-TRAN\n",
       "2141    ZMAT3-METH    ZMAT3-TRAN\n",
       "2142   ZNF274-METH   ZNF274-TRAN\n",
       "2143    ZNRF3-METH    ZNRF3-TRAN\n",
       "\n",
       "[2144 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-TRAN</td>\n",
       "      <td>ABL1-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-TRAN</td>\n",
       "      <td>ABL2-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "      <td>ACAA1-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-TRAN</td>\n",
       "      <td>ACACA-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-TRAN</td>\n",
       "      <td>ACACB-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>ZFYVE16-TRAN</td>\n",
       "      <td>ZFYVE16-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>ZFYVE9-TRAN</td>\n",
       "      <td>ZFYVE9-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>ZMAT3-TRAN</td>\n",
       "      <td>ZMAT3-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>ZNF274-TRAN</td>\n",
       "      <td>ZNF274-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>ZNRF3-TRAN</td>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               src          dest\n",
       "0        ABL1-TRAN     ABL1-PROT\n",
       "1        ABL2-TRAN     ABL2-PROT\n",
       "2       ACAA1-TRAN    ACAA1-PROT\n",
       "3       ACACA-TRAN    ACACA-PROT\n",
       "4       ACACB-TRAN    ACACB-PROT\n",
       "...            ...           ...\n",
       "2139  ZFYVE16-TRAN  ZFYVE16-PROT\n",
       "2140   ZFYVE9-TRAN   ZFYVE9-PROT\n",
       "2141    ZMAT3-TRAN    ZMAT3-PROT\n",
       "2142   ZNF274-TRAN   ZNF274-PROT\n",
       "2143    ZNRF3-TRAN    ZNRF3-PROT\n",
       "\n",
       "[2144 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# [Gene-METH - Gene]\n",
    "sorted_gene_methy = sorted_gene_methy_df['Gene'].tolist()\n",
    "sorted_gene_list = sorted_gene_tran_df['Gene'].tolist()\n",
    "sorted_gene_protein = sorted_gene_protein_df['Gene'].tolist()\n",
    "sorted_intersection = [gene_protein.replace('-PROT', '-TRAN') for gene_protein in sorted_gene_protein]\n",
    "gene_meth_edge_df = pd.DataFrame({'src': sorted_gene_methy, 'dest': sorted_gene_list})\n",
    "display(gene_meth_edge_df)\n",
    "# [Gene - Gene-PROT]\n",
    "gene_protein_edge_df = pd.DataFrame({'src': sorted_intersection, 'dest': sorted_gene_protein})\n",
    "display(gene_protein_edge_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2144\n",
      "4288\n"
     ]
    }
   ],
   "source": [
    "print(sorted_all_gene_name_dict['ABL1-TRAN'])\n",
    "print(sorted_all_gene_name_dict['ABL1-METH'])\n",
    "print(sorted_all_gene_name_dict['ABL1-PROT'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2144</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2145</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2146</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2147</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2148</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>4283</td>\n",
       "      <td>2139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>4284</td>\n",
       "      <td>2140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>4285</td>\n",
       "      <td>2141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>4286</td>\n",
       "      <td>2142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>4287</td>\n",
       "      <td>2143</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       src  dest\n",
       "0     2144     0\n",
       "1     2145     1\n",
       "2     2146     2\n",
       "3     2147     3\n",
       "4     2148     4\n",
       "...    ...   ...\n",
       "2139  4283  2139\n",
       "2140  4284  2140\n",
       "2141  4285  2141\n",
       "2142  4286  2142\n",
       "2143  4287  2143\n",
       "\n",
       "[2144 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>2139</td>\n",
       "      <td>6427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>2140</td>\n",
       "      <td>6428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>2141</td>\n",
       "      <td>6429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>2142</td>\n",
       "      <td>6430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>2143</td>\n",
       "      <td>6431</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2144 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       src  dest\n",
       "0        0  4288\n",
       "1        1  4289\n",
       "2        2  4290\n",
       "3        3  4291\n",
       "4        4  4292\n",
       "...    ...   ...\n",
       "2139  2139  6427\n",
       "2140  2140  6428\n",
       "2141  2141  6429\n",
       "2142  2142  6430\n",
       "2143  2143  6431\n",
       "\n",
       "[2144 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# replace gene name with gene number\n",
    "gene_meth_num_edge_df = gene_meth_edge_df.copy()\n",
    "gene_meth_num_edge_df['src'] = gene_meth_edge_df['src'].map(sorted_all_gene_name_dict)\n",
    "gene_meth_num_edge_df['dest'] = gene_meth_edge_df['dest'].map(sorted_all_gene_name_dict)\n",
    "display(gene_meth_num_edge_df)\n",
    "gene_protein_num_edge_df = gene_protein_edge_df.copy()\n",
    "gene_protein_num_edge_df['src'] = gene_protein_edge_df['src'].map(sorted_all_gene_name_dict)\n",
    "gene_protein_num_edge_df['dest'] = gene_protein_edge_df['dest'].map(sorted_all_gene_name_dict)\n",
    "display(gene_protein_num_edge_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7.3 Concat all edges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4306</td>\n",
       "      <td>5603</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4306</td>\n",
       "      <td>5604</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4306</td>\n",
       "      <td>5605</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4306</td>\n",
       "      <td>6221</td>\n",
       "      <td>Hippo signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4307</td>\n",
       "      <td>5603</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28838</th>\n",
       "      <td>6431</td>\n",
       "      <td>4992</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28839</th>\n",
       "      <td>6431</td>\n",
       "      <td>4993</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28840</th>\n",
       "      <td>6431</td>\n",
       "      <td>4994</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28841</th>\n",
       "      <td>6431</td>\n",
       "      <td>4995</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28842</th>\n",
       "      <td>6431</td>\n",
       "      <td>4996</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28843 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                        path\n",
       "0      4306  5603  Oxytocin signaling pathway\n",
       "1      4306  5604  Oxytocin signaling pathway\n",
       "2      4306  5605  Oxytocin signaling pathway\n",
       "3      4306  6221     Hippo signaling pathway\n",
       "4      4307  5603  Oxytocin signaling pathway\n",
       "...     ...   ...                         ...\n",
       "28838  6431  4992       Wnt signaling pathway\n",
       "28839  6431  4993       Wnt signaling pathway\n",
       "28840  6431  4994       Wnt signaling pathway\n",
       "28841  6431  4995       Wnt signaling pathway\n",
       "28842  6431  4996       Wnt signaling pathway\n",
       "\n",
       "[28843 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4306</td>\n",
       "      <td>5603</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4306</td>\n",
       "      <td>5604</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4306</td>\n",
       "      <td>5605</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4306</td>\n",
       "      <td>6221</td>\n",
       "      <td>Hippo signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4307</td>\n",
       "      <td>5603</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>2139</td>\n",
       "      <td>6427</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>2140</td>\n",
       "      <td>6428</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>2141</td>\n",
       "      <td>6429</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>2142</td>\n",
       "      <td>6430</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143</th>\n",
       "      <td>2143</td>\n",
       "      <td>6431</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33131 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       src  dest                        path\n",
       "0     4306  5603  Oxytocin signaling pathway\n",
       "1     4306  5604  Oxytocin signaling pathway\n",
       "2     4306  5605  Oxytocin signaling pathway\n",
       "3     4306  6221     Hippo signaling pathway\n",
       "4     4307  5603  Oxytocin signaling pathway\n",
       "...    ...   ...                         ...\n",
       "2139  2139  6427                         NaN\n",
       "2140  2140  6428                         NaN\n",
       "2141  2141  6429                         NaN\n",
       "2142  2142  6430                         NaN\n",
       "2143  2143  6431                         NaN\n",
       "\n",
       "[33131 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4288</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4289</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4290</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4291</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4292</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33126</th>\n",
       "      <td>6431</td>\n",
       "      <td>4992</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33127</th>\n",
       "      <td>6431</td>\n",
       "      <td>4993</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33128</th>\n",
       "      <td>6431</td>\n",
       "      <td>4994</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33129</th>\n",
       "      <td>6431</td>\n",
       "      <td>4995</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33130</th>\n",
       "      <td>6431</td>\n",
       "      <td>4996</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33131 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                   path             EdgeType\n",
       "0         0  4288          internal link  Gene-TRAN-Gene-PROT\n",
       "1         1  4289          internal link  Gene-TRAN-Gene-PROT\n",
       "2         2  4290          internal link  Gene-TRAN-Gene-PROT\n",
       "3         3  4291          internal link  Gene-TRAN-Gene-PROT\n",
       "4         4  4292          internal link  Gene-TRAN-Gene-PROT\n",
       "...     ...   ...                    ...                  ...\n",
       "33126  6431  4992  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33127  6431  4993  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33128  6431  4994  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33129  6431  4995  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33130  6431  4996  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[33131 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if selected_database == 'KEGG':\n",
    "    # filtered_up_num_df = filtered_up_kegg_df.copy()\n",
    "    filtered_up_num_df = filtered_up_kegg_path_df.copy()\n",
    "elif selected_database == 'BioGRID':\n",
    "    filtered_up_num_df = filtered_up_biogrid_df.copy()\n",
    "elif selected_database == 'STRING':\n",
    "    filtered_up_num_df = filtered_up_string_df.copy()\n",
    "\n",
    "# Add 'PROT' to the end of each gene name in the 'src' and 'dest' columns\n",
    "filtered_up_num_df['src'] = filtered_up_num_df['src'].apply(lambda x: x + '-PROT')\n",
    "filtered_up_num_df['dest'] = filtered_up_num_df['dest'].apply(lambda x: x + '-PROT')\n",
    "\n",
    "filtered_up_num_df['src'] = filtered_up_num_df['src'].map(sorted_all_gene_name_dict)\n",
    "filtered_up_num_df['dest'] = filtered_up_num_df['dest'].map(sorted_all_gene_name_dict)\n",
    "display(filtered_up_num_df)\n",
    "all_gene_edge_num_df = pd.concat([filtered_up_num_df, gene_meth_num_edge_df, gene_protein_num_edge_df])\n",
    "display(all_gene_edge_num_df)\n",
    "\n",
    "num_gene_edge = filtered_up_num_df.shape[0]\n",
    "num_gene_meth_edge = gene_meth_num_edge_df.shape[0]\n",
    "num_gene_protein_edge = gene_protein_num_edge_df.shape[0]\n",
    "edgetype_list = ['Gene-PROT-Gene-PROT'] * num_gene_edge + ['Gene-TRAN-Gene-METH'] * num_gene_meth_edge + ['Gene-TRAN-Gene-PROT'] * num_gene_protein_edge\n",
    "all_gene_edge_num_df['EdgeType'] = edgetype_list\n",
    "all_gene_edge_num_df = all_gene_edge_num_df.sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "all_gene_edge_num_df.fillna('internal link', inplace=True)\n",
    "display(all_gene_edge_num_df)\n",
    "all_gene_edge_num_df.to_csv(os.path.join(graph_output_folder, 'all-gene-edge-num.csv'), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-TRAN</td>\n",
       "      <td>ABL1-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-TRAN</td>\n",
       "      <td>ABL2-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "      <td>ACAA1-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-TRAN</td>\n",
       "      <td>ACACA-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-TRAN</td>\n",
       "      <td>ACACB-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33126</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>FZD5-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33127</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>FZD6-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33128</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>FZD7-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33129</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>FZD8-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33130</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>FZD9-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33131 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              src        dest                   path             EdgeType\n",
       "0       ABL1-TRAN   ABL1-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "1       ABL2-TRAN   ABL2-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "2      ACAA1-TRAN  ACAA1-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "3      ACACA-TRAN  ACACA-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "4      ACACB-TRAN  ACACB-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "...           ...         ...                    ...                  ...\n",
       "33126  ZNRF3-PROT   FZD5-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33127  ZNRF3-PROT   FZD6-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33128  ZNRF3-PROT   FZD7-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33129  ZNRF3-PROT   FZD8-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33130  ZNRF3-PROT   FZD9-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[33131 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# gene edge interactions without map\n",
    "all_gene_edge_df = all_gene_edge_num_df.copy()\n",
    "all_gene_edge_df = all_gene_edge_df.replace(sorted_all_gene_dict)\n",
    "\n",
    "num_gene_edge = filtered_up_num_df.shape[0]\n",
    "num_gene_meth_edge = gene_meth_edge_df.shape[0]\n",
    "num_gene_protein_edge = gene_protein_edge_df.shape[0]\n",
    "# all_gene_edge_df = all_gene_edge_df.sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "all_gene_edge_df.to_csv(os.path.join(graph_output_folder, 'all-gene-edge.csv'), index=False)\n",
    "display(all_gene_edge_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8.Load data into graph format"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.1 Form up the input samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "recommends the use of the ceradsc as the classfication for AD types\n",
    "\n",
    "* ceradsc\n",
    "* cogdx"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Balance data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "276 161\n",
      "275 162\n",
      "AD Female: 177\n",
      "AD Male: 98\n"
     ]
    }
   ],
   "source": [
    "survival_filtered\n",
    "\n",
    "count_female = (survival_filtered['msex'] == 0).sum()\n",
    "count_male = (survival_filtered['msex'] == 1).sum()\n",
    "print(count_female, count_male)\n",
    "\n",
    "count_AD = ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2)).sum()\n",
    "count_NOAD = ((survival_filtered['ceradsc'] == 3) | (survival_filtered['ceradsc'] == 4)).sum()\n",
    "count_AD, count_NOAD\n",
    "print(count_AD, count_NOAD)\n",
    "\n",
    "count_female_AD = ((survival_filtered['msex'] == 0) & ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2))).sum()\n",
    "count_male_AD = ((survival_filtered['msex'] == 1) & ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2))).sum()\n",
    "print('AD Female:', count_female_AD)\n",
    "print('AD Male:', count_male_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from sklearn.utils import resample\n",
    "# survival_filtered\n",
    "\n",
    "# df_majority = survival_filtered[survival_filtered.msex == 0]\n",
    "# df_minority = survival_filtered[survival_filtered.msex == 1]\n",
    "\n",
    "# df_majority_downsampled = resample(df_majority, \n",
    "#                                    replace=False,\n",
    "#                                    n_samples=len(df_minority),\n",
    "#                                    random_state=123)\n",
    "\n",
    "# df_balanced = pd.concat([df_majority_downsampled, df_minority])\n",
    "\n",
    "# count_zeros = (survival_filtered['msex'] == 0).sum()\n",
    "# count_ones = (survival_filtered['msex'] == 1).sum()\n",
    "# if count_zeros > count_ones:\n",
    "#     zeros_indices = survival_filtered[survival_filtered['msex'] == 0].index\n",
    "#     n_drop = count_zeros - count_ones\n",
    "#     drop_indices = np.random.choice(zeros_indices, n_drop, replace=False)\n",
    "#     survival_filtered = survival_filtered.drop(drop_indices)\n",
    "# survival_filtered = survival_filtered.reset_index(drop=True)\n",
    "# survival_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AD Female: 98\n",
      "AD Male: 98\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R7696135</td>\n",
       "      <td>66924745</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>27.000000</td>\n",
       "      <td>8.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R7738727</td>\n",
       "      <td>8132197</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>27.000000</td>\n",
       "      <td>4.083333</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R7516349</td>\n",
       "      <td>20156469</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>24.000000</td>\n",
       "      <td>13.000000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R9053141</td>\n",
       "      <td>21411459</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>82.866529774127315</td>\n",
       "      <td>83.110198494182072</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R9781891</td>\n",
       "      <td>83034844</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>78.143737166324442</td>\n",
       "      <td>78.444900752908964</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>7.450000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>R5525186</td>\n",
       "      <td>15121461</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>16.153846</td>\n",
       "      <td>5.833333</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>R8608442</td>\n",
       "      <td>10248033</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>88.191649555099247</td>\n",
       "      <td>88.468172484599592</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>17.916667</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>R2731324</td>\n",
       "      <td>50302554</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>7.750000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>R9473965</td>\n",
       "      <td>10262905</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>25.000000</td>\n",
       "      <td>9.583333</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>R6487961</td>\n",
       "      <td>50104684</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>85.086926762491444</td>\n",
       "      <td>85.379876796714584</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID    projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0       R7696135  66924745   MAP   0.0  16.0   1.0      2.0           33.0   \n",
       "1       R7738727   8132197   MAP   0.0  16.0   1.0      2.0           23.0   \n",
       "2       R7516349  20156469   ROS   0.0  20.0   1.0      2.0           23.0   \n",
       "3       R9053141  21411459   ROS   0.0  18.0   1.0      2.0           34.0   \n",
       "4       R9781891  83034844   MAP   0.0  15.0   1.0      2.0           34.0   \n",
       "..           ...       ...   ...   ...   ...   ...      ...            ...   \n",
       "191     R5525186  15121461   ROS   1.0  22.0   1.0      2.0           34.0   \n",
       "192     R8608442  10248033   ROS   1.0  19.0   1.0      2.0           44.0   \n",
       "193     R2731324  50302554   MAP   1.0  16.0   1.0      2.0           33.0   \n",
       "194     R9473965  10262905   ROS   1.0  18.0   1.0      2.0           24.0   \n",
       "195     R6487961  50104684   MAP   1.0  18.0   1.0      2.0           33.0   \n",
       "\n",
       "       age_at_visit_max           age_death  cts_mmse30_lv        pmi  \\\n",
       "0                   90+                 90+      27.000000   8.500000   \n",
       "1                   90+                 90+      27.000000   4.083333   \n",
       "2                   90+                 90+      24.000000  13.000000   \n",
       "3    82.866529774127315  83.110198494182072      29.000000        NaN   \n",
       "4    78.143737166324442  78.444900752908964      30.000000   7.450000   \n",
       "..                  ...                 ...            ...        ...   \n",
       "191                 90+                 90+      16.153846   5.833333   \n",
       "192  88.191649555099247  88.468172484599592      12.000000  17.916667   \n",
       "193                 90+                 90+      22.000000   7.750000   \n",
       "194                 90+                 90+      25.000000   9.583333   \n",
       "195  85.086926762491444  85.379876796714584      28.000000   4.000000   \n",
       "\n",
       "     braaksc  ceradsc  cogdx  dcfdx_lv  \n",
       "0        4.0      2.0    1.0       2.0  \n",
       "1        3.0      2.0    2.0       2.0  \n",
       "2        4.0      2.0    5.0       4.0  \n",
       "3        3.0      1.0    1.0       1.0  \n",
       "4        3.0      2.0    1.0       1.0  \n",
       "..       ...      ...    ...       ...  \n",
       "191      5.0      1.0    4.0       4.0  \n",
       "192      3.0      2.0    4.0       4.0  \n",
       "193      3.0      1.0    4.0       4.0  \n",
       "194      5.0      2.0    4.0       4.0  \n",
       "195      5.0      2.0    2.0       2.0  \n",
       "\n",
       "[196 rows x 16 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.utils import resample\n",
    "\n",
    "# Calculate the number of female and male AD samples in the original dataset\n",
    "count_female_AD = ((survival_filtered['msex'] == 0) & ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2)) ).sum()\n",
    "count_male_AD = ((survival_filtered['msex'] == 1) & ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2)) ).sum()\n",
    "\n",
    "# Get the female and male AD datasets\n",
    "df_female_AD = survival_filtered[(survival_filtered['msex'] == 0) & ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2))]\n",
    "df_male_AD = survival_filtered[(survival_filtered['msex'] == 1) & ((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2))]\n",
    "\n",
    "# Determine the number of samples after downsampling, taking the smaller value between female and male AD sample counts\n",
    "n_samples = min(count_female_AD, count_male_AD)\n",
    "\n",
    "# Downsample the female AD dataset\n",
    "df_female_AD_downsampled = resample(df_female_AD, \n",
    "                                    replace=False,\n",
    "                                    n_samples=n_samples,\n",
    "                                    random_state=123)\n",
    "\n",
    "# Downsample the male AD dataset\n",
    "df_male_AD_downsampled = resample(df_male_AD, \n",
    "                                  replace=False,\n",
    "                                  n_samples=n_samples,\n",
    "                                  random_state=123)\n",
    "\n",
    "# Combine the downsampled female and male AD datasets\n",
    "df_balanced_AD = pd.concat([df_female_AD_downsampled, df_male_AD_downsampled]).reset_index(drop=True)\n",
    "\n",
    "# Print the counts of the downsampled female and male AD samples\n",
    "count_female_AD_downsampled = len(df_female_AD_downsampled)\n",
    "count_male_AD_downsampled = len(df_male_AD_downsampled)\n",
    "print('AD Female:', count_female_AD_downsampled)\n",
    "print('AD Male:', count_male_AD_downsampled)\n",
    "\n",
    "# Overwrite the original survival_filtered with the balanced dataset\n",
    "survival_filtered = df_balanced_AD\n",
    "survival_filtered\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AD Female: 98\n",
      "AD Male: 98\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R7696135</td>\n",
       "      <td>66924745</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>27.000000</td>\n",
       "      <td>8.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R7738727</td>\n",
       "      <td>8132197</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>27.000000</td>\n",
       "      <td>4.083333</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R7516349</td>\n",
       "      <td>20156469</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>24.000000</td>\n",
       "      <td>13.000000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R9053141</td>\n",
       "      <td>21411459</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>82.866529774127315</td>\n",
       "      <td>83.110198494182072</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R9781891</td>\n",
       "      <td>83034844</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>78.143737166324442</td>\n",
       "      <td>78.444900752908964</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>7.450000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>R5525186</td>\n",
       "      <td>15121461</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>16.153846</td>\n",
       "      <td>5.833333</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>R8608442</td>\n",
       "      <td>10248033</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>88.191649555099247</td>\n",
       "      <td>88.468172484599592</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>17.916667</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>R2731324</td>\n",
       "      <td>50302554</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>7.750000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>R9473965</td>\n",
       "      <td>10262905</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>25.000000</td>\n",
       "      <td>9.583333</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>R6487961</td>\n",
       "      <td>50104684</td>\n",
       "      <td>MAP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>85.086926762491444</td>\n",
       "      <td>85.379876796714584</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID    projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0       R7696135  66924745   MAP   0.0  16.0   1.0      2.0           33.0   \n",
       "1       R7738727   8132197   MAP   0.0  16.0   1.0      2.0           23.0   \n",
       "2       R7516349  20156469   ROS   0.0  20.0   1.0      2.0           23.0   \n",
       "3       R9053141  21411459   ROS   0.0  18.0   1.0      2.0           34.0   \n",
       "4       R9781891  83034844   MAP   0.0  15.0   1.0      2.0           34.0   \n",
       "..           ...       ...   ...   ...   ...   ...      ...            ...   \n",
       "191     R5525186  15121461   ROS   1.0  22.0   1.0      2.0           34.0   \n",
       "192     R8608442  10248033   ROS   1.0  19.0   1.0      2.0           44.0   \n",
       "193     R2731324  50302554   MAP   1.0  16.0   1.0      2.0           33.0   \n",
       "194     R9473965  10262905   ROS   1.0  18.0   1.0      2.0           24.0   \n",
       "195     R6487961  50104684   MAP   1.0  18.0   1.0      2.0           33.0   \n",
       "\n",
       "       age_at_visit_max           age_death  cts_mmse30_lv        pmi  \\\n",
       "0                   90+                 90+      27.000000   8.500000   \n",
       "1                   90+                 90+      27.000000   4.083333   \n",
       "2                   90+                 90+      24.000000  13.000000   \n",
       "3    82.866529774127315  83.110198494182072      29.000000        NaN   \n",
       "4    78.143737166324442  78.444900752908964      30.000000   7.450000   \n",
       "..                  ...                 ...            ...        ...   \n",
       "191                 90+                 90+      16.153846   5.833333   \n",
       "192  88.191649555099247  88.468172484599592      12.000000  17.916667   \n",
       "193                 90+                 90+      22.000000   7.750000   \n",
       "194                 90+                 90+      25.000000   9.583333   \n",
       "195  85.086926762491444  85.379876796714584      28.000000   4.000000   \n",
       "\n",
       "     braaksc  ceradsc  cogdx  dcfdx_lv  \n",
       "0        4.0      2.0    1.0       2.0  \n",
       "1        3.0      2.0    2.0       2.0  \n",
       "2        4.0      2.0    5.0       4.0  \n",
       "3        3.0      1.0    1.0       1.0  \n",
       "4        3.0      2.0    1.0       1.0  \n",
       "..       ...      ...    ...       ...  \n",
       "191      5.0      1.0    4.0       4.0  \n",
       "192      3.0      2.0    4.0       4.0  \n",
       "193      3.0      1.0    4.0       4.0  \n",
       "194      5.0      2.0    4.0       4.0  \n",
       "195      5.0      2.0    2.0       2.0  \n",
       "\n",
       "[196 rows x 16 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter the dataset to keep only AD samples\n",
    "df_AD = survival_filtered[((survival_filtered['ceradsc'] == 1) | (survival_filtered['ceradsc'] == 2))]\n",
    "\n",
    "# Calculate the number of female and male AD samples\n",
    "count_female_AD = (df_AD['msex'] == 0).sum()\n",
    "count_male_AD = (df_AD['msex'] == 1).sum()\n",
    "\n",
    "# Print the counts of female and male AD samples\n",
    "print('AD Female:', count_female_AD)\n",
    "print('AD Male:', count_male_AD)\n",
    "\n",
    "# Overwrite the original survival_filtered with the AD dataset\n",
    "survival_filtered = df_AD.reset_index(drop=True)\n",
    "survival_filtered\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>msex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R7696135</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R7738727</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R7516349</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R9053141</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R9781891</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>R5525186</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>R8608442</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>R2731324</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>R9473965</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>R6487961</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID  msex\n",
       "0       R7696135   0.0\n",
       "1       R7738727   0.0\n",
       "2       R7516349   0.0\n",
       "3       R9053141   0.0\n",
       "4       R9781891   0.0\n",
       "..           ...   ...\n",
       "191     R5525186   1.0\n",
       "192     R8608442   1.0\n",
       "193     R2731324   1.0\n",
       "194     R9473965   1.0\n",
       "195     R6487961   1.0\n",
       "\n",
       "[196 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0. 1.]\n"
     ]
    }
   ],
   "source": [
    "survival_filtered_feature_df = survival_filtered.copy()\n",
    "\n",
    "# TODO\n",
    "# survival_filtered_feature_df = survival_filtered_feature_df[['individualID', 'ceradsc']]\n",
    "survival_filtered_feature_df = survival_filtered_feature_df[['individualID', 'msex']]\n",
    "display(survival_filtered_feature_df)\n",
    "\n",
    "nan_counts = survival_filtered_feature_df.isna().sum()  # or df.isnull()\n",
    "\n",
    "# TODO\n",
    "# print(survival_filtered_feature_df['ceradsc'].unique())\n",
    "print(survival_filtered_feature_df['msex'].unique())\n",
    "\n",
    "survival_filtered_feature_df.to_csv(os.path.join(graph_output_folder, 'survival-label.csv'), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.2 Randomize the input label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>msex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R2697774</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R3841825</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R4077358</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R6268135</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R7210604</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>R9354381</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>R5211056</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>R1028639</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>R8432213</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>R3741788</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    individualID  msex\n",
       "0       R2697774   1.0\n",
       "1       R3841825   1.0\n",
       "2       R4077358   1.0\n",
       "3       R6268135   1.0\n",
       "4       R7210604   1.0\n",
       "..           ...   ...\n",
       "191     R9354381   0.0\n",
       "192     R5211056   0.0\n",
       "193     R1028639   1.0\n",
       "194     R8432213   0.0\n",
       "195     R3741788   1.0\n",
       "\n",
       "[196 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Randomize the survival label\n",
    "def input_random(randomized, graph_output_folder):\n",
    "    # randomized = True\n",
    "    if randomized == True:\n",
    "        random_survival_filtered_feature_df = survival_filtered_feature_df.sample(frac = 1).reset_index(drop=True)\n",
    "        random_survival_filtered_feature_df.to_csv(os.path.join(graph_output_folder, 'random-survival-label.csv'), index=False)\n",
    "    else:\n",
    "        random_survival_filtered_feature_df = pd.read_csv(os.path.join(graph_output_folder, 'random-survival-label.csv'))\n",
    "    display(random_survival_filtered_feature_df)\n",
    "\n",
    "# TODO\n",
    "input_random(randomized=True, graph_output_folder=graph_output_folder)\n",
    "# input_random(randomized=False, graph_output_folder=graph_output_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.3 Split the randomized input into K-fold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "--------TRAIN-TEST SPLIT WITH TEST FROM 0 TO 39--------\n",
      "(39, 2)\n",
      "\n",
      "--------TRAIN-TEST SPLIT WITH TEST FROM 39 TO 78--------\n",
      "(39, 2)\n",
      "\n",
      "--------TRAIN-TEST SPLIT WITH TEST FROM 78 TO 117--------\n",
      "(39, 2)\n",
      "\n",
      "--------TRAIN-TEST SPLIT WITH TEST FROM 117 TO 156--------\n",
      "(39, 2)\n",
      "\n",
      "--------TRAIN-TEST SPLIT WITH TEST FROM 156 TO 196--------\n",
      "(40, 2)\n"
     ]
    }
   ],
   "source": [
    "# Split deep learning input into training and test\n",
    "def split_k_fold(k, graph_output_folder):\n",
    "    random_survival_filtered_feature_df = pd.read_csv(os.path.join(graph_output_folder, 'random-survival-label.csv'))\n",
    "    num_points = random_survival_filtered_feature_df.shape[0]\n",
    "    num_div = int(num_points / k)\n",
    "    num_div_list = [i * num_div for i in range(0, k)]\n",
    "    num_div_list.append(num_points)\n",
    "    # Split [random_survival_filtered_feature_df] into [k] folds\n",
    "    for place_num in range(k):\n",
    "        low_idx = num_div_list[place_num]\n",
    "        high_idx = num_div_list[place_num + 1]\n",
    "        print('\\n--------TRAIN-TEST SPLIT WITH TEST FROM ' + str(low_idx) + ' TO ' + str(high_idx) + '--------')\n",
    "        split_input_df = random_survival_filtered_feature_df[low_idx : high_idx]\n",
    "        split_input_df.to_csv(os.path.join(graph_output_folder, 'split-random-survival-label-' + str(place_num + 1) + '.csv'), index=False)\n",
    "        print(split_input_df.shape)\n",
    "\n",
    "split_k_fold(k=5, graph_output_folder=graph_output_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 8.4 Reprocess the edge_index file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4288</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4289</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4290</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4291</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4292</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33126</th>\n",
       "      <td>6431</td>\n",
       "      <td>4992</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33127</th>\n",
       "      <td>6431</td>\n",
       "      <td>4993</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33128</th>\n",
       "      <td>6431</td>\n",
       "      <td>4994</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33129</th>\n",
       "      <td>6431</td>\n",
       "      <td>4995</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33130</th>\n",
       "      <td>6431</td>\n",
       "      <td>4996</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33131 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                   path             EdgeType\n",
       "0         0  4288          internal link  Gene-TRAN-Gene-PROT\n",
       "1         1  4289          internal link  Gene-TRAN-Gene-PROT\n",
       "2         2  4290          internal link  Gene-TRAN-Gene-PROT\n",
       "3         3  4291          internal link  Gene-TRAN-Gene-PROT\n",
       "4         4  4292          internal link  Gene-TRAN-Gene-PROT\n",
       "...     ...   ...                    ...                  ...\n",
       "33126  6431  4992  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33127  6431  4993  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33128  6431  4994  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33129  6431  4995  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33130  6431  4996  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[33131 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4288</th>\n",
       "      <td>5603</td>\n",
       "      <td>4306</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4289</th>\n",
       "      <td>5604</td>\n",
       "      <td>4306</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4290</th>\n",
       "      <td>5605</td>\n",
       "      <td>4306</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4291</th>\n",
       "      <td>6221</td>\n",
       "      <td>4306</td>\n",
       "      <td>Hippo signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4292</th>\n",
       "      <td>5603</td>\n",
       "      <td>4307</td>\n",
       "      <td>Oxytocin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33126</th>\n",
       "      <td>4992</td>\n",
       "      <td>6431</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33127</th>\n",
       "      <td>4993</td>\n",
       "      <td>6431</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33128</th>\n",
       "      <td>4994</td>\n",
       "      <td>6431</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33129</th>\n",
       "      <td>4995</td>\n",
       "      <td>6431</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33130</th>\n",
       "      <td>4996</td>\n",
       "      <td>6431</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28843 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                        path             EdgeType\n",
       "4288   5603  4306  Oxytocin signaling pathway  Gene-PROT-Gene-PROT\n",
       "4289   5604  4306  Oxytocin signaling pathway  Gene-PROT-Gene-PROT\n",
       "4290   5605  4306  Oxytocin signaling pathway  Gene-PROT-Gene-PROT\n",
       "4291   6221  4306     Hippo signaling pathway  Gene-PROT-Gene-PROT\n",
       "4292   5603  4307  Oxytocin signaling pathway  Gene-PROT-Gene-PROT\n",
       "...     ...   ...                         ...                  ...\n",
       "33126  4992  6431       Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33127  4993  6431       Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33128  4994  6431       Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33129  4995  6431       Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "33130  4996  6431       Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[28843 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4288</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4289</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4290</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4291</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4292</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61839</th>\n",
       "      <td>6431</td>\n",
       "      <td>4996</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61840</th>\n",
       "      <td>6431</td>\n",
       "      <td>6072</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61841</th>\n",
       "      <td>6431</td>\n",
       "      <td>6073</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61842</th>\n",
       "      <td>6431</td>\n",
       "      <td>6074</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61843</th>\n",
       "      <td>6431</td>\n",
       "      <td>6075</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61844 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                   path             EdgeType\n",
       "0         0  4288          internal link  Gene-TRAN-Gene-PROT\n",
       "1         1  4289          internal link  Gene-TRAN-Gene-PROT\n",
       "2         2  4290          internal link  Gene-TRAN-Gene-PROT\n",
       "3         3  4291          internal link  Gene-TRAN-Gene-PROT\n",
       "4         4  4292          internal link  Gene-TRAN-Gene-PROT\n",
       "...     ...   ...                    ...                  ...\n",
       "61839  6431  4996  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61840  6431  6072  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61841  6431  6073  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61842  6431  6074  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61843  6431  6075  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[61844 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "graph_output_folder = 'ROSMAP-graph-data'\n",
    "gene_edge_num_df = pd.read_csv(os.path.join(graph_output_folder, 'all-gene-edge-num.csv'))\n",
    "src_gene_list = list(gene_edge_num_df['src'])\n",
    "dest_gene_list = list(gene_edge_num_df['dest'])\n",
    "edgetype_list = list(gene_edge_num_df['EdgeType'])\n",
    "path_list = list(gene_edge_num_df['path'])\n",
    "gene_edge_num_reverse_df = pd.DataFrame({'src': dest_gene_list, 'dest': src_gene_list, 'path': path_list, 'EdgeType': edgetype_list})\n",
    "gene_edge_num_reverse_wointernal_df = gene_edge_num_reverse_df[~gene_edge_num_reverse_df['path'].str.contains('internal link')]\n",
    "display(gene_edge_num_df)\n",
    "display(gene_edge_num_reverse_wointernal_df)\n",
    "gene_edge_num_all_df = pd.concat([gene_edge_num_df, gene_edge_num_reverse_wointernal_df]).drop_duplicates().sort_values(by=['src', 'dest']).reset_index(drop=True)\n",
    "display(gene_edge_num_all_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4288</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4289</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4290</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4291</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4292</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61839</th>\n",
       "      <td>6431</td>\n",
       "      <td>4996</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61840</th>\n",
       "      <td>6431</td>\n",
       "      <td>6072</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61841</th>\n",
       "      <td>6431</td>\n",
       "      <td>6073</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61842</th>\n",
       "      <td>6431</td>\n",
       "      <td>6074</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61843</th>\n",
       "      <td>6431</td>\n",
       "      <td>6075</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61844 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                   path             EdgeType\n",
       "0         0  4288          internal link  Gene-TRAN-Gene-PROT\n",
       "1         1  4289          internal link  Gene-TRAN-Gene-PROT\n",
       "2         2  4290          internal link  Gene-TRAN-Gene-PROT\n",
       "3         3  4291          internal link  Gene-TRAN-Gene-PROT\n",
       "4         4  4292          internal link  Gene-TRAN-Gene-PROT\n",
       "...     ...   ...                    ...                  ...\n",
       "61839  6431  4996  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61840  6431  6072  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61841  6431  6073  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61842  6431  6074  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61843  6431  6075  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[61844 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1-TRAN</td>\n",
       "      <td>ABL1-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABL2-TRAN</td>\n",
       "      <td>ABL2-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1-TRAN</td>\n",
       "      <td>ACAA1-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACA-TRAN</td>\n",
       "      <td>ACACA-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB-TRAN</td>\n",
       "      <td>ACACB-PROT</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61839</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>FZD9-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61840</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>RSPO1-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61841</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>RSPO2-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61842</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>RSPO3-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61843</th>\n",
       "      <td>ZNRF3-PROT</td>\n",
       "      <td>RSPO4-PROT</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61844 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              src        dest                   path             EdgeType\n",
       "0       ABL1-TRAN   ABL1-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "1       ABL2-TRAN   ABL2-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "2      ACAA1-TRAN  ACAA1-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "3      ACACA-TRAN  ACACA-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "4      ACACB-TRAN  ACACB-PROT          internal link  Gene-TRAN-Gene-PROT\n",
       "...           ...         ...                    ...                  ...\n",
       "61839  ZNRF3-PROT   FZD9-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61840  ZNRF3-PROT  RSPO1-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61841  ZNRF3-PROT  RSPO2-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61842  ZNRF3-PROT  RSPO3-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61843  ZNRF3-PROT  RSPO4-PROT  Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[61844 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(gene_edge_num_all_df)\n",
    "gene_edge_num_all_df.to_csv(os.path.join(graph_output_folder, 'gene_edge_num_all_df.csv'), index=False)\n",
    "\n",
    "gene_edge_name_all_df = gene_edge_num_all_df.replace(sorted_all_gene_dict)\n",
    "display(gene_edge_name_all_df)\n",
    "gene_edge_name_all_df.to_csv(os.path.join(graph_output_folder, 'gene_edge_name_all_df.csv'), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4288</th>\n",
       "      <td>4288</td>\n",
       "      <td>4705</td>\n",
       "      <td>ErbB signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4289</th>\n",
       "      <td>4288</td>\n",
       "      <td>4706</td>\n",
       "      <td>ErbB signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4290</th>\n",
       "      <td>4288</td>\n",
       "      <td>5683</td>\n",
       "      <td>Neurotrophin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4291</th>\n",
       "      <td>4288</td>\n",
       "      <td>5684</td>\n",
       "      <td>Neurotrophin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4292</th>\n",
       "      <td>4288</td>\n",
       "      <td>5685</td>\n",
       "      <td>Neurotrophin signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61839</th>\n",
       "      <td>6431</td>\n",
       "      <td>4996</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61840</th>\n",
       "      <td>6431</td>\n",
       "      <td>6072</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61841</th>\n",
       "      <td>6431</td>\n",
       "      <td>6073</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61842</th>\n",
       "      <td>6431</td>\n",
       "      <td>6074</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61843</th>\n",
       "      <td>6431</td>\n",
       "      <td>6075</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "      <td>Gene-PROT-Gene-PROT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57556 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        src  dest                            path             EdgeType\n",
       "4288   4288  4705          ErbB signaling pathway  Gene-PROT-Gene-PROT\n",
       "4289   4288  4706          ErbB signaling pathway  Gene-PROT-Gene-PROT\n",
       "4290   4288  5683  Neurotrophin signaling pathway  Gene-PROT-Gene-PROT\n",
       "4291   4288  5684  Neurotrophin signaling pathway  Gene-PROT-Gene-PROT\n",
       "4292   4288  5685  Neurotrophin signaling pathway  Gene-PROT-Gene-PROT\n",
       "...     ...   ...                             ...                  ...\n",
       "61839  6431  4996           Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61840  6431  6072           Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61841  6431  6073           Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61842  6431  6074           Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "61843  6431  6075           Wnt signaling pathway  Gene-PROT-Gene-PROT\n",
       "\n",
       "[57556 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>src</th>\n",
       "      <th>dest</th>\n",
       "      <th>path</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4288</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4289</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4290</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4291</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4292</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-PROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4283</th>\n",
       "      <td>4283</td>\n",
       "      <td>2139</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4284</th>\n",
       "      <td>4284</td>\n",
       "      <td>2140</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4285</th>\n",
       "      <td>4285</td>\n",
       "      <td>2141</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4286</th>\n",
       "      <td>4286</td>\n",
       "      <td>2142</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-METH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4287</th>\n",
       "      <td>4287</td>\n",
       "      <td>2143</td>\n",
       "      <td>internal link</td>\n",
       "      <td>Gene-TRAN-Gene-METH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4288 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       src  dest           path             EdgeType\n",
       "0        0  4288  internal link  Gene-TRAN-Gene-PROT\n",
       "1        1  4289  internal link  Gene-TRAN-Gene-PROT\n",
       "2        2  4290  internal link  Gene-TRAN-Gene-PROT\n",
       "3        3  4291  internal link  Gene-TRAN-Gene-PROT\n",
       "4        4  4292  internal link  Gene-TRAN-Gene-PROT\n",
       "...    ...   ...            ...                  ...\n",
       "4283  4283  2139  internal link  Gene-TRAN-Gene-METH\n",
       "4284  4284  2140  internal link  Gene-TRAN-Gene-METH\n",
       "4285  4285  2141  internal link  Gene-TRAN-Gene-METH\n",
       "4286  4286  2142  internal link  Gene-TRAN-Gene-METH\n",
       "4287  4287  2143  internal link  Gene-TRAN-Gene-METH\n",
       "\n",
       "[4288 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# remove internal link\n",
    "kegg_path_gene_edge_num_all_df = gene_edge_num_all_df[~gene_edge_num_all_df['path'].str.contains('internal link')]\n",
    "display(kegg_path_gene_edge_num_all_df)\n",
    "kegg_path_gene_edge_num_all_df.to_csv(os.path.join(graph_output_folder, 'keggpath-gene-edge-num-all.csv'), index=False)\n",
    "# keep only internal link\n",
    "internal_gene_edge_num_all_df = gene_edge_num_all_df[gene_edge_num_all_df['path'].str.contains('internal link')]\n",
    "display(internal_gene_edge_num_all_df)\n",
    "internal_gene_edge_num_all_df.to_csv(os.path.join(graph_output_folder, 'internal-gene-edge-num-all.csv'), index=False)\n",
    "\n",
    "kegg_path_gene_edge_name_all_df = kegg_path_gene_edge_num_all_df.replace(sorted_all_gene_dict)\n",
    "kegg_path_gene_edge_name_all_df.to_csv(os.path.join(graph_output_folder, 'keggpath-gene-edge-name-all.csv'), index=False)\n",
    "internal_gene_edge_name_all_df = internal_gene_edge_num_all_df.replace(sorted_all_gene_dict)\n",
    "internal_gene_edge_name_all_df.to_csv(os.path.join(graph_output_folder, 'internal-gene-edge-name-all.csv'), index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "win_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
